The function <code>strerror</code> returns a buffer that is only valid until the next call to <code>strerror</code>. In a multithreaded environment, you don't know when this next call will happen, which makes this function dangerous to call. You should use thread-safe alternatives, such as https://en.cppreference.com/w/c/string/byte/strerror[<code>strerror_s</code>] or <code>strerror_r</code>.

Note that <code>strerror_s</code> is defined in annex K of C11, so to have access to it, you need a standard library that supports it (this can be tested with the macro <code>__STDC_LIB_EXT1_</code>_), and you need to enable it by defining the macro <code>}}</code>_<code><code>_STDC_WANT_LIB_EXT1__</code> before including <code><string.h></code>.


== Noncompliant Code Example

----
char *msg = strerror(errno);{code}
----

== Compliant Solution

----
size_t size = strerrorlen_s(errno);
char *msg = malloc(size);
strerror_s(msg, size);{code}
----

