include::../description.adoc[]

== Noncompliant Code Example

----
Module Module1
    Sub Main()
        Dim foo() As String ' Noncompliant
    End Sub
End Module
----

== Compliant Solution

----
Module Module1
    Sub Main()
        Dim foo As String() ' Compliant
    End Sub
End Module
----
