The system field <code>SY-SUBRC</code> must be tested immediately after any statement setting this variable. Reading this variable informs on previous operation success or errors. Such errors should be handled properly so that the program continues in a consistent state.

This rule raises an issue when the field <code>SY-SUBRC</code> is not checked just after performing one of the following operations:

* Calling a function which can throw exceptions.
* Calling one of the file access operation <code>OPEN DATASET</code>, <code>READ DATASET</code> or <code>DELETE DATASET</code>.

<code>SY-SUBRC</code> check must be done either with the <code>CASE</code>, <code>IF</code> or <code>CHECK</code> statement.


== Noncompliant Code Example

In the following case nothing happens if the exceptions <code>NOT_FOUND</code> or <code>OTHERS</code> are raised:

----
CALL FUNCTION 'STRING_SPLIT'
  EXPORTING
    DELIMITER = ':'
    STRING = FELD
  IMPORTING
    HEAD =   HEAD
    TAIL = TAIL
  EXCEPTIONS
    NOT_FOUND = 1
    OTHERS = 2.
----


== Compliant Solution

----
CALL FUNCTION 'STRING_SPLIT'
  EXPORTING
    DELIMITER = ':'
    STRING = FELD
  IMPORTING
    HEAD =   HEAD
    TAIL = TAIL
  EXCEPTIONS
    NOT_FOUND = 1
    OTHERS = 2.
CASE SY-SUBRC.
  WHEN 1. ...
  WHEN 2. ...
  WHEN OTHER.
ENDCASE.
----


== Exceptions

No issue will be raised in the following cases:

* One or more <code>WRITE</code> operation are performed between the statement setting <code>SY-SUBRC</code> and its check. An exception will be however raised if the <code>WRITE</code> operation is a <code>WRITE ... TO</code>, as this will set <code>SY-SUBRC</code> too.
* <code>SY-SUBRC</code>'s value is assigned to a variable. We then assume that it will be checked later.

----
OPEN DATASET my_dataset FOR INPUT IN TEXT MODE ENCODING DEFAULT. " Compliant
WRITE 'Test'. " WRITE is accepted before checking SY-SUBRC
IF SY-SUBRC <> 0. 
    EXIT.
ENDIF.

OPEN DATASET my_dataset FOR INPUT IN TEXT MODE ENCODING DEFAULT. " Compliant
Tmp = SY-SUBRC. " Assigning SY-SUBRC value to a variable. We assume that it will be checked later.
IF Tmp <> 0.
    EXIT.
ENDIF.
----

