== Noncompliant Code Example

----
void f1()
{
  try
  {
    // ...
  }
  catch (...)
  {
    // Handle all exception types
  }
  catch (std::exception const &e)  // Noncompliant - handler will never be called
  {
  }
}
----
