You should avoid opening a cursor inside a <code>PERFORM</code> statement because doing so could impact performance, or lead to unexpected behavior if the the closing of the cursor is not defined in the same loop.


== Noncompliant Code Example

----
  PERFORM UNTIL (NOT DA-OK) OR (Y00CIA-CD-RET-PGM = ZERO)
    EXEC SQL OPEN C2
    END-EXEC
  END-PERFORM.
----

