include::../description.adoc[]

== Noncompliant Code Example

----
Integer divide(Integer numerator, Integer denominator) {
  return numerator / denominator;              // FIXME denominator value might be  0
}
----

include::../see.adoc[]
