include::../description.adoc[]

== Noncompliant Code Example

----
def hello(name):
    message = "Hello " + name # Noncompliant
    print(name)
for i in range(10):
    foo()
----

== Compliant Solution

----
def hello(name):
    message = "Hello " + name
    print(message)
for _ in range(10):
    foo()
----

== Exceptions

<code>_</code> as well as tuples will not raise an issue for this rule. The following examples are compliant:
----
for _ in range(10):     
    do_something()
username, login, password = auth
do_something_else(username, login)
----
