<p>Using the &lt;code&gt;OPEN&lt;/code&gt; file statement is costly, and therefore be avoided inside loops. Instead, the file can be saved into a buffer to increase performance.</p>
<h2>Noncompliant Code Example</h2>
<pre>
  PERFORM UNTIL (NOT DA-OK) OR (Y00CIA-CD-RET-PGM = ZERO)
     OPEN INPUT inventory-file
  END-PERFORM.
</pre>
