Primitives can be read and written to atomically. Except for <code>long</code> and <code>double</code>, that is. These 64-bit primitives must be marked <code>volatile</code> in multi-threaded environments, or swapped out for their atomic counterparts: <code>AtomicLong</code>, and <code>AtomicDouble</code> to guarantee that their updates are always visible to other threads.

Similarly, to ensure that updates to 32-bit primitives are visible to all threads, they should also be marked <code>volatile</code>.


== Noncompliant Code Example

----
long m = 0;  // Noncompliant

public void increment() {
  m++;
}
----


== Compliant Solution

----
volatile long m = 0;

public void increment() {
  m++;
}
----
or

----
AtomicLong m = new AtomicLong(0);

public void increment() {
  m.incrementAndGet();
}
----


== See

* https://www.securecoding.cert.org/confluence/x/PoFJAg[CERT, VNA00-J.] - Ensure visibility when accessing shared primitive variables
* https://www.securecoding.cert.org/confluence/x/8IBBAg[CERT, VNA05-J.] - Ensure atomicity when reading and writing 64-bit values

