include::../description.adoc[]

== Noncompliant Code Example

----
function numberOfMinutes($hours) {
  $seconds = 0;   // seconds is never used
  return hours * 60;
}
----

== Compliant Solution

----
function numberOfMinutes($hours) {
  return hours * 60;
}
----
