A test case without assertions ensures only that no exceptions are thrown. Beyond basic runnability, it ensures nothing about the behavior of the code under test.

Â 

This rule raises an issue when a function with an <code>@isTest</code> annotation does not call at least one of the following functions once:

* <code>System.assert</code>
* <code>System.assertEquals</code>
* <code>System.assertNotEquals</code>

== Noncompliant Code Example

----
@isTest()
class ApexTests {
    @isTest
    private static void mytest() { // Noncompliant
        MyClass.createTask();
    }
}
----

== Compliant Solution

----
@isTest()
class ApexTests {
    @isTest
    private static void mytest() {
        Task task = MyClass.createTask();
        System.assert("expected subject", task.subject, "task has not the expected subject");
    }
}
----
