The functions <code>memcpy</code>, <code>memmove</code> and <code>memset</code> can only be used for objects of trivially copyable types. This includes scalar types, arrays, and trivially copyable classes.

A class type is trivially copyable if:

* One or more of the following methods is trivial and the rest are deleted: copy constructor, move constructor, copy assignment operator, and move assignment operator, 
* It has a trivial, non-deleted destructor,
* It has trivially copyable members and base classes,
* It has no virtual functions.


== Noncompliant Code Example

----
class Shape {
public:
  int x;
  int y;
  virtual ~Shape(); // This makes the class non trivially copyable
};

void f(Shape *dest, Shape *source)
{
    memcpy(dest, source, sizeof Shape); // Noncompliant
}
----


== Compliant Solution

----
class Shape {
public:
  int x;
  int y;
  virtual ~Shape(); // This makes the class non trivially copyable
};

void f(Shape *dest, Shape *source)
{
    (*dest) = (*source);
}
----


