Loggers should be:
* <code>private</code>: not accessible outside of their parent classes. If another class needs to log something, it should instantiate its own logger.
* <code>static</code>: not dependent on an instance of a class (an object). When logging something, contextual information can of course be provided in the messages but the logger should be created at class level to prevent creating a logger along with each object.
* <code>const</code>: created once and only once per class.

== Noncompliant Code Example

With the default regular expression <code>LOG(?:GER)?</code>:
----
public const logger:ILogger = LogUtil.getLogger(MyClass);
----

== Compliant Solution

----
private static const LOG:ILogger = LogUtil.getLogger(MyClass);
----
