Once set, the value of a Hibernate <code>@Entity</code>'s <code>@Id</code> field/column should never be updated. Therefore, setters for such fields should always be <code>private</code>.


== Noncompliant Code Example

----
public class Book {

  @Id
  @GeneratedValue
  private int id;

  public void setId(int id) {  // Noncompliant
    this.id = id;
  }
----


== Compliant Solution

----
public class Book {

  @Id
  @GeneratedValue
  private int id;

  private void setId(int id) {
    this.id = id;
  }
----

