<p>If an attempt is made to use an identifier in a preprocessor directive, and that identifier has not been defined, the preprocessor will assume the value zero. &lt;code&gt;#ifdef&lt;/code&gt;, &lt;code&gt;#ifndef&lt;/code&gt; and &lt;code&gt;defined()&lt;/code&gt; are provided to test the existence of a macro, and are therefore excluded.</p>
<h2>Noncompliant Code Example</h2>
<pre>
#if x &lt; 0 // Noncompliant, x assumed to be zero as it is not defined
// ...
#endif
</pre>
<h2>Compliant Solution</h2>
<pre>
#ifdef x
#if x &lt; 0 // Compliant, x is guaranteed to be defined.
// ...
#endif
#endif

// ...

#define Y 12

// ...

#if Y &lt; 12 // Compliant, y is defined.
#endif
</pre>
<h2>See</h2>
<ul>
<li>MISRA C++:2008, 16-0-7</li>
</ul>
