It is often considered good practice at the end of an override to invoke <code>super</code>, but there are cases where according to the Apple developer documentation this should not be done.

* <code>updateLayer</code> - optimize the rendering of your view
* <code>loadView</code> - provide a <code>view</code> when <code>view</code> is <code>nil</code>
* <code>providePlaceholder</code> - provide a placeholder for a document returned by the Document Picker but not yet stored locally

In all cases, these are actions that should happen once and only once. Subsequently invoking <code>super</code> would see your desired result replaced (at best) by less specialized results.


== Noncompliant Code Example

----
class VC: UIMyViewController {
  override func loadView() {
    // ...  
    super.loadView()
  }
}
----


== Compliant Solution

----
class VC: UIMyViewController {
  override func loadView() {
    // ...  
  }
}
----

