Shared coding conventions allow teams to collaborate efficiently. This rule checks that all enumeration member names match a provided regular expression.

== Noncompliant Code Example

With default provided regular expression: <code>^[a-z][a-zA-Z0-9]*$</code>

----
enum SomeEnumeration {
    case SomeMember  // Non-Compliant
}
----

== Compliant Solution

----
enum SomeEnumeration {
    case someMember
}
----
