include::../description.adoc[]

== Noncompliant Code Example

With the default threshold value of 3
----
if ((($condition1 && $condition2) || ($condition3 && $condition4)) && $condition5) { ... } 
----

== Compliant Solution

----
if ( (my_first_condition() || my_second_condition()) && my_last_condition()) { ... }     
----
