<p>Automatic variables are those that are allocated automatically when program flow enters the objectâ€™s scope, and deallocated automatically when program flow leaves. Therefore returning the address of an automatic variable or object is an error because by the time the calling code attempts to use the value at the returned address, it no longer exists.</p>
<p>Unfortunately, this type of error is not always immediately evident. While the relevant memory has been deallocated, it may not have been overwritten by the time the object is dereferenced, thus leading to unpredictable behavior because sometimes the dereference is fully successful, sometimes it is partially successful (only parts of the object have been overwritten) and other times the dereference is a complete failure.</p>
<h2>Noncompliant Code Example</h2>
<pre>
int * divide (int numerator, int denominator) {
  int result = numerator / denominator;
  return (&amp;result);
}
</pre>
<h2>Compliant Solution</h2>
<pre>
int divide (int numerator, int denominator) {
  int result = numerator / denominator;
  return result;
}
</pre>
<h2>See</h2>
<ul>
<li>MISRA C++:2008, 7-5-1 - A function shall not return a reference or a pointer to an automatic variable (including parameters), defined within the function.</li>
<li><a href="https://www.securecoding.cert.org/confluence/x/bQ4">CERT, DCL30-C.</a> - Declare objects with appropriate storage durations</li>
</ul>
