Since Oracle 11.2, <code>RELIES_ON</code> has been deprecated because the dependencies of result cache-enabled functions are automatically computed.


== Noncompliant Code Example

----
CREATE OR REPLACE FUNCTION foo RETURN PLS_INTEGER RESULT_CACHE RELIES_ON(DUAL) AS -- Noncompliant
BEGIN
  RETURN 0;
END;
/

DROP FUNCTION foo;
----


== Compliant Solution

----
CREATE OR REPLACE FUNCTION foo RETURN PLS_INTEGER RESULT_CACHE AS
BEGIN
  RETURN 0;
END;
/

DROP FUNCTION foo;
----

