Having all branches in a <code>CASE</code>, <code>IF</code> or <code>IIF</code> chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be a <code>CASE</code>/<code>IF</code>/<code>IIF</code> chain at all.

== Noncompliant Code Example

----
IF @x < 25 -- Noncompliant
  PRINT 'A'
ELSE IF @x < 10
  PRINT 'A'
ELSE
  PRINT 'A'

SELECT
  CASE col1 -- Noncompliant
    WHEN 1 THEN 'A'
    WHEN 2 THEN 'A'
    ELSE        'A'
  END,
  IIF(col1 < 25, 'A', 'A') -- Noncompliant
 FROM table1
----

== Exceptions

This rule does not apply to <code>IF</code>/<code>CASE</code> chains without <code>ELSE</code> clauses.

----
IF @x < 25 -- no issue, this could have been done on purpose to make the code more readable
  PRINT 'A'
ELSE IF @x > 10
  PRINT 'A'
----
