<p>All reserved words should be written using the same case to ensure consistency in the code.
This rule checks that reserved words are all in lower case.</p>
<h2>Noncompliant Code Example</h2>
<pre>
begin
  null;
  NULL; -- Noncompliant
end;
/
</pre>
