When using a <code>PRINTER</code> defined file, an <code>INFDS</code> keyword should be used to read the file's line count from <code>*PSSR</code> positions 367-368.


== Noncompliant Code Example

----
 FCSO602R1  O    E             PRINTER USROPN 
 or ... 
 FCSO602R1  O    E             PRINTER USROPN 
 F                                     INFDS(WSFD01) 
 F                                     INFSR(*PSSR) 

 D WSFD01          DS 
 D  W1ST01           *STATUS 
 D  W1SPNO               123    124B 0 
----


== Compliant Solution

----
 FCSO602R1  O    E             PRINTER USROPN 
 F                                     INFDS(WSFD01) 
 F                                     INFSR(*PSSR) 

 D WSFD01          DS 
 D  W1ST01           *STATUS 
 D  W1SPNO               123    124B 0 
 D  W1LNCT               367    368B 0 
 D  W1PGNO               369    372B 0 
----

