include::../description.adoc[]

== Noncompliant Code Example

----
public class Foo
{
    public const double Version = 1.0;           // Noncompliant
}
----

== Compliant Solution

----
public class Foo
{
    public static double Version 
    {
      get { return 1.0; }
    }
}
----
