Taking the address of a library function is not something robust: The library might make changes to a function that are compatible with a normal use of a function, but not with taking its address (for instance, adding a parameter with a default value, or adding an overload to an overload set). More specifically, the standard library has stated that there would be no barrier against such changes, and that for stability users should not take the address of standard library functions.


== Noncompliant Code Example

----
int main() {
  std::unique_ptr<FILE, int(*)(FILE*)> fp(
    fopen("test.txt", "r"),
    fclose); // Non compliant, address of fclose is implicitly taken
  // Work with fp
}
----


== Compliant Solution

----
int main() {
  std::unique_ptr<FILE, int(*)(FILE*)> fp(
    fopen("test.txt", "r"),
    [](FILE*file){return fclose(file);});
  // Work with fp
}
----


== See

* https://wg21.link/P0921 - Document of the C++ standardization committee about change management in the standard library.


