<p>There is no reason to use Javadoc-specific tags in non-Javadoc comments, and doing to could be confusing.</p>
<h2>Noncompliant Code Example</h2>
<pre>
// FIXME should be moved to {@link ServerPluginRepository#uninstall(String)}  &lt;-- Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
// FIXME should be moved to ServerPluginRepository#uninstall(String)
</pre>
