There is no reason to use Javadoc-specific tags in non-Javadoc comments, and doing to could be confusing.


== Noncompliant Code Example

----
// FIXME should be moved to {@link ServerPluginRepository#uninstall(String)}  <-- Noncompliant
----


== Compliant Solution

----
// FIXME should be moved to ServerPluginRepository#uninstall(String)
----

