The class <code>java.util.zip.GZIPInputStream</code> is already buffering its input while reading. Thus passing a <code>java.io.BufferedInputStream</code> to a <code>java.util.zip.GZIPInputStream</code> is redundant. It is more efficient to directly pass the original input stream to <code>java.util.zip.GZIPInputStream</code>.

Note that the default buffer size of <code>GZIPInputStream</code> is not the same as the one in <code>BufferedInputStream</code>. Configure it if need be.

This rule raises an issue when a <code>java.util.zip.GZIPInputStream</code> reads from a <code>java.io.BufferedInputStream</code>.


== Noncompliant Code Example

----
import java.io.*;
import java.util.zip.GZIPInputStream;

public class Noncompliant {

    void deflateFile(final File file) throws IOException {
        try (
            FileInputStream fileStream = new FileInputStream(file);
            BufferedInputStream bufferedStream = new BufferedInputStream(fileStream);
            InputStream input = new GZIPInputStream(bufferedStream); // Noncompliant
        ) {
            // process the input
        }
    }
}
----


== Compliant Solution

----
import java.io.*;
import java.util.zip.GZIPInputStream;
public class Compliant {

    void deflateFile(final File file) throws IOException {
        try (
            FileInputStream fileStream = new FileInputStream(file);
            InputStream input = new GZIPInputStream(fileStream);
        ) {
            // process the input
        }
    }
}
----


== See

* Java Performance Tuning Guide - http://java-performance.info/java-io-bufferedinputstream-and-java-util-zip-gzipinputstream/[java.io.BufferedInputStream and java.util.zip.GZIPInputStream]

