Jump statements (<code>return</code>, <code>break</code>, <code>continue</code>, and <code>raise</code>) move control flow out of the current code block. So any statements that come after a jump are dead code.

== Noncompliant Code Example

----
def fun(a):
  i = 10
  return i + a       # Noncompliant 
  i += 1             # this is never executed
----

== Compliant Solution

----
def fun(a):
  i = 10
  return i + a
----

include::../see.adoc[]
