<p>Correctly updating a &lt;code&gt;static&lt;/code&gt; field from a non-static method is tricky to get right and could easily lead to bugs if there are multiple class instances  and/or multiple threads in play. Ideally, &lt;code&gt;static&lt;/code&gt; fields are only updated from &lt;code&gt;synchronized static&lt;/code&gt; methods.</p>
<p>This rule raises an issue each time a &lt;code&gt;static&lt;/code&gt; field is updated from a non-static method.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public class MyClass {

  private static int count = 0;

  public void doSomething() {
    //...
    count++;  // Noncompliant
  }
}
</pre>
