Using certain features of regular expressions, it is possible to create regular expressions that can never match or contain subpatterns that can never match. Since a pattern or sub-pattern that can never match any input is pointless, this is a sign that the pattern does not work as intended and needs to be fixed.

This rule finds such regular expressions and subpatterns. Possible causes of this could be:

* A https://www.regular-expressions.info/possessive.html[possessive quantifier] (such as <code>*+</code>) is applied to a sub-pattern that matches the same thing as (or a superset of) the pattern that follows the quantifier. In that case the subsequent pattern will always fail because anything it could match would be already consumed by the possessive quantifier, causing the whole pattern to fail. In that case possible fixes might be to make the quantifier non-possessive or to rewrite the sub-patterns so that they do not overlap.
* A positive lookahead is followed by a pattern that does not match it or a negative lookahead is followed by a pattern that does match it (similarly for lookbehinds preceded by other patterns).
* A back reference refers to a capturing group that will never be matched before the back reference


== Noncompliant Code Example

=== Possessive Quantifiers

----
[a-z]*+a
----

=== Lookahead

----
^(?:[a-z]+\d+(?=a))*$
----

(when the lookahead for <code>a</code> succeeds, the following <code>$</code> will fail).

=== Backreference

----
\1(.)
----


== Compliant Solution

=== Possessive Quantifiers

----
[a-z]*a
----
(removed the possessive quantifier)

or

----
[b-z]*+a
----
(changed the character class to not overlap with the following pattern)

=== Lookahead

----
^(?:[a-z]+\d+(?=a|$))*$
----

or

----
^(?:[a-z]+\d+(?:a[a-z]*\d+)*)?$
----

=== Backreference

----
(.)\1
----


