This variant of the <code>REFRESH</code> statement is deprecated and should be avoided.

This <code>REFRESH</code> statement initializes the internal table <code>itab</code>, reads several rows from the database table <code>dbtab</code>, and adds their contents to the internal table <code>itab</code>. A <code>SELECT</code> statement should be used instead.


== Noncompliant Code Example

----
TABLES t100. 
DATA itab TYPE STANDARD TABLE OF t100.

t100-sprsl = 'E'. 
t100-arbgb = 'BC'.

REFRESH itab FROM TABLE t100.
----


== Compliant Solution

----
DATA itab TYPE STANDARD TABLE OF t100.

SELECT * 
       FROM t100 
       INTO TABLE itab 
       WHERE sprsl = 'E' AND 
             arbgb LIKE 'BC%'. 
----

