Browsers best support the <code>window</code> mode for the <code>wmode</code> parameter, also in terms of accessibility.
As it is the default mode, it is acceptable to either not specify a <code>wmode</code> parameter altogether, or to set it explicitly to <code>window</code>.


== Noncompliant Code Example

----
<object classid="clsid:d27cdb6e-ae6d-11cf-96b8-444553540000" width="550" height="400">
  <param name="movie" value="movie_name.swf" />
  <param name="wmode" value="direct" />                              <!-- Non-Compliant -->
</object>

<embed src="movie_name.swf"
       width="550"
       height="400"
       wmode="direct"                                                                 <!-- Non-Compliant -->
       type="application/x-shockwave-flash"
       pluginspage="http://www.macromedia.com/go/getflashplayer" />
----


== Compliant Solution

----
<object classid="clsid:d27cdb6e-ae6d-11cf-96b8-444553540000" width="550" height="400">
  <param name="movie" value="movie_name.swf" />
</object>

<embed src="movie_name.swf"
       width="550"
       height="400"
       type="application/x-shockwave-flash"
       pluginspage="http://www.macromedia.com/go/getflashplayer" />
----

or

----
<object classid="clsid:d27cdb6e-ae6d-11cf-96b8-444553540000" width="550" height="400">
  <param name="movie" value="movie_name.swf" />
  <param name="wmode" value="window" />
</object>

<embed src="movie_name.swf"
       width="550"
       height="400"
       wmode="window"
       type="application/x-shockwave-flash"
       pluginspage="http://www.macromedia.com/go/getflashplayer" />
----

