<p>A &lt;code&gt;readObject&lt;/code&gt; method is written when a &lt;code&gt;Serializable&lt;/code&gt; object needs special handling to be rehydrated from file. It should be the case that the object being created by &lt;code&gt;readObject&lt;/code&gt; is only visible to the thread that invoked the method, and the &lt;code&gt;synchronized&lt;/code&gt; keyword is not needed, and using &lt;code&gt;synchronized&lt;/code&gt; anyway is just confusing. If this is not the case, the method should be refactored to make it the case.</p>
<h2>Noncompliant Code Example</h2>
<pre>
private synchronized void readObject(java.io.ObjectInputStream in)
     throws IOException, ClassNotFoundException { // Noncompliant
  //...
}
</pre>
<h2>Compliant Solution</h2>
<pre>
private void readObject(java.io.ObjectInputStream in)
     throws IOException, ClassNotFoundException { // Compliant
  //...
}
</pre>
