A <code>readObject</code> method is written when a <code>Serializable</code> object needs special handling to be rehydrated from file. It should be the case that the object being created by <code>readObject</code> is only visible to the thread that invoked the method, and the <code>synchronized</code> keyword is not needed, and using <code>synchronized</code> anyway is just confusing. If this is not the case, the method should be refactored to make it the case.


== Noncompliant Code Example

----
private synchronized void readObject(java.io.ObjectInputStream in)
     throws IOException, ClassNotFoundException { // Noncompliant
  //...
}
----


== Compliant Solution

----
private void readObject(java.io.ObjectInputStream in)
     throws IOException, ClassNotFoundException { // Compliant
  //...
}
----

