include::../description.adoc[]

== Noncompliant Code Example

----
try {
  file.open("test.txt");
} catch (...) {  // Noncompliant
  // ...
}
----

== Compliant Solution

----
try {
  file.open("test.txt");
} catch (std::ifstream::failure e) {
  // ...
}
----

== Exceptions

There are cases though where you want to catch all exceptions, because no exceptions should be allowed to escape the function, and generic <code>catch</code> handlers are excluded from the rule:
* In the main function
* In a <code>noexcept</code> function
* In an <code>extern "C"</code> function

Additionally, if the <code>catch</code> handler is throwing an exception (either the same as before, with <code>throw;</code> or a new one that may make more sense to the callers of the function), or is never exiting (because it calls a <code>noreturn</code> function, for instance <code>exit</code>), then the accurate type of the exception usually does not matter any longer: this case is excluded too.
