include::../description.adoc[]

== Noncompliant Code Example

----
int divide(int numerator, int denominator) {
  return numerator / denominator;              // FIXME denominator value might be  0
}
----

include::../see.adoc[]
