<p>Nested &lt;code&gt;switch&lt;/code&gt; structures are difficult to understand because you can easily confuse the &lt;code&gt;when&lt;/code&gt; blocks of an inner &lt;code&gt;switch&lt;/code&gt; as belonging to an outer statement. Therefore nested &lt;code&gt;switch&lt;/code&gt; statements should be avoided.</p>
<p>Specifically, you should structure your code to avoid the need for nested &lt;code&gt;switch&lt;/code&gt; statements, but if you cannot, then consider moving the inner &lt;code&gt;switch&lt;/code&gt; to another function.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public void foo(Integer i, Integer j) {
  switch on i {
    when 1 {System.debug(' 1'); }
    when 2 { System.debug(' 2'); }
    when -3 {
      switch on j {  // Noncompliant
        when 1 {System.debug(' 3');}
        when else {System.debug(' 4');}
      }
    }
  }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
public void foo(Integer i, Integer j) {
  switch on i {
    when 1 {System.debug(' 1'); }
    when 2 { System.debug(' 2'); }
    when -3 {bar(j);}
  }
}

public void bar(Integer j) {
  switch on j {
    when 1 {System.debug(' 3');}
    when else {System.debug(' 4');}
  }
}
</pre>
