<p>Nested &lt;code&gt;switch&lt;/code&gt; structures are difficult to understand because you can easily confuse the cases of an inner &lt;code&gt;switch&lt;/code&gt; as belonging to an outer statement. Therefore nested &lt;code&gt;switch&lt;/code&gt; statements should be avoided.</p>
<p>Specifically, you should structure your code to avoid the need for nested &lt;code&gt;switch&lt;/code&gt; statements, but if you cannot, then consider moving the inner &lt;code&gt;switch&lt;/code&gt; to another function.</p>
<h2>Noncompliant Code Example</h2>
<pre>
function foo(n, m) {
  switch (n) {
    case 0:
      switch (m) {  // Noncompliant; nested switch
        // ...
      }
    case 1:
      // ...
    default:
      // ...
  }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
function foo(n, m) {
  switch (n) {
    case 0:
      bar(m);
    case 1:
      // ...
    default:
      // ...
  }
}

function bar(m) {
  switch(m) {
    // ...
  }
}
</pre>
