<p>Nested &lt;code&gt;switch&lt;/code&gt; structures are difficult to understand because you can easily confuse the cases of an inner &lt;code&gt;switch&lt;/code&gt; as belonging to an outer statement. Therefore nested &lt;code&gt;switch&lt;/code&gt; statements should be avoided.</p>
<p>Specifically, you should structure your code to avoid the need for nested &lt;code&gt;switch&lt;/code&gt; statements, but if you cannot, then consider moving the inner &lt;code&gt;switch&lt;/code&gt; to another function.</p>
<h2>Noncompliant Code Example</h2>
<pre>
switch($a) {
  case "foo":
    switch($b) { // Noncompliant
      case "bar":
        doSomething();
      break;
    }
  break;
}
</pre>
<h2>Compliant Solution</h2>
<pre>
function handleFoo($foo) {
  switch($foo) {
    case "bar":
      doSomething();
    break;
  }
}

switch($a) {
  case "foo":
    handleFoo($b);
  break;
}
</pre>
