include::../description.adoc[]

== Noncompliant Code Example

----
Class MyException
    Inherits Exception

    Public Sub MyException()
        If bad_thing Then
            Throw New Exception("A bad thing happened")
        End If
    End Sub
End Class
----
