In C++, the literal 0 is both an integer type and the null-pointer-constant. To meet developer expectations, <code>NULL</code> should be used as the null-pointer-constant, and 0 for the integer zero.


== Noncompliant Code Example

----
#include <cstddef>

void f1 ( int32_t );
void f2 ( int32_t * );
void f3 ( )
{ 
  f1 ( 0 ); // Compliant, integer expected
  f2 ( 0 ); // Noncompliant, 0 used as the null pointer constant
  f2 ( NULL ); // Compliant, pointer expected
}
----


== See

* MISRA C++:2008, 4-10-2 - Literal zero (0) shall not be used as the null-pointer-constant.

