By default, Hibernate <code>session</code> flushing is set to <code>FlushMode.AUTO</code>, and is called from <code>Transaction.commit</code>, <code>Session.flush</code> and before some queries are executed. Setting it to <code>FlushMode.COMMIT</code>, <code>FlushMode.NEVER</code>, or <code>FlushMode.MANUAL</code> could mean that parts of your application get stale data, so you should be sure of what you're doing before you use any of these modes.

This rule raises an issue when flush mode is explicitly set to any of these modes.


== Noncompliant Code Example

----
Session session = sessionFactory.openSession();
session.setFlushMode(FlushMode.NEVER);  // Noncompliant
----

