<p>Using &lt;code&gt;Collection.size()&lt;/code&gt; to test for emptiness works, but using &lt;code&gt;Collection.isEmpty()&lt;/code&gt; makes the code more readable and can be more performant. The time complexity of any &lt;code&gt;isEmpty()&lt;/code&gt; method implementation should be &lt;code&gt;O(1)&lt;/code&gt; whereas some implementations of &lt;code&gt;size()&lt;/code&gt; can be &lt;code&gt;O(n)&lt;/code&gt;.</p>
<h2>Noncompliant Code Example</h2>
<pre>
if (myCollection.size() == 0) {  // Noncompliant
  /* ... */
}
</pre>
<h2>Compliant Solution</h2>
<pre>
if (myCollection.isEmpty()) {
  /* ... */
}
</pre>
