<p>Using &lt;code&gt;count()&lt;/code&gt; to test for emptiness works, but using &lt;code&gt;empty()&lt;/code&gt; makes the code more readable and can be more performant</p>
<h2>Noncompliant Code Example</h2>
<pre>
if (count($a) &gt; 0) { // Noncompliant
 echo $a[0];
}
</pre>
<h2>Compliant Solution</h2>
<pre>
if (!empty($a)) {
  echo $a[0];
}
</pre>
