Classes can register themselves to receive notifications using <code>NotificationCenter.add</code>. Having done so, it seems suspicious that a class would opt to stop receiving notifications before de-initialization. For that reason, this rule raises an issue when <code>NotificationCenter.default.removeObserver(self)</code> is called anywhere but in <code>deinit</code>


== Noncompliant Code Example

----
class MyClass {
  func doTheThing() {
    //...
    NotificationCenter.default.removeObserver(self)  // Noncompliant
  }
}
----


== Compliant Solution

----
class MyClass {
  func doTheThing() {
    //...
  }

  func deinit() {
    NotificationCenter.default.removeObserver(self)
  }
}
----


