<p>Aligning common keywords in a series of statements makes the code easier to read. therefore, it is better to align the &lt;code&gt;TO&lt;/code&gt; keywords in a series of successive MOVE statements.</p>
<h2>Noncompliant Code Example</h2>
<pre>
MOVE "Hi There" TO field
MOVE temp TO b
MOVE 123 TO item
</pre>
<h2>Compliant Solution</h2>
<pre>
MOVE "Hi There" TO field
MOVE temp       TO b
MOVE 123        TO item
</pre>
