include::../description.adoc[]

== Noncompliant Code Example

----
import java.io.IOException;
import javax.servlet.http.HttpServletRequest;

public void runUnsafe(HttpServletRequest request) throws IOException {
  String folder = request.getParameter("folder");

  String cmd = "mkdir " + folder;

  Runtime.getRuntime().exec(cmd);
}
----

== Compliant Solution

----
import java.io.IOException;
import javax.servlet.http.HttpServletRequest;

import org.owasp.esapi.ESAPI;
import org.owasp.esapi.codecs.WindowsCodec;

public void runSafe(HttpServletRequest request) throws IOException {
  String folder = request.getParameter("folder");

  String cmd = "mkdir " + ESAPI.encoder().encodeForOS(new WindowsCodec(), folder);

  Runtime.getRuntime().exec(cmd);
}
----

include::../see.adoc[]
