It is possible to use <code>await</code> on values which are not <code>Promise</code>s, but it's useless and misleading. The point of <code>await</code> is to pause execution until the <code>Promise</code>'s asynchronous code has run to completion. With anything other than a <code>Promise</code>, there's nothing to wait for.

This rule raises an issue when an <code>await</code>ed value is guaranteed not to be a <code>Promise</code>.


== Noncompliant Code Example

----
let x = 42;
await x; // Noncompliant
----


== Compliant Solution

----
let x = new Promise(resolve => resolve(42));
await x;

let y = p ? 42 : new Promise(resolve => resolve(42));
await y;
----


