Multiple imports from the same module should be merged together to improve readability.


== Noncompliant Code Example

----
import { B1 } from 'b';
import { B2 } from 'b'; // Noncompliant
----


== Compliant Solution

----
import { B1, B2 } from 'b';
----


