include::../description.adoc[]

== Noncompliant Code Example

----
if (condition1) {
  do_something();
} else if (condition2) {
  do_something_else();
} 
----

== Compliant Solution

----
if (condition1) {
  do_something();
} else if (condition2) {
  do_something_else();
} else {
  throw new InvalidArgumentException('message');
}
----

include::../see.adoc[]
