<p>Because the &lt;code&gt;is&lt;/code&gt; operator performs a cast if the object is not null, using &lt;code&gt;is&lt;/code&gt; to check type and then casting the same argument to that type, necessarily performs two casts. The same result can be achieved more efficiently with a single cast using &lt;code&gt;as&lt;/code&gt;, followed by a null-check.</p>
<h2>Noncompliant Code Example</h2>
<pre>
if (x is Fruit)  // Noncompliant
{
  var f = (Fruit)x; // or x as Fruit
  // ...
}
</pre>
<h2>Compliant Solution</h2>
<pre>
var f = x as Fruit;
if (f != null)
{
  // code
}
</pre>
