Because the <code>is</code> operator performs a cast if the object is not null, using <code>is</code> to check type and then casting the same argument to that type, necessarily performs two casts. The same result can be achieved more efficiently with a single cast using <code>as</code>, followed by a null-check.


== Noncompliant Code Example

----
if (x is Fruit)  // Noncompliant
{
  var f = (Fruit)x; // or x as Fruit
  // ...
}
----


== Compliant Solution

----
var f = x as Fruit;
if (f != null)
{
  // code
}
----

