Return of boolean literal statements wrapped into <code>If-Then-Else</code> ones should be simplified.

== Noncompliant Code Example

----
Function MyFunction
  If condition Then
    MyFunction = True
  Else
    MyFunction = False
  End If
End Function
----

== Compliant Solution

----
Function MyFunction
  MyFunction = condition
End Function
----
