<code>DELETE FROM dbtab</code> without a <code>WHERE</code> condition deletes all the entries of the table. Check whether dataset to be deleted can be limited by a suitable WHERE condition.

== Noncompliant Code Example

----
DELETE FROM COUNTRIES.
----

== Compliant Solution

----
DELETE FROM COUNTRIES WHERE CODE = country_code.
----
