<p>Explicitly defining a cursor as read-only can improve performance by avoiding table locking. This allows other SQL requests to execute in parallel. Therefore when a cursor will only be used to read data, without modifying anything, the &lt;code&gt;FOR READ ONLY&lt;/code&gt; clause or its synonyn, &lt;code&gt;FOR FETCH ONLY&lt;/code&gt;, should be used.</p>
<p>Conversely when a cursor will modify data, that too should be specified using the &lt;code&gt;FOR UPDATE&lt;/code&gt; clause.</p>
<p>In short, itâ€™s better to always explicitly define the purpose of the cursor with help of the &lt;code&gt;FOR READ ONLY&lt;/code&gt;, &lt;code&gt;FOR FETCH ONLY&lt;/code&gt; or &lt;code&gt;FOR UPDATE&lt;/code&gt; clauses.</p>
<h2>Noncompliant Code Example</h2>
<pre>
EXEC SQL DECLARE CMAJ_0A CURSOR
  FOR SELECT C_BQ
    FROM       S1ORDCOU
    WHERE C_BQ = :TORD-C-BQ
END-EXEC
</pre>
<h2>Compliant Solution</h2>
<pre>
EXEC SQL DECLARE CMAJ_0A CURSOR
  FOR SELECT C_BQ
    FROM       S1ORDCOU
    WHERE C_BQ = :TORD-C-BQ
  FOR READ ONLY
END-EXEC
</pre>
