Arrays in JavaScript have several methods for filtering, mapping or folding that require a callback. Not having a return statement in such a callback function is most likely a mistake.
This rule applies for the following methods of an array:

* <code>Array.from</code>
* <code>Array.prototype.every</code>
* <code>Array.prototype.filter</code>
* <code>Array.prototype.find</code>
* <code>Array.prototype.findIndex</code>
* <code>Array.prototype.map</code>
* <code>Array.prototype.reduce</code>
* <code>Array.prototype.reduceRight</code>
* <code>Array.prototype.some</code>
* <code>Array.prototype.sort</code>


== Noncompliant Code Example

----
var merged = arr.reduce(function(a, b) {
  a.concat(b);
}); // Noncompliant: No return statement
----


== Compliant Solution

----
var merged = arr.reduce(function(a, b) {
  return a.concat(b);
});
----


