<p>A lambda consisting only of a method call with exactly the same parameters passed to the lambda itself is unneeded. Simply call the enclosed method directly, instead.</p>
<h2>Noncompliant Code Example</h2>
<pre>
def add(x, y):
    return x + y

add_lambda = lambda x, y: add(x, y)  # Noncompliant

print(add_lambda(1, 2))
</pre>
<h2>Compliant Solution</h2>
<pre>
def add(x, y):
    return x + y

print(add(1, 2))
</pre>
