There can't be any good reason to do a full table scan on large database tables due to the cost of such operation and the scalability issue that might raise. This rule raises an issue when a <code>SELECT</code> statement doesn't use at least one indexed column in its <code>WHERE</code> clause.

*Note* That this rule raises issues only when a database catalog is provided during the SonarQube analysis.


== Noncompliant Code Example

----
SELECT * FROM USERS WHERE NAME = :name  -- non compliant when NAME column is not indexed
----


== Exceptions

This rule ignores <code>SELECT</code> statements not having any <code>WHERE</code> clauses; they are covered by other rules: S1590, SQL.SelectWithNoWhereClauseCheck

