include::../description.adoc[]

== Noncompliant Code Example

With the default threshold value of 3:
----
if (((condition1 && condition2) || (condition3 && condition4)) && condition5) { ... } 
----

== Compliant Solution

----
if ( (myFirstCondition() || mySecondCondition()) && myLastCondition()) { ... }     
----

== Exceptions

No issue is reported inside <code>equals</code> methods, because it is common to compare all the fields of a class for equality inside this kind of method.
