Naming conventions allow teams to collaborate effectively. This rule checks that exception names match a given regular expression.


== Noncompliant Code Example

With the default regular expression <code>[a-zA-Z]([a-zA-Z0-9_]*[a-zA-Z0-9])?</code>:

----
DECLARE
  my-Exception_ EXCEPTION; -- Noncompliant
BEGIN
  NULL;
END;
/
----


== Compliant Solution

----
DECLARE
  myException EXCEPTION;
BEGIN
  NULL;
END;
/
----

