There's no point in including the default value of a column in an insert statement. It simply clutters the code to no additional benefit.

*Note* that this rule raises issues only when a database catalog is provided during the SonarQube analysis.


== Noncompliant Code Example

With the table <code>PRODUCT</code> having a column <code>INV_COUNT</code> with default <code>0</code>:
----
EXEC SQL
  INSERT INTO PRODUCT
  (
    NAME,
    INV_COUNT  -- Noncompliant
  )
  VALUES
  (
    :PROD-NAME,
    0  -- this is the default value
  )
END-EXEC
----


== Compliant Solution

----
EXEC SQL
  INSERT INTO PRODUCT
  (
    NAME
  )
  VALUES
  (
    :PROD-NAME
  )
END-EXEC
----


