include::../description.adoc[]

== Noncompliant Code Example

With a maximum number of 4 parameters:

----
public void doSomething(int param1, int param2, int param3, String param4, long param5) {
...
}
----

== Compliant Solution

----
public void doSomething(int param1, int param2, int param3, String param4) {
...
}
----

== Exceptions

Methods annotated with Spring's <code>@RequestMapping</code> (and related shortcut annotations, like <code>@GetRequest</code>) or <code>@JsonCreator</code> may have a lot of parameters, encapsulation being possible. Such methods are therefore ignored.
