Having robust documentation just makes life easier. That applies to XSD's as well. Each one should include a description of its intended use, and perhaps versioning.

This rule raises an issue when there's neither a comment before the <code><xs:schema></code> tag, nor <code><xsd:annotation><xsd:documentation>...</xsd:documentation></xsd:annotation></code> just inside the <code><xs:schema></code> element.


== Noncompliant Code Example

----
<xs:schema targetNamespace="http://www.codeSamples.com/fruit"
           xmlns:xs="http://www.w3.org/2001/XMLSchema"
           elementFormDefault="qualified" version="1">
  <xs:element name="fruit">
    ...
  </xs:element>
</xs:schema>
----


== Compliant Solution

----
<-- This schema is intended... -->
<xs:schema targetNamespace="http://www.codeSamples.com/fruit"
           xmlns:xs="http://www.w3.org/2001/XMLSchema"
           elementFormDefault="qualified" version="1">
  <xs:element name="fruit">
    ...
  </xs:element>
</xs:schema>
----
or

----
<xs:schema targetNamespace="http://www.codeSamples.com/fruit"
           xmlns:xs="http://www.w3.org/2001/XMLSchema"
           elementFormDefault="qualified" version="1">
  <xsd:annotation>
    <xsd:documentation>
      This schema is intended...
    </xsd:documentation:
  </xsd:annotation>
  <xs:element name="fruit">
    ...
  </xs:element>
</xs:schema>
----

