<p>The use of positional notation to describe the overlapping of one field on another can quickly become confusing, particularly in large, or complicated data structures. Instead, the &lt;code&gt;OVERLAY&lt;/code&gt; keyword should be used to make such overlaps more readily identifiable.</p>
<h2>Noncompliant Code Example</h2>
<pre>
     D Message         DS                  QUALIFIED
     D   Id                           7
      * Noncompliant
     D   Type                  1      3
</pre>
<h2>Compliant Solution</h2>
<pre>
     D Message         DS                  QUALIFIED
     D   Id                           7
     D   Type                         3    OVERLAY(Id)
</pre>
