<code>RAISE_APPLICATION_ERROR</code> may only be called with an error code from <code>-20,000</code> to <code>-20,999</code>, which is the range reserved for application errors. When called with another value, Oracle raises the exception: <code>ORA-21000: error number argument to raise_application_error of 0 is out of range.</code>


== Noncompliant Code Example

----
BEGIN
  RAISE_APPLICATION_ERROR(0, 'This is an application error'); -- Non-Compliant - raises ORA-21000
END;
/
----


== Compliant Solution

----
BEGIN
  RAISE_APPLICATION_ERROR(-20000, 'This is an application error'); -- Compliant
END;
/
----

