<p>Defining variables in the minimum block scope possible reduces the visibility of those variables and therefore reduces the possibility that these identifiers will be used accidentally. A corollary of this is that global objects (including singleton function objects) shall be used in more than one function.</p>
<h2>Noncompliant Code Example</h2>
<pre>
void f ( int32_t k )
{
  int32_t i = k * k; // Noncompliant, visibility could be reduced
  {
    std::cout &lt;&lt; i &lt;&lt; std::endl;
  }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
void f ( int32_t k )
{
  {
    int32_t i = k * k; // Compliant
    std::cout &lt;&lt; i &lt;&lt; std::endl;
  }
}
</pre>
<h2>See</h2>
<ul>
<li>MISRA C++:2008, 3-4-1</li>
</ul>
