include::../description.adoc[]

== Noncompliant Code Example

----
func add(x, y int) int {                   
	return x + y // Noncompliant
	z := x + y // dead code
}
----

== Compliant Solution

----
func add(x, y int) int {
	return x + y // Compliant
}
----

include::../see.adoc[]
