Assistive technologies, such as screen readers, will not be able to render <code><object></code> elements, in such cases it is the content of the <code><object></code> which is provided to the user. This alternative content needs to be accessible or the screen readers won't be able to use it. For example, if an <code><img></code> is used it must contain an \<code>alt</code> attribute (see corresponding rule Web:ImgWithoutAltCheck).

This rule raises an issue when an <code><object></code> tag does not have any alternative content.


== Noncompliant Code Example

----
<object></object> <!-- Noncompliant -->

<object>
  <object></object> <!-- Noncompliant -->
</object>
----


== Compliant Solution

----
<object>This application shows the simulation of two particles colliding</object>

<object>
  <img src="flower.png" alt="Flower growing in a pot" />
</object>

<object>
  <object>
    This application shows the simulation of two particles colliding
  </object>
</object>
----


== See

* https://www.w3.org/TR/WCAG20-TECHS/H53.html[WCAG2, H53] - Using the body of the object element
* https://www.w3.org/WAI/WCAG21/quickref/?versions=2.0#qr-text-equiv-all[WCAG2, 1.1.1] - Non-text Content
* https://www.w3.org/WAI/WCAG21/quickref/?versions=2.0#qr-media-equiv-audio-desc[WCAG2, 1.2.3] - Audio Description or Media Alternative (Prerecorded)
* https://www.w3.org/WAI/WCAG21/quickref/?versions=2.0#qr-media-equiv-text-doc[WCAG2, 1.2.8] - Media Alternative (Prerecorded)

