<p>Shared naming conventions allow teams to collaborate efficiently. This rule raises an issue when a test class name does not match the provided regular expression.</p>
<h2>Noncompliant Code Example</h2>
<p>With the default value: &lt;code&gt;^Test|IT)[a-zA-Z0-9]+|[A-Z][a-zA-Z0-9]*(Test|IT|TestCase|ITCase$&lt;/code&gt;</p>
<pre>
[TestClass]
public class Foo // Noncompliant
{
</pre>
<h2>Compliant Solution</h2>
<pre>
[TestClass]
public class FooTest
{
</pre>
