Exceptions types should provide the following constructors:

* <code>public MyException()</code>
* <code>public MyException(string)</code>
* <code>public MyException(string, Exception)</code>
* <code>protected</code> or <code>private MyException(SerializationInfo, StreamingContext)</code>

That fourth constructor should be <code>protected</code> in unsealed classes, and <code>private</code> in sealed classes.

Not having this full set of constructors can make it difficult to handle exceptions.


== Noncompliant Code Example

----
using System;

namespace MyLibrary
{
  public class MyException // Noncompliant: several constructors are missing
  {
    public MyException()
    {
    }
  }
}
----


== Compliant Solution

----
using System;
using System.Runtime.Serialization;

namespace MyLibrary
{
  public class MyException : Exception
  {
      public MyException()
      {
      }

      public MyException(string message)
          :base(message)
      {
      }

      public MyException(string message, Exception innerException)
          : base(message, innerException)
      {
      }

      protected MyException(SerializationInfo info, StreamingContext context)
          : base(info, context)
      {
      }
  }
}
----


