include::../description.adoc[]

== Noncompliant Code Example

----
function compute($a, $a, $c) { // Noncompliant
}
----

== Compliant Solution

----
function compute($a, $b, $c) { // Compliant
}
----
