<p>A general &lt;code&gt;catch&lt;/code&gt; block seems like an efficient way to handle multiple possible exceptions. Unfortunately, it traps all exception types, casting too broad a net, and perhaps mishandling extraordinary cases. Instead, specific exception sub-types should be caught.</p>
<h2>Noncompliant Code Example</h2>
<pre>
try:
  result = do_the_thing(a)
except:  #Noncompliant; no specific exception caught
  result = None

try:
  result = do_the_thing(a)
except Exception:  #Noncompliant; too generic to be the only exception type caught
  result = None
</pre>
<h2>Compliant Solution</h2>
<pre>
try:
  result = do_the_thing(a)
except ZeroDivisionError:
  result = None
except TypeError:
  result = -1
</pre>
