include::../description.adoc[]

== Noncompliant Code Example

----
var CryptoJS = require("crypto-js");

var hash = CryptoJS.MD5("Message");
...
var hash = CryptoJS.SHA1("Message");
----

== Compliant Solution

----
var hash = CryptoJS.SHA256("Message");
----

include::../see.adoc[]
