The <code>LONG</code> and <code>LONG RAW</code> datatypes are deprecated and Oracle recommends to migrate them to the LOB datatypes <code>CLOB</code>, <code>NCLOB</code> or <code>BLOB</code>.


== Noncompliant Code Example

----
CREATE TABLE images(
  data LONG RAW
);
----


== Compliant Solution

----
CREATE TABLE images(
  data BLOB
);
----


