Shared naming conventions allow teams to collaborate efficiently. This rule raises an issue when a test method name does not match the provided regular expression.


== Noncompliant Code Example

With the default value: <code>^test[A-Z][a-zA-Z0-9]*$</code>

----
@Test
public void foo() {  // Noncompliant
  //...
}
----


== Compliant Solution

----
@Test
public void testFoo() {
  // ...
}
----

