String data types, such as <code>VARCHAR2</code> or <code>NVARCHAR2</code> require a size constraint. Omitting the size results in the exception <code>PLS-00215: String length constraints must be in range (1 .. 32767)</code> being raised.


== Noncompliant Code Example

----
DECLARE
  foo VARCHAR2; -- Noncompliant - raises PLS-00215
BEGIN
  NULL;
END;
/
----


== Compliant Solution

----
DECLARE
  foo VARCHAR2(42); -- Compliant
BEGIN
  NULL;
END;
/
----

