Using <code>EXIT</code> and <code>CHECK</code> in <code>SELECT</code> statements to stop the execution of <code>SELECT</code> loop is an expensive and ineffective way to filter data. Filtering should be part of the <code>SELECT</code> loop themselves. Most of the time conditions located in a <code>CHECK</code> statement should be moved to the <code>WHERE</code> clause, and the <code>EXIT</code> statement should typically be replaced by an <code>UP TO 1 ROW</code> clause.


== Noncompliant Code Example

----
SELECT * FROM SBOOK INTO SBOOK_WA.
  CHECK: SBOOK_WAS-CARRID = 'LH' AND SBOOK_WAS-CONNID = '0400'. "Noncompliant
ENDSELECT.
----


== Compliant Solution

----
SELECT * FROM SBOOK INTO SBOOK_WA WHERE CARRID = 'LH' AND CONNID = '0400'.
ENDSELECT.
----

