<p>A &lt;code&gt;catch&lt;/code&gt; clause that only rethrows the caught exception has the same effect as omitting the &lt;code&gt;catch&lt;/code&gt; altogether and letting it bubble up automatically, but with more code and the additional detriment of leaving maintainers scratching their heads.</p>
<p>Such clauses should either be eliminated or populated with the appropriate logic.</p>
<h2>Noncompliant Code Example</h2>
<pre>
do {
    try loadImage(imagePath)
} catch { // Noncompliant
    throw error
}
</pre>
<h2>Compliant Solution</h2>
<pre>
do {
    try loadImage(imagePath)
} catch {
    handleImageError(error)
    throw error
}
</pre>
<p>or</p>
<pre>
try loadImage(imagePath)
</pre>
