<p>A &lt;code&gt;CATCH&lt;/code&gt; clause that only rethrows the caught exception has the same effect as omitting the &lt;code&gt;CATCH&lt;/code&gt; altogether and letting it bubble up automatically, but with more code and the additional detriment of leaving maintainers scratching their heads.</p>
<p>Such clauses should either be eliminated or populated with the appropriate logic.</p>
<h2>Noncompliant Code Example</h2>
<pre>
BEGIN TRY
    SELECT 1/0;
END TRY
BEGIN CATCH -- Noncompliant
    THROW;
END CATCH;
</pre>
<h2>Compliant Solution</h2>
<pre>
BEGIN TRY
    SELECT 1/0;
END TRY
BEGIN CATCH
    EXECUTE usp_GetErrorInfo;
    THROW;
END CATCH;
</pre>
<p>or</p>
<pre>
SELECT 1/0;
</pre>
