Shared naming conventions allow teams to collaborate effectively. This rule raises an issue when the a pom's <code>groupId</code> does not match the provided regular expression.


== Noncompliant Code Example

With the default regular expression: <code>(com|org)(\.[a-z][a-z-0-9]*)+</code>

----
<project ...>
  <groupId>myCo</groupId>  <!-- Noncompliant -->

  <!-- ... -->
</project>
----


== Compliant Solution

----
<project ...>
  <groupId>com.myco</groupId>

  <!-- ... -->
</project>
----


