Using a <code>NUMBER</code> to store an integer is less performant than using a <code>PLS_INTEGER</code>. <code>PLS_INTEGER</code>s require less storage than <code>NUMBER</code>s, and benefit from the use of hardware math, as opposed to the library math required for <code>NUMBER</code>s. Even more performant is the <code>SIMPLE_INTEGER</code> subtype of <code>PLS_INTEGER</code>. However, changing to either of these types is only appropriate under certain circumstances.

<code>PLS_INTEGER</code> is only a candidate for <code>NUMBER</code> with a scale of up to 9.
<code>SIMPLE_INTEGER</code> has the same size limitation, in addition to it's <code>NOT NULL</code> constraint and lack of overflow checking.

This rule raises an issue when a <code>NUMBER</code> is declared with a scale of 9 or less.


== Noncompliant Code Example

----
DECLARE
    son NUMBER(1);      -- Noncompliant
    rumbo NUMBER(9);  -- Noncompliant
    conga Number(10);   -- Ignored; falls outside the PLS_INTEGER range
    compalsa PLS_INTEGER;
----


== Compliant Solution

----
DECLARE
    son SIMPLE_INTEGER;
    rumbo PLS_INTEGER;
    conga Number(10);   -- Ignored; falls outside the PLS_INTEGER range
    compalsa PLS_INTEGER;
----


