The use of virtual base classes can introduce a number of undefined and potentially confusing behaviours. The use of virtual bases is not recommended.


== Noncompliant Code Example

----
class B {};
class D: public virtual B {}; // Noncompliant, B is a virtual base
----


== See

* MISRA C++:2008, 10-1-1 - Classes should not be derived from virtual bases.

