Even if all browsers are fault-tolerant, HTML tags should be closed to prevent any unexpected behavior.


== Noncompliant Code Example

----
<html>
  <head>
    <title>Test Page    <!-- Noncompliant; title not closed -->
  <!-- Noncompliant; head not closed -->
  <body>
    <em>Emphasized Text  <!-- Noncompliant; em not closed -->
  <!-- Noncompliant; body not closed -->
</html>  
----


== Compliant Solution

----
<html>
  <head>
    <title>Test Page</title>
  </head>
  <body>
    <em>Emphasized Text</em>
  </body>
</html>  
----

