Regardless of the logging framework in use (logback, log4j, commons-logging, java.util.logging, ...), loggers should be:

* <code>private</code>: never be accessible outside of its parent class. If another class needs to log something, it should instantiate its own logger.
* <code>static</code>: not be dependent on an instance of a class (an object). When logging something, contextual information can of course be provided in the messages but the logger should be created at class level to prevent creating a logger along with each object.
* <code>final</code>: be created once and only once per class.

== Noncompliant Code Example

With a default regular expression of <code>LOG(?:GER)?</code>:
----
public Logger logger = LoggerFactory.getLogger(Foo.class);  // Noncompliant
----

== Compliant Solution

----
private static final Logger LOGGER = LoggerFactory.getLogger(Foo.class);
----

== Exceptions

Variables of type <code>org.apache.maven.plugin.logging.Log</code> are ignored.
