<p>There is no good reason to keep an empty and therefore valueless section. Such sections should be removed.</p>
<h2>Noncompliant Code Example</h2>
<pre>
FIRST SECTION.
  MOVE A TO B.

SECOND SECTION.  *&gt; Noncompliant; empty

THIRD SECTION.

someParagraph.
  DISPLAY B.
</pre>
<h2>Compliant Solution</h2>
<pre>
FIRST SECTION.
  MOVE A TO B.

THIRD SECTION.

someParagraph.
  DISPLAY B.
</pre>
