The use of operators pairs ( <code>=\+</code> or <code>=\-</code>) where the reversed, single operator was meant (<code>+=</code> or <code>-=</code>) will compile and run, but not produce the expected results.

This rule raises an issue when <code>=+</code> or <code>=-</code> is used without any spacing between the two operators and when there is at least one whitespace character after.

== Noncompliant Code Example

----
Dim target As Integer = -5
Dim num As Integer = 3

target =- num ' Noncompliant; target = -3. Is that really what's meant?
target =+ num ' Noncompliant; target = 3
----

== Compliant Solution

----
Dim target As Integer = -5
Dim num As Integer = 3

target = -num ' Compliant; intent to assign inverse value of num is clear
target += num
----
