include::../description.adoc[]

== Noncompliant Code Example

----
UPDATE books
SET title = 'unknown'
WHERE title = NULL -- Noncompliant 
----

== Compliant Solution

----
UPDATE books
SET title = 'unknown'
WHERE title IS NULL
----
