Both <code>FixedSizeList</code> from the Commons library, and the list returned from <code>Arrays.asList</code> offer <code>add</code> and <code>remove</code> methods (as required by the <code>List</code> interface they implement), but neither truly supports their use. Both list types have fixed lengths and will throw errors if an <code>add</code> or <code>remove</code> method, or any of their variations, is called.

This rule raises an issue when one of the following methods is invoked on a <code>FixedSizeList</code> instance:

* <code>add</code>
* <code>addAll</code>
* <code>clear</code>
* <code>remove</code>
* <code>removeAll</code>


== Noncompliant Code Example

----
List<String> strings = Arrays.asList("Hello");
strings.add("world");  // Noncompliant
----


== Compliant Solution

----
List<String> strings = Arrays.asList("Hello", "world");
----

