include::../description.adoc[]

== Noncompliant Code Example

----
void save() {
  try {
    saveDocument();
  } catch (const std::exception& ex) {
  }
}
----

== Compliant Solution

----
void save() {
  try {
    saveDocument();
  } catch (const std::exception& ex) {
    log << "Exception while saving the document: " << ex.what();
  }
}
----

include::../see.adoc[]
