<p>Fields should not be part of an API, and therefore should always be private. Indeed, they cannot be added to an interface for instance, and validation cannot be added later on without breaking backward compatibility. Instead, developers should encapsulate their fields into properties. Explicit property getters and setters can be introduced for validation purposes or to smooth the transition to a newer system.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public class Foo
{
  public int MagicNumber = 42;
}
</pre>
<h2>Compliant Solution</h2>
<pre>
public class Foo
{
  public int MagicNumber
  {
    get { return 42; }
  }
}
</pre>
<p>or</p>
<pre>
public class Foo
{
  private int MagicNumber = 42;
}
</pre>
<h2>Exceptions</h2>
<p>&lt;code&gt;struct&lt;/code&gt;s are ignored, as are &lt;code&gt;static&lt;/code&gt; and &lt;code&gt;const&lt;/code&gt; fields in classes.
Further, an issue is only raised when the real accessibility is &lt;code&gt;public&lt;/code&gt;, taking into account the class accessibility.</p>
