<p>Specifying the parameters to a procedure with a &lt;code&gt;PLIST&lt;/code&gt; makes that procedure unusable from free-format code. Instead, prototypes should be used - both when defining a procedure and when calling it. They have the additional benefit of allowing you to use keywords such as &lt;code&gt;Const&lt;/code&gt; to better-specify how parameters are passed to a procedure. Further, the use of a prototype instead of a &lt;code&gt;PLIST&lt;/code&gt; is cleaner and  more consistent with the code required for subprocedures.</p>
<h2>Noncompliant Code Example</h2>
<pre>
      * Noncompliant; PLIST specified
     C     *ENTRY        PLIST
     C                   PARM                    ZipCode           5 0
     C                   PARM                    City             20
...
      * Noncompliant; PLIST used in call
     C                   CALL      'OTHERPROG'
     C                   PARM                    ZipCode
     C                   PARM                    City
</pre>
<h2>Compliant Solution</h2>
<pre>
     D MYPROG          PR
     D  ZipCode                       5I   Const
     D  City                         20A   Const
     D MYPROG          PI
     D  ZipCode                       5I   Const
     D  City                         20A   Const
...
     D OTHERPROG       PR                  ExtPgm('OtherProgram')
     D  ZipCode                       5I   Const
     D  City                         20A   Const
      /free
        OTHERPROG(ZipCode:City);
      /end-free
</pre>
