<p>Using a scalar function or an arithmetic expression in a &lt;code&gt;WHERE&lt;/code&gt; condition can prevent the database from using indexes on the relevant column(s), and could therefore lead to performance issues.</p>
<h2>Noncompliant Code Example</h2>
<pre>
SELECT * FROM MY_TABE WHERE C2 = C3 + :HostVar1  -- Noncompliant

SELECT * FROM MY_TABLE WHERE YEAR(BIRTHDATE) &gt; 2000  -- Noncompliant
</pre>
