Using a scalar function or an arithmetic expression in a <code>WHERE</code> condition can prevent the database from using indexes on the relevant column(s), and could therefore lead to performance issues. 


== Noncompliant Code Example

----
SELECT * FROM MY_TABE WHERE C2 = C3 + :HostVar1  -- Noncompliant

SELECT * FROM MY_TABLE WHERE YEAR(BIRTHDATE) > 2000  -- Noncompliant
----

