This rule checks that the specified attributes are not present in HTML tags.


== Noncompliant Code Example

For a disallowed attribute list of: <code>a.name</code>:

----
<a href="blah.com" name="Blah link"> <!-- Noncompliant; name attribute is used -->
----


== Compliant Solution

----
<a href="blah.com">
----

