In Swift, the semicolon (<code>;</code>) is optional as a statement separator, but omitting semicolons can be confusing.

== Noncompliant Code Example

----
var x = 1
----

== Compliant Solution

----
var x = 1;
----
