<p>88-level variables, also known as "condition name" variables, represent possible values of the "conditional variables" theyâ€™re tied to. An unused "condition name" variable is dead code. Such variables should be removed to increase the maintainability of the program.</p>
<h2>Noncompliant Code Example</h2>
<pre>
01 COLOR PIC X.
  88 COL-YELLOW VALUE 'Y'.
  88 COL-GREEN VALUE 'G'. *&gt; Noncompliant; not used
  88 COL-RED VALUE 'R'.

* ...
IF COL-YELLOW
* ...
END-IF
IF COL-RED
* ...
END-IF
</pre>
<h2>Compliant Solution</h2>
<pre>
01 COLOR PIC X.
  88 COL-YELLOW VALUE 'Y'.
  88 COL-RED VALUE 'R'.

* ...
IF COL-YELLOW
* ...
END-IF
IF COL-RED
* ...
END-IF
</pre>
