<p>By default, the parameter mode is &lt;code&gt;IN&lt;/code&gt;. However, specifying it explicitly makes the code easier to read.</p>
<h2>Noncompliant Code Example</h2>
<pre>
SET SERVEROUTPUT ON

DECLARE
  PROCEDURE printName(name VARCHAR2) AS -- Noncompliant; relies on default mode
  BEGIN
    DBMS_OUTPUT.PUT_LINE('name: ' || name);
  END;

BEGIN
  printName('Foo');
END;
/
</pre>
<h2>Compliant Solution</h2>
<pre>
SET SERVEROUTPUT ON

DECLARE
  PROCEDURE printName(name IN VARCHAR2) AS
  BEGIN
    DBMS_OUTPUT.PUT_LINE('name: ' || name);
  END;
BEGIN
  printName('Foo');
END;
/
</pre>
