include::../description.adoc[]

== Noncompliant Code Example

----
       IDENTIFICATION DIVISION.
       PROGRAM-ID. foo.

       PROCEDURE DIVISION.

      * Noncompliant, 4 operands are found, higher than the maximum allowed 3
           IF WS-FOO(1) = 1 OR
              WS-FOO(2) = 2 OR
              WS-FOO(3) = 3 OR
              WS-BAR = 4 OR
              WS-BAZ = 5 OR
              WS-QUX = 42
           END-IF.
       END PROGRAM foo.
----
