<p>&lt;code&gt;mktemp&lt;/code&gt;, &lt;code&gt;mkstemp&lt;/code&gt;, &lt;code&gt;mkstemps&lt;/code&gt; or &lt;code&gt;mkdtemp&lt;/code&gt; are useful functions to create temporary files and directories. They accept a &lt;code&gt;template&lt;/code&gt; parameter which defines the path where the file should be created and which should contain at least six trailing "X"s to ensure the uniqueness of the filename.</p>
<h2>Noncompliant Code Example</h2>
<pre>
mkstemp("/tmp/file_XXXX"); // Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
mkstemp("/tmp/file_XXXXXX"); // Compliant
</pre>
