The point of using an optional is to signal that the value may be <code>nil</code> and to provide graceful ways of dealing with it if it is <code>nil</code>. While implicitly unwrapped optionals still provide means of dealing with <code>nil</code> values, they also signal that the value won't be <code>nil</code>, and unwrap it automatically. In addition to sending a decidedly mixed signal, this could lead to runtime errors if the value ever is <code>nil</code>. 

It is safest, and clearest to use either an optional or a plain type and avoid the boggy middle ground of implicitly unwrapped optionals.


== Noncompliant Code Example

----
var greeting : String!  // Noncompliant

println(greeting)  // At this point the value is nil. Runtime error results
----


== Compliant Solution

----
var greeting : String?

if let howdy = greeting {
  println(howdy)
}
----


