The use of DOM attributes that are only supported by specific browsers can doom your website to obsolescence as the browser market shifts or the vendor that introduced the extension abandons it. Instead, it's always best to stick to standard attributes so that your site works for as broad a cross section of users as possible.

This rule raises an issue when <code>innerText</code> is used.


== Noncompliant Code Example

----
function setData(element) {
  element.innerText = "Hello World!"; // Noncompliant
}
----


== Compliant Solution

----
function setData(element) {
  element.textContent = "Hello World!";
}
----

