include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression: ^([A-Z0-9_]*|[a-z0-9_]*)$
----
DEFINE MyMacro.
  ...
END-OF-DEFINITION.
----

== Compliant Solution

----
DEFINE my_macro.
  ...
END-OF-DEFINITION.
----
