<p>In Swift, the semicolon (&lt;code&gt;;&lt;/code&gt;) is optional as a statement separator, but omitting semicolons can be confusing.</p>
<h2>Noncompliant Code Example</h2>
<pre>
var x = 1
</pre>
<h2>Compliant Solution</h2>
<pre>
var x = 1;
</pre>
