<p>Shared coding conventions allow teams to collaborate efficiently. This rule checks that all record field names match the provided regular rexpression.</p>
<h2>Noncompliant Code Example</h2>
<p>With the default regular expression &lt;code&gt;[a-zA-Z](_?[a-zA-Z0-9])*&lt;/code&gt;:</p>
<pre>
DECLARE
  TYPE my_type IS RECORD(
    foo__bar PLS_INTEGER   -- Non-Compliant
  );
BEGIN
  NULL;
END;
/
</pre>
<h2>Compliant Solution</h2>
<pre>
DECLARE
  TYPE my_type IS RECORD(
    foo_bar PLS_INTEGER    -- Compliant
  );
BEGIN
  NULL;
END;
/
</pre>
