include::../description.adoc[]

== Noncompliant Code Example

----
SET FOO TO FALSE

IF FOO IS TRUE   *> Noncompliant
  DISPLAY "..."           *> never executed
END-IF.

IF FOO IS NOT TRUE OR BAR IS TRUE *> Noncompliant; BAR is never evaluated
  DISPLAY "..."
END-IF.
----

include::../see.adoc[]
