Developers should use the <code>FROM ... OUTER JOIN</code> syntax rather than the Oracle join operator (<code>+</code>). The reason is that outer join queries that use <code>+</code> are subject to several restrictions which do not apply to the <code>FROM ... OUTER JOIN</code> syntax. For instance, a <code>WHERE</code> condition containing the <code>+</code> operator cannot be combined with another condition using the <code>OR</code> logical operator.


== Noncompliant Code Example

----
BEGIN
  -- Noncompliant
  SELECT *
    INTO employeesArray
    FROM employee, department
    WHERE employee.DepartmentID = department.ID(+);
END;
/
----


== Compliant Solution

----
BEGIN
  -- Compliant
  SELECT *
    INTO employeesArray
    FROM employee LEFT OUTER JOIN department
    ON employee.DepartmentID = department.ID;
END;
/
----

