include::../description.adoc[]

== Noncompliant Code Example

----
FUNCTION my_function.
  ...
ENDFUNCTION. 
----

== Compliant Solution

----
* here is  
* my comment
FUNCTION my_function.
  ...
ENDFUNCTION. 
----
