include::../description.adoc[]

== Noncompliant Code Example

----
foo#: proc options(main); /* Non-Compliant */
end;
----

== Compliant Solution

----
foo: proc options(main);  /* Compliant */
end;
----
