<h2>Noncompliant Code Example</h2>
<pre>
var i, j int = 1, 2
</pre>
<h2>Compliant Solution</h2>
<pre>
var i int = 1
var j int = 2
</pre>
<h2>See</h2>
<ul>
<li><a href="https://www.securecoding.cert.org/confluence/x/7wHEAw">CERT, DCL52-J.</a> - Do not declare more than one variable per declaration</li>
<li><a href="https://www.securecoding.cert.org/confluence/x/VgU">CERT, DCL04-C.</a> - Do not declare more than one variable per declaration</li>
</ul>
