<p>Declaring multiple variable on one line is difficult to read.</p>
<h2>Noncompliant Code Example</h2>
<pre>
DECLARE @aaa AS INTEGER = 5, @bbb AS INTEGER = 42, @ccc AS CHAR(3) = 'foo'  -- Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
DECLARE @aaa AS INTEGER = 5
DECLARE @bbb AS INTEGER = 42
DECLARE @ccc AS CHAR(3) = 'foo'
</pre>
<h2>See</h2>
<ul>
<li><a href="https://www.securecoding.cert.org/confluence/x/7wHEAw">CERT, DCL52-J.</a> - Do not declare more than one variable per declaration</li>
<li><a href="https://www.securecoding.cert.org/confluence/x/VgU">CERT, DCL04-C.</a> - Do not declare more than one variable per declaration</li>
</ul>
