As soon as the <code>yield</code> keyword is used the enclosing method or function becomes a generator. Thus <code>yield</code> should never be used in a function or method which is not intended to be a generator.

This rule raises an issue when <code>yield</code> or <code>yield from</code> are used in a function or method which is not a generator because:

* the function/method's return type annotation is not [<code>typing.Generator[...]</code>|https://docs.python.org/3/library/typing.html#typing.Generator]
* it is a special method which can never be a generator (ex: <code>\_\_init\_\_</code>).


== Noncompliant Code Example

----
class A:
    def __init__(self, value):
        self.value = value
        yield value  # Noncompliant

def mylist2() -> List[str]:
    yield ['string']  # Noncompliant. Return should be used instead of yield

def generator_ok() -> Generator[int, float, str]:
    sent = yield 42
    return '42'
----


== Compliant Solution

----
class A:
    def __init__(self, value):
        self.value = value

def mylist2() -> List[str]:
    return ['string']

def generator_ok() -> Generator[int, float, str]:
    sent = yield 42
    return '42'
----


== See

* https://docs.python.org/3/library/typing.html[Python documentation - Support for type hints]
* https://docs.python.org/3/reference/datamodel.html#special-method-names[Python documentation - Special method names]

