In order to make your site usable by as many people as possible, subtitles should be provided for videos.

This rule raises an issue when a <code>video</code> does not include at least one <code><track/></code> tag with the <code>kind</code> attribute set to <code>captions</code>, or <code>descriptions</code> or at the very least <code>subtitles</code>.
Note that the <code>subtitles</code> kind is not meant for accessibility but for translation. The kind <code>captions</code> targets people with hearing impairment, whereas <code>descriptions</code> targets people with vision impairment.


== Noncompliant Code Example

----
<video id="video" controls preload="metadata">
   <source src="resources/myvideo.mp4" type="video/mp4">
   <source src="resources/myvideo.webm" type="video/webm">
</video>
----


== Compliant Solution

----
<video id="video" controls preload="metadata">
   <source src="resources/myvideo.mp4" type="video/mp4">
   <source src="resources/myvideo.webm" type="video/webm">
   <track label="English" kind="captions" srclang="en" src="resources/myvideo-en.vtt" default>
   <track label="Deutsch" kind="captions" srclang="de" src="resources/myvideo-de.vtt">
   <track label="EspaÃ±ol" kind="captions" srclang="es" src="resources/myvideo-es.vtt">
</video>
----


