<p>Enumerations have implementation-defined representation and so should not be used in arithmetic contexts.</p>
<h2>Noncompliant Code Example</h2>
<pre>
enum { COLOUR_0, COLOUR_1, COLOUR_2, COLOUR_COUNT } colour;
if ( COLOUR_0 == colour ) { ... }
if ( ( COLOUR_0 + COLOUR_1 ) == colour ) { ... } // Noncompliant, arithmetic used
if ( colour &lt; COLOUR_COUNT ) { ... }
</pre>
<h2>See</h2>
<ul>
<li>MISRA C++:2008, 4-5-2 - Expressions with type enum shall not be used as operands to builtin operators other than the subscript operator [ ], the assignment operator =, the equality operators == and !=, the unary &amp; operator, and the relational operators &lt;, ‚áê, &gt;, &gt;=</li>
</ul>
