Since C++03, a member function that is contained within a class definition is by definition <code>inline</code>, so an using an <code>inline</code> specifier on such functions is redundant.


== Noncompliant Code Example

----
class Foo {
  inline void method() { // Noncompliant
    // ...
  }
};
----


== Compliant Solution

----
class Foo {
  void method() {
    // ...
  }
};
----


