<p>You should avoid closing a &lt;code&gt;CURSOR&lt;/code&gt; inside a &lt;code&gt;PERFORM&lt;/code&gt; statement, because it could impact performance or lead to unexpected behavior if the cursor was not opened in the same loop.</p>
<h2>Noncompliant Code Example</h2>
<pre>
 PERFORM UNTIL (NOT DA-OK)
       OR (Y00CIA-CD-RET-PGM = ZERO)
    EXEC SQL CLOSE C2
    END-EXEC
  END-PERFORM.
</pre>
