Procedures, unlike functions, do not return values. The <code>RETURN</code> statement therefore, when used within a procedure, is used to prematurely end the procedure. However, having multiple exit points (i.e. more than the <code>END</code> of the procedure itself), increases the complexity of the procedure and makes it harder to understand and debug.

== Noncompliant Code Example

----
SET SERVEROUTPUT ON

DECLARE
  PROCEDURE prcoedureWithReturn AS
  BEGIN
    RETURN; -- Noncompliant

    DBMS_OUTPUT.PUT_LINE('prcoedureWithReturn called'); -- This is actually unreachable
  END;
BEGIN
  prcoedureWithReturn;
END;
/
----
