When <code>@Overrides</code> of <code>synchronized</code> methods are not themselves <code>synchronized</code>, the result can be improper synchronization as callers rely on the thread-safety promised by the parent class.


== Noncompliant Code Example

----
public class Parent {

  synchronized void foo() {
    //...
  }
}

public class Child extends Parent {

 @Override
  public foo () {  // Noncompliant
    // ...
    super.foo();
  }
}
----


== Compliant Solution

----
public class Parent {

  synchronized void foo() {
    //...
  }
}

public class Child extends Parent {

  @Override
  synchronized foo () {
    // ...
    super.foo();
  }
}
----


== See

* https://www.securecoding.cert.org/confluence/x/XgAZAg[CERT, TSM00-J] - Do not override thread-safe methods with methods that are not thread-safe

