<p>Python does not check returned types by default, however some methods are expected to return a specific type otherwise builtin functions will fail. Developers can also use type hints to specify which types they expect. Not following these type hints will most probably result in a runtime error.</p>
<p>This rule raises an issue when a function or method:</p>
<ul>
<li>does not return or yield a value of the expected type.</li>
<li>returns a value when no value should be returned.</li>
<li>returns no value when a return value is expected.</li>
</ul>
<h2>Noncompliant Code Example</h2>
<pre>
class MyInt:
    def __init__(self):
        return self  # Noncompliant. __init__ should return nothing. This will raise a TypeError.

    def __int__(self):
        return 3.0  # Noncompliant. __int__ should always return an integer

int(MyInt())  # This will fail with "TypeError: __int__ returned non-int (type float)"

def hello() -&gt; str:
    return 42  # Noncompliant. Function's type hint asks for a string return value

def should_return_a_string(condition) -&gt; str:
    if condition:
        return "a string"
    # Noncompliant. The function returns None if the condition is not met

def generator_noncompliant() -&gt; Generator[int, float, str]:
    sent = yield '42'  # Noncompliant
    return 42  # Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
class MyInt:
    def __int__(self):
        return 3

int(MyInt())

def hello() -&gt; str:
    return "Hello"

def should_return_a_string() -&gt; str:
    return "a string"

def generator_ok() -&gt; Generator[int, float, str]:
    sent = yield 42
    return '42'
</pre>
<h2>See</h2>
<ul>
<li><a href="https://docs.python.org/3/library/typing.html">Python documentation - Support for type hints</a></li>
<li><a href="https://docs.python.org/3/reference/datamodel.html#special-method-names">Python documentation - Special method names</a></li>
</ul>
