<p>Using boxed type suggests that &lt;code&gt;null&lt;/code&gt; is a possible value for the variable. Use of the primitive type should be preferred if this is not the case to avoid any confusion about possible values variable can contain.</p>
<h2>Noncompliant Code Example</h2>
<pre>
Integer x = 5;
</pre>
