This rule allows you to track the usage of the <code>SuppressMessage</code> attributes and <code>#pragma warning disable</code> mechanism.

== Noncompliant Code Example

----
[SuppressMessage("", "S100")]
...

#pragma warning disable S100
...
#pragma warning restore S100
----
