<p>Non-static variables initialized with &lt;code&gt;INZ()&lt;/code&gt; are only reliably initialized on the first run through the code in a  "terminate and stay resident" (TSR) program. Subsequent calls to the program within the same job do not re-initialize the variable with the value from &lt;code&gt;INZ()&lt;/code&gt; unless the last record indicator is set at the end of the program.</p>
<p>Without setting the last record indicator, the assumptions the code makes about the variableâ€™s initial value will be wrong every time but one, potentially leading to bad program behavior.</p>
<h2>Noncompliant Code Example</h2>
<p>{code}     D TESTINZ         PR                  extpgm('TESTINZ')
     D  Parm1                        15A
     D  Parm2                        15A   CONST</p>
<pre>
DMUTVAR           S             15A   INZ('ABC')  // Noncompliant; *inlr not set
</pre>
<pre>
D TESTINZ         PI
D  Parm1                        15A
D  Parm2                        15A   CONST
 /Free
  IF %PARMS &gt; 1 and Parm2 &lt;&gt; '';
    MUTVAR=Parm2;
  ENDIF;
  DSPLY(E) ('MUTVAR:' + MUTVAR);
  Parm1=MUTVAR;
  return;
  /End-free
</pre>
<pre>

</pre>
