<p>Nesting &lt;code&gt;try&lt;/code&gt;/&lt;code&gt;catch&lt;/code&gt; or &lt;code&gt;@try&lt;/code&gt;/&lt;code&gt;@catch&lt;/code&gt; blocks severely impacts the readability of source code because it makes it too difficult to understand which block will catch which exception.</p>
<p>This C++ example also applies to Objective-C.</p>
<h2>Noncompliant Code Example</h2>
<pre>
try {
  try {                                     // Noncompliant
    doSomething();
  } catch (RuntimeException e) {
    /* Ignore */
  }

  doSomethingElse();
} catch (Exception e) {
  /* ... */
}
</pre>
<h2>Compliant Solution</h2>
<pre>
try {
  dedicatedMethod();                        // Compliant
  doSomethingElse();
} catch (Exception e) {
  /* ... */
}

/* ... */

private void dedicatedMethod() {
  try {                                     // Compliant
    doSomething();
  } catch (RuntimeException e) {
    /* Ignore */
  }
}
</pre>
