Swing interfaces should be constructed and shown from the Swing event dispatch thread. Doing so from any other thread, such as from <code>main</code> risks deadlocks since you run the risk of multiple threads accessing things which are inherently not thread-safe.

Instead, use <code>SwingUtilities.invokeLater</code> or <code>SwingUtilities.invokeAndWait</code> to kick off a <code>new Runnable</code> that handles your GUI creation.


== Noncompliant Code Example

----
public static void main(String args[]) {
  makeGui(); // Noncompliant
}

public void makeGui() {
  JFrame frame = new JFrame();
  // ...
  frame.show();
}
----


== Compliant Solution

----
public static void main(String args[]) {
  SwingUtilities.invokeAndWait(new Runnable() {
    public void run() {
      makeGui();
    }
  }
}

public void makeGui() {
  JFrame frame = new JFrame();
  // ...
  frame.show();
}
----


