include::../description.adoc[]

== Noncompliant Code Example

----
void fun() {
  label: doSomething();
}
----

== Compliant Solution

----
void fun() {
  doSomething();
}
----

== See

* MISRA C:2012, 2.6 - A function should not contain unused label declarations
* https://www.securecoding.cert.org/confluence/x/NYA5[CERT, MSC12-C.] - Detect and remove code that has no effect or is never executed
