Dictionary unpacking requires an objects with methods <code>\_\_getitem\_\_</code> and <code>keys</code> or <code>\_\_getitem\_\_</code> and <code>\_\_getattr\_\_</code>. This is the case for any https://docs.python.org/3/glossary.html#term-mapping[mapping] object such as <code>dict</code>. Using an object which doesn't have these methods will raise a <code>TypeError</code>.


== Noncompliant Code Example

----
class A:
    pass

dict(**A())  # Noncompliant
{'a': 10, 'b': 20, **A()}  # Noncompliant 
----


== Compliant Solution

----
class A:
  def __getitem__(self, key):
      return 2

  def keys(self):
      return ['1','2','3']

dict(**A())  # => {'1': 2, '2': 2, '3': 2}
{'a': 10, 'b': 20, **A()}  # => {'a': 10, 'b': 20, '1': 2, '2': 2, '3': 2}
----

