<p>Sharing some naming conventions is a key point to make it possible for a team to efficiently collaborate. This rule allows to check that all interface names match a provided regular expression.</p>
<h2>Noncompliant Code Example</h2>
<p>With the default regular expression &lt;code&gt;YIF_[A-Za-z]+&lt;/code&gt;:</p>
<pre>
INTERFACE MyInterface. "Noncompliant
...
ENDINTERFACE.
</pre>
<h2>Compliant Solution</h2>
<pre>
INTERFACE YIF_myinterface.
...
ENDINTERFACE.
</pre>
