<p>Sharing some naming conventions is a key point to make it possible for a team to efficiently collaborate. This rule allows to check that all protocol names match a provided regular expression.</p>
<h2>Noncompliant Code Example</h2>
<p>With the default regular expression &lt;code&gt;^[A-Z][a-zA-Z0-9]*$&lt;/code&gt;:</p>
<pre>
public protocol myProtocol {...} // Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
public protocol MyProtocol {...}
</pre>
