include::../description.adoc[]

== Noncompliant Code Example

----
public class Program
{
    public void WriteMatrix(int[][] matrix) // Non-Compliant
    {
    }
}
----

== Compliant Solution

----
public class Matrix
{
    // ...
}

public class Program
{
    public void WriteMatrix(Matrix matrix) // Compliant
    {
    }
}
----
