<code>CASE</code> can contain a <code>WHEN OTHERS</code> clause for various reasons: to handle unexpected values, to show that all the cases were properly considered.
For readability purpose, to help a developer to quickly find the default behavior of a <code>CASE</code> statement, it is recommended to put the <code>WHEN OTHERS</code> clause at the end of the <code>CASE</code> statement. This rule raises an issue if the <code>WHEN OTHERS</code> clause is not the last one of the <code>CASE</code>'s cases.

== Noncompliant Code Example

----
CASE SY-INDEX.
  WHEN OTHERS.   // Noncompliant; WHEN OTHERS should be last statement
    WRITE 'Unexpected result'
  WHEN ONE.
    WRITE  'One'. 
  WHEN 2.
    WRITE   'Two'.
ENDCASE.
----

== Compliant Solution

----
CASE SY-INDEX. 
  WHEN ONE.
    WRITE  'One'. 
  WHEN 2.
    WRITE   'Two'.
  WHEN OTHERS. // Compliant
    WRITE 'Unexpected result'
ENDCASE.


----
