The PHP 4 method of declaring a variable, using the <code>var</code> keyword, was deprecated in early versions of PHP 5. Even though it's not considered deprecated in the most recent versions, it's nonetheless not best practice to use it. When <code>var</code> does appear, it is interpreted as a synonym for <code>public</code> and treated as such. Therefore <code>public</code> should be used instead.

From the PHP Manual:
{quote}
The PHP 4 method of declaring a variable with the var keyword is still supported for compatibility reasons (as a synonym for the public keyword). In PHP 5 before 5.1.3, its usage would generate an E_STRICT warning.
{quote}


== Noncompliant Code Example

----
<?php
class Foo
{
    var $bar = 1;
}
----


== Compliant Solution

----
<?php
class Foo
{
    public $bar = 1;
}
----

