include::../description.adoc[]

== Noncompliant Code Example

----
class Foo 
{
  public int myField;

  public void DoSomething() 
  {
    int myField = 0;  // Noncompliant
    ...
  }
}
----

include::../see.adoc[]
