<code>NullReferenceException</code> should be avoided, not caught. Any situation in which <code>NullReferenceException</code> is explicitly caught can easily be converted to a <code>null</code> test, and any behavior being carried out in the catch block can easily be moved to the "is null" branch of the conditional.

== Noncompliant Code Example

----
public int GetLengthPlusTwo(string str) 
{
    int length = 2;
    try 
    {
       length += str.Length;
    }
    catch (NullReferenceException e) 
    {
        log.info("argument was null");
    }
    return length;
}
----

== Compliant Solution

----
public int GetLengthPlusTwo(string str) 
{
    int length = 2;

    if (str != null) 
    {
        length += str.Length;
    }
    else 
    {
        log.info("argument was null");
    }
    return length;
}
----

include::../see.adoc[]
