To ensure future code portability, obsolete keywords should not be used. The following keywords were declared obsolete in the COBOL ANSI-85 standard and removed in the ISO/IEC 1989:2002 standard: 

* Paragraphs: <code>AUTHOR</code>, <code>INSTALLATION</code>, <code>DATE-WRITTEN</code>, <code>DATE-COMPILED</code>, <code>SECURITY</code>
* Clauses: <code>DATA RECORD(S)</code>, <code>LABEL RECORD(S)</code>, <code>MEMORY SIZE</code>, <code>MULTIPLE FILE (TAPE)</code>, <code>RERUN</code>, <code>VALUE OF</code>, <code>CODE SEGMENT-LIMIT</code>
* Statements: <code>ALTER</code>, <code>ENTER</code>, <code>STOP</code> _literal_, <code>GO TO</code> without an argument
* Phrases: <code>REVERSED</code> phrase of the <code>OPEN</code> statement
* Special registers: <code>DEBUG-ITEM</code>
* Sections: Debugging sections
* Declarative: <code>USE FOR DEBUGGING</code>

The following keywords were declared obsolete in the ISO/IEC 1989:2002 standard:

* Phrase: <code>DEBUGGING MODE</code>
* Clause: <code>PADDING CHARACTER</code>


== Noncompliant Code Example

----
IDENTIFICATION DIVISION.
PROGRAM-ID.  AcceptAndDisplay.
AUTHOR.  Michael Coughlan.  *> Noncompliant
----


== Compliant Solution

----
IDENTIFICATION DIVISION.
PROGRAM-ID.  AcceptAndDisplay.
----

