<p>When a function parameter has a nullable type, e.g., parameter &lt;code&gt;param&lt;/code&gt; in &lt;code&gt;f(?int param)&lt;/code&gt;, it must be explicitly provided in every function call. A nullable-type parameter has no default value.</p>
<h2>Noncompliant Code Example</h2>
<pre>
function f(?int param) {}
f();
</pre>
<h2>Compliant Solution</h2>
<pre>
function f(?int param) {}
f(0);
</pre>
