When a function parameter has a nullable type, e.g., parameter <code>param</code> in <code>f(?int param)</code>, it must be explicitly provided in every function call. A nullable-type parameter has no default value.


== Noncompliant Code Example

----
function f(?int param) {}
f();
----


== Compliant Solution

----
function f(?int param) {}
f(0);
----


