include::../description.adoc[]

== Noncompliant Code Example

----
void foo() {
  outer: //label is not used.
  for(int i = 0; i<10; i++) {
    break;
  }
}
----

== Compliant Solution

----
void foo() {
  for(int i = 0; i<10; i++) {
    break;
  }
}
----

include::../see.adoc[]
