Code is clearest when each statement has its own line. Nonetheless, it is a common pattern to combine on the same line an <code>if</code> and its resulting _then_ statement. However, when an <code>if</code> is placed on the same line as the closing <code>}</code> from a preceding <code>else</code> or <code>elseif</code>, it is either an error - <code>else</code> is missing - or the invitation to a future error as maintainers fail to understand that the two statements are unconnected.

== Noncompliant Code Example

----
if ($condition1) {
  // ...
} if ($condition2) {  // Noncompliant
  //...
}
----

== Compliant Solution

----
if ($condition1) {
  // ...
} elseif ($condition2) { 
  //...
}
----
Or
----
if ($condition1) {
  // ...
} 

if ($condition2) {
  //...
}
----
