This includes file and I/O functions <code>fgetpos</code>, <code>fopen</code>, <code>ftell</code>, <code>gets</code>, <code>perror</code>, <code>remove</code>, <code>rename</code>, etc.

Streams and file I/O have a large number of unspecified, undefined and implementation-defined behaviors associated with them.


== Noncompliant Code Example

----
#include <cstdio>  /* Noncompliant */

void fn()
{
  char_t array[10];
  gets(array);           /* Can lead to buffer over-run */
}
----


== See

* MISRA C++:2008, 27-0-1 - The stream input/output library _<cstdio>_ shall not be used.

