Shadowing fields or <code>enum</code> <code>case</code>s with a local variable is a bad practice that reduces code readability: It makes it confusing to know whether the field or the variable is being used.

== Noncompliant Code Example

----
public class Foo {
  public var myField:Int = 0

  public func doSomething() {
    var myField = 0
    ...
  }
}
----

include::../see.adoc[]
