Using keywords as variable names may yield incomprehensible code, and should be avoided.

== Noncompliant Code Example

----
DATA:  wa_struct TYPE struct,
       name       TYPE string,
       dob          TYPE string,
       aliases     TYPE string,    " ALIASES is a keyword
----

== Compliant Solution

----
DATA:  wa_struct TYPE struct,
       name         TYPE string,
       dob            TYPE string,
       alt_names  TYPE string,
----
