<p>An incorrectly defined platform invoke method (marked with &lt;code&gt;System.Runtime.InteropServices.DllImportAttribute&lt;/code&gt;) can lead to runtime exceptions because of issues such as a misnamed function, faulty mapping of parameter and return value data types, and incorrect field specifications. If available, it is a lot less error prone, not to mention more secure and quite simpler, to call the equivalent managed method.</p>
<p>This rule raises an issue when a platform invoke method is called and a method with the equivalent functionality exists in the .Net framework library.</p>
<h2>Noncompliant Code Example</h2>
<pre>
using System;
using System.Runtime.InteropServices;
using System.Text;

namespace MyLibrary
{
   public class Foo
   {
      public Foo() {}

      [DllImport("kernel32.dll", CharSet = CharSet.Unicode, SetLastError = true)]
      internal static extern int ExpandEnvironmentStrings(
         string lpSrc, StringBuilder lpDst, int nSize);

      public void Bar()
      {
         string environmentVariable = "%TEMP%";
         StringBuilder expandedVariable = new StringBuilder(100);

         ExpandEnvironmentStrings( // Noncompliant
            environmentVariable,
            expandedVariable,
            expandedVariable.Capacity);
      }
   }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
using System;
using System.Text;

namespace MyLibrary
{
   public class Foo
   {
      public Foo() {}

      public void Bar()
      {
         string environmentVariable = "%TEMP%";
         StringBuilder expandedVariable = new StringBuilder(100);

         Environment.ExpandEnvironmentVariables(environmentVariable);
      }
   }
}
</pre>
