Because Visual Basic 6 does not automatically unload <code>Form</code> s after a program terminates, it is possible for them to remain in memory. Even after <code>Unload</code> ing, they may stay in memory if references remain. Therefore <code>Form</code> s should be explicitly <code>Unload</code> ed, and their references set to <code>Nothing</code>.


== Noncompliant Code Example

----
dim f1 as new form
f1.show
...
f1.hide
----


== Compliant Solution

----
dim f1 as new form
f1.show
...
f1.hide
Unload f1
Set f1 = Nothing
----

