include::../description.adoc[]

== Noncompliant Code Example

----
class MyClass {
  private:
    int foo = 42;  // Noncompliant, foo is unused

  public:
    int compute(int a) {
      return a * 42;
    }
};
----

== Compliant Solution

----
class MyClass {
  public:
    int compute(int a) {
      return a * 42;
    }
};
----
