<p>It is needlessly complex to invert the result of a boolean comparison. The opposite comparison should be made instead.</p>
<h2>Noncompliant Code Example</h2>
<pre>
IF NOT x &lt;&gt; y THEN   -- Noncompliant
  -- ...
END IF;
</pre>
<h2>Compliant Solution</h2>
<pre>
IF x = y THEN
  -- ...
END IF;
</pre>
