Initializing a variable to zero or <code>nil</code> in an <code>init</code> method is completely redundant; the compiler takes care of that for you. Therefore initializing class instance variables to 0 or <code>nil</code> is simply wasted keystrokes.

This rule applies to methods that return an <code>id</code> and have names that begin with "init".


== Noncompliant Code Example

----
- (id)initWithBlah:(NSString *)blah {
    self = [super init];
    if (self) {
      _count = 0;  // Noncompliant;
      _blah = [blah copy];
    }
    return self;
}
----


== Compliant Solution

----
- (id)initWithBlah:(NSString *)blah {
    self = [super init];
    if (self) {
      _blah = [blah copy];
    }
    return self;
}
----

