https://docs.python.org/3/library/exceptions.html#SystemExit[<code>SystemExit</code>]Â is raised when https://docs.python.org/3/library/sys.html#sys.exit[<code>sys.exit()</code>] is called. https://docs.python.org/3/library/exceptions.html#KeyboardInterrupt[<code>KeyboardInterrupt</code>] is raised when the user asks the program to stop by pressing interrupt keys. Both exceptions are expected to propagate up until the application stops.

In order to avoid catching <code>SystemExit</code> and <code>KeyboardInterrupt</code> by mistake https://www.python.org/dev/peps/pep-0352/#exception-hierarchy[PEP-352] created the root class <code>BaseException</code> from which <code>SystemExit</code>, <code>KeyboardInterrupt</code> and <code>Exception</code> derive. Thus developers can use <code>except Exception:</code> without preventing the software from stopping.

The <code>GeneratorExit</code> class also derives from <code>BaseException</code> as it is not really an error and is not supposed to be caught by user code.

As said in https://docs.python.org/3/library/exceptions.html#BaseException[Python's documentation], user-defined exceptions are not supposed to inherit directly from <code>BaseException</code>. They should instead inherit from <code>Exception</code> or one of its subclasses.

This rule raises an issue when a class derives from one of the following exception classes: <code>BaseException</code>, <code>KeyboardInterrupt</code>, <code>SystemExit</code> or <code>GeneratorExit</code>.


== Noncompliant Code Example

----
class MyException(BaseException):  # Noncompliant
    pass

class MyException(GeneratorExit):  # Noncompliant
    pass

class MyException(KeyboardInterrupt):  # Noncompliant
    pass

class MyException(SystemExit):  # Noncompliant
    pass
----


== Compliant Solution

----
class MyException(Exception):
    pass
----


== See

* PEP 352 -- https://www.python.org/dev/peps/pep-0352/#exception-hierarchy-changes[Required Superclass for Exceptions]
* Python Documentation - https://docs.python.org/3/library/exceptions.html#BaseException[BaseException class]

