include::../description.adoc[]

== Noncompliant Code Example

----
whileLoopLabel: while x > 0 {    // Noncompliant
    x -= 1
}
----

== Compliant Solution

----
while x > 0 {
    x -= 1
}
----

include::../see.adoc[]
