include::../description.adoc[]

== Noncompliant Code Example

----
function get_title($gender, $is_married) {
  return $gender == "MALE" ? "Mr. " : ($is_married ? "Mrs. " : "Miss ");  // Noncompliant
}
----

== Compliant Solution

----
function get_title($gender, $is_married) {
  if ($gender == "MALE") {
    return "Mr. ";
  }
  return $is_married ? "Mrs. " : "Miss ";
}
----
