The long suffix should always be written in uppercase, i.e. 'L', as the lowercase 'l' can easily be confused with the digit one '1'.


== Noncompliant Code Example

----
return 10l  // Noncompliant; easily confused with one zero one
----


== Compliant Solution

----
return 10L
----


