<p>Includes &lt;code&gt;time&lt;/code&gt;, &lt;code&gt;strftime&lt;/code&gt;. This library is associated with clock times. Various aspects are implementation dependent or unspecified, such as the formats of times. If any of the facilities of &lt;code&gt;time.h&lt;/code&gt; are used, then the exact implementation for the compiler being used must be determined, and a deviation raised.</p>
<h2>Noncompliant Code Example</h2>
<pre>
#include &lt;time.h&gt;  /* Noncompliant */
</pre>
<h2>See</h2>
<ul>
<li>MISRA C:2004, 20.12 - The time handling functions of library &lt;time.h&gt; shall not be used.</li>
<li>MISRA C:2012, 21.10 - The Standard Library time and date functions shall not be used</li>
</ul>
