While both <code>new</code> and <code>Create</code> will yield the same <code>Tuple</code>, the <code>Create</code> method automatically infers the argument types, yielding sleeker, more readable code.


== Noncompliant Code Example

----
var fastAnimal = new Tuple<string, int>("Cheetah", 93);  // Noncompliant
----


== Compliant Solution

----
var fastAnimal = Tuple.Create("Cheetah", 93);
----

