Shared coding conventions allow teams to collaborate efficiently. This rule checks that all enumeration values match a provided regular expression.

== Noncompliant Code Example

With default provided regular expression:
----
enum SomeEnumeration {
    some  // Non-Compliant
};
----

== Compliant Solution

----
enum SomeEnumeration {
    SOME
};
----
