An <code>async</code> function always wraps the return value in a <code>Promise</code>. Using <code>return await</code> is therefore redundant.


== Noncompliant Code Example

----
async function foo() {
  // ...
}

async function bar() {
  // ...
  return await foo(); // Noncompliant
}
----


== Compliant Solution

----
async function foo() {
  // ...
}

async function bar() {
  // ...
  return foo();
}
----

