<p>A Single Abstract Method (SAM) interface is a Java interface containing only one method. The Java API is full of SAM interfaces, such as &lt;code&gt;java.lang.Runnable&lt;/code&gt;, &lt;code&gt;java.awt.event.ActionListener&lt;/code&gt;, &lt;code&gt;java.util.Comparator&lt;/code&gt; and &lt;code&gt;java.util.concurrent.Callable&lt;/code&gt;. SAM interfaces have a special place in Java 8 because they can be implemented using Lambda expressions or Method references.</p>
<p>Using &lt;code&gt;@FunctionalInterface&lt;/code&gt; forces a compile break when an additional, non-overriding abstract method is added to a SAM, which would break the use of Lambda implementations.</p>
<p><strong>Note</strong> that this rule is automatically disabled when the projectâ€™s &lt;code&gt;sonar.java.source&lt;/code&gt; is lower than &lt;code&gt;8&lt;/code&gt;.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public interface Changeable&lt;T&gt; {
  public void change(T o);
}
</pre>
<h2>Compliant Solution</h2>
<pre>
@FunctionalInterface
public interface Changeable&lt;T&gt; {
  public void change(T o);
}
</pre>
