include::../description.adoc[]

== Noncompliant Code Example

----
$name = $_GET["name"];
echo "Welcome $name"; // Noncompliant
----

== Compliant Solution

----
$name = $_GET["name"];
$safename = htmlspecialchars($name);
echo "Welcome $safename";
----

include::../see.adoc[]
