According to the documentation,
bq. A program may produce unpredictable results if it attempts to distinguish two references to equal values of a value-based class, whether directly via reference equality or indirectly via an appeal to synchronization...

This is because value-based classes are intended to be wrappers for value types, which will be primitive-like collections of data (similar to <code>struct</code>s in other languages) that will come in future versions of Java.

{quote} Instances of a value-based class ... 
* do not have accessible constructors, but are instead instantiated through factory methods which make no committment as to the identity of returned instances;
{quote}

Which means that you can't be sure you're the only one trying to lock on any given instance of a value-based class, opening your code up to contention and deadlock issues.

Under Java 8 breaking this rule may not actually break your code, but there are no guarantees of the behavior beyond that.

This rule raises an issue when a known value-based class is used for synchronization. That includes all the classes in the <code>java.time</code> package except <code>Clock</code>; the date classes for alternate calendars, <code>HijrahDate</code>, <code>JapaneseDate</code>, <code>MinguoDate</code>, <code>ThaiBuddhistDate</code>; and the optional classes: <code>Optional</code>, <code>OptionalDouble</code>, <code>OptionalLong</code>, <code>OptionalInt</code>.

*Note* that this rule is automatically disabled when the project's <code>sonar.java.source</code> is lower than <code>8</code>.


== Noncompliant Code Example

----
Optional<Foo> fOpt = doSomething();
synchronized (fOpt) {  // Noncompliant
  // ...
}
----


== See

* https://docs.oracle.com/javase/8/docs/api/java/lang/doc-files/ValueBased.html[Value-based classes]

