include::../description.adoc[]

== Noncompliant Code Example

----
IF @x > 0 SET @x = 0; IF @y > 0 SET @y = 0; -- Noncompliant
----

== Compliant Solution

----
IF @x > 0 SET @x = 0;
IF @y > 0 SET @y = 0;
----
