<p>There is no need to declare the same dependency or plugin twice in a project. In fact, doing so is likely to cause errors in the future when maintainers try to change or upgrade the plugin or dependency.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    &lt;dependency&gt;
      &lt;groupId&gt;com.mygroup&lt;/groupId&gt;
      &lt;artifactId&gt;myartifact&lt;/artifactId&gt;
      &lt;version&gt;1.0&lt;/version&gt;
      &lt;scope&gt;runtime&lt;/scope&gt;
    &lt;/dependency&gt;
    &lt;!-- ... --&gt;
    &lt;dependency&gt;  &lt;!-- Noncompliant --&gt;
      &lt;groupId&gt;com.mygroup&lt;/groupId&gt;
      &lt;artifactId&gt;myartifact&lt;/artifactId&gt;
      &lt;version&gt;1.0&lt;/version&gt;
      &lt;type&gt;jar&lt;/type&gt;
    &lt;/dependency&gt;
</pre>
