<p>A class with only &lt;code&gt;private&lt;/code&gt; constructors canâ€™t be instantiated, thus, it seems to be pointless code.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public class MyClass // Noncompliant
{
  private MyClass() { ... }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
public class MyClass
{
  public MyClass() { ... }
}
</pre>
<h2>Exceptions</h2>
<p>Classes that themselves access their private constructors (singletons or smart enums) are ignored. Classes with only &lt;code&gt;static&lt;/code&gt; members are also ignored because they are covered by Rule S1118.</p>
