include::../description.adoc[]

== Noncompliant Code Example

----
function fun($a) {
  $i = 10;
  return $i + $a;
  $i++;             // dead code
}
----

== Compliant Solution

----
function fun($a) {
  $i = 10;
  return $i + $a;
}
----

include::../see.adoc[]
