Identifiers declared with https://www.ibm.com/support/knowledgecenter/en/SSY2V3_5.1.0/com.ibm.ent.pl1.zos.doc/lr/value.html[<code>STATIC</code>] and <code>INIT</code> are initialized only once and may be later modified. Modifying such an identifier will fail at runtime when a CICS application is https://www.ibm.com/support/knowledgecenter/en/SSGMGV_3.1.0/com.ibm.cics.ts31.doc/dfhp3/dfhp3r3.htm[executed as reentrant]. It is safer to declare the value as https://www.ibm.com/support/knowledgecenter/en/SSY2V3_5.1.0/com.ibm.ent.pl1.zos.doc/lr/value.html[a named constant] (using the keyword <code>VALUE</code>) because any modification will be reported at compilation time.

This rule raises an issue when an identifier is declared as <code>STATIC</code> with <code>INIT</code> or <code>INITIAL</code> and has one of the following types:

* CHARACTER/CHAR
* BINARY/BIN
* DECIMAL/DEC
* FLOAT
* POINTER/PTR
* OFFSET


== Noncompliant Code Example

----
dcl returnCodeOk fixed bin(15) static init (00); /* Noncompliant */
/* or */
dcl actions char(05) init('12345') static; /* Noncompliant */
dcl actions_in char(01);
....
valid_action = (verify(action_in, actions) = 0);
----


== Compliant Solution

----
dcl returnCodeOk fixed bin(15) value (00);
/* or */
dcl actions char(05) value('12345');
dcl actions_in char(01);
....
valid_action = (verify(action_in, actions) = 0);
----

