Theoretically, <code>CASE</code>, <code>COALESCE</code>, and <code>IIF</code> evaluate conditions sequentially, stopping with the first one that evaluates to <code>TRUE</code>. In reality, these expressions evaluate aggregate functions and non-native service calls such as <code>CONTAIN</code> and <code>FREETEXT</code> first, and _then_ pass the results into the expression. That means that if you're relying on short-circuit behavior to avoid runtime errors with these arguments, you will not get it. You can work around the issue by wrapping such calls in a subselect.


== Noncompliant Code Example

----
SELECT 
  CASE 
    WHEN @i = 0 
      THEN 1 
    ELSE MIN(1/@i)  -- Noncompliant
  END; 
----


== Compliant Solution

----
SELECT 
  CASE 
    WHEN @i = 0 
      THEN 1 
    ELSE (SELECT MIN(1/@i))
  END; 
----

