According to the Java Language Specification:

{quote}
Unnamed packages are provided by the Java platform principally for convenience when developing small or temporary applications or when just beginning development.
{quote}

To enforce this best practice, classes located in default package can no longer be accessed from named ones since Java 1.4.


== Noncompliant Code Example

----
public class MyClass { /* ... */ }
----


== Compliant Solution

----
package org.example;

public class MyClass{ /* ... */ }
----

