According to the C++ standard, <code>this</code> can never be null, so comparisons of the two are pointless at best. At worst, because of compiler optimizations, such comparisons could lead to null pointer dereferences or obscure, difficult-to-diagnose errors in production.

This rule raises an issue when <code>this</code> is compared to <code>nullptr</code> or 0 or anything <code>#defined</code> as <code>nullptr</code> or 0, such as <code>NULL</code> in most environments.


== Noncompliant Code Example

----
class MyClass {
  string name;

  string GetName() {
    if (this != 0) {  // Noncompliant
      return name;
    }
    return 0;
  }
}
----


== Compliant Solution

----
class MyClass {
  string name;

  string GetName() {
    return name;
  }
}
----


