A results indicator should be used on a <code>CHAIN</code> statement because it leads to cleaner, more readable, and more correct code. Using an indicator results in the conditions <code>record not found</code>, and <code>end of file</code> being automatically handled correctly, without the need for further error handling. 


== Noncompliant Code Example

----
C     K0GMCP        CHAIN     SSGMCPR
----


== Compliant Solution

----
C     K0GMCP        CHAIN     SSGMCPR                            81 
----


== Exceptions

Free-form <code>CHAIN</code> statements cannot have indicators, and are ignored.

