include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression <code>^[A-Z][a-zA-Z0-9]*$</code>:

----
interface myInterface {...} // Noncompliant
----

== Compliant Solution

----
interface MyInterface {...}
----
