<p>For better readability, do not put more than one statement on a single line.</p>
<h2>Noncompliant Code Example</h2>
<pre>
WRITE 'Hello World'. WRITE 'You can use multiple statements on a line'.
</pre>
<h2>Compliant Solution</h2>
<pre>
WRITE 'Hello World'.
WRITE 'You can use multiple statements on a line'.
WRITE 'But you shouldn't.'.
</pre>
