<p>For better readability, do not put more than one statement on a single line.</p>
<h2>Noncompliant Code Example</h2>
<pre>
MOVE '1' TO OCTET-1 MOVE '2' TO OCTET-2
</pre>
<h2>Compliant Solution</h2>
<pre>
MOVE '1' TO OCTET-1
MOVE '2' TO OCTET-2
</pre>
<h2>Exceptions</h2>
<p>Control flow statements with a single nested statement are ignored.</p>
<pre>
IF x &gt; 0 THEN DISPLAY "positive". *&gt; Compliant
</pre>
