<p>For better readability, do not put more than one statement on a single line.</p>
<h2>Noncompliant Code Example</h2>
<pre>
foo: proc options(main);
  declare i fixed decimal init(42); put list (i); /* Noncompliant - there are two statements */
end;
</pre>
<h2>Compliant Solution</h2>
<pre>
foo: proc options(main);
  /* Compliant */
  declare i fixed decimal init(42);
  put list (i);
end;
</pre>
