<p>For better readability, do not put more than one statement on a single line.</p>
<h2>Noncompliant Code Example</h2>
<pre>
if someCondition { doSomething()}
...
var result = doSomething(); return result
</pre>
<h2>Compliant Solution</h2>
<pre>
if someCondition {
  doSomething()
}
...
var result = doSomething()
return result
</pre>
<h2>Exceptions</h2>
<p>Variable declaration with initialising code block and closure expressions containing a single statement are ignored.</p>
<pre>
var x : Int { return 0 }                                       // Variable declaration with initialising code block
doSomething({ (x: Int, y: Int) -&gt; Bool in return x &gt; y }, 5)   // Closure expression
</pre>
