Using a named constant to refer to an indicator makes the content of the field clearer, and therefore makes the code easier to read and maintain. 

== Noncompliant Code Example

----
      /Free

       If *In(25);  // Noncompliant
        // Process contents...
       EndIf;
----

== Compliant Solution

----
     D accountTotal    c                   25

      /Free

       If *In(accountTotal);
        // Process contents...
       EndIf;
----
