<p>Declaring a variable only to immediately return or throw it is a bad practice.
Some developers argue that the practice improves code readability, because it enables them to explicitly name what is being returned. However, this variable is an internal implementation detail that is not exposed to the callers of the method. The method name should be sufficient for callers to know exactly what will be returned.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public long ComputeDurationInMilliseconds()
{
  long duration = (((hours * 60) + minutes) * 60 + seconds ) * 1000 ;
  return duration;
}

public void DoSomething()
{
  ApplicationException myException = new ApplicationException();
  throw myException;
}
</pre>
<h2>Compliant Solution</h2>
<pre>
public long ComputeDurationInMilliseconds()
{
  return (((hours * 60) + minutes) * 60 + seconds ) * 1000 ;
}

public void DoSomething()
{
  throw new ApplicaitonException();
}
</pre>
