include::../description.adoc[]

== Noncompliant Code Example

https://github.com/jwt-dotnet/jwt[jwt-dotnet] library:
----
Dim decodedtoken1 As String = decoder.Decode(token, secret, verify:= false) ' Noncompliant: signature should be verified

Dim decodedtoken2 As String = new JwtBuilder().
    WithSecret(secret).
    Decode(forgedtoken1) ' Noncompliant: signature should be verified
----

== Compliant Solution

https://github.com/jwt-dotnet/jwt[jwt-dotnet] library:
----
Dim decodedtoken1 As String = decoder.Decode(forgedtoken1, secret, verify:= true) ' Compliant

Dim decodedtoken2 As String = new JwtBuilder().
    WithSecret(secret).
    MustVerifySignature().
    Decode(token) ' Compliant
----

include::../see.adoc[]
