Iterators are useful to deal with data inside a container: they point to one of its element and can be incremented or decremented to access other elements of this container. However, it will be undefined behavior if they  try to access data out of bounds:
- You should not try to get an iterator which would be before <code>begin()</code>
- You should not try to get an iterator which would be after <code>end()</code>
- You should not try to dereference <code>end()</code>: ranges are semi open, which means that <code>begin()</code> is the location of the first element, but <code>end()</code> is a location past-the-end of the container, and does not correspond to any data.


== Noncompliant Code Example

----
void test(std::vector<int> const &v) {
  auto iter = v.end();
  auto x = *iter; // Noncompliant; access out of range
  // ...
}
----


