Having the same implementation in every branch of an <code>if</code> chain is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be an <code>if</code> chain at all.

== Noncompliant Code Example

----
if b == 0:  # Noncompliant
    doOneMoreThing()
elif b == 1:
    doOneMoreThing()
else:
    doOneMoreThing()

b = 4 if a > 12 else 4  # Noncompliant

----

== Exceptions

This rule does not apply to <code>if</code> chains without <code>else</code>-s.

----
if b == 0:  # no issue, this could have been done on purpose to make the code more readable
    doOneMoreThing()
elif b == 1:
    doOneMoreThing()
----
