<p>Using trailing closure syntax for the last parameter in a call is often the most elegant way to handle it. But if the call requires multiple function-type arguments, the use of a trailing closure can be messy and confusing. In such cases, itâ€™s better to pass closure expressions as normal arguments.</p>
<h2>Noncompliant Code Example</h2>
<pre>
var x = complexOperation(
  arg: 2,
  op1: {$0 + 10}
) {$0 * $0}
</pre>
<h2>Compliant Solution</h2>
<pre>
var x = complexOperation(
  arg: 2,
  op1: {$0 + 10},
  op2: {$0 * $0}
)
</pre>
