include::../description.adoc[]

== Noncompliant Code Example

----
for (var i:int = 0; i < 10; i++) {
  var temp:MyObj = new MyObject();  // Noncompliant
  //...  
}
----

== Compliant Solution

----
var temp:MyObj = new MyObject();
for (var i:int = 0; i < 10; i++) {
  //...  
}
----
