<p>Labeled loops are useful, especially when the code is badly indented, to match the begin and end of each loop. This rule raises an issue when the end of a labeled loop is unlabeled.</p>
<h2>Noncompliant Code Example</h2>
<pre>
BEGIN
  &lt;&lt;myLoopLabel1&gt;&gt;
  LOOP
    EXIT;
  END LOOP; -- Noncompliant; this labeled loop has no ending label

  LOOP
    EXIT;
  END LOOP; -- Compliant; not a labeled loop
END;
/
</pre>
<h2>Compliant Solution</h2>
<pre>
BEGIN
  &lt;&lt;myLoopLabel1&gt;&gt;
  LOOP
    EXIT;
  END LOOP myLoopLabel1;

  LOOP
    EXIT;
  END LOOP;
END;
/
</pre>
