With the advent of .NET framework version 2, certain practices have become obsolete.
In particular, exceptions should now extend <code>System.Exception</code> instead of <code>System.ApplicationException</code>. Similarly, generic collections should be used instead of the older, non-generic, ones. Finally when creating an XML view, you should not extend <code>System.Xml.XmlDocument</code>.
This rule raises an issue when an externally visible type extends one of these types:
* <code>System.ApplicationException</code>
* <code>System.Xml.XmlDocument</code>
* <code>System.Collections.CollectionBase</code>
* <code>System.Collections.DictionaryBase</code>
* <code>System.Collections.Queue</code>
* <code>System.Collections.ReadOnlyCollectionBase</code>
* <code>System.Collections.SortedList</code>
* <code>System.Collections.Stack</code>


== Noncompliant Code Example

----
using System;
using System.Collections;

namespace MyLibrary
{
  public class MyCollection : CollectionBase  // Noncompliant
  {
  }
}
----


== Compliant Solution

----
using System;
using System.Collections;

namespace MyLibrary
{
  public class MyCollection : Collection<T>
  {
  }
}
----


