<p>When a &lt;code&gt;static&lt;/code&gt; constructor serves no other purpose that initializing &lt;code&gt;static&lt;/code&gt; fields, it comes with an unnecessary performance cost because the compiler generates a check before each &lt;code&gt;static&lt;/code&gt; method or instance constructor invocation.</p>
<p>Instead, inline initialization is highly recommended.</p>
<h2>Noncompliant Code Example</h2>
<pre>
namespace myLib
{
  public class Foo
  {
    static int i;
    static string s;

    static Foo() // Noncompliant
    {
      i = 3;
      ResourceManager sm =  new ResourceManager("strings", Assembly.GetExecutingAssembly());
      s = sm.GetString("mystring");
    }
  }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
namespace myLib
{
  public class Foo
  {
    static int i =3;
    static string s = InitString();

    static string InitString()
    {
      ResourceManager sm = new ResourceManager("strings", Assembly.GetExecutingAssembly());
      return sm.GetString("mystring");
    }
  }
}
</pre>
