While <code>NOT IN</code> can be far more efficient than <code>NOT EXISTS</code> in a query, it may return misleading results if the column in question contains null values.


== Noncompliant Code Example

----
SELECT COUNT(*) FROM emp 
  WHERE empno NOT IN ( SELECT mgr FROM emp );
----


== Compliant Solution

----
SELECT COUNT(*) FROM emp T1
   WHERE NOT EXISTS ( SELECT NULL FROM emp T2 WHERE t2.mgr = t1.empno );
----

