Class members are initialized in the order in which they are declared in the class, not the order in which they appear in the class initializer list. To avoid errors caused by order-dependent initialization, the order of members in the initialization list should match the order in which members are declared in a class.


== Noncompliant Code Example

----
#include <iostream>

class C {
public:
  int x;
  int y;

  C(int i) : y(i), x(y + 1) { }  // Noncompliant
};

int main() {
  C c(1);
  std::cout << c.x << " " << c.y << std::endl;  // prints 1 1
}
----


== See

* https://www.securecoding.cert.org/confluence/x/VADiAQ[CERT, OOP53-CPP.] - Write constructor member initializers in the canonical order
* https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#c47-define-and-initialize-member-variables-in-the-order-of-member-declaration[C++ Core Guidelines C.47]: Define and initialize member variables in the order of member declaration

