include::../description.adoc[]

== Noncompliant Code Example

----
if x == 0 {
  doSomething()
} else if x == 1 {
  doSomethingElse()
} 
----

== Compliant Solution

----
if x == 0 {
  doSomething()
} else if x == 1 {
  doSomethingElse()
} else {
  NSException(name:"IllegalStateException", reason:"Unreachable else clause is reached", userInfo:nil).raise()
}
----

include::../see.adoc[]
