<p>Function pointer syntax can be hard on the eyes, particularly when one function is used as a parameter to another. Providing and using a &lt;code&gt;typedef&lt;/code&gt; instead (or a &lt;code&gt;using&lt;/code&gt; in C++) can make code easier to read, and should be preferred.</p>
<h2>Noncompliant Code Example</h2>
<pre>
extern void (*signal(int, void(*)(int)))(int);
</pre>
<h2>Compliant Solution</h2>
<pre>
typedef void (*SignalHandler)(int signum);
extern SignalHandler signal(int signum, SignalHandler handler);
</pre>
