Using a unique <code>KLIST</code> to access files ensures that the correct key is used at all times and creates a standard method of accessing the data throughout the code, simplifying maintenance and improving readability. 

This rule flags non-compliant instances of <code>CHAIN</code>, <code>DELETE</code>, <code>READE</code>, <code>READPE</code>, <code>SETGT</code>, and <code>SETLL</code> operations.


== Noncompliant Code Example

----
C     'D001 B'      CHAIN     KYCUSTN                            20
... 
C     DEPT          CHAIN     KYCUSTN                            20
----


== Compliant Solution

----
C     DEPTKY        KLIST
C                   KFLD                    DEPT
C                   KFLD                    SHIFT 

C     DEPTKY        CHAIN     KYCUSTN                            20
----

