When allocating memory with <code>malloc</code>, <code>calloc</code> and <code>realloc</code> it is important to make sure that the size of the allocated memory is compatible with the receiver type.


== Noncompliant Code Example

----
void f() {
  ...
  int* ptr1 = malloc(sizeof(short)); // Noncompliant
  ...
}
----


== Compliant Solution

----
void f() {
  ...
  int* ptr1 = malloc(sizeof(int));
  ...
}
----

