Declaring multiple variable on one line is difficult to read.

== Noncompliant Code Example

----
Module Module1
  Public Const AAA As Integer = 5, BBB = 42, CCC As String = "foo"  ' Noncompliant
End Module
----

== Compliant Solution

----
Module Module1
  Public Const AAA As Integer = 5
  Public Const BBB = 42
  Public Const CCC as String = "foo"
End Module
----

include::../see.adoc[]
