The W3C specifications define the valid CSS properties. Only the official and browser-specific properties should be used to get the expected impact in the final rendering.

This rule ignores:
- <code>$sass</code>, <code>@less</code>, and <code>var(--custom-property)</code> variable syntaxes.
- vendor-prefixed properties (e.g., <code>-moz-align-self</code>, <code>-webkit-align-self</code>).


== Noncompliant Code Example

----
a {
  colour: blue; /* Noncompliant; colour is not part of the specifications */
}
----


== Compliant Solution

----
a {
  color: blue;
}
----

