<p>If a private variable is declared but not used in the program, it can be considered dead code and should therefore be removed. This will improve maintainability because developers will not wonder what the variable is used for.</p>
<h2>Noncompliant Code Example</h2>
<pre>
Private Foo as Integer

Function Compute(A As Integer)
   Compute = A * 42
End Function
</pre>
<h2>Compliant Solution</h2>
<pre>
Function Compute(A As Integer)
   Compute = A * 42
End Function
</pre>
