include::../description.adoc[]

== Noncompliant Code Example

----
<Export(GetType(ISomeType))>
Public Class SomeType  // Noncompliant; doesn't implement 'ISomeType'.
End Class
----

== Compliant Solution

----
<Export(GetType(ISomeType))>
Public Class SomeType
    Inherits ISomeType
End Class
----
