<p>Shared coding conventions allow teams to collaborate efficiently. This rule checks that all function names match a provided regular expression.</p>
<h2>Noncompliant Code Example</h2>
<p>With default provided regular expression: ^(<strong>|[a-z0-9_]</strong>)$</p>
<pre>
FUNCTION MyFunction.
...
ENDFUNCTION.
</pre>
<h2>Compliant Solution</h2>
<pre>
FUNCTION MY_FUNCTION.
...
ENDFUNCTION.
</pre>
