<p>Shared coding conventions allow teams to collaborate efficiently. This rule checks that all function names match a provided regular expression.</p>
<h2>Noncompliant Code Example</h2>
<pre>
function MyFunc() {
  // ...
}
</pre>
<h2>Compliant Solution</h2>
<pre>
function myFunc() {
  // ...
}
</pre>
