<p>Shared coding conventions allow teams to collaborate efficiently. This rule checks that all function names match a provided regular expression.</p>
<h2>Noncompliant Code Example</h2>
<pre>
foo#: proc options(main); /* Non-Compliant */
end;
</pre>
<h2>Compliant Solution</h2>
<pre>
foo: proc options(main);  /* Compliant */
end;
</pre>
