<p>Shared coding conventions allow teams to collaborate efficiently. This rule checks that all function names match a provided regular expression.</p>
<h2>Noncompliant Code Example</h2>
<p>With default provided regular expression: ^[A-Z][a-zA-Z0-9_]*$</p>
<pre>
Private Sub doSomething()
...
End Sub
</pre>
<h2>Compliant Solution</h2>
<pre>
Private Sub DoSomething()
...
End Sub
</pre>
