include::../description.adoc[]

== Noncompliant Code Example

----
try {
  saveDocument();
} catch (const std::exception& e) { // Noncompliant
  throw;
}
----

== Compliant Solution

----
try {
  saveDocument();
} catch (const std::exception& e) { // Compliant
  log << e.what();
  throw;
}
----
or

----
saveDocument();
----
