include::../description.adoc[]

== Noncompliant Code Example

----
public int Foo
{
    get
    {
        throw new Exception(); // Noncompliant
    }
}
----

== Compliant Solution

----
public int Foo
{
    get
    {
        return 42;
    }
}
----

include::../exceptions.adoc[]
