<code>StringBuffer</code> and <code>StringBuilder</code> instances that are <code>append</code>ed but never <code>toString</code>ed needlessly clutter the code, and worse are a drag on performance. Either they should be removed, or the missing <code>toString</code> call added.


== Noncompliant Code Example

----
public void doSomething(List<String> strings) {

  StringBuilder sb = new StringBuilder();  // Noncompliant
  sb.append("Got: ");
  for (String str : strings) {
    sb.append(str).append(", ");
    // ...
  }
}
----


== Compliant Solution

----
public void doSomething(List<String> strings) {

  for (String str : strings) {
    // ...
  }
}
----
or

----
public void doSomething(List<String> strings) {

  StringBuilder sb = new StringBuilder();
  sb.append("Got: ");
  for (String str : strings) {
    sb.append(str).append(", ");
    // ...
  }

  LOGGER.info(sb.toString);
}
----


== Exceptions

This rule ignores <code>StringBuffer</code>s and <code>StringBuilder</code>s that are passed as method arguments on the grounds that they are likely <code>toString</code>ed there.

