According to the PSR2 coding standard:
{quote}
The keyword <code>elseif</code> SHOULD be used instead of <code>else if</code> so that all control keywords look like single words.
{quote}


== Noncompliant Code Example

----
if ($expr1) {
  ...
} else if ($expr2) {
  ...
} else {...}
----


== Compliant Solution

----
if ($expr1) {
  ...
} elseif ($expr2) {
  ...
} else {...}
----

