Python has no pre/post increment/decrement operator. For instance, <code>x++</code> and <code>x--</code> will fail to parse. More importantly, <code>++x</code> and <code>--x</code> will do nothing. To increment a number, simply write <code>x += 1</code>.


== Noncompliant Code Example

----
++x # Noncompliant
----


== Compliant Solution

----
x += 1
----

