<p>The goal of this coding rule is to make it obvious at first look that a class &lt;code&gt;extends&lt;/code&gt; some other classes and/or &lt;code&gt;implements&lt;/code&gt; some interfaces. The names of extended classes or implemented interfaces can be located on next lines.</p>
<h2>Noncompliant Code Example</h2>
<pre>
&lt;?php
class ClassName
extends ParentClass {...}
</pre>
<h2>Compliant Solution</h2>
<pre>
&lt;?php
class ClassName extends ParentClass {...}
</pre>
<p>or</p>
<pre>
&lt;?php
class ClassName extends
ParentClass {...}
</pre>
