include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression <code>^[a-z]+(\.[a-z][a-z0-9]*)*$</code>:
----
package org.Example {  // Noncompliant
...
}
----

== Compliant Solution

----
package org.example { 
...
}
----
