<p>The W3C specifications say comments should be defined using &lt;code&gt;/* …​ */&lt;/code&gt;. The use of &lt;code&gt;//&lt;/code&gt; is not supported on all browsers and can lead to unexpected results.</p>
<h2>Noncompliant Code Example</h2>
<pre>
// some comment
a { color: pink; }
</pre>
<h2>Compliant Solution</h2>
<pre>
/* some comment */
a { color: pink; }
</pre>
<h2>Exceptions</h2>
<p>This rule ignores single line comments in &lt;code&gt;less&lt;/code&gt; and &lt;code&gt;scss&lt;/code&gt; files.</p>
