The W3C specifications say comments should be defined using <code>/* ... */</code>. The use of <code>//</code> is not supported on all browsers and can lead to unexpected results.


== Noncompliant Code Example

----
// some comment
a { color: pink; }
----


== Compliant Solution

----
/* some comment */
a { color: pink; }
----


== Exceptions

This rule ignores single line comments in <code>less</code> and <code>scss</code> files.


