<p>Theoretically, the use of the &lt;code&gt;static&lt;/code&gt; keyword on the size of an array parameter means you can assume that only arrays of at least that size will be passed as arguments to the function. I.e. a function parameter of &lt;code&gt;int my_array[static 10]&lt;/code&gt; means that &lt;code&gt;my_array&lt;/code&gt; will always be <em>at least</em> 10 elements long. If it is not, the behavior is undefined.</p>
<p>In practice, the use of &lt;code&gt;static&lt;/code&gt; on the size of an array parameter means the compiler might issue a warning if a noncompliant array is passed to the function - a warning that might or might not be ignored. Therefore, in practice the use of &lt;code&gt;static&lt;/code&gt; on an array parameterâ€™s size merely lends a false sense of security, and &lt;code&gt;static&lt;/code&gt; should not be used in this context.</p>
<p>Note that for some compiler/processor combinations, more efficient code can be generated when &lt;code&gt;static&lt;/code&gt; is used, but these combinations are limited, and the benefit does not outweigh the cost.</p>
<h2>Noncompliant Code Example</h2>
<pre>
int total (int size, int my_array[static 10]) {...}
</pre>
<h2>Compliant Solution</h2>
<pre>
int total (int size, int my_array[10]) {...}
</pre>
<h2>See</h2>
<ul>
<li>MISRA C:2012, 17.6 - The declaration of an array parameter shall not contain the static keyword between the []</li>
</ul>
