There's almost no point in returning results from a trigger. Because there is generally no expectation that triggers will return anything, any such results are likely to be ignored nearly all the time. 

This rule raises an issue when <code>PRINT</code> is used inside a trigger, and when <code>SELECT</code>, or <code>FETCH</code> is used inside a trigger without the retrieved data subsequently being used inside the trigger itself.


== Noncompliant Code Example

----
ALTER TRIGGER TRG_myTrigger
...
  PRINT @diagnostic_message -- Noncompliant
----

