<p>&lt;code&gt;Thread.yield&lt;/code&gt; is intended to hint to the processor that the current thread is willing to suspended in favor of another thread. Unfortunately, it doesnâ€™t have the same results across platforms, thus marring the cross-platform compatibility of any application that uses it.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public void doSomething() {
  // ...
  Thread.yield();  // Noncompliant
  // ...
}
</pre>
