include::../description.adoc[]

== Noncompliant Code Example

----
CREATE TABLE notnull(col1 INTEGER NOT NULL)
INSERT notnull VALUES (@value) -- FIXME ensure @value is not null
----

include::../see.adoc[]
