<p>The problem with invoking &lt;code&gt;Thread.start()&lt;/code&gt; in a constructor is that youâ€™ll have a confusing mess on your hands if the class is ever extended because the superclass' constructor will start the thread before the child class has truly been initialized.</p>
<p>This rule raises an issue any time &lt;code&gt;start&lt;/code&gt; is invoked in the constructor of a non-&lt;code&gt;final&lt;/code&gt; class.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public class MyClass {

  Thread thread = null;

  public MyClass(Runnable runnable) {
    thread = new Thread(runnable);
    thread.start(); // Noncompliant
  }
}
</pre>
<h2>See</h2>
<ul>
<li><a href="https://www.securecoding.cert.org/confluence/x/ZQIRAg">CERT, TSM02-J.</a> - Do not use background threads during class initialization</li>
</ul>
