Restricting the number of <code>CONTINUE</code>, <code>EXIT</code> and <code>CHECK</code> statements in a loop is done in the interest of good structured programming.

One <code>CONTINUE</code>, <code>EXIT</code> and <code>CHECK</code> statement is acceptable in a loop, since it facilitates optimal coding. If there is more than one, the code should be refactored to increase readability.

== Noncompliant Code Example

----
DO counter TIMES.
  IF sy-index = 2.
    CONTINUE.
  ENDIF.
  IF sy-index = 10.
    EXIT.
  ENDIF.
  WRITE sy-index.
ENDDO.
----
