<code>\_\_all\_\_</code> is used to define the list of module's names that should be imported when <code>from package import *</code> is used. For that reason, it may only contain strings.


== Noncompliant Code Example

----
class MyClass:
    pass

__all__ = [
    MyClass  # Noncompliant
]
----


== Compliant Solution

----
class MyClass:
    pass

__all__ = [
    "MyClass"
]
----


== See

* https://docs.python.org/3/tutorial/modules.html#importing-from-a-package[Python documentation  - Importing * From a Package]


