<p>When constraints enforce a limited set of values for a column, conditioning a SQL statement on a value outside that set is simply an error.</p>
<h2>Noncompliant Code Example</h2>
<pre>
...
CHECK(SIZE = ’S’ OR SIZE = ’M’ OR SIZE = 'L')
...

SELECT id, name, price
FROM product
WHERE size != 'XXL'  -- Noncompliant; XXL is not a valid value, so clause has no effect

SELECT id, name, price
FROM product
WHERE size = 'XXL'  -- Noncompliant; XXL is not a valid value, so no rows selected
</pre>
