include::../description.adoc[]

== Noncompliant Code Example

----
$i = $a + $b; // Noncompliant; calculation result not used before value is overwritten
$i = compute();
----

== Compliant Solution

----
$i = $a + $b;
$i += compute();
----

== Exceptions

This rule ignores initializations to -1, 0, 1, <code>null</code>, <code>true</code>, <code>false</code>, <code>""</code>, <code>[]</code> and <code>array()</code>.

include::../see.adoc[]
