<p>&lt;code&gt;WHERE&lt;/code&gt; clause conditions that reinforce or contradict the definitions of their columns are useless; they are always either unconditionally true or unconditionally false. For instance, thereâ€™s no point in including &lt;code&gt;AND column IS NOT NULL&lt;/code&gt; if the column is defined as non-null.</p>
<p><strong>Noteworthy</strong>
This rule raises issues only when a <strong>Data Dictionary</strong> is provided during the analysis. See&nbsp;https://docs.sonarqube.org/display/PLUG/Data+Dictionary</p>
<h2>Noncompliant Code Example</h2>
<pre>
CREATE TABLE product
(id INT,
  name VARCHAR(6) NOT NULL,
  mfg_name VARCHAR(6),
  mfg_id INT
  ...

SELECT name, price
FROM product
WHERE name is not null -- Noncompliant; always true. This column is NOT NULL
  AND mfg_name = 'Too long name' -- Noncompliant; always false. This column can contain only 6 characters
</pre>
