include::../description.adoc[]

== Noncompliant Code Example

----
// in web.xml
<web-app  ...>
  <!--  ...  -->
  <login-config>
    <auth-method>BASIC</auth-method>
  </login-config>
</web-app>
----

include::../see.adoc[]
