include::../description.adoc[]

== Noncompliant Code Example

----
from flask import request, send_file

@app.route('/download')
def download():
    file = request.args['file']
    return send_file("static/%s" % file, as_attachment=True) # Noncompliant
----

== Compliant Solution

----
from flask import request, send_from_directory

@app.route('/download')
def download():
    file = request.args['file']
    return send_from_directory('static', file) # Compliant
----

include::../see.adoc[]
