<p>This rule allows you to check for certain types of comments before each method. For example, you may want to check for a Doxygen @author comment before each method.</p>
<h2>Noncompliant Code Example</h2>
<p>With regular expression: &lt;code&gt;//! @brief[ \t][a-zA-Z ]&lt;/code&gt;:</p>
<pre>
bool TEST_CLASS_OK::Method(int nArg)  // Noncompliant
{
  return true;
}{code}
</pre>
<h2>Compliant Solution</h2>
<pre>
//-------------------------------------------------------------------------
//! @brief    Method
//-------------------------------------------------------------------------
bool TEST_CLASS_OK::Method(int nArg)
{
  return true;
}
</pre>
