C++ allows the traditional C-style casts  [E.G. <code>(int) f</code>] and functional notation casts [E.G. <code>int(f)</code>], but adds its own forms:

* <code>static_cast<type>(expression)</code>
* <code>const_cast<type>(expression)</code>
* <code>dynamic_cast<type>(expression)</code>
* <code>reinterpret_cast<type>(expression)</code>

C-style casts and functional notation casts are largely functionally equivalent. However, when they do not invoke a converting constructor, C-style casts are capable of performing dangerous conversions between unrelated types and of changing a variable's <code>const</code>-ness. Attempt to do these things with an explicit C++-style cast, and the compiler will catch the error. Use a C-style or functional notation cast, and it cannot.

Additionally, C++-style casts are preferred because they are visually striking. The visual subtlety of a C-style or functional cast may mask that a cast has taken place, but a C++-style cast draws attention to itself, and makes the the programmer's intention explicit.


== Noncompliant Code Example

----
#include <iostream>

class Base { };

class Derived: public Base
{
public:
  int a;
};

void DoSomethingElse(Derived *ptr)
{
  ptr->a = 42;
}

void DoSomething(const Base *ptr)
{
  Derived* derived = (Derived*)ptr; // Noncompliant; inadvertently removes constness
  DoSomethingElse(derived);
}

int main(int argc, char* argv[])
{
  Derived *ptr = new Derived();
  ptr->a = 1337;

  DoSomething(ptr);

  std::cout << ptr->a << std::endl; /* 1337 was expected, but 42 is printed */

  return 0;
}
----


== Compliant Solution

----
/* ... */

void DoSomething(const Base *ptr)
{
  /* error: static_cast from type 'const Base*' to type 'Derived*' casts away qualifiers */
  Derived* derived = static_cast<Derived*>(ptr); // Compliant. Compile fails with above error
  DoSomethingElse(derived);
}

/* ... */
----


== Exceptions

Void casts and explicit constructor calls are allowed.


== See

* MISRA C++:2008, 5-2-4 - C-style casts (other than void casts) and functional notation casts (other than explicit constructor calls) shall not be used.

