<p>Marking a variable with &lt;code&gt;IBOutlet&lt;/code&gt; allows it to be connected with a Storyboard component through the Interface Builder. Allowing such a variable to be accessed outside the class, may result in other classes making assignments that override the automatic dependency injection from the Storyboard itself.</p>
<h2>Noncompliant Code Example</h2>
<pre>
  @IBOutlet var label: UILabel!  // Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
  @IBOutlet private var label: UILabel!
</pre>
