include::../description.adoc[]

== Noncompliant Code Example

----
if len(myArr) >= 0 { ... }

if len(myArr) < 0 { ... }

var result = len(myArr) >= 0
----

== Compliant Solution

----
if len(myArr) > 0 { ... }

if len(myArr) < 42 { ... }
----
