Shared coding conventions allow teams to collaborate effectively. This rule checks that classes extending <code>React.PureComponent</code> or <code>React.Component</code> are named with UpperCamelCase (aka PascalCase).


== Noncompliant Code Example

----
export default class my-super-component extends React.PureComponent { // Noncompliant
 ...
}
----


== Compliant Solution

----
export default class MySuperComponent extends React.PureComponent {
 ...
}
----

