include::../description.adoc[]

== Noncompliant Code Example

----
private int Divide(int numerator, int denominator) 
{
    return numerator / denominator;  // FIXME denominator value might be  0
}
----

include::../see.adoc[]
