include::../description.adoc[]

== Noncompliant Code Example

----
func main() {
  v1 := (true && false) && (true && false) // Noncompliant
}
----

== Compliant Solution

----
func main() {
  v1 := (true && false) // Compliant
}
----

== Exceptions

This rule ignores <code>*</code>, <code>+</code>, <code><<</code> and <code>=</code>. 

include::../see.adoc[]
