Throwing such general exceptions as <code>Exception</code>, <code>SystemException</code>, <code>ApplicationException</code>, <code>IndexOutOfRangeException</code>, <code>NullReferenceException</code>, <code>OutOfMemoryException</code> and <code>ExecutionEngineException</code> prevents calling methods from handling true, system-generated exceptions differently than application-generated errors. 

== Noncompliant Code Example

----
Public Sub DoSomething(obj As Object)
  If obj Is Nothing Then
    ' Noncompliant
    Throw New NullReferenceException("obj")
  End If
  ' ...
End Sub
----

== Compliant Solution

----
Public Sub DoSomething(obj As Object)
  If obj Is Nothing Then
    Throw New ArgumentNullException("obj")
  End If
  ' ...
End Sub
----

include::../see.adoc[]
