PHP's "variable variables" feature (dynamically-named variables) is temptingly powerful, but can lead to unmaintainable code. 


== Noncompliant Code Example

----
$var = 'foo';
$$var = 'bar';      //Noncompliant
$$$var = 'hello';  //Noncompliant

echo $foo; //will display 'bar'
echo $bar; //will display 'hello'
----

