include::../description.adoc[]

== Noncompliant Code Example

With the default threshold of 3:
----
IF @x='Yes'
  SELECT ...
    FROM ...
    WHERE field='Yes'
...
...
IF @x='Yes'
  ...
----

== Compliant Solution

----
DECLARE @Yes VARCHAR(3) = 'Yes'
IF @x=@Yes
  SELECT ...
    FROM ...
    WHERE field=@Yes
...
...
IF @x=@Yes
  ...
----

include::../exceptions.adoc[]
