<p>While the ternary operator is pleasingly compact, its use can make code more difficult to read. It should therefore be avoided in favor of the more verbose &lt;code&gt;if&lt;/code&gt;/&lt;code&gt;else&lt;/code&gt; structure.</p>
<h2>Noncompliant Code Example</h2>
<pre>
printf("%s", (i &gt; 10 ? "yes" : "no"));
</pre>
<h2>Compliant Solution</h2>
<pre>
if (i &gt; 10) {
  printf("yes");
} else {
  printf("no");
}
</pre>
