<p>ORACLE 11g introduced the &lt;code&gt;SIMPLE_INTEGER&lt;/code&gt; data type, which is a sub-type of &lt;code&gt;PLS_INTEGER&lt;/code&gt;, and covers the same range. There are three main differences between the two types:</p>
<ul>
<li>&lt;code&gt;SIMPLE_INTEGER&lt;/code&gt; is always &lt;code&gt;NOT NULL&lt;/code&gt;. So when the value of the declared variable is never going to be null, you can declare it as &lt;code&gt;SIMPLE_INTEGER&lt;/code&gt;.</li>
<li>You will never face a numeric overflow using &lt;code&gt;SIMPLE_INTEGER&lt;/code&gt; because this data type wraps around without giving any error.</li>
<li>The &lt;code&gt;SIMPLE_INTEGER&lt;/code&gt; data type gives a major performance boost over &lt;code&gt;PLS_INTEGER&lt;/code&gt; when the code is compiled in "NATIVE" mode, because arithmetic operations on &lt;code&gt;SIMPLE_INTEGER&lt;/code&gt; type are performed directly at the hardware level.</li>
</ul>
<h2>Noncompliant Code Example</h2>
<pre>
DECLARE
  v1 PLS_INTEGER; -- Noncompliant
  v2 VARCHAR2(10);
BEGIN
  NULL;
END;
/
</pre>
<h2>Compliant Solution</h2>
<pre>
DECLARE
  v1 SIMPLE_INTEGER := 42;
  v2 VARCHAR2(10);
BEGIN
  NULL;
END;
/
</pre>
