When using the <code>Stream</code> API, call chains should be simplified as much as possible to improve readability and maintainability.

This rule raises an issue when one of the following substitution can be made:
||Original||Preferred||
|<code>stream.collect(counting())</code>|<code>stream.count()</code>|
|<code>stream.collect(maxBy(comparator))</code>|<code>stream.max(comparator)</code>|
|<code>stream.collect(minBy(comparator))</code>|<code>stream.min(comparator)</code>|
|<code>stream.collect(mapping(mapper))</code>|<code>stream.map(mapper).collect()</code>|
|<code>stream.collect(reducing(...))</code>|<code>stream.reduce(...)</code>|
|<code>stream.collect(summingInt(mapper))</code>|<code>stream.mapToInt(mapper).sum()</code>|
|<code>stream.collect(summingLong(mapper))</code>|<code>stream.mapToLong(mapper).sum()</code>|
|<code>stream.collect(summingDouble(mapper))</code>|<code>stream.mapToDouble(mapper).sum()</code>|


== Noncompliant Code Example

----
int count = stream.collect(counting());  // Noncompliant
----


== Compliant Solution

----
int count = stream.count();
----


