Type assertion can be done in two ways: with <code>as MyType</code> or with <code><MyType></code>. But since there is an ambiguity in the latter when using JSX and there is no ambiguity in the former, <code>as</code> is preferred.


== Noncompliant Code Example

----
var foo = <any>"foo";  // Noncompliant
----


== Compliant Solution

----
var foo = "foo" as any;
----

