<p>The use of parentheses, even those not required to enforce a desired order of operations, can clarify the intent behind a piece of code. But redundant pairs of parentheses could be misleading, and should be removed.</p>
<h2>Noncompliant Code Example</h2>
<pre>
COMPUTE x = (y / 2 + 1).  *&gt; Compliant even if the parenthesis are ignored by the compiler
COMPUTE y = 2 * ((x + 1)).  *&gt; Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
COMPUTE x = (y / 2 + 1).
COMPUTE y = 2 * (x + 1).
</pre>
