Various aspects of <code>ctime</code> are implementation-defined or unspecified, such as the formats of times. 


== Noncompliant Code Example

----
#include <ctime>  /* Noncompliant */

void f()
{
  clock();
}
----


== See

* MISRA C++:2008, 18-0-4 - The time handling functions of library <ctime> shall not be used.

