HTML-style comments are not part of EcmaScript specification, and should not be used.


== Noncompliant Code Example

----
<!-- Noncompliant -->
----


== Compliant Solution

----
// Compliant
/* Compliant */
----

