<p>Since C++11, type aliases can be declared via &lt;code&gt;using&lt;/code&gt; or &lt;code&gt;typedef&lt;/code&gt;. &lt;code&gt;using&lt;/code&gt; should be preferred as  more readable because you see the new name/alias first.
In addition, &lt;code&gt;using&lt;/code&gt; can be templated, which makes it applicable to more situations than &lt;code&gt;typedef&lt;/code&gt;.</p>
<h2>Noncompliant Code Example</h2>
<pre>
typedef void (*FunctionPointerType)(int);
</pre>
<h2>Compliant Solution</h2>
<pre>
using FunctionPointerType = void (*)(int);
</pre>
<h2>See</h2>
<ul>
<li><a href="https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#t43-prefer-using-over-typedef-for-defining-aliases">C++ Core Guidelines - T.43</a> - Prefer using over typedef for defining aliases</li>
</ul>
