Since C++11, type aliases can be declared via <code>using</code> or <code>typedef</code>. <code>using</code> should be preferred as  more readable because you see the new name/alias first.
In addition, <code>using</code> can be templated, which makes it applicable to more situations than <code>typedef</code>.


== Noncompliant Code Example

----
typedef void (*FunctionPointerType)(int);
----


== Compliant Solution

----
using FunctionPointerType = void (*)(int);
----


== See

* https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#t43-prefer-using-over-typedef-for-defining-aliases[C++ Core Guidelines - T.43] - Prefer using over typedef for defining aliases

