This rule applies whenever an <code>If</code> statement is followed by one or more <code>ElseIf</code> statements; the final <code>ElseIf</code> should be followed by an <code>Else</code> statement.

The requirement for a final <code>Else</code> statement is defensive programming.
The <code>Else</code> statement should either take appropriate action or contain a suitable comment as to why no action is taken. This is consistent with the requirement to have a final <code>Case Else</code> clause in a <code>Select Case</code> statement.

== Noncompliant Code Example

----
If x = 0 Then
    DoSomething()
ElseIf x = 1 Then
    DoSomethingElse()
End If
----

== Compliant Solution

----
If x = 0 Then
    DoSomething()
ElseIf x = 1 Then
    DoSomethingElse()
Else
    Throw New ArgumentException("...")
End If
----

== Exceptions

None

include::../see.adoc[]
