In order to be callable, a python class should implement the https://docs.python.org/3/reference/datamodel.html#object.__call__[<code>\_\_call\_\_</code>] method.

This rule raises an issue when a non-callable object is called.


== Noncompliant Code Example

----
class MyClass:
    pass

myvar = MyClass()
myvar()  # Noncompliant

none_var = None
none_var()  # Noncompliant
----


== Compliant Solution

----
class MyClass:
    def __call__(self):
        print("called")

myvar = MyClass()
myvar()
----


== See

* https://docs.python.org/3/reference/datamodel.html#object.__call__[Python documentation - \_\_call\_\_ method]

