As stated per effective java : 
{quote}
Varargs methods are a convenient way to define methods that require a variable number of arguments, but they should not be overused. They can produce confusing results if used inappropriately.
{quote}

== Noncompliant Code Example

----
void fun ( String... strings )	// Noncompliant
{
  // ...
}
----
