Most COBOL environments do not support recursive <code>PERFORM</code> calls, since they can cause unpredictable results. This rule raises an issue when recursive <code>PERFORM</code> calls are used. 

== Noncompliant Code Example

----
PARAGRAPH1.
  PERFORM PARAGRAPH2.

PARAGRAPH2.
  PERFORM PARAGRAPH3.

PARAGRAPH3.
  PERFORM PARAGRAPH1.
----

== Compliant Solution

----
PARAGRAPH1.
  PERFORM PARAGRAPH2.

PARAGRAPH2.
  PERFORM PARAGRAPH3.

PARAGRAPH3.
  DISPLAY "THIS IS PARAGRAPH3".
----
