Explicitly specifying a <code>void</code> parameter list is required in C, but optional in C++. Using <code>void</code> for a parameter-less function decreases its readability. The at-a-glance impression is that the function _does_ take a parameter, and it takes a second look to ascertain that it does not. Therefore the more compact notation is preferred.


== Noncompliant Code Example

----
int fun(void);

int fun(void) {
  ...
}
----


== Compliant Solution

----
int fun();

int fun() {
  ...
}
----


