<p>When multiple tables are involved in a query, using table aliases helps to make it more understandable and keeps it short.</p>
<h2>Noncompliant Code Example</h2>
<pre>
BEGIN
  SELECT
    name,
    firstname,
    location
  INTO employeesArray
  FROM employee -- Noncompliant - should be aliased
  INNER JOIN department -- Noncompliant - should be aliased
  ON employee.DepartmentID = department.ID;
END;
/
</pre>
<h2>Compliant Solution</h2>
<pre>
BEGIN
  SELECT
    empl.name,
    empl.firstname,
    dpt.location
  INTO employeesArray
  FROM employee empl
  INNER JOIN department dpt
  ON empl.DepartmentID = dpt.ID;
END;
/
</pre>
