<p>This rule is a more precise version of S1308 preventing the use of &lt;code&gt;GO TO&lt;/code&gt;. The flow of a program is already complicated to understand with simple &lt;code&gt;GO TO&lt;/code&gt;s. Itâ€™s even worse when a &lt;code&gt;GO TO&lt;/code&gt; is executed conditionally like this is the case with &lt;code&gt;GO TO DEPENDING ON&lt;/code&gt;.</p>
<h2>Noncompliant Code Example</h2>
<pre>
PROCEDURE DIVISION.
   ...
   GO TO PARAGRAPH-10
         PARAGRAPH-20
         PARAGRAPH-30
   DEPENDING ON WS-PARA-NUMBER *&gt; Noncompliant
   ...
</pre>
<h2>Compliant Solution</h2>
<pre>
PROCEDURE DIVISION.
   ...
   EVALUATE WS-PARA-NUMBER
      WHEN 1
         PERFORM PARAGRAPH-10
      WHEN 2
         PERFORM PARAGRAPH-20
      WHEN 3
         PERFORM PARAGRAPH-30
      WHEN OTHER
         PERFORM PARAGRAPH-99
   END-EVALUATE
   ...
</pre>
