<p>Testing &lt;code&gt;%PARMS&lt;/code&gt; before the use of one or more parameters is a clear indication in the code that at least some parameters are treated as optional. However, failure to actually mark those parameters as such with &lt;code&gt;options( *nopass )&lt;/code&gt; denies callers the ability to treat them as optional. Since all parameters after the first one explicitly declared &lt;code&gt;options( *nopass )&lt;/code&gt; are also implicitly optional, this marking is only required once. However, for clarity you should consider explicitly marking all optional parameters.</p>
<h2>Noncompliant Code Example</h2>
<pre>
       DCL-PR VIOLATION1 EXTPGM;  // Noncompliant
         LASTNAME Char(10);
         FIRSTNAME Char(10);
       END-PR;

       DCL-PI VIOLATION1;
         LASTNAME Char(10);
         FIRSTNAME Char(10);
       END-PI;

       dsply FIRSTNAME;
       IF %PARMS &gt; 1;
         dsply LASTNAME;
       EndIf;
       *inlr = *on;
</pre>
<h2>Compliant Solution</h2>
<pre>
       DCL-PR VIOLATION1 EXTPGM;
         LASTNAME Char(10);
         FIRSTNAME Char(10)  options( *nopass );
       END-PR;

       DCL-PI VIOLATION1;
         LASTNAME Char(10);
         FIRSTNAME Char(10)  options( *nopass );
       END-PI;

       dsply FIRSTNAME;
       IF %PARMS &gt; 1;
         dsply LASTNAME;
       EndIf;
       *inlr = *on;
</pre>
