<p>Having two clauses in a &lt;code&gt;when&lt;/code&gt; statement or two branches in an &lt;code&gt;if&lt;/code&gt; chain with the same implementation is at best duplicate code, and at worst a coding error. If the same logic is truly needed for both instances, then in an &lt;code&gt;if&lt;/code&gt; chain they should be combined, or for a &lt;code&gt;when&lt;/code&gt;, duplicates should be refactored.</p>
<h2>Noncompliant Code Example</h2>
<pre>
fun s1871(x: Int) {
    when (x) {
        1 -&gt; {
            val y = x / 2
            print(y)
        }
        2 -&gt; {
            val y = x / 2
            print(y)
        }
    }
}
</pre>
<h2>Exceptions</h2>
<p>Blocks in an &lt;code&gt;if&lt;/code&gt; chain that contain a single line of code are ignored, as are blocks in a &lt;code&gt;when&lt;/code&gt; statement that contain a single line of code with or without a following &lt;code&gt;break&lt;/code&gt;.</p>
<pre>
if (a == 1) {
    doSomething()  //no issue, usually this is done on purpose to increase the readability
} else if (a == 2) {
    doSomethingElse()
} else {
    doSomething()
}
</pre>
<p>But this exception does not apply to &lt;code&gt;if&lt;/code&gt; chains without &lt;code&gt;else&lt;/code&gt;-s, or to &lt;code&gt;when&lt;/code&gt;-es without &lt;code&gt;else&lt;/code&gt; clauses when all branches have the same single line of code. In case of &lt;code&gt;if&lt;/code&gt; chains with &lt;code&gt;else&lt;/code&gt;-s, or of &lt;code&gt;when&lt;/code&gt;-es with default clauses, rule S3923 raises a bug.</p>
<pre>
if (a == 1) {
  doSomething()  //Noncompliant, this might have been done on purpose but probably not
} else if (a == 2) {
  doSomething()
}
</pre>
