<p>Having two branches in the same &lt;code&gt;if&lt;/code&gt; structure  with the same implementation is at best duplicate code, and at worst a coding error. If the same logic is truly needed for both instances, then they should be combined.</p>
<h2>Noncompliant Code Example</h2>
<pre>
if 0 &lt;= a &lt; 10:
    do_first()
    do_second()
elif 10 &lt;= a &lt; 20:
    do_the_other_thing()
elif 20 &lt;= a &lt; 50:
    do_first()         # Noncompliant; duplicates first condition
    do_second()
</pre>
<h2>Exceptions</h2>
<p>Blocks in an &lt;code&gt;if&lt;/code&gt; chain that contain a single line of code are ignored.</p>
<pre>
if 0 &lt;= a &lt; 10:
    do_first()
elif 10 &lt;= a &lt; 20:
    do_the_other_thing()
elif 20 &lt;= a &lt; 50:
    do_first()         # no issue, usually this is done on purpose to increase the readability
</pre>
<p>But this exception does not apply to &lt;code&gt;if&lt;/code&gt; chains without &lt;code&gt;else&lt;/code&gt;-s when all branches have the same single line of code. In case of &lt;code&gt;if&lt;/code&gt; chains with &lt;code&gt;else&lt;/code&gt;-s rule S3923 raises a bug.</p>
<pre>
if 0 &lt;= a &lt; 10:
    do_first()
elif 20 &lt;= a &lt; 50:
    do_first()         # Noncompliant, this might have been done on purpose but probably not
</pre>
