<p>&lt;code&gt;&lt;stdlib.h&gt;&lt;/code&gt;'s &lt;code&gt;abort&lt;/code&gt;, &lt;code&gt;exit&lt;/code&gt;, &lt;code&gt;getenv&lt;/code&gt;, and &lt;code&gt;system&lt;/code&gt; have implementation-defined behaviors, and should therefore be avoided.</p>
<h2>Noncompliant Code Example</h2>
<pre>
#include &lt;stdlib.h&gt;

void f( ) {
  exit(0); // Noncompliant
}
</pre>
<h2>See</h2>
<ul>
<li>MISRA C:2004, 20.11 - The library functions abort, exit, getenv and system from library &lt;stdlib.h&gt; shall not be used.</li>
<li>MISRA C++:2008, 18-0-3 - The library functions abort, exit, getenv and system from library &lt;cstdlib&gt; shall not be used.</li>
<li>MISRA C:2012, 21.8 - The library functions abort, exit, getenv and system of &lt;stdlib.h&gt; shall not be used</li>
</ul>
