Any extensible class might have subclasses located in a different package. When that happens, the use of <code>this.getClass().getResource</code> with a relative path would mean that the resource isn't found for the child class.

Instead, use an absolute path or make the class <code>final</code>.


== Noncompliant Code Example

----
public class MyClass {

private URL url = null;

  public MyClass(){
    this.url = this.getClass().getResource("file.txt");  // Noncompliant
  }
----

