<p>The Java Persistence API specification imposes only a conditional requirement that &lt;code&gt;@Entity&lt;/code&gt; classes be &lt;code&gt;Serializable&lt;/code&gt;:
bq. If an entity instance is to be passed by value as a detached object (e.g., through a remote interface), the entity class must implement the &lt;code&gt;Serializable&lt;/code&gt; interface.</p>
<p>But itâ€™s best practice to make all such classes &lt;code&gt;Serializable&lt;/code&gt; from the start. So this rule raises an issue when an &lt;code&gt;@Entity&lt;/code&gt; does not implement &lt;code&gt;Serializable&lt;/code&gt;.</p>
<h2>Noncompliant Code Example</h2>
<pre>
@Entity
pubic class Person {  // Noncompliant
  private String fname;
  // ...
</pre>
<h2>Compliant Solution</h2>
<pre>
@Entity
pubic class Person implements Serializable {
  private String fname;
  // ...
</pre>
