The Java Persistence API specification imposes only a conditional requirement that <code>@Entity</code> classes be <code>Serializable</code>:
bq. If an entity instance is to be passed by value as a detached object (e.g., through a remote interface), the entity class must implement the <code>Serializable</code> interface.

But it's best practice to make all such classes <code>Serializable</code> from the start. So this rule raises an issue when an <code>@Entity</code> does not implement <code>Serializable</code>.


== Noncompliant Code Example

----
@Entity
pubic class Person {  // Noncompliant
  private String fname;
  // ...
----


== Compliant Solution

----
@Entity
pubic class Person implements Serializable {
  private String fname;
  // ...
----


