<p>Shared coding conventions allow teams to collaborate efficiently. This rule checks that all constant names match a provided regular expression.</p>
<h2>Noncompliant Code Example</h2>
<p>With the default regular expression &lt;code&gt;^[A-Z]<strong>(_[A-Z0-9]+)</strong>$&lt;/code&gt;:</p>
<pre>
define("const1", true);

class Foo {
    const const2 = "bar";
}
</pre>
<h2>Compliant Solution</h2>
<pre>
define("CONST1", true);

class Foo {
    const CONST2 = "bar";
}
</pre>
