<p>Shared coding conventions allow teams to collaborate efficiently. This rule checks that all constant names match a provided regular expression.</p>
<h2>Noncompliant Code Example</h2>
<p>With the default regular expression &lt;code&gt;[a-zA-Z]([a-zA-Z0-9_]*[a-zA-Z0-9])?&lt;/code&gt;:</p>
<pre>
DECLARE
  constant_ CONSTANT PLS_INTEGER := 42; -- Noncompliant
BEGIN
  NULL;
END;
/
</pre>
<h2>Compliant Solution</h2>
<pre>
DECLARE
  constant CONSTANT PLS_INTEGER := 42; -- Compliant
BEGIN
  NULL;
END;
/
</pre>
