<p>Shared coding conventions allow teams to collaborate efficiently. This rule checks that all constant names match a provided regular expression.</p>
<h2>Noncompliant Code Example</h2>
<p>With the default regular expression &lt;code&gt;^[a-z][a-zA-Z0-9]*$&lt;/code&gt;:</p>
<pre>
let Pi = 3.14
</pre>
<h2>Compliant Solution</h2>
<pre>
let pi = 3.14
</pre>
