include::../description.adoc[]

== Noncompliant Code Example

With the default threshold value of 3:

----
if ((condition1 && condition2) || (condition3 && condition4)) && condition5
  ...
end
----

== Compliant Solution

----
if (myFirstCondition() || mySecondCondition()) && myLastCondition()
  ...
end
----
