When a <code>SELECT</code> returns <code>null</code> from a nullable column, the relevant host variable isn't updated; it simply retains its previous value. The only way you'll ever know the column value was <code>null</code> is to check the relevant null indicator included in the <code>SELECT</code> for a negative (<code>null</code>) value.

This rule raises an issue when a <code>SELECT</code> omits a <code>null</code> indicator for a nullable column. 

*Note* that this rule raises issues only when a database catalog is provided during the SonarQube analysis.


== Noncompliant Code Example

With the table <code>PRODUCT</code> having a nullable column <code>NAME</code>:
----
EXEC SQL
  SELECT 
    PROD_ID, 
    NAME
  INTO  
    :P-ID,
    :P-NAME                 -- Noncompliant; No null indicator
  FROM PRODUCT
END-EXEC
----


== Compliant Solution

----
EXEC SQL
  SELECT 
    PROD_ID, 
    NAME
  INTO  
    :P-ID,
    :P-NAME :P-NAME-NULL  -- Compliant
  FROM PRODUCT
END-EXEC
----

