<p>&lt;code&gt;@ComponentScan&lt;/code&gt; is used to find which Spring &lt;code&gt;@Component&lt;/code&gt; beans (&lt;code&gt;@Service&lt;/code&gt; or &lt;code&gt;@Repository&lt;/code&gt; or &lt;code&gt;Controller&lt;/code&gt;) are available in the classpath so they can be used in the application context. This is a convenient feature especially when you begin a new project but it comes with the drawback of slowing down the application start-up time especially when the application becomes bigger (ie: it references a large JAR file, or it references a significant number of JAR files, or the base-package refers to a large amount of .class files).</p>
<p>&lt;code&gt;@ComponentScan&lt;/code&gt; should be replaced by an explicit list of Spring beans loaded by &lt;code&gt;@Import&lt;/code&gt;.</p>
<p>The interface &lt;code&gt;@SpringBootApplication&lt;/code&gt; is also considered by this rule because it is annotated with &lt;code&gt;@ComponentScan&lt;/code&gt;.</p>
<h2>Noncompliant Code Example</h2>
<pre>
@ComponentScan
public class MyApplication {
...
}

@SpringBootApplication
public class MyApplication {
...
}
</pre>
<h2>Compliant Solution</h2>
<pre>
@Configuration
@Import({
        DispatcherServletAutoConfiguration.class,
        ErrorMvcAutoConfiguration.class,
        HttpEncodingAutoConfiguration.class,
        HttpMessageConvertersAutoConfiguration.class,
        MultipartAutoConfiguration.class,
        ServerPropertiesAutoConfiguration.class,
        PropertyPlaceholderAutoConfiguration.class,
        WebMvcAutoConfiguration.class
})
public class MyApplication {
...
}
</pre>
<h2>See</h2>
<ul>
<li><a href="https://sites.google.com/site/appsdevelopersindia/google-blog/optimizingspringframeworkforappengineapplications">Optimizing Spring Framework for App Engine Applications</a></li>
</ul>
