<p>&lt;code&gt;switch&lt;/code&gt; can contain a &lt;code&gt;default&lt;/code&gt; clause for various reasons: to handle unexpected values, to show that all the cases were properly considered.
For readability purpose, to help a developer to quickly find the default behavior of a &lt;code&gt;switch&lt;/code&gt; statement, it is recommended to put the &lt;code&gt;default&lt;/code&gt; clause at the end of the &lt;code&gt;switch&lt;/code&gt; statement. This rule raises an issue if the &lt;code&gt;default&lt;/code&gt; clause is not the first or the last one of the &lt;code&gt;switch&lt;/code&gt;'s cases.</p>
<h2>Noncompliant Code Example</h2>
<pre>
switch (param) {
  case 0:
    doSomething();
    break;
  default: // default clause should be the first or last one
    error();
    break;
  case 1:
    doSomethingElse();
    break;
}
</pre>
<h2>Compliant Solution</h2>
<pre>
switch (param) {
  case 0:
    doSomething();
    break;
  case 1:
    doSomethingElse();
    break;
  default:
    error();
    break;
}
</pre>
<h2>See</h2>
<ul>
<li>MISRA C:2004, 15.3 - The final clause of a switch statement shall be the default clause</li>
<li>MISRA C++:2008, 6-4-6 - The final clause of a switch statement shall be the default-clause</li>
<li>MISRA C:2012, 16.4 - Every switch statement shall have a default label</li>
<li>MISRA C:2012, 16.5 - A default label shall appear as either the first or the last switch label of a switch statement</li>
</ul>
