<p>&lt;code&gt;switch&lt;/code&gt; can contain a &lt;code&gt;default&lt;/code&gt; clause for various reasons: to handle unexpected values, to show that all the cases were properly considered.
For readability purpose, to help a developer to quickly find the default behavior of a &lt;code&gt;switch&lt;/code&gt; statement, it is recommended to put the &lt;code&gt;default&lt;/code&gt; clause at the end of the &lt;code&gt;switch&lt;/code&gt; statement. This rule raises an issue if the &lt;code&gt;default&lt;/code&gt; clause is not the first or the last one of the &lt;code&gt;switch&lt;/code&gt;'s cases.</p>
<h2>Noncompliant Code Example</h2>
<pre>
switch (param)
{
    case 0:
      DoSomething();
      break;
    default: // default clause should be the first or last one
      Error();
      break;
    case 1:
      DoSomethingElse();
      break;
}
</pre>
<h2>Compliant Solution</h2>
<pre>
switch (param)
{
    case 0:
      DoSomething();
      break;
    case 1:
      DoSomethingElse();
      break;
    default:
      Error();
      break;
}
</pre>
