<p>&lt;code&gt;switch&lt;/code&gt; can contain a &lt;code&gt;default&lt;/code&gt; clause for various reasons: to handle unexpected values, to show that all the cases were properly considered.
For readability purpose, to help a developer to quickly find the default behavior of a &lt;code&gt;switch&lt;/code&gt; statement, it is recommended to put the &lt;code&gt;default&lt;/code&gt; clause at the beginning or the end of the &lt;code&gt;switch&lt;/code&gt; statement. This rule raises an issue if the &lt;code&gt;default&lt;/code&gt; clause is not the first or the last one of the &lt;code&gt;switch&lt;/code&gt;'s cases.</p>
<h2>Noncompliant Code Example</h2>
<pre>
switch tag {
case 0, 1, 2, 3:
	foo()
default:  // Noncompliant; default case should be the first or last one
	qix()
case 4, 5, 6, 7:
	bar()
}
</pre>
<h2>Compliant Solution</h2>
<pre>
switch tag {
default:
	qix() // Compliant; default is the first one
case 0, 1, 2, 3:
	foo()
case 4, 5, 6, 7:
	bar()
}
</pre>
<pre>
switch tag {
case 0, 1, 2, 3:
	foo()
case 4, 5, 6, 7:
	bar()
default:
	qix() // Compliant; default is the last one
}
</pre>
