<p>&lt;code&gt;SELECT&lt;/code&gt; can contain an &lt;code&gt;OTHERWISE&lt;/code&gt; clause for various reasons: to handle unexpected values, to show that all the cases were properly considered.
For readability purpose, to help a developer to quickly find the default behavior of a &lt;code&gt;SELECT&lt;/code&gt; statement, it is recommended to put the &lt;code&gt;OTHERWISE&lt;/code&gt; clause at the end of the &lt;code&gt;SELECT&lt;/code&gt; statement. This rule raises an issue if the &lt;code&gt;OTHERWISE&lt;/code&gt; clause is not the first or the last one of the &lt;code&gt;SELECT&lt;/code&gt;'s cases.</p>
<h2>Noncompliant Code Example</h2>
<pre>
foo: proc options(main);
  declare i fixed decimal init (42);

  select (i);
    when (0) put list ('i = 0');
    otherwise put list ('i = 2'); /* Non-Compliant */
    when (1) put list ('i = 1');
  end;
end;
</pre>
<h2>Compliant Solution</h2>
<pre>
foo: proc options(main);
  declare i fixed decimal init (42);

  select (i); /* Compliant */
    when (0) put list ('i = 0');
    when (1) put list ('i = 1');
    otherwise put list ('i = 2');
  end;
end;
</pre>
