Unsigned right shift (<code>>>></code>) fills from the left with 0's, versus signed right shift (<code>>></code>), which fills with the sign bit. There's no point in using <code>>>></code> unless you care about the upper bits of the result, so casting a <code>>>></code>-ed value to a smaller type (which discards the upper bits) is likely a mistake. Either <code>>></code> should have been used, or the cast was made in error.


== Noncompliant Code Example

----
int i = -1;
short s = (short)i >>> 4; /// Noncompliant
----


== See

* https://www.securecoding.cert.org/confluence/x/SAHEAw[CERT, NUM52-J.] - Be aware of numeric promotion behavior 

