Making a public constant just <code>const</code> as opposed to <code>static const</code> leads to duplicating its value for every instance of the class, uselessly increasing the amount of memory required to execute the application.

== Noncompliant Code Example

----
public class Myclass 
{
  public const THRESHOLD:int = 3;   
}
----

== Compliant Solution

----
public class Myclass 
{
  public static const THRESHOLD:int = 3;   
}
----
