This rule applies whenever an <code>if</code> statement is followed by one or more <code>elsif</code> statements; the final <code>elsif</code> should be followed by an <code>else</code> statement.

The requirement for a final <code>else</code> statement is defensive programming.
The <code>else</code> statement should either take appropriate action or contain a suitable comment as to why no action is taken. This is consistent with the requirement to have a final <code>else</code> clause in a <code>case</code> statement.

== Noncompliant Code Example

----
if x == 0                                      
  doSomething
elsif x == 1
  doSomethingElse
end
----

== Compliant Solution

----
if x == 0                                      
  doSomething
elsif x == 1
  doSomethingElse
else
  raise 'An error has occured'
end
----

include::../see.adoc[]
