include::../description.adoc[]

== Noncompliant Code Example

----
if (true) 
{  
  DoSomething(); 
}
...
if (false) 
{
  DoSomethingElse(); 
}
----

== Compliant Solution

----
DoSomething(); 
...
----

include::../see.adoc[]
