Classes with only <code>private</code> constructors should be marked <code>final</code> to prevent any mistaken extension attempts.


== Noncompliant Code Example

----
public class PrivateConstructorClass {  // Noncompliant
  private PrivateConstructorClass() {
    // ...
  }

  public static int magic(){
    return 42;
  }
}
----


== Compliant Solution

----
public final class PrivateConstructorClass {  // Compliant
  private PrivateConstructorClass() {
    // ...
  }

  public static int magic(){
    return 42;
  }
}
----

