include::../description.adoc[]

== Noncompliant Code Example

----
Public Sub Foo()
    Thread.CurrentThread.Suspend() ' Noncompliant
    Thread.[Resume]() ' Noncompliant
End Sub
----

include::../see.adoc[]
