include::../description.adoc[]

== Noncompliant Code Example

----
FUNCTION f.
  DATA: LOCAL_1 LIKE BAR.
  DATA: LOCAL_2 LIKE BAR. "Noncompliant

  SELECT * FROM LOCAL_1.

ENDFUNCTION.
----

== Compliant Solution

----
FUNCTION f.
  DATA: LOCAL_1 LIKE BAR.

  SELECT * FROM LOCAL_1.

ENDFUNCTION.
----
