In a Zen-like manner, "NULL" is never equal to anything, even itself. Therefore comparisons using equality operators will always return <code>False</code>, even when the value actually <code>IS NULL</code>.

For that reason, comparison operators should never be used to make comparisons with <code>NULL</code>; <code>IS NULL</code> and <code>IS NOT NULL</code> should be used instead.

== Noncompliant Code Example

----
UPDATE books
SET title = 'unknown'
WHERE title = NULL -- Noncompliant 
----

== Compliant Solution

----
UPDATE books
SET title = 'unknown'
WHERE title IS NULL
----
