This statement deletes all rows of an internal table itab. This <code>REFRESH</code> statement is deprecated and usage should be avoided.


== Noncompliant Code Example

----
REFRESH itab.
----


== Compliant Solution

----
CLEAR itab.
----

