<p>To perform calculations when specifying a CSS property &lt;code&gt;calc()&lt;/code&gt; function can be used. This function takes single expression as parameter. When writing this expression some rules must be respected:</p>
<ul>
<li>no empty &lt;code&gt;calc()&lt;/code&gt;</li>
<li>there should be an operator between the arguments, spacing should be respected</li>
<li>there should not be any division by zero</li>
<li>the resolved type should be valid for where the expression is placed</li>
</ul>
<p>Otherwise &lt;code&gt;calc()&lt;/code&gt; function will be invalid and the entire rule using it will be ignored.</p>
<h2>Noncompliant Code Example</h2>
<pre>
.btn {
  border: solid black 1px;
  width: calc(100% 80px);  /* Noncompliant */
}
</pre>
<h2>Compliant Solution</h2>
<pre>
.btn {
  border: solid black 1px;
  width: calc(100% - 80px);
}
</pre>
