The use of C headers and therefore C functions in a C++ program, is sometimes necessary, but should be avoided in favor of C++ headers and functions.


== Noncompliant Code Example

----
#include <string.h>
----


== Compliant Solution

----
#include <cstring>
#include <string>
----


== See

* MISRA C++:2008, 18-0-1 - The C library shall not be used.

