Surrounding your operators with whitespace in operator declarations will help maintainers derive meaning from what might otherwise look like a meaningless jumble of punctuation.


== Noncompliant Code Example

----
func <*>(a: MyClass, b: MyClass) -> Boolean { // Noncompliant
----


== Compliant Solution

----
func <*> (a: MyClass, b: MyClass) -> Boolean {
----

