For fixed-length values, a <code>CHAR</code> field occupies the same amount of disk space as a <code>VARCHAR2</code> field, but for variable-length values <code>CHAR</code> fields use more storage space and make searching more difficult by right-padding values with whitespaces. Therefore <code>VARCHAR2</code> fields are preferred. Similarly, <code>NCHAR</code> should be replaced by <code>NVARCHAR2</code>.

Note that for 1-character fields, <code>CHAR</code> is naturally equivalent to <code>VARCHAR2</code>, but the latter is still preferred for consistency.


== Noncompliant Code Example

----
DECLARE
  var1 CHAR; -- Noncompliant

  var2 CHAR(42); -- Noncompliant

  var3 NCHAR; -- Noncompliant

  var4 NCHAR(42); -- Noncompliant
BEGIN
  NULL;
END;
/
----


== Compliant Solution

{code}DECLARE
  var1 VARCHAR2(42); 

  var2 VARCHAR2(42);

  var3 NVARCHAR2(42); 

  var4 NVARCHAR2(42);
BEGIN
  NULL;
END;
/

----
----

