<p>According to the Java Language Specification:</p>
<p>{quote}
It is permitted, but discouraged as a matter of style, to redundantly specify the &lt;code&gt;public&lt;/code&gt; and/or &lt;code&gt;abstract&lt;/code&gt; modifier for a method declared in an interface.
{quote}</p>
<h2>Noncompliant Code Example</h2>
<pre>
public interface Task{
  public abstract void execute();
}
</pre>
<h2>Compliant Solution</h2>
<pre>
public interface Task{
  void execute();
}
</pre>
