According to the Java Language Specification:

{quote}
It is permitted, but discouraged as a matter of style, to redundantly specify the <code>public</code> and/or <code>abstract</code> modifier for a method declared in an interface.
{quote}


== Noncompliant Code Example

----
public interface Task{
  public abstract void execute();
}
----


== Compliant Solution

----
public interface Task{
  void execute();
}
----

