include::../description.adoc[]

== Noncompliant Code Example

----
function setName(name) {
    name = name;
}
----

== Compliant Solution

----
function setName(name) {
    this.name = name;
}
----

include::../see.adoc[]
