The requirement for a final <code>ELSE</code> clause is defensive programming. The clause should either take appropriate action, or contain a suitable comment as to why no action is taken.

== Noncompliant Code Example

----
SELECT
  CASE category
    WHEN 'A' THEN 21
    WHEN 'B' THEN 33
END shipping_cost
FROM product
----

== Compliant Solution

----
SELECT
  CASE category
    WHEN 'A' THEN 21
    WHEN 'B' THEN 33
    ELSE 42
END shipping_cost
FROM product
----

include::../see.adoc[]
