<p>&lt;code&gt;SYNCHRONIZE&lt;/code&gt; was introduced by Oracle as a recovery mechanism in case there is a loss of synchronization between what is in memory and what is displayed. It works but it comes with a price; each call to &lt;code&gt;SYNCHRONIZE&lt;/code&gt; generates a network round-trip between the server and the Forms client. Most of the time it should be avoided or used with caution as explicitly stated in the Oracle Forms documentation.</p>
<h2>Noncompliant Code Example</h2>
<pre>
SYNCHRONIZE;
</pre>
