Classes without <code>public static</code> members cannot be used without being instantiated, but classes with only <code>private</code> constructors cannot be instantiated. When a class has only <code>private</code> constructors and no <code>static</code> members, it is useless and should be removed or refactored.


== Noncompliant Code Example

----
public class MyClass {  // Noncompliant

  private MyClass () {
  }

  public double getPi(){
    return 3.14;
  }
}
----


== Compliant Solution

----
public class MyClass {  

  public MyClass () {
  }

  public double getPi(){
    return 3.14;
  }
}
----
or

----
public class MyClass {  

  private MyClass () {
  }

  public static double getPi(){
    return 3.14;
  }
}
----


