include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression: <code>^[a-z][a-zA-Z0-9]*$</code>:

----
void DoSomething (void);
----

== Compliant Solution

----
void doSomething (void);
----
