The deprecated MLSLABEL datatype is still supported only for backwards compatibility with Trusted Oracle, and since Oracle8, the only valid value it can hold is NULL. Thus, the usage of this type should be progressively removed.


== Noncompliant Code Example

----
DECLARE
  foo MLSLABEL; -- Noncompliant
BEGIN
  NULL;
END;
/
----

