== Noncompliant Code Example

----
public void doSomething(int max) {
  for (int i = 0; i < max; i++) {
    doSynchronized(i);  // Noncompliant
  }
}

public synchronized void doSynchronized(int val) {
  // ...
}
----