<code>CAST(... AS CHAR/VARCHAR)</code> can be a source of incompatibilities between database versions: the behavior of <code>CAST</code> may not be the same depending on the version of the database system. Such incompatibilities can cause unexpected output from applications that <code>CAST</code> decimal data to <code>CHAR</code> or <code>VARCHAR</code>, it's therefore best to avoid using <code>CAST(... AS CHAR/VARCHAR)</code>.


== Noncompliant Code Example

----
  DELETE product
  WHERE CAST(status_code AS CHAR(2)) = '42' -- Noncompliant
----


