include::../description.adoc[]

== Noncompliant Code Example

----
Dim o As Object = 5
Dim x As Integer = 5

x = o >> 5 ' Noncompliant
x = x >> o ' Noncompliant
----

== Exceptions

This rule does not raise when the left or the right expression is <code>Nothing</code>.

----
x = Nothing >> 5
x = 5 >> Nothing
----
