Looking for a given substring starting from a specified offset can be achieved by such code: <code>str.Substring(startIndex).IndexOf(char1)</code>. This works well, but it creates a new <code>string</code> for each call to the <code>Substring</code> method. When this is done in a loop, a lot of <code>strings</code> are created for nothing, which can lead to performance problems if <code>str</code> is large.

To avoid performance problems, <code>string.Substring(startIndex)</code> should not be chained with the following methods:
- <code>IndexOf</code>
- <code>IndexOfAny</code>
- <code>LastIndexOf</code>
- <code>LastIndexOfAny</code>

For each of these methods, another method with an additional parameter is available to specify an offset.
Using these methods gives the same result while avoiding the creation of additional <code>String</code> instances.

== Noncompliant Code Example

----
str.Substring(StartIndex).IndexOf(char1); // Noncompliant; a new string is going to be created by "Substring"
----

== Compliant Solution

----
str.IndexOf(char1, startIndex);
----
