include::../description.adoc[]

== Noncompliant Code Example

----
func divide(numerator:Int, denominator:Int) -> Int {
  return numerator / denominator              // FIXME denominator value might be  0
}
----

include::../see.adoc[]
