include::../description.adoc[]

== Noncompliant Code Example

----
if (condition)  // Noncompliant
  executeSomething()
----

== Compliant Solution

----
if (condition) {
  executeSomething()
}
----

include::../see.adoc[]
