<p>This rule allows banning certain PHP functions.</p>
<h2>Noncompliant Code Example</h2>
<p>Given a parameter of &lt;code&gt;foo&lt;/code&gt; for this rule:</p>
<pre>
$arr=array("apple", "pear","banana");
echo foo($arr);  // Noncompliant
</pre>
