== Noncompliant Code Example

----
// Header.hpp
namespace                  // Noncompliant
{
  extern int32_t x;
}
----

----
// File1.cpp
#include "Header.cpp"

namespace
{
  int32_t x;
}

void fn_a(void)
{
  x = 42;
}
----

----
// File2.cpp
#include "Header.cpp"

namespace
{
  int32_t x;  // this is a different x than in File1.cpp
}

void fn_b(void)
{
  fn_a();                  // Is expected to initialize "x" to 42
  if (x == 42)             // But does not, as there are 2 distinct "x" variables
  {
  }
}
----
