<p>Deprecated system tables and views are those that have been retained temporarily for backward compatibility, but which will eventually be removed from the language. In effect, deprecation announces a grace period to allow the smooth transition from the old features to the new ones.</p>
<p>This rule raises an issue when system tables or views are used. Catalog tables and views should be used instead.</p>
<h2>Noncompliant Code Example</h2>
<pre>
SELECT name FROM syscolumns -- Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
SELECT name FROM sys.columns
</pre>
