<p>Using unvalidated XML can expose an application to several types of XML-related attacks:</p>
<ul>
<li>XML injection - the structure and/or contents of the document are changed by the addition of extra elements</li>
<li>XML entity expansion - the size of the document is ballooned by injecting elements that expand exponentially, exhausing server resources</li>
<li>XML external entity injection - entities are injected which result dynamically including data from another source, such as a sensitive file on the filesystem.</li>
</ul>
<p>This rule checks that method parameters are not unconditionally used directly in XML.</p>
<h2>Noncompliant Code Example</h2>
<pre>
private string writeXml(OleDbDataReader odr) {

  XmlTextWriter xTWriter = null;
  string key = String.Empty;
  string value = String.Empty;

  try
  {
    while(oDR.Read())
    {
      key = odr.GetValue(0);
      value = odr.GetValue(1);
      xTWriter.WriteElementString(key, value);  // Noncompliant
    }
  }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
private string writeXml(OleDbDataReader odr) {

  XmlTextWriter xTWriter = null;
  string key = String.Empty;
  string value = String.Empty;

  try
  {
    while(oDR.Read())
    {
      key = System.SecurityElement.Escape(odr.GetValue(0));
      value = System.SecurityElement.Escape(odr.GetValue(1));
      xTWriter.WriteElementString(key, value);
    }
  }
}
</pre>
<h2>See</h2>
<ul>
<li><a href="https://www.owasp.org/index.php/Top_10-2017_A1-Injection">OWASP Top 10 2017 Category A1</a> - Injection</li>
<li><a href="https://www.owasp.org/index.php/Top_10-2017_A4-XML_External_Entities_(XXE)">OWASP Top 10 2017 Category A4</a> - XML External Entities (XXE)</li>
<li><a href="https://www.securecoding.cert.org/confluence/x/EwAZAg">CERT, IDS51-J.</a> - Properly encode or escape output</li>
</ul>
