This rule raised an issue when a component is used in the <code>render()</code> method and not <code>import</code>ed or defined. 


== Noncompliant Code Example

----
<Rating value={this.props.measure.reliability_rating} />;
----


== Compliant Solution

----
import Rating from './Rating'
[...]
<Rating value={this.props.measure.reliability_rating} />
----

