<p>Salesforce provides a library of Aggregate functions like COUNT() which efficiently process the data records in the database and return the count of records which meet the SOQL Query criteria. Using size() function instead of COUNT() involves more processing and is less efficient.</p>
<p>This rule raises an issue when the result of an SOQL query is only used to call &lt;code&gt;size()&lt;/code&gt; on it.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public class fooClass{
    public static fooMethod {
        integer i = [SELECT Id FROM Case].size();  // Noncompliant
    }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
public class fooClass{
    public static fooMethod {
        integer i = [SELECT COUNT() FROM Case];
    }
}
</pre>
<h2>See</h2>
<ul>
<li><a href="https://developer.salesforce.com/docs/atlas.en-us.soql_sosl.meta/soql_sosl/sforce_api_calls_soql_select_count.htm">COUNT() and COUNT(fieldName)</a></li>
</ul>
