include::../description.adoc[]

== Noncompliant Code Example

----
class MyException: Exception
{
    public void MyException()
    {
         if (bad_thing) 
         {
             throw new Exception("A bad thing happened");  // Noncompliant
          }
    }
}
----
