<p>In property and indexer &lt;code&gt;set&lt;/code&gt; methods, and in event &lt;code&gt;add&lt;/code&gt; and &lt;code&gt;remove&lt;/code&gt; methods, the implicit &lt;code&gt;value&lt;/code&gt; parameter holds the value the accessor was called with. Not using the &lt;code&gt;value&lt;/code&gt; means that the accessor ignores the caller’s intent which could cause unexpected results at runtime.</p>
<h2>Noncompliant Code Example</h2>
<pre>
private int count;
public int Count
{
  get { return count; }
  set { count = 42; } // Noncompliant
}
</pre>
<h2>Compliant Solution</h2>
<pre>
private int count;
public int Count
{
  get { return count; }
  set { count = value; }
}
</pre>
<p>or</p>
<pre>
public int Count
{
  get { return count; }
  set { throw new InvalidOperationException(); }
}
</pre>
<h2>Exceptions</h2>
<p>This rule doesn’t raise an issue when the setter is empty and part of the implementation of an &lt;code&gt;interface&lt;/code&gt; . The assumption is that this part of the interface is not meaningful to that particular implementation. A good example of that would be a "sink" logger that discards any logs.</p>
