:source-highlighter: prettify
:source-language: c#
:source-indent: 0

Because `printf` format strings are interpreted at runtime, rather than validated by the compiler, they can contain errors that result in the wrong strings being created. This rule statically validates the correlation of `printf` format strings to their arguments.

// TODO add a MACRO to replace reference to rule S2275
The related rule S2275 is about errors that will create undefined behavior, while this rule is about errors that produce an unexpected string.

== Exceptions
This rule will only work if the format string is provided as a string literal.

== Noncompliant Code Example
----
include::noncompliant.c[lines=4..5]
----

== Compliant Solution
----
include::compliant.c[lines=4..5]
----

== See
include::../see.adoc[]