<p>Any statement after a &lt;code&gt;STOP RUN&lt;/code&gt; or &lt;code&gt;GOBACK&lt;/code&gt; is unreachable and therefore dead code which should be removed.</p>
<h2>Noncompliant Code Example</h2>
<pre>
PARAGRAPH1.
  MOVE A TO B.
  STOP RUN.
  MOVE B TO C.
</pre>
<p>or</p>
<pre>
PARAGRAPH1.
  MOVE A TO B.
  GOBACK.
  MOVE B TO C.
</pre>
<h2>Compliant Solution</h2>
<pre>
PARAGRAPH1.
  MOVE A TO B.
  MOVE B TO C.
  STOP RUN.
</pre>
<p>or</p>
<pre>
PARAGRAPH1.
  MOVE A TO B.
  MOVE B TO C.
  GOBACK.
</pre>
