Shared naming conventions allow teams to collaborate efficiently. This rule checks that all attribute names match a provided regular expression.


== Noncompliant Code Example

With the default regular expression:

----
<xs:element name="fruit">
  ...
  <xs:attribute name="FLESH_COLOR" type="xs:string">     <!-- Noncompliant -->
</xs:element>
----


== Compliant Solution

----
<xs:element name="fruit">
  ...
  <xs:attribute name="fleshColor" type="xs:string">
</xs:element>
----

