Having all branches in a <code>CASE</code> or <code>IF</code> chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be a <code>CASE</code>/<code>IF</code> chain at all.

== Noncompliant Code Example

----
IF a >0.
    doSomething.
ELSE IF b> 0.
    doSomething.
ELSE.
    doSomething.
ENDIF.

CASE i.
  WHEN 1 OR 3.
    doSomething.
  WHEN 2.
    doSomething.
  WHEN OTHERS.
        doSomething.
ENDCASE.
----

== Exceptions

Note that this rule does not apply to <code>IF</code> chains without final <code>ELSE</code>-s, or to <code>CASE</code>-s without <code>WHEN OTHERS</code> clauses.

----
IF a >0.   "no issue, this could have been done on purpose to make the code more readable
    doSomething.
ELSEIF b> 0.
    doSomething.
ENDIF.
----
