

== Noncompliant Code Example

----
SELECT * 
       FROM persons 
       INTO newyorkers 
       WHERE city = 'NEW YORK'
----

== Compliant Solution

----
SELECT firstname, lastname 
       FROM persons 
       INTO newyorkers
       WHERE city = 'NEW YORK'
----
