Shared coding conventions allow teams to collaborate effectively. While it is possible to omit the specification letter before a <code>/COPY</code> statement, it is advisable not to do so because including a specification on the line indicates to other developers what type of statements will be added to the program by the copybook. Further, using a specification letter on the line makes it clearer to those who might be skimming the file that the line is not just a comment, but does contain code.


== Noncompliant Code Example

----
      /COPY MBR1
----


== Compliant Solution

----
     I/COPY MBR1
----

