<p>By convention, constructor function names should start with an upper case letter as a reminder that they should be called with the &lt;code&gt;new&lt;/code&gt; keyword.</p>
<p>A function is considered to be a constructor when it sets all of its arguments as object properties, and returns no value.</p>
<h2>Noncompliant Code Example</h2>
<pre>
function person (firstName, middleInitial, lastName) {  // Noncompliant
  this.firstName = firstName;
  this.middleInitial = middleInitial;
  this.lastName = lastName;
}
</pre>
<h2>Compliant Solution</h2>
<pre>
function Person (firstName, middleInitial, lastName) {
  this.firstName = firstName;
  this.middleInitial = middleInitial;
  this.lastName = lastName;
}
</pre>
