include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression <code>@[a-zA-Z0-9_]*</code>:
----
CREATE PROCEDURE proc1
    @@var1 INT -- Noncompliant
AS
BEGIN
    DECLARE @@var2 INT; -- Noncompliant
END
----
