include::../description.adoc[]

== Noncompliant Code Example

With the default threshold value 3

----
  if ((condition1 && condition2) || (condition3 && condition4)) && condition5 { ... }
----

== Compliant Solution

----
  if (myFirstCondition() || mySecondCondition()) && myLastCondition() { ... }
----
