<p>Multi-line comments are more readable when each line is aligned using the "*" character. At most one violation is created for each comment</p>
<h2>Noncompliant Code Example</h2>
<pre>
/*
this line is not aligned and ugly Non-Compliant
no violation is created on this line, even though is it also bad
 */

/* this is Compliant */
</pre>
<h2>Compliant Solution</h2>
<pre>
/*
 * this is much better Compliant
 */

/* this is Compliant */
</pre>
