Multi-line comments are more readable when each line is aligned using the "*" character. At most one violation is created for each comment


== Noncompliant Code Example

----
/*
this line is not aligned and ugly Non-Compliant
no violation is created on this line, even though is it also bad
 */

/* this is Compliant */
----


== Compliant Solution

----
/*
 * this is much better Compliant
 */

/* this is Compliant */
----

