<p>Although unnecessary wonâ€™t change anything to the produced application, removing them:</p>
<ul>
<li>Will help readability and maintenance.</li>
<li>Will help reduce the number of items in the IDE auto-completion list when coding.</li>
<li>May avoid some name collisions.</li>
<li>May improve compilation time because the compiler has fewer namespaces to look-up when it resolves types.</li>
</ul>
<h2>Noncompliant Code Example</h2>
<pre>
Imports System.Collections.Generic // Noncompliant - unnecessary using

Module Module1
    Sub Main(path As String)
        File.ReadAllLines(path);
    End Sub
End Module
</pre>
<h2>Compliant Solution</h2>
<pre>
Imports System.IO

Module Module1
    Sub Main(path As String)
        File.ReadAllLines(path);
    End Sub
End Module
</pre>
