<code>getClass</code> should not be used for synchronization in non-<code>final</code> classes because child classes will synchronize on a different object than the parent or each other, allowing multiple threads into the code block at once, despite the <code>synchronized</code> keyword.

Instead, hard code the name of the class on which to synchronize or make the class <code>final</code>.


== Noncompliant Code Example

----
public class MyClass {
  public void doSomethingSynchronized(){
    synchronized (this.getClass()) {  // Noncompliant
      // ...
    }
  }
----


== Compliant Solution

----
public class MyClass {
  public void doSomethingSynchronized(){
    synchronized (MyClass.class) {
      // ...
    }
  }
----


== See

* https://www.securecoding.cert.org/confluence/x/bwCaAg[CERT, LCK02-J.] - Do not synchronize on the class object returned by getClass()

