If a SQL <code>TABLE</code> is declared but not used in the program, it can be considered dead code and should therefore be removed. This will improve maintainability because developers will not wonder what the variable is used for.


== Noncompliant Code Example

----
EXEC SQL
 DECLARE DSN8B10.DEPT TABLE  -- Noncompliant
   ( ...  ) 
END-EXEC.
----

