<p>Externally described files standardize file access, and result in simpler, easier to read code.</p>
<h2>Noncompliant Code Example</h2>
<pre>
     FEPBLMSL2  IF   F  100        DISK    INFSR(*PSSR)
     F*
     F*
</pre>
<h2>Compliant Solution</h2>
<pre>
     F*
     FIPBDCCP   IF   E           K DISK    INFSR(*PSSR)
     F*

     FEP210     CF   E             WORKSTN INFSR(*PSSR)
     F                                     SFILE(EP21003S:RELN03)


     FEP471R1   O    E             PRINTER INFSR(*PSSR)
     F                                     USROPN
     F                                     INFDS(W1SF01)
</pre>
