<p>The &lt;code&gt;defined&lt;/code&gt; preprocessing directive is used in the context of &lt;code&gt;#if&lt;/code&gt; and &lt;code&gt;#elif&lt;/code&gt; expressions to see whether a given identifier has been defined as a macro. It returns a value of 0 (false) or 1 (true), and has two valid forms, &lt;code&gt;defined IDENTIFIER&lt;/code&gt; and &lt;code&gt;defined ( IDENTIFIER )&lt;/code&gt;. Since it is essentially a macro existence check, it cannot take expressions as arguments.</p>
<p>Note that since
&lt;code&gt;#if defined AN_IDENTIFIER&lt;/code&gt;
is equivalent to
&lt;code&gt;#ifdef AN_IDENTIFIER&lt;/code&gt;
&lt;code&gt;defined&lt;/code&gt; is most useful when there are multiple arguments to check, E.G.
&lt;code&gt;#if defined AAA || defined BBB&lt;/code&gt;</p>
<h2>Noncompliant Code Example</h2>
<pre>
#if defined ( X &gt; Y ) // Noncompliant; expressions not allowed
</pre>
<h2>Compliant Solution</h2>
<pre>
#if defined X &amp;&amp; defined Y &amp;&amp; X &gt; Y
</pre>
<h2>See</h2>
<ul>
<li>MISRA C:2004, 19.14 - The defined preprocessor operator shall only be used in one of the two standard forms.</li>
<li>MISRA C++:2008, 16-1-1 - The defined preprocessor operator shall only be used in one of the two standard forms.</li>
</ul>
