The <code>defined</code> preprocessing directive is used in the context of <code>#if</code> and <code>#elif</code> expressions to see whether a given identifier has been defined as a macro. It returns a value of 0 (false) or 1 (true), and has two valid forms, <code>defined IDENTIFIER</code> and <code>defined ( IDENTIFIER )</code>. Since it is essentially a macro existence check, it cannot take expressions as arguments.

Note that since
<code>#if defined AN_IDENTIFIER</code>
is equivalent to
<code>#ifdef AN_IDENTIFIER</code>
<code>defined</code> is most useful when there are multiple arguments to check, E.G.
<code>#if defined AAA || defined BBB</code>


== Noncompliant Code Example

----
#if defined ( X > Y ) // Noncompliant; expressions not allowed
----


== Compliant Solution

----
#if defined X && defined Y && X > Y
----


== See

* MISRA C:2004, 19.14 - The defined preprocessor operator shall only be used in one of the two standard forms.
* MISRA C++:2008, 16-1-1 - The defined preprocessor operator shall only be used in one of the two standard forms.

