<p>It is well documented that while &lt;code&gt;HttpClient&lt;/code&gt; implements &lt;code&gt;IDisposable&lt;/code&gt;, it is intended to be reused, which means not wrapping it in a &lt;code&gt;using&lt;/code&gt; statement.</p>
<p>As noted in the documentation:
{quote}
HttpClient is intended to be instantiated once and re-used throughout the life of an application. Instantiating an HttpClient class for every request will exhaust the number of sockets available under heavy loads. This will result in SocketException errors.
{quote}</p>
<p>This rule raises an issue when an instance of a &lt;code&gt;HttpClient&lt;/code&gt; is created within a &lt;code&gt;using&lt;/code&gt; statement.</p>
<h2>Noncompliant Code Example</h2>
<pre>
namespace MyLibrary
{
    public class Foo
    {
        public async Task Bar()
        {
            using (var client = new HttpClient()) // Noncompliant
            {
                await client.GetAsync("https://www.sonarqube.org/");
            }
        }
    }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
namespace MyLibrary
{
    public class Foo
    {
        private static HttpClient client = new HttpClient();

        public async Task Bar()
        {
            await client.GetAsync("https://www.sonarqube.org/");
        }
    }
}
</pre>
<h2>See</h2>
<ul>
<li><a href="https://msdn.microsoft.com/en-us/library/system.net.http.httpclient(v=vs.110).aspx?f=255&amp;mspperror=-2147217396#Anchor_5">HttpClient Class</a></li>
<li><a href="https://docs.microsoft.com/en-us/aspnet/web-api/overview/advanced/calling-a-web-api-from-a-net-client">Call a Web API From a .NET Client (C#)</a></li>
<li><a href="https://github.com/mspnp/performance-optimization/blob/master/ImproperInstantiation/docs/LoadTesting.md">Microsoft Patterns &amp; Practices</a></li>
</ul>
