<p>Having duplicated font names doesnâ€™t help to read the font declaration and may be an indicator the author of the line was not sure how to configure it. This rule raises an issue when &lt;code&gt;font&lt;/code&gt; or &lt;code&gt;font-family&lt;/code&gt; properties contain a duplicated font name. This rule ignores &lt;code&gt;$sass&lt;/code&gt;, &lt;code&gt;@less&lt;/code&gt;, and &lt;code&gt;var(--custom-property)&lt;/code&gt; variable syntaxes.</p>
<h2>Noncompliant Code Example</h2>
<pre>
a {
  font-family: 'Georgia', Georgia, serif; /* Noncompliant; 'Georgia' is duplicated */
}
</pre>
<h2>Compliant Solution</h2>
<pre>
a {
  font-family: Georgia, serif;
}
</pre>
