<code>fixed bin(31)</code> type has been optimized to run faster than <code>int</code> especially when used in loops. For this reason, <code>fixed bin(31)</code> should be preferred to <code>int</code>.


== Noncompliant Code Example

----
dcl i int(2) init (0);
do-loop i to 20 by 1; 
...
end-loop;
----


== Compliant Solution

----
dcl i fixed bin(31) init (0);
do-loop i to 20 by 1; 
...
end-loop;
----


