String data types (<code>char</code>, <code>varchar</code>, <code>nchar</code>, <code>nvarchar</code>) default to a size of 1 if no size is specified in the declaration. For <code>char</code> and <code>nchar</code> this is confusing at best, but it is most probably a mistake for <code>varchar</code> and <code>nvarchar</code>.

This rule raises an issue when no size is specified for <code>varchar</code> or <code>nvarchar</code>.


== Noncompliant Code Example

----
DECLARE @myStr varchar;  -- Noncompliant
----


== Compliant Solution

----
DECLARE @myStr varchar(255);
----


