<p>Shared coding conventions allow teams to collaborate effectively. Typing operation codes, keywords, specification codes, compiler directives and built-in functions in upper case makes them more immediately recognizable as such, thereby making the code easier to read.</p>
<h2>Noncompliant Code Example</h2>
<pre>
     c     test12        plist
     c                   parm                    asdf01
     c                   parm                    asdf02
     c                   parm                    @pdq
</pre>
<pre>
      /free
        if x = *blank;
          read a;
        endif;
      /end-free
</pre>
<h2>Compliant Solution</h2>
<pre>
     C     test12        PLIST
     C                   PARM                    asdf01
     C                   PARM                    asdf02
     C                   PARM                    @pdq
</pre>
<pre>
      /free
        IF x = *BLANK;
          READ a;
        ENDIF;
      /end-free
</pre>
