<p>This rule allows banning certain statements.</p>
<h2>Noncompliant Code Example</h2>
<p>Given a parameter of &lt;code&gt;CANCEL&lt;/code&gt; for this rule:</p>
<pre>
DISPLAY "Cancelling action".
CANCEL PROGRAM1. *&gt; Noncompliant
</pre>
