This rule allows banning certain statements. 


== Noncompliant Code Example

Given a parameter of <code>CANCEL</code> for this rule: 

----
DISPLAY "Cancelling action".
CANCEL PROGRAM1. *> Noncompliant
----


