The <code>delete</code> operator can be used to remove a property from any object. Arrays are objects, so the <code>delete</code> operator can be used here too, but if it is, a hole will be left in the array because the indexes/keys won't be shifted to reflect the deletion. 

The proper method for removing an element at a certain index would be:

* <code>Array.prototype.splice</code> - add/remove elements from the array 
* <code>Array.prototype.pop</code> - add/remove elements from the end of the array 
* <code>Array.prototype.shift</code> - add/remove elements from the beginning of the array


== Noncompliant Code Example

----
var myArray = ['a', 'b', 'c', 'd'];

delete myArray[2];  // Noncompliant. myArray => ['a', 'b', undefined, 'd']
console.log(myArray[2]); // expected value was 'd' but output is undefined
----


== Compliant Solution

----
var myArray = ['a', 'b', 'c', 'd'];

// removes 1 element from index 2
removed = myArray.splice(2, 1);  // myArray => ['a', 'b', 'd']
console.log(myArray[2]); // outputs 'd'
----


