Reading a non-existent property on an object always returns <code>undefined</code>. Doing so is usually an error; either in the name of the property or the type of the variable being accessed.

If an attempt is made to access properties of a primitive, the primitive is automatically encased in a primitive-wrapper object for the operation. But being "promoted" to an object doesn't mean that the primitive will actually have properties to access. The wrapper object still won't have the non-existent property and <code>undefined</code> will be returned instead. 

This rule raises an issue when an attempt is made to access properties of a primitive. Thus this rule should only be activated when you don't use monkey patching for standard objects, like <code>Number</code>, <code>Boolean</code> and <code>String</code>. 


== Noncompliant Code Example

----
x = 42;
y = x.length;   // Noncompliant, Number type doesn't have "length" property
----


== Exceptions

The Ember framework introduces a few extensions to <code>String</code>. Since it is a widely used package, the following <code>String</code> properties will not trigger this rule even though they are not built-in:

* <code>camelize</code>
* <code>capitalize</code>
* <code>classify</code>
* <code>dasherize</code>
* <code>decamelize</code>
* <code>fmt</code>
* <code>loc</code>
* <code>underscore</code>
* <code>w</code>

