<p>Regardless of scope, no identifier should be re-used across any files in the system.</p>
<h2>Noncompliant Code Example</h2>
<pre>
struct air_speed
{
  uint16_t speed; /* knots */
} * x;

struct gnd_speed
{
  uint16_t speed; /* mph */ // Not Compliant - speed is in different units
} * y;

x-&gt;speed = y-&gt;speed; // Error: unit conversion required
</pre>
<h2>Compliant Solution</h2>
<pre>
struct air_speed
{
  uint16_t knotSpeed; /* knots */
} * x;

struct gnd_speed
{
  uint16_t mphSpeed; /* mph */ // Compliant, different name
} * y;

x-&gt;knotSpeed = y-&gt;mphSpeed; // An error can be deducted from the name of the fields.
</pre>
<h2>Exceptions</h2>
<p>Where an identifier name is used in a header file, and that header file is included in multiple source files, this rule is not violated. The use of a rigorous naming convention can support the implementation of this rule.</p>
<h2>See</h2>
<ul>
<li>MISRA C:2004, 5.7</li>
</ul>
