In the absence of enclosing <code>BEGIN...END</code> block, the line immediately after a conditional is the one that is conditionally executed. By both convention and good practice, such lines are indented. In the absence of both <code>BEGIN...END</code> block and indentation the intent of the original programmer is entirely unclear and perhaps not actually what is executed. Additionally, such code is highly likely to be confusing to maintainers.

== Noncompliant Code Example

----
IF @condition  -- Noncompliant
EXEC doTheThing;
EXEC doTheOtherThing;
EXEX somethingElseEntirely;

EXEC foo;
----

== Compliant Solution

----
IF @condition
BEGIN
  EXEC doTheThing;
  EXEC doTheOtherThing;
  EXEX somethingElseEntirely;
END; 

EXEC foo;
----
Or

----
IF @condition
  EXEC doTheThing;
EXEC doTheOtherThing;
EXEX somethingElseEntirely;

EXEC foo;
----
