Sharing naming conventions allows teams to collaborate efficiently. This rule checks that type names match a provided regular expression.


== Noncompliant Code Example

With the default regular expression:

----
<xs:complexType name="fruit">    <!-- Noncompliant -->
  ...
</xs:complexType>
----


== Compliant Solution

----
<xs:complexType name="fruitType">
  ...
</xs:complexType>
----

