<p>Sharing naming conventions allows teams to collaborate efficiently. This rule checks that type names match a provided regular expression.</p>
<h2>Noncompliant Code Example</h2>
<p>With the default regular expression:</p>
<pre>
&lt;xs:complexType name="fruit"&gt;    &lt;!-- Noncompliant --&gt;
  ...
&lt;/xs:complexType&gt;
</pre>
<h2>Compliant Solution</h2>
<pre>
&lt;xs:complexType name="fruitType"&gt;
  ...
&lt;/xs:complexType&gt;
</pre>
