Appending <code>String.valueOf()</code> to a <code>String</code> decreases the code readability.
The argument passed to <code>String.valueOf()</code> should be directly appended instead.


== Noncompliant Code Example

----
public void display(int i){
  System.out.println("Output is " + String.valueOf(i));    // Noncompliant
}
----


== Compliant Solution

----
public void display(int i){
  System.out.println("Output is " + i);                    // Compliant
}
----


