<p>Creating temporary primitive wrapper objects only for &lt;code&gt;String&lt;/code&gt; conversion or the use of the &lt;code&gt;compareTo&lt;/code&gt; method is inefficient.
Instead, the static &lt;code&gt;toString()&lt;/code&gt; or &lt;code&gt;compare&lt;/code&gt; method of the primitive wrapper class should be used.</p>
<h2>Noncompliant Code Example</h2>
<pre>
new Integer(myInteger).toString();  // Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
Integer.toString(myInteger);        // Compliant
</pre>
