Creating temporary primitive wrapper objects only for <code>String</code> conversion or the use of the <code>compareTo</code> method is inefficient.
Instead, the static <code>toString()</code> or <code>compare</code> method of the primitive wrapper class should be used.


== Noncompliant Code Example

----
new Integer(myInteger).toString();  // Noncompliant
----


== Compliant Solution

----
Integer.toString(myInteger);        // Compliant
----


