Explicitly declaring a function's return data type makes it easier to use the function correctly. 

== Noncompliant Code Example

----
Function BinarySearch(. . .) 
    [. . .]
End Function
----

== Compliant Solution

----
Function BinarySearch(. . .) As Boolean
    [. . .]
End Function
----

include::../see.adoc[]
