include::../description.adoc[]

== Noncompliant Code Example

----
class Ball {
    String color="red";  // Noncompliant
}
enum A {
  B;
  int a;
}
----

== Compliant Solution

----
class Ball {
    private String color="red";  // Compliant
}
enum A {
  B;
  private int a;
}
----

== Exceptions

Members annotated with Guava's <code>@VisibleForTesting</code> annotation are ignored, as it indicates that visibility has been purposely relaxed to make the code testable.

----
class Cone {
  @VisibleForTesting
  Logger logger; // Compliant
}
----
