<p>In Transact-SQL, the semicolon statement terminator is in most cases optional. Therefore many developers donâ€™t use semicolons. However, in some situations missing semicolons may yield insidious errors.</p>
<p>Semicolons are required by the ANSI standard, and Microsoft <a href="https://docs.microsoft.com/en-us/sql/t-sql/language-elements/transact-sql-syntax-conventions-transact-sql">recommends</a> the consistent usage of semicolons and might make semicolons mandatory in a future version of SQL Server. Also, semicolons make the code more portable.</p>
<h2>Noncompliant Code Example</h2>
<p>In the code sample below, the exception will never be rethrown because &lt;code&gt;SELECT ERROR_MESSAGE() THROW&lt;/code&gt; will be interpreted as a column alias.</p>
<pre>
BEGIN TRY
  BEGIN TRAN;
  SELECT 1/0 AS AnException;
  COMMIT;
END TRY
BEGIN CATCH
  SELECT ERROR_MESSAGE()    -- Noncompliant; no exception will be thrown
  THROW
END CATCH
</pre>
<h2>Compliant Solution</h2>
<pre>
BEGIN TRY
  BEGIN TRAN;
  SELECT 1/0 AS AnException;
  COMMIT;
END TRY
BEGIN CATCH
  SELECT ERROR_MESSAGE();
  THROW;
END CATCH
</pre>
