Every <code>AUTHORITY-CHECK</code> statement sets the fields <code>SY-SUBRC</code> (also accessible as <code>SYST-SUBRC</code>) to the authorization check result. Thus <code>SY-SUBRC</code> value should be checked just after every <code>AUTHORITY-CHECK</code> statement.


== Noncompliant Code Example

----
AUTHORITY-CHECK OBJECT 'S_MYOBJ' "Noncompliant
    ID 'ID1' FIELD myvalue.
----


== Compliant Solution

----
AUTHORITY-CHECK OBJECT 'S_MYOBJ'  "Compliant
    ID 'ID1' FIELD myvalue.

  IF sy-subrc <> 0. 
    MESSAGE 'NOT AUTHORIZED' TYPE 'E'. 
  ENDIF. 
----


== Exceptions

No issue will be raised in the following cases:
* One or more <code>WRITE</code> operation are performed between the <code>AUTHORITY-CHECK</code> statement and <code>SY-SUBRC</code> check. An exception will be however raised if the <code>WRITE</code> operation is a <code>WRITE ... TO</code> statement, as this will set again <code>SY-SUBRC</code>.
* <code>SY-SUBRC</code>'s value is assigned to a variable. We then assume that it will be checked later.

----
AUTHORITY-CHECK OBJECT 'S_MYOBJ'  "Compliant
    ID 'ID1' FIELD myvalue.
WRITE 'Test' " WRITE is accepted before checking SY-SUBRC
IF SY-SUBRC <> 0. 
    EXIT.
ENDIF.

AUTHORITY-CHECK OBJECT 'S_MYOBJ'  "Compliant
    ID 'ID1' FIELD myvalue.
Tmp = SY-SUBRC " Assigning SY-SUBRC value to a variable. We assume that it will be checked later.
IF Tmp <> 0.
    EXIT.
ENDIF.
----

