<p>Trailing closure syntax can only be used with the last argument to a function call. Place a function type parameter anywhere else in the list and you limit the options of the caller.</p>
<h2>Noncompliant Code Example</h2>
<pre>
func foo(p1: Int-&gt;Int, p2: Int){  // Noncompliant; p1 should come at the end
  print(p1(p2))
}

foo({a in a * 2}, 42) // Trailing closure syntax can't be used here
</pre>
<h2>Compliant Solution</h2>
<pre>
func foo(p2: Int, p1: Int-&gt;Int){
  print(p1(p2))
}

foo(42) {a in a * 2}
</pre>
