<p>The ability to map a class to a database table has made database interaction in Java a lot easier. But map multiple classes to the same table and youâ€™ll end up with a classic case of the left hand not knowing what the right hand is doing. In the worse case scenario, it could lead to serious data corruption.</p>
<p>This rule raises an issue when multiple classes are mapped to the same table with Hibernate or JPA annotations.</p>
<h2>Noncompliant Code Example</h2>
<pre>
@Entity  // implicitly mapped to "point" table
public class Point {
  // ..
}

@Entity
@Table(name = "point")  // Noncompliant
public class Spot {
  // ...
}
</pre>
