<p>The number of &lt;code&gt;RECORDS&lt;/code&gt; or &lt;code&gt;CHARACTERS&lt;/code&gt; specified in a &lt;code&gt;BLOCK CONTAINS&lt;/code&gt; clause is used to determine block size. Specify &lt;code&gt;10 RECORDS&lt;/code&gt;, and the block will be exactly 10x the length of the record. But that may not be the right size, depending on the environment. Instead, it is considered a best practice to specify &lt;code&gt;0 RECORDS&lt;/code&gt;, so the block size will be calculated automatically.</p>
<h2>Noncompliant Code Example</h2>
<pre>
       FD OUTFILE1
           BLOCK CONTAINS 32760 RECORDS     &gt;* Noncompliant
           RECORDING MODE V.
       FD OUTFILE2
           BLOCK CONTAINS 1024 CHARACTERS.  &gt;* Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
       FD OUTFILE1
           BLOCK CONTAINS 0 RECORDS
           RECORDING MODE V.
       FD OUTFILE2
           BLOCK CONTAINS 0 RECORDS.
</pre>
<h2>Exceptions</h2>
<p>&lt;code&gt;0 CHARACTERS&lt;/code&gt; is compliant.</p>
