The number of <code>RECORDS</code> or <code>CHARACTERS</code> specified in a <code>BLOCK CONTAINS</code> clause is used to determine block size. Specify <code>10 RECORDS</code>, and the block will be exactly 10x the length of the record. But that may not be the right size, depending on the environment. Instead, it is considered a best practice to specify <code>0 RECORDS</code>, so the block size will be calculated automatically.


== Noncompliant Code Example

----
       FD OUTFILE1
           BLOCK CONTAINS 32760 RECORDS     >* Noncompliant
           RECORDING MODE V.
       FD OUTFILE2
           BLOCK CONTAINS 1024 CHARACTERS.  >* Noncompliant
----


== Compliant Solution

----
       FD OUTFILE1
           BLOCK CONTAINS 0 RECORDS
           RECORDING MODE V.
       FD OUTFILE2
           BLOCK CONTAINS 0 RECORDS.
----


== Exceptions

<code>0 CHARACTERS</code> is compliant.

