include::../description.adoc[]

== Noncompliant Code Example

----
void Notify(string company, string office = "QJZ") // Noncompliant
{
}
----

== Compliant Solution

----
void Notify(string company)
{
  Notify(company, "QJZ");
}
void Notify(string company, string office)
{
}
----

include::../exceptions.adoc[]
