<p>When &lt;code&gt;sizeof&lt;/code&gt; is used on a pointer, it returns the size of the pointer (4 or 8 bytes, depending on architecture and OS), not the size of the pointed-to memory area. Therefore, division of that value is not likely to yield what was intended.</p>
<h2>Noncompliant Code Example</h2>
<pre>
char* message = "hello world";
sizeof(message) / 4; // Noncompliant
</pre>
