<p>According to the SAP documentation:
{quote}
System functions are only intended for internal usage. Incompatible changes and further development is possible at any time and without warning or notice.
{quote}</p>
<p>So calling system C functions using a &lt;code&gt;CALL&lt;/code&gt; statement should be avoided.</p>
<h2>Noncompliant Code Example</h2>
<pre>
CALL 'MULTIPLY' ID 'P1'  FIELD '9999'
                ID 'P2'  FIELD '9999'
                ID 'RES' FIELD RESULT.
</pre>
