According to the SAP documentation:
{quote}
System functions are only intended for internal usage. Incompatible changes and further development is possible at any time and without warning or notice.
{quote}

So calling system C functions using a <code>CALL</code> statement should be avoided.


== Noncompliant Code Example

----
CALL 'MULTIPLY' ID 'P1'  FIELD '9999' 
                ID 'P2'  FIELD '9999' 
                ID 'RES' FIELD RESULT. 
----

