<p>Calling an operator in python is equivalent to calling a special method (except for the identity operator &lt;code&gt;is&lt;/code&gt;). Python provides a set of built-in operations. It is for example possible to add two integers: &lt;code&gt;1 + 2&lt;/code&gt;. It is however not possible to add a string and an integer: &lt;code&gt;1 + "2"&lt;/code&gt; and such an operation will raise a &lt;code&gt;TypeError&lt;/code&gt;.</p>
<p>It is possible to define how an operator will behave with a custom class by defining the corresponding special method. See python documentation for a complete list of operators and their methods: <a href="https://docs.python.org/3/reference/datamodel.html?highlight=">https://docs.python.org/3/reference/datamodel.html?highlight=</a><em>exit</em>%20special%20methods#emulating-numeric-types[arithmetic and bitwise operators], <a href="https://docs.python.org/3/reference/datamodel.html?highlight=">https://docs.python.org/3/reference/datamodel.html?highlight=</a><em>exit</em>%20special%20methods#object.<em>lt</em>[comparison operators].</p>
<p>For symmetrical binary operators you need to define two methods so that the order of operands doesnâ€™t matter, ex: &lt;code&gt;_\_add\__&lt;/code&gt; and &lt;code&gt;\__radd\_\_&lt;/code&gt;.</p>
<p>This rule raises an issue when an operator is used on incompatible types. Types are considered incompatible if no built-in operations between those types exist and none of the operands has implemented the corresponding special methods.</p>
<h2>Noncompliant Code Example</h2>
<pre>
class Empty:
    pass

class Add:
    def __add__(self, other):
        return 42

1 + 2
1 + "2"  # Noncompliant
Empty() + 1  # Noncompliant
Add() + 1
1 + Add()  # Noncompliant
Add() + Empty()
Empty() + Add()  # Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
class Empty:
    pass

class Add:
    def __add__(self, other):
        return 42

    def __radd__(self, other):
        return 42

Add() + 1
1 + Add()
Add() + Empty()
Empty() + Add()
</pre>
<h2>See</h2>
<ul>
<li>Python documentation - <a href="https://docs.python.org/3/reference/datamodel.html?highlight=">https://docs.python.org/3/reference/datamodel.html?highlight=</a><em>exit</em>%20special%20methods#object.<em>lt</em>[Rich comparison methods]</li>
<li>Python documentation - <a href="https://docs.python.org/3/reference/datamodel.html?highlight=">https://docs.python.org/3/reference/datamodel.html?highlight=</a><em>exit</em>%20special%20methods#emulating-numeric-types[Emulating numeric types]</li>
</ul>
