Calling an operator in python is equivalent to calling a special method (except for the identity operator <code>is</code>). Python provides a set of built-in operations. It is for example possible to add two integers: <code>1 + 2</code>. It is however not possible to add a string and an integer: <code>1 + "2"</code> and such an operation will raise a <code>TypeError</code>.

It is possible to define how an operator will behave with a custom class by defining the corresponding special method. See python documentation for a complete list of operators and their methods: https://docs.python.org/3/reference/datamodel.html?highlight=__exit__%20special%20methods#emulating-numeric-types[arithmetic and bitwise operators], https://docs.python.org/3/reference/datamodel.html?highlight=__exit__%20special%20methods#object.__lt__[comparison operators].

For symmetrical binary operators you need to define two methods so that the order of operands doesn't matter, ex: <code>\_\_add\_\_</code> and <code>\_\_radd\_\_</code>.

This rule raises an issue when an operator is used on incompatible types. Types are considered incompatible if no built-in operations between those types exist and none of the operands has implemented the corresponding special methods.


== Noncompliant Code Example

----
class Empty:
    pass

class Add:
    def __add__(self, other):
        return 42

1 + 2
1 + "2"  # Noncompliant
Empty() + 1  # Noncompliant
Add() + 1
1 + Add()  # Noncompliant
Add() + Empty()
Empty() + Add()  # Noncompliant
----


== Compliant Solution

----
class Empty:
    pass

class Add:
    def __add__(self, other):
        return 42

    def __radd__(self, other):
        return 42

Add() + 1
1 + Add()
Add() + Empty()
Empty() + Add()
----


== See

* Python documentation - https://docs.python.org/3/reference/datamodel.html?highlight=__exit__%20special%20methods#object.__lt__[Rich comparison methods]
* Python documentation - https://docs.python.org/3/reference/datamodel.html?highlight=__exit__%20special%20methods#emulating-numeric-types[Emulating numeric types]

