<p>Variables declared with &lt;code&gt;const&lt;/code&gt; cannot be modified. Unfortunately, attempts to do so donâ€™t always raise an error; in a non-ES2015 environment, such an attempt might simply be ignored.</p>
<h2>Noncompliant Code Example</h2>
<pre>
const pi = "yes, please";
pi = 3.14;  // Noncompliant
</pre>
