<p>There are several reasons for a function or closure not to have a body:</p>
<ul>
<li>It is an unintentional omission, and should be fixed to prevent unexpected behavior in production.</li>
<li>It is not yet, or never will be, supported. In this case an exception should be thrown.</li>
<li>The function is an intentionally-blank override. In this case a nested comment should explain the reason for the blank override.</li>
</ul>
<h2>Noncompliant Code Example</h2>
<pre>
func fun(p1:Int) {
}
</pre>
<h2>Compliant Solution</h2>
<pre>
func fun(p1:Int) {
  var a = doSomething(p1)
  var threshold = 42
  if a &gt; threshold {
    // ...
  }
}
</pre>
<p>or</p>
<pre>
func fun(p1:Int) {
  // Intentionally unimplemented...
}
</pre>
