The result of an expression with an arithmetic operator <code>/, *, %, \+\+, --, -, \+=, -=, *=, /=, %=, \+</code> or unary operator <code>+, -</code> when at least one operand is <code>Object</code> or <code>Undefined</code> will be always a <code>NaN</code> (Not a Number).


== Noncompliant Code Example

----
x = [1, 2];
var y = x / 4;  //Noncompliant
----


== Exceptions

* <code>Date</code> operands: they are implicitly converted to numbers.
* The binary <code>+</code> operator with <code>Object</code> operand (concatenation).

