Because <code>CX_ROOT</code> is the base exception type, catching it directly probably casts a wider net than you intended. Catching <code>CX_ROOT</code> could mask far more serious system errors that your <code>CATCH</code> logic was intended to deal with.

Some smaller, more specific exception type should be caught instead.


== Noncompliant Code Example

----
  try.
    if ABS( NUMBER ) > 100.
      write / 'Number is large'.
    endif.
  catch CX_ROOT into OREF.
    write / OREF->GET_TEXT( ).
  endtry.
----


== Compliant Solution

----
  try.
    if ABS( NUMBER ) > 100.
      write / 'Number is large'.
    endif.
  catch CX_SY_ARITHMETIC_ERROR into OREF.
    write / OREF->GET_TEXT( ).
  endtry.
----

