Collection classes from the <code>java.util.concurrent</code> package have their own concurrency control mechanisms. A concurrent collection is thread-safe, but not governed by a single exclusion lock, therefore using an instance of such a class for synchronization is, at best unnecessary and at worst likely to have unintended consequences. 

This rule raises an issue when a synchronization lock is used on an instance of one of the collection classes from the <code>java.util.concurrent</code> package.


== Noncompliant Code Example

----
public class Foo {
  private ConcurrentHashMap<String,String> map = new ConcurrentHashMap<>();

  public void bar() {
    synchronized(map) {  // Noncompliant
      map.put("foo", "bar");
    }
    // Do something ...
  }
}
----


== Compliant Solution

----
public class Foo {
  private ConcurrentHashMap<String,String> map = new ConcurrentHashMap<>();

  public void bar() {
    map.put("foo", "bar");
    // Do something ...
  }
}
----


