<p>When directly subclassing &lt;code&gt;java.io.OutputStream&lt;/code&gt; or &lt;code&gt;java.io.FilterOutputStream&lt;/code&gt;, the only requirement is that you implement the method &lt;code&gt;write(int)&lt;/code&gt;. However most uses for such streams don’t write a single byte at a time and the default implementation for &lt;code&gt;write(byte[],int,int)&lt;/code&gt; will call &lt;code&gt;write(int)&lt;/code&gt; for every single byte in the array which can create a lot of overhead and is utterly inefficient. It is therefore strongly recommended that subclasses provide an efficient implementation of &lt;code&gt;write(byte[],int,int)&lt;/code&gt;.</p>
<p>This rule raises an issue when a direct subclass of &lt;code&gt;java.io.OutputStream&lt;/code&gt; or &lt;code&gt;java.io.FilterOutputStream&lt;/code&gt; doesn’t provide an override of &lt;code&gt;write(byte[],int,int)&lt;/code&gt;.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public class MyStream extends OutputStream { // Noncompliant
    private FileOutputStream fout;

    public MyStream(File file) throws IOException {
        fout = new FileOutputStream(file);
    }

    @Override
    public void write(int b) throws IOException {
        fout.write(b);
    }

    @Override
    public void close() throws IOException {
        fout.write("\n\n".getBytes());
        fout.close();
        super.close();
    }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
public class MyStream extends OutputStream {
    private FileOutputStream fout;

    public MyStream(File file) throws IOException {
        fout = new FileOutputStream(file);
    }

    @Override
    public void write(int b) throws IOException {
        fout.write(b);
    }

    @Override
    public void write(byte[] b, int off, int len) throws IOException {
        fout.write(b, off, len);
    }

    @Override
    public void close() throws IOException {
        fout.write("\n\n".getBytes());
        fout.close();
        super.close();
    }
}
</pre>
<h2>Exceptions</h2>
<p>This rule doesn’t raise an issue when the class is declared &lt;code&gt;abstract&lt;/code&gt;.</p>
