<p>The use of &lt;code&gt;virtual&lt;/code&gt; base classes can introduce a number of undefined and potentially confusing behaviours. Therefore, a base class shall only be declared &lt;code&gt;virtual&lt;/code&gt; if that base class is to be used as a common base class in a diamond hierarchy.</p>
<h2>Noncompliant Code Example</h2>
<pre>
class A {};
class B1: public virtual A {}; // Compliant, A is a common base for C
class B2: public virtual A {}; // Compliant, A is a common base for C
class C: public B1, B2 {};
class D: public virtual A {}; // Noncompliant, D is not part of a diamond-shaped hierarchy.
</pre>
<h2>See</h2>
<ul>
<li>MISRA C++:2008, 10-1-2 - A base class shall only be declared virtual if it is used in a diamond hierarchy.</li>
</ul>
