<p>Shared coding conventions allow teams to collaborate efficiently. For maximum readability, this rule checks that the levels, names and PICTURE clauses for all items of the same level and which are subordinate to the same item start in the same column.</p>
<h2>Noncompliant Code Example</h2>
<pre>
01  ZONE1.
    03  ZONE2  PIC X(10).
    03   ZONE3 PIC X(10).    *&gt; Noncompliant; name out of line
     03 ZONE4  PIC X(10).    *&gt; Noncompliant; level out of line
    03  ZONE5   PIC X(10).   *&gt; Noncompliant; PIC out of line
</pre>
<h2>Compliant Solution</h2>
<pre>
01  ZONE1.
    03 ZONE2  PIC X(10).
    03 ZONE3  PIC X(10).
    03 ZONE4  PIC X(10).
    03 ZONE5  PIC X(10).
</pre>
