include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression <code>^[a-z][a-zA-Z0-9]*$</code>:

----
class MyClass {
  int my_field;
};
----

== Compliant Solution

----
class MyClass {
  int myField;
};
----
