Arrow functions <code>=></code> use a syntax similar to certain comparison operators (<code><=</code>, <code>>=</code>). This can create confusion when used in certain contexts.


== Noncompliant Code Example

----
let foo = (x) => x ? "Africa" : "Asia";   // Noncompliant
----


== Compliant Solution

----
let foo = (x) => { return x ? "Africa" : "Asia"; };
----


