<p>Host variables are used for passing data back and forth from the database to a COBOL program via SQL. In order to keep that transmission clear and free of errors, the format of each host variable should match its corresponding database column type.</p>
<p>This rule raises an issue when column and corresponding host variable donâ€™t match in terms of numeric vs character data, and when the host variable is smaller than the column width.</p>
<h2>Noncompliant Code Example</h2>
<pre>
TODO
</pre>
<h2>Compliant Solution</h2>
<pre>
TODO
</pre>
