<p>Where the developer uses new versions of standard library macros or objects (e.g. to enhance functionality or add checks of input values), the modified macro or object shall have a new name.</p>
<p>This is to avoid any confusion as to whether a standard macro or object, or a modified version of them, is being used.</p>
<h2>Noncompliant Code Example</h2>
<pre>
#define NULL ( a &gt; b ) // Noncompliant
</pre>
<h2>See</h2>
<ul>
<li>MISRA C:2004, 20.2</li>
<li>MISRA C++:2008, 17-0-2</li>
<li><a href="https://www.securecoding.cert.org/confluence/x/FAGTAw">CERT, DCL01-J.</a> - Do not reuse public identifiers from the Java Standard Library</li>
</ul>
