<p>According to the Java &lt;code&gt;Comparable.compareTo(T o)&lt;/code&gt; documentation:</p>
<p>{quote}
It is strongly recommended, but not strictly required that &lt;code&gt;(x.compareTo(y)==0) == (x.equals(y))&lt;/code&gt;.
Generally speaking, any class that implements the Comparable interface and violates this condition should clearly indicate this fact.
The recommended language is "Note: this class has a natural ordering that is inconsistent with equals."
{quote}</p>
<p>If this rule is violated, weird and unpredictable failures can occur.
For example, in Java 5 the &lt;code&gt;PriorityQueue.remove()&lt;/code&gt; method relied on &lt;code&gt;compareTo()&lt;/code&gt;, but since Java 6 it has relied on &lt;code&gt;equals()&lt;/code&gt;.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public class Foo implements Comparable&lt;Foo&gt; {
  @Override
  public int compareTo(Foo foo) { /* ... */ }      // Noncompliant as the equals(Object obj) method is not overridden
}
</pre>
<h2>Compliant Solution</h2>
<pre>
public class Foo implements Comparable&lt;Foo&gt; {
  @Override
  public int compareTo(Foo foo) { /* ... */ }      // Compliant

  @Override
  public boolean equals(Object obj) { /* ... */ }
}
</pre>
