<p>Sharing naming conventions allows teams to collaborate efficiently. This rule checks that targetNamespaces match a provided regular expression.</p>
<h2>Noncompliant Code Example</h2>
<p>With the default regular expression: <a href="http://www+">http://www+</a></p>
<pre>
&lt;xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema"
           elementFormDefault="qualified" version="1"
            targetNamespace="fruit"&gt;  &lt;!-- Noncompliant --&gt;
...
</pre>
<h2>Compliant Solution</h2>
<pre>
&lt;xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema"
           elementFormDefault="qualified" version="1"
           targetNamespace="http://www.codeSamples.com/fruit"&gt;
...
</pre>
