When the last argument to a function is a closure, it's possible and sometimes desirable to write that closure after the function's parentheses. This is called a _trailing_ closure. In order to help distinguish a trailing closure from an independent code block, it is best to begin the closure on the same line as the function call.


== Noncompliant Code Example

----
funWithClosureArgument()
{ // Noncompliant; looks like an independent code block
   print("Hello world")
}
----


== Compliant Solution

----
funWithClosureArgument() { 
   print("Hello world")
}
----


