<code>SELECT</code> queries that return too many columns may be complex or difficult to maintain. 

This rule identifies queries that <code>SELECT</code> more than the specified number of columns.


== Noncompliant Code Example

With a threshold of 6:
----
BEGIN
  SELECT id, name, firstname, gender, height, weight, age -- Noncompliant
    INTO peopleArray
    FROM people
    WHERE age > 60;
END;
/
----

