<p>Each file is considered an "external" module. The use of the &lt;code&gt;module&lt;/code&gt; keyword creates an internal module, and was used before the ECMAScript 2015 addition of &lt;code&gt;namespace&lt;/code&gt;s for the same purpose. Now that &lt;code&gt;namespace&lt;/code&gt; is available, the use of &lt;code&gt;module&lt;/code&gt; is deprecated because it does the same thing, and its use could confuse maintainers unaware of the history of the language.</p>
<h2>Noncompliant Code Example</h2>
<pre>
module myMod {  // Noncompliant
  // ...
}
</pre>
<h2>Compliant Solution</h2>
<pre>
namespace myMod {
  // ...
}
</pre>
