<code>Optional</code> value can hold either a value or not. The value held in the <code>Optional</code> can be accessed using the <code>get()</code> method, but it will throw a 
<code>NoSuchElementException</code> if there is no value present. To avoid the exception, calling the <code>isPresent()</code> or <code>! isEmpty()</code> method should always be done before any call to <code>get()</code>.

Alternatively, note that other methods such as <code>orElse(...)</code>, <code>orElseGet(...)</code> or <code>orElseThrow(...)</code> can be used to specify what to do with an empty <code>Optional</code>.

== Noncompliant Code Example

----
Optional<String> value = this.getOptionalValue();

// ...

String stringValue = value.get(); // Noncompliant
----

== Compliant Solution

----
Optional<String> value = this.getOptionalValue();

// ...

if (value.isPresent()) {
  String stringValue = value.get();
}
----

or

----
Optional<String> value = this.getOptionalValue();

// ...

String stringValue = value.orElse("default");
----

include::../see.adoc[]
