Data members and member functions cannot be defined as external, although entire objects can. When a member is declared as <code>extern</code>, the compiler simply ignores the keyword, making it both extraneous and confusing.


== Noncompliant Code Example

----
class C {
  void fun();
};

extern void C::fun() { // Noncompliant
}
----


== Compliant Solution

----
class C {
  void fun();
};

void C::fun() {
}
----

