<p>HTML &lt;table&gt; elements should not be used for layout purpose as it can confuse screen readers. It is recommended to use CSS instead.</p>
<p>This rule raises an issue on every &lt;code&gt;&lt;table&gt;&lt;/code&gt; element containing a &lt;code&gt;role&lt;/code&gt;&nbsp;attribute set to &lt;code&gt;"presentation"&lt;/code&gt; or &lt;code&gt;"none"&lt;/code&gt;, which is how <a href="https://www.w3.org/WAI/tutorials/tables/tips/">W3C recommends</a> marks layout tables.</p>
<h2>Noncompliant Code Example</h2>
<pre>
&lt;table role="presentation"&gt;
&lt;!-- ... --&gt;
&lt;/table&gt;
</pre>
<h2>See</h2>
<ul>
<li><a href="https://www.w3schools.com/html/html_layout.asp">HTML Layouts</a></li>
</ul>
