The multi-line comment syntax <code>/* ... */</code> should not be used for single line comments; the <code>--</code> syntax is more appropriate.


== Noncompliant Code Example

----
/*
 These comment lines are Compliant
 comment 1
 comment 2
*/

/*
 This comment is also Compliant
 */

/* This comment is Noncompliant */
----


== Compliant Solution

----
/*
 These comment lines are Compliant
 comment 1
 comment 2
*/

/*
 This comment is also Compliant
 */

-- This comment is compliant 
----

