<p>The following predefined variables are deprecated and should be replaced by the new versions:
||Replace||With||
|$HTTP_SERVER_VARS|$_SERVER|
|$HTTP_GET_VARS|$_GET|
|$HTTP_POST_VARS|$_POST|
|$HTTP_POST_FILES|$_FILES|
|$HTTP_SESSION_VARS|$_SESSION|
|$HTTP_ENV_VARS|$_ENV|
|$HTTP_COOKIE_VARS|$_COOKIE|
|$php_errormsg|error_get_last()|</p>
<h2>Noncompliant Code Example</h2>
<pre>
echo 'Name parameter value: ' . $HTTP_GET_VARS["name"];
</pre>
<h2>Compliant Solution</h2>
<pre>
echo 'Name parameter value: ' . $_GET["name"];
</pre>
