<p>Declaring a &lt;code&gt;NUMBER&lt;/code&gt; variable without any precision wastes memory because Oracle supports up to 38 decimal digits by default (or the maximum supported by your system, whichever is less). If you donâ€™t need that large a value, you should specify whatever matches your needs. This will save memory and provide extra integrity checking on input.</p>
<p>This rule also applies to some &lt;code&gt;NUMBER&lt;/code&gt; subtypes as well: &lt;code&gt;NUMERIC&lt;/code&gt;, &lt;code&gt;DEC&lt;/code&gt;, and &lt;code&gt;DECIMAL&lt;/code&gt;.</p>
<h2>Noncompliant Code Example</h2>
<pre>
DECLARE
  var1 NUMBER; -- Noncompliant
  var2 NUMERIC; -- Noncompliant
BEGIN
  NULL;
END;
/
</pre>
<h2>Compliant Solution</h2>
<pre>
DECLARE
  var1 NUMBER(9,2);
  var2 NUMERIC(4,0);
BEGIN
  NULL;
END;
/
</pre>
