<p>Developers may define a list named &lt;code&gt;_\_all\_\_&lt;/code&gt; in a module to limit the names imported from it by wildcard imports (&lt;code&gt;from mymodule import *&lt;/code&gt;). This list can only reference defined names, otherwise an &lt;code&gt;AttributeError&lt;/code&gt; will be raised when the module is imported.</p>
<h2>Noncompliant Code Example</h2>
<pre>
from mymodule import my_func

__all__ = ["unknown_func"]  # Noncompliant. "unknown_func" is undefined
</pre>
<h2>Compliant Solution</h2>
<pre>
from mymodule import my_func

__all__ = ["my_func"]
</pre>
<h2>See</h2>
<ul>
<li><a href="https://docs.python.org/3/tutorial/modules.html#importing-from-a-package">Python documentation  - Importing * From a Package</a></li>
</ul>
