Developers may define a list named <code>\_\_all\_\_</code> in a module to limit the names imported from it by wildcard imports (<code>from mymodule import *</code>). This list can only reference defined names, otherwise an <code>AttributeError</code> will be raised when the module is imported.


== Noncompliant Code Example

----
from mymodule import my_func

__all__ = ["unknown_func"]  # Noncompliant. "unknown_func" is undefined
----


== Compliant Solution

----
from mymodule import my_func

__all__ = ["my_func"]
----


== See

* https://docs.python.org/3/tutorial/modules.html#importing-from-a-package[Python documentation  - Importing * From a Package]


