It can be extremely confusing when a <code>for</code> condition tests a variable which is not updated inside the <code>for</code> post statement.

== Noncompliant Code Example

----
for i := 1; i <= 10; j++ { // Noncompliant
	// ...
}
----

== Compliant Solution

----
for i := 1; i <= 10; i++ {
	// ...
}
----
