Since ANSI SQL-92, explicit joins using the <code>JOIN</code> keyword have been possible, and are preferred. Therefore table joins should be done with help of the one of the following clauses: <code>JOIN</code>, <code>INNER  JOIN</code>, <code>LEFT OUTER  JOIN</code>, <code>RIGHT OUTER JOIN</code>, and <code>FULL OUTER  JOIN</code>. The old way to join tables is deprecated and should not be used anymore.


== Noncompliant Code Example

----
SELECT *
FROM PARTS, PRODUCTS
WHERE PARTS.PROD = PRODUCTS.PROD
----


== Compliant Solution

----
SELECT *
FROM PARTS
INNER JOIN PRODUCTS ON PARTS.PROD = PRODUCTS.PROD
----

