The <code>asm</code> declaration is available to all C++ implementations, allowing a consistent mechanism to be used.

However, the parameters to <code>asm</code> are still implementation-defined.


== Noncompliant Code Example

----
void Delay ( void )
{
#pragma asm
  "NOP" // Noncompliant
#pragma endasm
}
----


== Compliant Solution

----
void Delay ( void )
{
  asm ( "NOP" ); // Compliant
}
----


== See

* MISRA C++ 2008, 7-4-2

