<p>Calling the &lt;code&gt;+&lt;/code&gt;, &lt;code&gt;-&lt;/code&gt; or &lt;code&gt;NOT&lt;/code&gt; prefix operator twice does nothing: the second invocation undoes the first. Such mistakes are typically caused by accidentally double-tapping the key in question without noticing.</p>
<p>Either this is a bug, if the operator was actually meant to be called once, or misleading if done on purpose.</p>
<h2>Noncompliant Code Example</h2>
<pre>
IF NOT ( NOT foo = 5 ) THEN  -- Noncompliant
  value := ++1;              -- Noncompliant
END IF;
</pre>
<h2>Compliant Solution</h2>
<pre>
IF foo = 5 THEN  -- Compliant
  value := +1;       -- Compliant
END IF;
</pre>
