For read-only computed properties and subscript declarations, the <code>get</code> keyword and its braces are optional, and should be omitted for the sake of brevity.


== Noncompliant Code Example

----
struct Magic {
  var number:Int {
    get {  // Noncompliant
      return 42
    }
  }
}
----


== Compliant Solution

----
struct Magic {
  var number:Int {
    return 42
  }
}
----


