<p>For read-only computed properties and subscript declarations, the &lt;code&gt;get&lt;/code&gt; keyword and its braces are optional, and should be omitted for the sake of brevity.</p>
<h2>Noncompliant Code Example</h2>
<pre>
struct Magic {
  var number:Int {
    get {  // Noncompliant
      return 42
    }
  }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
struct Magic {
  var number:Int {
    return 42
  }
}
</pre>
