include::../description.adoc[]

== Noncompliant Code Example

----
if (x == 0) {
  doSomething();
} else if (x == 1) {
  doSomethingElse();
}
----

== Compliant Solution

----
if (x == 0) {
  doSomething();
} else if (x == 1) {
  doSomethingElse();
} else {
  error();
}
----

== Exceptions

When all branches of an <code>if</code>-<code>else if</code> end with <code>return</code>, <code>break</code> or <code>throw</code>, the code that comes after the <code>if</code> implicitly behaves as if it was in an <code>else</code> clause. This rule will therefore ignore that case.

== See

* MISRA C:2004, 14.10 - All if...else if constructs shall be terminated with an else clause.
* MISRA C++:2008, 6-4-2 - All if...else if constructs shall be terminated with an else clause.
* MISRA C:2012, 15.7 - All if...else if constructs shall be terminated with an else statement
* https://www.securecoding.cert.org/confluence/x/YgE[CERT, MSC01-C.] - Strive for logical completeness
* https://www.securecoding.cert.org/confluence/x/PQHRAw[CERT, MSC57-J.] - Strive for logical completeness
