When using Backbone.js, the use of <code>has</code> to check attribute existence is clearer and more readable than the use of <code>get</code>, and is therefore preferred.


== Noncompliant Code Example

----
if (myModel.get('myProperty')) { ... }
----


== Compliant Solution

----
if (myModel.has('myProperty') { ... }
----

