Call stacks containing lot of <code>PERFORM</code> statements is a key ingredient for making what's known as "Spaghetti code".
Such code is hard to read, refactor and therefore maintain.

This rule supports both sections and paragraphs. 


== Noncompliant Code Example

With a threshold of 3:

----
PERFORM FIRST.

FIRST.
  PERFORM SECOND.

SECOND.
  PERFORM THIRD.

THIRD.
  PERFORM FOURTH.  *> Noncompliant

FOURTH.
  DISPLAY something.
----


