Having two <code>WHEN</code> clauses in the same <code>EVALUATE</code> statement or two branches in the same <code>IF</code> structure with the same implementation is at best duplicate code, and at worst a coding error. If the same logic is truly needed for both instances, then they should be combined, for an <code>IF</code> structure or one should fall through to the other for an <code>EVALUATE</code>.

== Noncompliant Code Example

----
EVALUATE X
  WHEN 1
    MOVE A TO B.
    PERFORM SECTION1
  WHEN 2
    MOVE A TO C.
    PERFORM SECTION2
  WHEN 3              *> Noncompliant; duplicates WHEN 1's implementation
    MOVE A TO B.
    PERFORM SECTION1
END-EVALUATE.

IF X = 1
  MOVE A TO B.
  PERFORM SECTION1
ELSE
  IF X > 10
    PERFORM SECTION2
  ELSE                *> Noncompliant
    MOVE A TO B.
    PERFORM SECTION1
  END-IF
END-IF.
----

== Exceptions

Blocks in an <code>IF</code> chain that contain a single line of code are ignored, as are blocks in a <code>EVALUATE</code> statement that contain a single line of code.

----
EVALUATE X
  WHEN 1
    PERFORM SECTION1
  WHEN 2
    PERFORM SECTION2
  WHEN 3              *> no issue, usually this is done on purpose to increase the readability
    PERFORM SECTION1
END-EVALUATE.
----

But this exception does not apply to <code>IF</code> chains without <code>ELSE</code>-s, or to <code>EVALUATE</code>-s without <code>WHEN OTHER</code> clauses when all branches have the same single line of code. In case of <code>IF</code> chains with <code>ELSE</code>-s, or of <code>EVALUATE</code>-es with <code>WHEN OTHER</code> clauses, rule S3923 raises a bug. 

----
EVALUATE X
  WHEN 1
    PERFORM SECTION1
  WHEN 3              *> Noncompliant, this might have been done on purpose but probably not
    PERFORM SECTION1
END-EVALUATE.
----
