<p>Shared coding conventions allow teams to collaborate efficiently. This rule checks that cursor parameters match the provided regular expression.</p>
<h2>Noncompliant Code Example</h2>
<p>With the default regular expression &lt;code&gt;[a-zA-Z]([a-zA-Z0-9_]*[a-zA-Z0-9])?&lt;/code&gt;:</p>
<pre>
CREATE TABLE employee(
  name VARCHAR2(42)
);

DECLARE
  CURSOR mycursor2(Employee-name-parameter_ VARCHAR2) RETURN employee%ROWTYPE; -- Noncompliant

  CURSOR mycursor2(Employee-name-parameter_ VARCHAR2) RETURN employee%ROWTYPE IS SELECT * FROM employee WHERE name = Employee-name-parameter_; -- Noncompliant
BEGIN
  NULL;
END;
/

DROP TABLE employee;
</pre>
<h2>Compliant Solution</h2>
<pre>
CREATE TABLE employee(
  name VARCHAR2(42)
);

DECLARE
  CURSOR mycursor2(employeeNameParameter VARCHAR2) RETURN employee%ROWTYPE;

  CURSOR mycursor2(employeeNameParameter VARCHAR2) RETURN employee%ROWTYPE IS SELECT * FROM employee WHERE name = employeeNameParameter;
BEGIN
  NULL;
END;
/

DROP TABLE employee;
</pre>
