include::../description.adoc[]

== Noncompliant Code Example

----
public string GetTitle(Person p)
{
  return p.Gender == Gender.MALE ? "Mr. " : p.IsMarried ? "Mrs. " : "Miss ";  // Noncompliant
}
----

== Compliant Solution

----
public string GetTitle(Person p) 
{
  if (p.Gender == Gender.MALE) 
  {
    return "Mr. ";
  }
  return p.IsMarried ? "Mrs. " : "Miss ";
}
----
