COBOL programs that include <code>EXEC SQL INCLUDE</code> statements must be pre-processed with an IBM DB2 precompiler or coprocessor before compilation. The precompiler does not require a period after the closing <code>END-EXEC</code>, but the coprocessor does. For portability, a period should always be used.


== Noncompliant Code Example

----
EXEC SQL INCLUDE ... END-EXEC  *> Noncompliant; '.' missing
----


== Compliant Solution

----
EXEC SQL INCLUDE ... END-EXEC.
----

