Under the covers, Simple <code>CASE</code> expressions are evaluated as searched <code>CASE</code> expressions. That is,

----
CASE @foo
WHEN 1 THEN 'a'
WHEN 2 THEN 'b'
----

is actually evaluated as

----
CASE 
WHEN @foo = 1 THEN 'a'
WHEN @foo = 2 THEN 'b'
----

In most situations the difference is inconsequential, but when the input expression isn't fixed, for instance if <code>RAND()</code> is involved, it is likely to yield unexpected results. For that reason, it is better to evaluate the input expression once, assign it to a variable, and use the variable as the <code>CASE</code>'s input expression.

This rule raises an issue when any of the following is used in a <code>CASE</code> input expression: <code>RAND</code>, <code>NEWID</code>, <code>CRYPT_GEN_RANDOM</code>.


== Noncompliant Code Example

----
CASE CONVERT(SMALLINT, RAND()*@foo)  -- Noncompliant
WHEN 1 THEN 'a'
WHEN 2 THEN 'b'
----


== Compliant Solution

----
DECLARE @bar SMALLINT = CONVERT(SMALLINT, RAND()*@foo) 
CASE @bar
WHEN 1 THEN 'a'
WHEN 2 THEN 'b'
----


