include::../description.adoc[]

== Noncompliant Code Example

----
func printTen() {                                                                                                                                
	myNumber := 010 // Noncompliant. myNumber will hold 8, not 10 - was this really expected?
	fmt.Println(myNumber)
}
----

== Compliant Solution

----
func printTen() {
	myNumber := 10
	fmt.Println(myNumber)
}
----

include::../see.adoc[]
