By convention, a file that exports only one class, function, or constant should be named for that class, function or constant. Anything else may confuse maintainers.

== Noncompliant Code Example

----
// file path: myclass.js  -- Noncompliant
class MyClass {
  // ...
}
export default MyClass;
----

== Compliant Solution

----
// file path: MyClass.js
class MyClass {
  // ...
}
export default MyClass;
----

== Exceptions

Case, underscores ( <code>_</code> ) and dashes  (<code>-</code>) are ignored from the name comparison.
