include::../description.adoc[]

== Noncompliant Code Example

----
class Foo: 
  foo = '' 

  def getFoo(self):
    ...

foo = Foo()
foo.getFoo() # what does this return?
----

== Compliant Solution

----
class Foo:
  name = ''

  def getName(self):
    ...

foo = Foo()
foo.getName()
----
