 

Declaring a function <code>constexpr</code> makes it implicitly inline.

In that situation, explicitly using the <code>inline</code> keyword would be redundant, and might lead to confusion if it's used in some cases but not others. It's better to simply omit it.


== Noncompliant Code Example

----
inline constexpr int addOne(int n) { return n+1; }
----


== Compliant Solution

----
constexpr int addOne(int n) { return n+1; }
----
 

