<p>In Python 3, attempting to raise an object which does not derive from BaseException will raise a &lt;code&gt;TypeError&lt;/code&gt;. In Python 2 it is possible to raise old-style classes but this shouldn’t be done anymore in order to be compatible with Python 3.</p>
<p>If you are about to create a custom Exception class, note that custom exceptions should inherit from &lt;code&gt;Exception&lt;/code&gt;, not &lt;code&gt;BaseException&lt;/code&gt;. &lt;code&gt;Exception&lt;/code&gt; allows people to catch all exceptions except the ones explicitly asking the interpreter to stop, such as &lt;code&gt;KeyboardInterrupt&lt;/code&gt; and <a href="https://docs.python.org/3/library/exceptions.html#GeneratorExit">&lt;code&gt;GeneratorExit&lt;/code&gt;</a> which is not an error. See <a href="https://www.python.org/dev/peps/pep-0352/#exception-hierarchy-changes">PEP 352</a> for more information.</p>
<p>This rule raises an issue when an object which doesn’t derive from BaseException is raised.</p>
<h2>Noncompliant Code Example</h2>
<pre>
raise "Something went wrong"  # Noncompliant

class A:
    pass

raise A  # Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
class MyError(Exception):
    pass

raise MyError("Something went wrong")
raise MyError
</pre>
<h2>See</h2>
<ul>
<li><a href="https://docs.python.org/3/tutorial/errors.html">Python documentation - Errors and Exceptions</a></li>
<li><a href="https://www.python.org/dev/peps/pep-0352/#exception-hierarchy-changes">PEP 352 - Required Superclass for Exceptions</a></li>
</ul>
