<p>By definition, primitive types are not Objects and so they canâ€™t be &lt;code&gt;null&lt;/code&gt;. Adding &lt;code&gt;@CheckForNull&lt;/code&gt; or &lt;code&gt;@Nullable&lt;/code&gt; on primitive types adds confusion and is useless.</p>
<p>This rule raises an issue when &lt;code&gt;@CheckForNull&lt;/code&gt; or &lt;code&gt;@Nullable&lt;/code&gt; is set on a method returning a primitive type: byte, short, int, long, float, double, boolean, char.</p>
<h2>Noncompliant Code Example</h2>
<pre>
@CheckForNull
boolean isFoo() {
 ...
}
</pre>
<h2>Compliant Solution</h2>
<pre>
boolean isFoo() {
 ...
}
</pre>
