By definition, primitive types are not Objects and so they can't be <code>null</code>. Adding <code>@CheckForNull</code> or <code>@Nullable</code> on primitive types adds confusion and is useless.

This rule raises an issue when <code>@CheckForNull</code> or <code>@Nullable</code> is set on a method returning a primitive type: byte, short, int, long, float, double, boolean, char.


== Noncompliant Code Example

----
@CheckForNull
boolean isFoo() {
 ...
}
----


== Compliant Solution

----
boolean isFoo() {
 ...
}
----


