<p>Duplicated string literals make the process of refactoring error-prone, since you must be sure to update all occurrences.
On the other hand, constants can be referenced from many places, but only need to be updated in a single place.</p>
<h2>Noncompliant Code Example</h2>
<p>With the default threshold of 3:</p>
<pre>
WRITE: / 'Firstname'.
*...
WRITE: / 'Firstname'.
*...
WRITE: / 'Firstname'.
</pre>
<h2>Compliant Solution</h2>
<pre>
CONSTANTS: fname    TYPE c LENGTH 9 VALUE 'Firstname',

WRITE: / fname.
*...
WRITE: / fname.
*...
WRITE: / fname.
</pre>
<h2>Exceptions</h2>
<p>To prevent generating some false-positives, literals having less than 5 characters are excluded.</p>
