<p>Duplicated string literals make the process of refactoring error-prone, since you must be sure to update all occurrences.
On the other hand, constants can be referenced from many places, but only need to be updated in a single place.</p>
<h2>Noncompliant Code Example</h2>
<p>With the default threshold of 3:</p>
<pre>
BEGIN
  prepare('action1');
  execute('action1');
  release('action1');
END;
/
</pre>
<h2>Compliant Solution</h2>
<pre>
DECLARE
  action CONSTANT VARCHAR2(7) := 'action1';
BEGIN
  prepare(action);
  execute(action);
  release(action);
END;
/
</pre>
<h2>Exceptions</h2>
<p>To prevent generating some false-positives, literals having less than 5 characters are excluded.</p>
