<p>Duplicated string literals make the process of refactoring error-prone, since you must be sure to update all occurrences.
On the other hand, constants can be referenced from many places, but only need to be updated in a single place.</p>
<h2>Noncompliant Code Example</h2>
<p>With the default threshold of 3:</p>
<pre>
prepare("a message")  // Noncompliant; duplicated 3 times
execute("a message")
release("a message")
</pre>
<h2>Compliant Solution</h2>
<pre>
let message = "a message"

prepare(message)
execute(message)
release(message)
</pre>
<h2>Exceptions</h2>
<p>To prevent generating some false-positives, literals having 5 or less characters are excluded as well as literals containing only letters, digits and '_'.</p>
