Jump statements (<code>return</code>, <code>break</code> and <code>continue</code>) move control flow out of the current code block. So any statements that come after a jump are dead code. 

== Noncompliant Code Example

----
fun foo(a: Int): Int {
  var i = 10;
  return a + i;       // Noncompliant 
  i++;                // dead code
}
----

== Compliant Solution

----
fun foo(a: Int): Int {
  var i = 10;
  return a + i;
}
----

include::../see.adoc[]
