<p>Every section should be commented to explain its goal and how it works. This comment can be placed either just before or just after the section label.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    UNCOMMENTED-SECTION SECTION.
</pre>
<h2>Compliant Solution</h2>
<pre>
* Some comments
    CORRECTLY-COMMENTED-SECTION SECTION.

    ANOTHER-CORRECTLY-COMMENTED-SECTION SECTION.
* Some comments
</pre>
