<p>The output of &lt;code&gt;DBMS_OUTPUT.PUT_LINE&lt;/code&gt; is not always visible, for example when &lt;code&gt;SERVEROUTPUT&lt;/code&gt; is set to &lt;code&gt;OFF&lt;/code&gt;. Moreover, there is no standardized way to specify the importance of the message. It is better to use a logging mechanism instead.</p>
<h2>Noncompliant Code Example</h2>
<pre>
SET SERVEROUTPUT ON

BEGIN
  DBMS_OUTPUT.PUT_LINE('An error occured'); -- Noncompliant
END;
/
</pre>
