The output of <code>DBMS_OUTPUT.PUT_LINE</code> is not always visible, for example when <code>SERVEROUTPUT</code> is set to <code>OFF</code>. Moreover, there is no standardized way to specify the importance of the message. It is better to use a logging mechanism instead.


== Noncompliant Code Example

----
SET SERVEROUTPUT ON

BEGIN
  DBMS_OUTPUT.PUT_LINE('An error occured'); -- Noncompliant
END;
/
----

