<p>Testing equality or nullness with JUnitâ€™s &lt;code&gt;assertTrue()&lt;/code&gt; or &lt;code&gt;assertFalse()&lt;/code&gt;  should be simplified to the corresponding dedicated assertion.</p>
<h2>Noncompliant Code Example</h2>
<pre>
Assert.assertTrue(a.equals(b));
Assert.assertTrue(a == b);
Assert.assertTrue(a == null);
Assert.assertTrue(a != null);
Assert.assertFalse(a.equals(b));
</pre>
<h2>Compliant Solution</h2>
<pre>
Assert.assertEquals(a, b);
Assert.assertSame(a, b);
Assert.assertNull(a);
Assert.assertNotNull(a);
Assert.assertNotEquals(a, b);
</pre>
