Testing equality or nullness with JUnit's <code>assertTrue()</code> or <code>assertFalse()</code>  should be simplified to the corresponding dedicated assertion.


== Noncompliant Code Example

----
Assert.assertTrue(a.equals(b));
Assert.assertTrue(a == b);
Assert.assertTrue(a == null);
Assert.assertTrue(a != null);
Assert.assertFalse(a.equals(b));
----


== Compliant Solution

----
Assert.assertEquals(a, b);
Assert.assertSame(a, b);
Assert.assertNull(a);
Assert.assertNotNull(a);
Assert.assertNotEquals(a, b);
----

