Java 8 adds <code>Comparator.comparing</code> to allow the creation of a single-value comparator to be shorthanded into a single call. This cleaner syntax should be preferred.

*Note* that this rule is automatically disabled when the project's <code>sonar.java.source</code> is lower than <code>8</code>.


== Noncompliant Code Example

----
Comparator<Foo> compartor = (foo1, foo2) -> foo.getName().compareTo(foo2.getName());  // Noncompliant
----


== Compliant Solution

----
Comparator<Foo> compartor = Comparator.comparing(Foo::getName);
----

