include::../description.adoc[]

== Noncompliant Code Example

----
If GetTrue() Or GetFalse() Then ' Noncompliant; both sides evaluated
End If
----

== Compliant Solution

----
If GetTrue() OrElse GetFalse() Then ' true short-circuit logic
End If
----

include::../see.adoc[]
