In the interests of making code as usable as possible, interfaces and delegates with generic parameters should use the <code>out</code> and <code>in</code> modifiers when possible to make the interfaces and delegates covariant and contravariant, respectively.

The <code>out</code> keyword can be used when the type parameter is used only as a return type in the interface or delegate. Doing so makes the parameter covariant, and allows interface and delegate instances created with a sub-type to be used as instances created with a base type. The most notable example of this is <code>IEnumerable<out T></code>, which allows the assignment of an <code>IEnumerable<string></code> instance to an <code>IEnumerable<object></code> variable, for instance.

The <code>in</code> keyword can be used when the type parameter is used only as a method parameter in the interface or a parameter in the delegate. Doing so makes the parameter contravariant, and allows interface and delegate instances created with a base type to be used as instances created with a sub-type. I.e. this is the inversion of covariance. The most notable example of this is the <code>Action<in T></code> delegate, which allows the assignment of an <code>Action<object></code> instance to a <code>Action<string></code> variable, for instance.


== Noncompliant Code Example

----
interface IConsumer<T>  // Noncompliant
{
    bool Eat(T fruit);
}
----


== Compliant Solution

----
interface IConsumer<in T>
{
    bool Eat(T fruit);
}
----

