<p>When a generic class is declaring wildcard with an upper bound that is &lt;code&gt;final&lt;/code&gt;, this generic class is not generic at all because it only accepts one and only one type at runtime: the one that is &lt;code&gt;final&lt;/code&gt;. Instead of using &lt;code&gt;Generics&lt;/code&gt;, itâ€™s simpler to directly use the concrete &lt;code&gt;final&lt;/code&gt; class.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public static &lt;T extends String&gt; T getMyString() { // Noncompliant; String is a "final" class and so can't be extended
 [...]
}
</pre>
<h2>Compliant Solution</h2>
<pre>
public static String getMyString() { // Compliant
  [...]
}
</pre>
