When a generic class is declaring wildcard with an upper bound that is <code>final</code>, this generic class is not generic at all because it only accepts one and only one type at runtime: the one that is <code>final</code>. Instead of using <code>Generics</code>, it's simpler to directly use the concrete <code>final</code> class.


== Noncompliant Code Example

----
public static <T extends String> T getMyString() { // Noncompliant; String is a "final" class and so can't be extended
 [...]
}
----


== Compliant Solution

----
public static String getMyString() { // Compliant
  [...]
}
----

