include::../description.adoc[]

== Noncompliant Code Example

----
Public Class [MyClass]
    Private Sub New() // Noncompliant
        ' ...
    End Sub
End Class
----

== Compliant Solution

----
Public Class [MyClass]
    Private Sub New()
        ' ...
    End Sub
End Class
----

include::../exceptions.adoc[]
