When multiple variables are <code>DECLARE</code>d in a single statement, parentheses must be used around the variable list to ensure the variable attributes in the statement are applied to all the variables in the declaration. Without parentheses, the attributes are applied only to the last variable in the list.


== Noncompliant Code Example

----
DECLARE X,Y FIXED BIN(15);
----


== Compliant Solution

----
DECLARE (X,Y) FIXED BIN(15);
----

