Access modifiers define which classes can access properties, variables, methods, and other classes. If an access modifier is not specified, the access level defaults to <code>internal</code>, which grants access to all classes in the same package. This may be what is intended, but it should be specified explicitly to avoid confusion.

Available access modifiers are:
* <code>internal</code> - access allowed within the same package
* <code>private</code> - access allowed only within the same class
* <code>protected</code> - access allowed to the class and its child classes
* <code>public</code> - unfettered access by all

== Noncompliant Code Example

----
function checkResources():Boolean { 
  ...
  return true;
}
----

== Compliant Solution

----
public function checkResources():Boolean { 
  ...
  return true;
}
----
