<p>The &lt;code&gt;result_cache&lt;/code&gt; Oracle hint can vastly improve performance, but it comes at the cost of extra memory consumption, so one should double-check that the gain in performance is significant, and avoid overusing this feature in general.</p>
<h2>Noncompliant Code Example</h2>
<pre>
SELECT /*+ result_cache */ * FROM DUAL;  -- Noncompliant
</pre>
