The <code>result_cache</code> Oracle hint can vastly improve performance, but it comes at the cost of extra memory consumption, so one should double-check that the gain in performance is significant, and avoid overusing this feature in general.


== Noncompliant Code Example

----
SELECT /*+ result_cache */ * FROM DUAL;  -- Noncompliant
----


