The forms <code><></code> and <code>!=</code> are equivalent. But in Python 2.7.3 the <code><></code> form is considered obsolete.


== Noncompliant Code Example

----
return a <> b # Noncompliant
----


== Compliant Solution

----
return a != b
----

