<p>This rule checks that the specified child elements are present inside the specified parent elements.</p>
<h2>Noncompliant Code Example</h2>
<p>Given a parent/child combination of &lt;code&gt;&lt;head&gt;&lt;/code&gt;/&lt;code&gt;&lt;title&gt;&lt;/code&gt;:</p>
<pre>
&lt;html&gt;
  &lt;head&gt;
  &lt;/head&gt;  &lt;!-- Noncompliant; no title element --&gt;
  &lt;body&gt;
    ...
</pre>
<h2>Compliant Solution</h2>
<pre>
&lt;html&gt;
  &lt;head&gt;
    &lt;title&gt;My Page&lt;/title&gt;
  &lt;/head&gt;
  &lt;body&gt;
    ...
</pre>
