include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression <code>^[A-Z][a-zA-Z0-9]*$</code>:
----
class my_class
  ...
end
----

== Compliant Solution

----
class MyClass
  ...
end
----
