include::../description.adoc[]

== Noncompliant Code Example

----
sampleString  = "Hello World" : MsgBox(sampleString)
----

== Compliant Solution

----
sampleString  = "Hello World"
MsgBox(sampleString)
----
