<p>Calling &lt;code&gt;DELETE ADJACENT DUPLICATES&lt;/code&gt; won’t reliably do any good if the table hasn’t first been sorted to put duplicates side by side, since the &lt;code&gt;ADJACENT&lt;/code&gt; part of the command looks for multiple rows side-by-side with the same content.</p>
<h2>Noncompliant Code Example</h2>
<pre>
DELETE ADJACENT DUPLICATES FROM ITAB COMPARING LAND.
</pre>
<h2>Compliant Solution</h2>
<pre>
SORT ITAB BY LAND.
DELETE ADJACENT DUPLICATES FROM ITAB COMPARING LAND.
</pre>
