Content that doesn't change or that doesn't change often should be included using a mechanism which won't try to interpret it. Specifically, <code><%@ include file="..." %></code>, which includes the file in the JSP servlet translation phase (i.e. it happens once), should be used instead of <code><jsp:include page="..." /></code>, which includes the page on the file, when the content is being served to the user.


== Noncompliant Code Example

----
<jsp:include page="header.jsp">  <!-- Noncompliant -->
----


== Compliant Solution

----
<%@ include file="header.jsp" %>
----

