<p>Shared naming conventions allow teams to collaborate effectively. This rule raises an issue when a pomâ€™s &lt;code&gt;artifactId&lt;/code&gt; does not match the provided regular expression.</p>
<h2>Noncompliant Code Example</h2>
<p>With the default regular expression: &lt;code&gt;[a-z][a-z-0-9]+&lt;/code&gt;</p>
<pre>
&lt;project ...&gt;
  &lt;artifactId&gt;My_Project&lt;/artifactId&gt;  &lt;!-- Noncompliant --&gt;

  &lt;!-- ... --&gt;
&lt;/project&gt;
</pre>
<h2>Compliant Solution</h2>
<pre>
&lt;project ...&gt;
  &lt;artifactId&gt;my-project&lt;/artifactId&gt;

  &lt;!-- ... --&gt;
&lt;/project&gt;
</pre>
