Shared naming conventions allow teams to collaborate effectively. This rule raises an issue when a pom's <code>artifactId</code> does not match the provided regular expression.


== Noncompliant Code Example

With the default regular expression: <code>[a-z][a-z-0-9]+</code>
----
<project ...>
  <artifactId>My_Project</artifactId>  <!-- Noncompliant -->

  <!-- ... -->
</project>
----


== Compliant Solution

----
<project ...>
  <artifactId>my-project</artifactId>

  <!-- ... -->
</project>
----


