include::../description.adoc[]

== Noncompliant Code Example

----
private void called(int foo) {
  foo==1; // Noncompliant
  if (foo==1) {
    System.out.println("foo\n");
  }
}

public void caller(String [ ] args {
  called(2);
  return 0;
}
----

== Compliant Solution

----
private void called(int foo) {
  foo=1;
  if (foo==1) {
    System.out.println("foo\n");
  }
}

public void caller(String [ ] args {
  called(2);
  return 0;
}
----

include::../see.adoc[]
