<p>Even though the &lt;code&gt;ORDER BY&lt;/code&gt; clause supports using column numbers, doing so makes the code difficult to read and maintain. Therefore the use of column names is preferred.</p>
<h2>Noncompliant Code Example</h2>
<pre>
SELECT FIRST_NAME, LAST_NAME, REGION
FROM PERSONS
ORDER BY 2, 1
</pre>
<h2>Compliant Solution</h2>
<pre>
SELECT FIRST_NAME, LAST_NAME, REGION
FROM PERSONS
ORDER BY LAST_NAME, FIRST_NAME
</pre>
