<p>While the assignment of default parameter values is typically a good thing, it can go very wrong very quickly when mutable objects are used. Thatâ€™s because a new instance of the object <em>is not</em> created for each function invocation. Instead, all invocations share the same instance, and the changes made for one caller are made for all!</p>
<h2>Noncompliant Code Example</h2>
<pre>
def get_attr_array(obj, arr=[]):  # Noncompliant
  props = (name for name in dir(obj) if not name.startswith('_'))
  arr.extend(props)  # after only a few calls, this is a big array!
  return arr
</pre>
<h2>Compliant Solution</h2>
<pre>
def get_attr_array(obj, arr):
  props = (name for name in dir(obj) if not name.startswith('_'))
  arr.extend(props)
  return arr
</pre>
