<p>The &lt;code&gt;READ TABLE …​ WITH KEY …​&lt;/code&gt; statement performs a linear search of &lt;code&gt;STANDARD&lt;/code&gt; tables, which is very inefficient in most cases.</p>
<p>This rule raises an issue when a &lt;code&gt;READ TABLE …​ WITH KEY …​&lt;/code&gt; statement does not finish with &lt;code&gt;BINARY SEARCH&lt;/code&gt;. No issue will be raised for &lt;code&gt;HASHED&lt;/code&gt; and &lt;code&gt;SORTED&lt;/code&gt; tables.</p>
<h2>Noncompliant Code Example</h2>
<pre>
TYPES BEGIN OF t_mytable,
    myfield TYPE i
END OF t_mytable.

DATA myworkarea TYPE t_mytable.

DATA mytable TYPE STANDARD TABLE OF t_mytable.

SORT mytable BY myfield.

READ TABLE mytable
    WITH KEY myfield = 42
    INTO myworkarea. " Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
TYPES BEGIN OF t_mytable,
    myfield TYPE i
END OF t_mytable.

DATA myworkarea TYPE t_mytable.

DATA mytable TYPE STANDARD TABLE OF t_mytable.

SORT mytable BY myfield.

READ TABLE mytable
    WITH KEY myfield = 42
    INTO myworkarea
    BINARY SEARCH. " Compliant

DATA my_hashed_table TYPE HASHED TABLE OF t_mytable
    WITH UNIQUE KEY myfield.

DATA my_sorted_table TYPE SORTED TABLE OF t_mytable
    WITH UNIQUE KEY myfield.

READ TABLE my_hashed_table
    WITH KEY myfield = 42
    INTO myworkarea. " Compliant

READ TABLE my_sorted_table
    WITH KEY myfield = 42
    INTO myworkarea. " Compliant
</pre>
