Leaving an <code>ON-ERROR</code> block empty means that the exception in question is neither handled nor passed forward to callers for handling at a higher level. Suppressing errors rather than handling them could lead to unpredictable system behavior and should be avoided.

== Noncompliant Code Example

----
      /free
        monitor;
          // ...
          on-error *FILE; // Noncompliant
        endmon;
      /end-free
----

== Compliant Solution

----
      /free
        monitor;
          // ...
          on-error *FILE;
            handleFileError();
        endmon;
      /end-free
----

== Exceptions

When a block contains a comment, it is not considered to be empty.

include::../see.adoc[]
