<p>The difference between &lt;code&gt;private&lt;/code&gt; and &lt;code&gt;protected&lt;/code&gt; visibility is that child classes can see and use &lt;code&gt;protected&lt;/code&gt; members, but they cannot see &lt;code&gt;private&lt;/code&gt; ones. Since a &lt;code&gt;final&lt;/code&gt; class will have no children, marking the members of a &lt;code&gt;final&lt;/code&gt; class &lt;code&gt;protected&lt;/code&gt; is confusingly pointless.</p>
<h2>Noncompliant Code Example</h2>
<pre>
class C final {
protected:  // Noncompliant
  void fun();
};
</pre>
<h2>Compliant Solution</h2>
<pre>
class C final {
private:
  void fun();
};
</pre>
<h2>Exceptions</h2>
<p>When overriding a base class function, it is common practice to keep the same visibility as for the base class. This rule ignores &lt;code&gt;protected&lt;/code&gt; functions in a &lt;code&gt;final&lt;/code&gt; class that are overrides of a base class &lt;code&gt;protected&lt;/code&gt; function.</p>
