include::../description.adoc[]

== Noncompliant Code Example

----
CREATE PROCEDURE say_hello(name VARCHAR2) AS -- Noncompliant; name is not used
BEGIN
  DBMS_OUTPUT.PUT_LINE('Hello World');
END;
/
----

== Compliant Solution

----
CREATE PROCEDURE say_hello(name VARCHAR2) AS -- Compliant
BEGIN
  DBMS_OUTPUT.PUT_LINE('Hello ' || name);
END;
/
----

include::../see.adoc[]
