include::../description.adoc[]

== Noncompliant Code Example

----
func isOdd(x:Int) -> Bool {
  return x % 2 == 1  // Noncompliant; if x is negative, x % 2 == -1
}
----

== Compliant Solution

----
func isOdd(x:Int) -> Bool {
  return x % 2 != 0
}
----
or
----
func isOdd(x:Int) -> Bool {
  return abs(x % 2) == 1
}
----

include::../see.adoc[]
