In Java 8, underscore (<code>_</code>) cannot be used as a parameter to a lambda, and by Java 9, it's a keyword. To ensure future portability (not to mention current readability), it should not be used now as an identifier.

== Noncompliant Code Example

----
private String _;  // Noncompliant
----

== Compliant Solution

----
private String foo;
----
