<p>If an &lt;code&gt;enum&lt;/code&gt; memberâ€™s name contains the word "reserved" it implies it is not currently used and will be change in the future. However changing an &lt;code&gt;enum&lt;/code&gt; member is a breaking change and can create significant problems. There is no need to reserve an &lt;code&gt;enum&lt;/code&gt; member since a new member can be added in the future, and such an addition will usually not be a breaking change.</p>
<p>This rule raises an issue when the name of an enumeration member contains "reserved".</p>
<h2>Noncompliant Code Example</h2>
<pre>
using System;

namespace MyLibrary
{
  public enum Color
  {
        None,
        Red,
        Orange,
        Yellow,
        ReservedColor  // Noncompliant
    }
}
</pre>
