<p>Using the value of a pointer to a &lt;code&gt;FILE&lt;/code&gt; object after the associated file is closed is undefined behavior.</p>
<h2>Noncompliant Code Example</h2>
<pre>
void fun() {
  FILE * pFile;
  pFile = fopen(fileName, "w");

  if (condition) {
    fclose(pFile);
    // ...
  }

  fclose(pFile); // Noncompliant, the file has already been closed
}
</pre>
<h2>Compliant Solution</h2>
<pre>
void fun() {
  FILE * pFile;
  pFile = fopen(fileName, "w");

  if (condition) {
    // ...
  }

  fclose(pFile);
}
</pre>
<h2>See</h2>
<ul>
<li><a href="https://www.securecoding.cert.org/confluence/x/KAGQBw">CERT, FIO46-C.</a> - Do not access a closed file</li>
</ul>
