Using the value of a pointer to a <code>FILE</code> object after the associated file is closed is undefined behavior.


== Noncompliant Code Example

----
void fun() {
  FILE * pFile;
  pFile = fopen(fileName, "w");

  if (condition) {
    fclose(pFile);
    // ...
  }

  fclose(pFile); // Noncompliant, the file has already been closed
}
----


== Compliant Solution

----
void fun() {
  FILE * pFile;
  pFile = fopen(fileName, "w");

  if (condition) {
    // ...
  }

  fclose(pFile);
}
----


== See

* https://www.securecoding.cert.org/confluence/x/KAGQBw[CERT, FIO46-C.] - Do not access a closed file

