include::../description.adoc[]

== Noncompliant Code Example

----
func getTitle(person: Person) -> String {
  return person.gender == Gender.MALE ? "Mr. " : person.married ? "Mrs. " : "Miss ";  // Noncompliant
}
----

== Compliant Solution

----
func getTitle(person: Person) -> String {
  let title: String;
  if (person.gender == Gender.MALE) {
     title = "Mr. ";
  }
  else {
     title = person.married ? "Mrs. " : "Miss "; 
  }
  return title;
}
----
