The use of <code>UNDERLINE</code> makes normal text printed to the screen difficult to read, and prevents titles from being read at all. For these reasons, <code>UNDERLINE</code> should not be used.


== Noncompliant Code Example

----
     R DETAIL1
      * Noncompliant; next line contains UNDERLINE
       NAME          25      2  2UNDERLINE
       ADD1          25      3  2
     R DETAIL3
       ADD2          25         2SPACEB(1)
----


== Compliant Solution

----
     R DETAIL1
       NAME          25      2  2
       ADD1          25      3  2
     R DETAIL3
       ADD2          25         2SPACEB(1)
----

