<p>Although the &lt;code&gt;WHERE&lt;/code&gt; condition is optional in a &lt;code&gt;SELECT&lt;/code&gt; statement, for performance and security reasons, a &lt;code&gt;WHERE&lt;/code&gt; clause should always be specified to prevent reading the whole table.</p>
<h2>Noncompliant Code Example</h2>
<pre>
SELECT * FROM db_persons INTO us_persons.
</pre>
<h2>Compliant Solution</h2>
<pre>
SELECT * FROM db_persons INTO us_persons WHERE country IS 'US'.
</pre>
<h2>Exceptions</h2>
<p>&lt;code&gt;SELECT SINGLE&lt;/code&gt; and &lt;code&gt;UP TO 1 ROWS&lt;/code&gt; result in only one record being read, so such &lt;code&gt;SELECT&lt;/code&gt;s are ignored by this rule.</p>
<pre>
SELECT SINGLE * FROM db_persons INTO us_persons.

SELECT * FROM db_persons UP TO 1 ROWS INTO us_persons.
</pre>
