<p>Even though the C++ standard defines both "Primary" and "Alternative" operators, it is <em>not</em> a good idea to use the alternatives. Developers seeing an alphabetical name expect a variable, a function, a class, a namespace…​ in short, anything but an operator, and they will be confused at best by code that uses such operators.</p>
<p>|| Primary || Alternative ||
| &amp;&amp; | and |
| &amp;= | and_eq |
| &amp; | bitand |
| \| | bitor |
| ~ | compl |
| ! | not |
| != | not_eq |
| \|\| | or |
| \|= | or_eq |
| ^ | xor |
| ^= | xor_eq |</p>
<h2>Noncompliant Code Example</h2>
<pre>
if (not valid or error) { // Noncompliant
  /* ... */
}
</pre>
<h2>Compliant Solution</h2>
<pre>
if (!valid || error) {
  /* ... */
}
</pre>
