<p>Shared coding conventions allow teams to collaborate effectively. To improve readability, blank lines in a program should not have end-of-line comments.</p>
<h2>Noncompliant Code Example</h2>
<p>In this example, the second line is blank except for the end-of-line comment, obscuring readability. The seventh line is commented-out, so an end-of-line comment is acceptable.</p>
<pre>
     I/COPY VALSRC,VALTEST                                                FIX001
                                                                          FIX001
      ** Data structure for subroutine VALDS                              FIX001
     I            DS                                                      FIX001
     I                                        5   60MMA                   FIX001
     I                                        7   80DDA                   FIX001
      *                                                                   FIX001
     I            DS                                                      FIX001
     I                                        5   60MMB                   FIX001
     I                                        7   80DDB                   FIX001
</pre>
<h2>Compliant Solution</h2>
<pre>
     I/COPY VALSRC,VALTEST                                                FIX001

      ** Data structure for subroutine VALDS                              FIX001
     I            DS                                                      FIX001
     I                                        5   60MMA                   FIX001
     I                                        7   80DDA                   FIX001
      *                                                                   FIX001
     I            DS                                                      FIX001
     I                                        5   60MMB                   FIX001
     I                                        7   80DDB                   FIX001
</pre>
