<p>Typically you want to use &lt;code&gt;using&lt;/code&gt; to create a local &lt;code&gt;IDisposable&lt;/code&gt; variable; it will trigger disposal of the object when control passes out of the blockâ€™s scope. The exception to this rule is when your method returns that &lt;code&gt;IDisposable&lt;/code&gt;. In that case &lt;code&gt;using&lt;/code&gt; disposes of the object before the caller can make use of it, likely causing exceptions at runtime. So you should either remove &lt;code&gt;using&lt;/code&gt; or avoid returning the &lt;code&gt;IDisposable&lt;/code&gt;.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public FileStream WriteToFile(string path, string text)
{
  using (var fs = File.Create(path)) // Noncompliant
  {
    var bytes = Encoding.UTF8.GetBytes(text);
    fs.Write(bytes, 0, bytes.Length);
    return fs;
  }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
public FileStream WriteToFile(string path, string text)
{
  var fs = File.Create(path);
  var bytes = Encoding.UTF8.GetBytes(text);
  fs.Write(bytes, 0, bytes.Length);
  return fs;
}
</pre>
