For code compliant with C++98 or C++03 standards, declaring overriding virtual functions with the <code>virtual</code> keyword removes the need to check the base class to determine whether a function is virtual.


== Noncompliant Code Example

----
class Base
{
  virtual void f();
};
class Derived : public Base
{
  void f(); // Noncompliant, implicitly declared "virtual"
};
----


== Compliant Solution

----
class Base
{
  virtual void f();
};
class Derived : public Base
{
  virtual void f(); // Compliant, explicitly declared "virtual"
};
----


== See

* MISRA C++:2008, 10-3-2 - Each overriding virtual function shall be declared with the virtual keyword.

=== See also

* For code compliant with C++11 and above see S3471

