<p>Declaring a parameter without specifying its data type leaves the compiler to assign the type that seems the most appropriate - whether itâ€™s what you need or not. Therefore you should always specify the data type.</p>
<h2>Noncompliant Code Example</h2>
<pre>
Sub SubComputeArea(Length, TheWidth)
  [. . .]
End Sub
</pre>
<h2>Compliant Solution</h2>
<pre>
Sub SubComputeArea(Length As Integer, TheWidth As Integer)
  [. . .]
End Sub
</pre>
