include::../description.adoc[]

== Noncompliant Code Example

----
IF something.   
  IF somethingElse.
    WRITE / 'hello'.
  ENDIF.
ENDIF.
----

== Compliant Solution

----
IF something and somethingElse.  
  WRITE / 'hello'.
ENDIF.
----
