Shared coding conventions allow teams to collaborate efficiently. This rule checks that all cursor names match the provided regular expression.


== Noncompliant Code Example

With the default regular expression, <code>[a-zA-Z]([a-zA-Z0-9_]*[a-zA-Z0-9])?</code>:

----
CREATE TABLE employee(
  name VARCHAR2(42)
);

DECLARE
  CURSOR myCursor_ RETURN employee%ROWTYPE; -- Noncompliant

  CURSOR myCursor_ RETURN employee%ROWTYPE IS SELECT * FROM employee; -- Noncompliant
BEGIN
  NULL;
END;
/

DROP TABLE employee;
----


== Compliant Solution

----
CREATE TABLE employee(
  name VARCHAR2(42)
);

DECLARE
  CURSOR myCursor RETURN employee%ROWTYPE;

  CURSOR myCursor RETURN employee%ROWTYPE IS SELECT * FROM employee;
BEGIN
  NULL;
END;
/

DROP TABLE employee;
----

