<p>If you call &lt;code&gt;GetType()&lt;/code&gt; on a &lt;code&gt;Type&lt;/code&gt; variable, the return value will always be &lt;code&gt;typeof(System.Type)&lt;/code&gt;. So thereâ€™s no real point in making that call. The same applies to passing a type argument to &lt;code&gt;IsInstanceOfType&lt;/code&gt;. In both cases the results are entirely predictable.</p>
<h2>Noncompliant Code Example</h2>
<pre>
var type = typeof(int);
var ttype = type.GetType(); //Noncompliant, always typeof(System.Type)

var s = "abc";

if (s.GetType().IsInstanceOfType(typeof(string))) //Noncompliant; false
{ /* ... */ }
</pre>
<h2>Compliant Solution</h2>
<pre>
var s = "abc";

if (s.GetType().IsInstanceOfType("ssss"))
{ /* ... */ }
</pre>
