When using the Backbone.js framework, the array of models inside a collection, <code>collection.models</code>, should not be accessed directly. Doing so bypasses the listeners set on the collection and could put your application in a bad state.

Instead, use <code>get</code>, <code>at</code> or the underscore methods.


== Noncompliant Code Example

----
myCollection.models.forEach(function (model) { });
----


== Compliant Solution

----
myCollection.forEach(function (model) {  });
----

