While working with bitwise operators <code>&</code> or <code>|</code> it is easy to make a typo andÂ write the equivalent logical operators <code>&&</code> or <code>||</code>. This rule is raising issues when the right operand of a logical expression  <code>&&</code> or <code>||</code> is a constant of integral type, as the developer probably meant to use the corresponding bitwise operator <code>&</code> or <code>|</code>.


== Noncompliant Code Example

----
int fun(int a) {
  return a || 4; // Noncompliant, did you mean to use bitwise operator '|'?
}
----


== Compliant Solution

----
int fun(int a) {
 return a | 4;
}
----


