<p>The point of declaring an optional property or parameter is to make explicit the fact that it might contain no valid value, i.e. &lt;code&gt;null&lt;/code&gt; or &lt;code&gt;undefined&lt;/code&gt;. Using a non-null assertion (the &lt;code&gt;!.&lt;/code&gt; operator) will lead to a runtime error if the optional does contain &lt;code&gt;null&lt;/code&gt; or &lt;code&gt;undefined&lt;/code&gt;. Even if the value is tested first, itâ€™s still considered a bad practice to use a non-null assertion.</p>
<h2>Noncompliant Code Example</h2>
<pre>
function doTheThing(foo?: Foo) {
  let s = foo!.bar;  // Noncompliant
}
</pre>
<h2>Compliant Solution</h2>
<pre>
function doTheThing(foo?: Foo) {
  if (foo) {
    let s = foo.bar;
  }
}
</pre>
