include::../description.adoc[]

== Noncompliant Code Example

----
<html>                                    <!-- Noncompliant -->
...
</html>
----

== Compliant Solution

----
<!-- Copyright 2013 SonarSource SA -->
<html>
...
</html>
----
