include::../description.adoc[]

== Noncompliant Code Example

----
if (x != undefined) {
  if (y === 2) {
    // ...
  }
}
----

== Compliant Solution

----
if (x != undefined && y === 2) {
  // ...
}
----
