The <code>void</code> operator evaluates its argument and unconditionally returns <code>undefined</code>. It can be useful in pre-ECMAScript 5 environments, where <code>undefined</code> could be reassigned, but generally, its use makes code harder to understand.


== Noncompliant Code Example

----
void (function() {
   ...
}());
----


== Compliant Solution

----
(function() {
   ...
}());
----


== Exceptions

No issue is raised when <code>void 0</code> is used in place of <code>undefined</code>. 

----
if (parameter === void 0) {...}
----

