Using <code>FETCH FIRST n ROWS ONLY</code> in a <code>SELECT</code> without ordering the results from which the "n first" results are chosen will return a seemingly random set of rows, and is surely a mistake.

== Noncompliant Code Example

----
SELECT fname, lname, city
  FROM people
  WHERE city IS NOT NULL
  FETCH FIRST 10 ROWS ONLY; -- Noncompliant selects 10 random rows
----

== Compliant Solution

----
SELECT fname, lname, city
  FROM people
  WHERE city IS NOT NULL
  ORDER BY birthdate DESC
  FETCH FIRST 10 ROWS ONLY;
----
