This rule raises an issue when a configured Java package or class is used in a JSP file.


== Noncompliant Code Example

Given a disallowed list of <code>java.sql,ArrayList</code>:

----
<%@ page import="java.sql.*" %>     <!-- Noncompliant -->
<% java.util.ArrayList clients; %>  <!-- Noncompliant -->
<% java.lang.String name; %>        <!-- Compliant -->
----

