<p>Complex types can be restricted by means of an &lt;code&gt;xs:restriction&lt;/code&gt; element. But &lt;code&gt;restriction&lt;/code&gt; is anti-inheritance: the derived type has less information than the base type. Further, restriction causes the redefinition of the complex types elements and attributes, which leads to fragility in your XML schema; any change to a super type will require corresponding changes in all restricting subtypes, even those in other schemas. For example, if the base type defines a mandatory "color" element of type &lt;code&gt;xs:int&lt;/code&gt;, the restricting type must also define a "color" element whose type cannot be, say, &lt;code&gt;xs:string&lt;/code&gt;.</p>
<p>&lt;code&gt;xs:extension&lt;/code&gt;, on the other hand, allows the extending element to inherit the base class' elements and add new ones.</p>
<h2>Noncompliant Code Example</h2>
<pre>
&lt;xs:complexType name="Fruit&gt;
  &lt;xs:sequence&gt;
    &lt;xs:element name="name" type="xs:string"/&gt;
    &lt;xs:element name="color" type="xs:string" minOccurs="0"/&gt;
  &lt;/xs:sequence&gt;
&lt;/xs:complexType&gt;
&lt;xs:complexType name="UncoloredFruit"&gt;
  &lt;xs:complexContent&gt;
    &lt;xs:restriction base="Fruit"&gt;  &lt;!-- Noncompliant --&gt;
      &lt;xs:sequence&gt;
        &lt;xs:element name="name" type="xs:string"/&gt;
      &lt;/xs:sequence&gt;
    &lt;/xs:restriction&gt;
  &lt;/xs:complexContent&gt;
&lt;/xs:complexType&gt;
</pre>
<h2>Compliant Solution</h2>
<pre>
&lt;xs:complexType name="FruitBase&gt;
  &lt;xs:sequence&gt;
    &lt;xs:element name="name" type="xs:string"/&gt;
  &lt;/xs:sequence&gt;
&lt;/xs:complexType&gt;
&lt;xs:complexType name="ColoredFruit&gt;
  &lt;xs:sequence&gt;
    &lt;!-- Bottom-up design, using aggregation --&gt;
    &lt;xs:element name="base" type="FruitBase"/&gt;
    &lt;xs:element name="color" type="xs:string" minOccurs="0"/&gt;
  &lt;/xs:sequence&gt;
&lt;/xs:complexType&gt;
</pre>
