<p>A common anti-pattern is to check that a key exists in a dictionary before retrieving its corresponding value and providing a default value otherwise. This pattern works but is less readable than the equivalent call to the built-in dictionary method "get()" with a default value.</p>
<p>Note that if a default value is set for every key of the dictionary it is possible to use pythonâ€™s defaultdict instead.</p>
<p>This rule raises an issue when a key presence is checked before retrieving its value or providing a default value. It only raises an issue when the default value is a hard-coded string, number, list, dictionary or tuple. Computed values will not raise an issue as they can have side-effects.</p>
<h2>Noncompliant Code Example</h2>
<pre>
result = "default"
if "missing" in mydict:
    result = mydict["missing"]  # Noncompliant

if "missing" in mydict:
    result = mydict["missing"]  # Noncompliant
else:
    result = "default"

if "missing" in mydict:
    result = mydict["missing"]  # Compliant. No issue is raised as generate_value() might have some side-effect.
else:
    result = generate_value()
</pre>
<h2>Compliant Solution</h2>
<pre>
result = mydict.get("missing", "default")

# OR, if "default" is the default value for every key

from collections import defaultdict
mydict = defaultdict(lambda: "default")
result =  mydict["missing"]
</pre>
