<p>A thread acquiring a lock on an object that can be accessed across application domain boundaries runs the risk of being blocked by another thread in a different application domain. Objects that can be accessed across application domain boundaries are said to have weak identity. Types with weak identity are:</p>
<ul>
<li>&lt;code&gt;MarshalByRefObject&lt;/code&gt;</li>
<li>&lt;code&gt;ExecutionEngineException&lt;/code&gt;</li>
<li>&lt;code&gt;OutOfMemoryException&lt;/code&gt;</li>
<li>&lt;code&gt;StackOverflowException&lt;/code&gt;</li>
<li>&lt;code&gt;String&lt;/code&gt;</li>
<li>&lt;code&gt;MemberInfo&lt;/code&gt;</li>
<li>&lt;code&gt;ParameterInfo&lt;/code&gt;</li>
<li>&lt;code&gt;Thread&lt;/code&gt;</li>
</ul>
<h2>Noncompliant Code Example</h2>
<pre>
using System;
using System.Threading;

namespace MyLibrary
{
  class Foo
  {
    string myString = "foo";

    void Bar()
    {
      lock(myString) { } // Noncompliant
    }
  }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
using System;
using System.Threading;

namespace MyLibrary
{
  class Foo
  {
    string myString = "foo";
    private readonly Object thisLock = new Object();

    void Bar()
    {
      lock(thisLock) { } // Compliant
    }
  }
}
</pre>
