A thread acquiring a lock on an object that can be accessed across application domain boundaries runs the risk of being blocked by another thread in a different application domain. Objects that can be accessed across application domain boundaries are said to have weak identity. Types with weak identity are:
* <code>MarshalByRefObject</code>
* <code>ExecutionEngineException</code>
* <code>OutOfMemoryException</code>
* <code>StackOverflowException</code>
* <code>String</code>
* <code>MemberInfo</code>
* <code>ParameterInfo</code>
* <code>Thread</code>


== Noncompliant Code Example

----
using System;
using System.Threading;

namespace MyLibrary
{
  class Foo
  {
    string myString = "foo";

    void Bar()
    {
      lock(myString) { } // Noncompliant
    }
  }
}
----


== Compliant Solution

----
using System;
using System.Threading;

namespace MyLibrary
{
  class Foo
  {
    string myString = "foo";
    private readonly Object thisLock = new Object();  

    void Bar()
    {
      lock(thisLock) { } // Compliant
    }
  }
}
----

