<p>The &lt;code&gt;SELECT&lt;/code&gt; statement should be used only to clearly define some new branches in the control flow. As soon as a &lt;code&gt;WHEN&lt;/code&gt; clause contains too many statements this highly decreases the readability of the overall control flow statement. In such case, the content of &lt;code&gt;WHEN&lt;/code&gt; clause should be extracted in a dedicated subroutine.</p>
<h2>Noncompliant Code Example</h2>
<pre>
C                   SELECT
C                   WHEN      X=1
C                   MOVE      A             B
C                   MOVE      C             D
C                   MOVE      E             F
C                   MOVE      J             K
C                   MOVE      L             M
C                   WHEN      X=2
C                   ....
C                   ENDSL
 /free
   select;
     when X=1;
       B = A;
       D = C;
       F = E;
       K = J;
       M = L;
     when X=2;
       ...
   endsl;
 /end-free
</pre>
<h2>Compliant Solution</h2>
<pre>
C                   SELECT
C                   WHEN      X=1
C                   EXSR      SR011
C                   WHEN      X=2
C                   ....
C                   ENDSL
C                   ...
C     SR011         BEGSR
C                   MOVE      A             B
C                   MOVE      C             D
C                   MOVE      E             F
C                   MOVE      J             K
C                   MOVE      L             M
C                   ENDSR
 /free
   select;
     when X=1;
       exsr SR011;
     when X=2;
       ...
   endsl;
 /end-free
</pre>
