<p>Similar to uncalled functions, un-instantiated class and function templates are a potential source of noise and they may be symptomatic of a more serious problem such as missing paths.</p>
<p>Note: Even though a given class template may be instantiated many times, it is possible that some of its member functions are never instantiated.</p>
<h2>Noncompliant Code Example</h2>
<pre>
template &lt; typename T &gt;
class Sample
{
public:
  void inst_mem ( ) { }
  void uninst_mem ( ) { } // Noncompliant, never instantiated
};

Sample&lt;int64_t&gt; s;
s.inst_mem ( ); // Call to s.inst_mem instantiates the member.
// s.uninst_mem is not called within the program and is not instantiated.
</pre>
<h2>Exceptions</h2>
<p>This rule does not apply to libraries.</p>
<h2>See</h2>
<ul>
<li>MISRA C++:2008, 14-7-1</li>
</ul>
