AssertJ assertions methods targeting the same object can be chained instead of using multiple <code>assertThat</code>. It avoids duplication and increases the clarity of the code.

This rule raises an issue when multiples <code>assertThat</code> target the same tested value.


== Noncompliant Code Example

----
assertThat(someList).hasSize(3);
assertThat(someList).contains("something");
----


== Compliant Solution

----
assertThat(someList)
  .hasSize(3)
  .contains("something");
----

