include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression: <code>^[a-z][a-zA-Z0-9]*$</code>:

----
function DoSomething(){...} 
----

== Compliant Solution

----
function doSomething(){...} 
----

== Exceptions

Methods with an <code>@inheritdoc</code> annotation, as well as magic methods (<code>__construct()</code>, <code>__destruct()</code>, <code>__call()</code>, <code>__callStatic()</code>, <code>__get()</code>, <code>__set()</code>, <code>__isset()</code>, <code>__unset()</code>, <code>__sleep()</code>, <code>__wakeup()</code>, <code>__toString()</code>, <code>__invoke()</code>, <code>__set\_state()</code>, <code>__clone()</code>, <code>__debugInfo()</code>) are ignored.

----
function __construct(){...}
function __destruct(){...}

/**
 * {@inheritdoc}
 */
function myFunc(){...}
----
