== Noncompliant Code Example

----
function sayHello(name) {
  console.log("hello " + name);  // ignored
}

function madLib(verb, noun) {
  console.log("I really " + verb + " one or two " + noun);  // Noncompliant
}
----
