Regular expressions can have an https://en.wikipedia.org/wiki/Regular_expression#Implementations_and_running_times[exponential execution time] depending on the pattern and the length of the input string. The example below, for instance,  can lead to a denial of service of the application:
* Pattern: <code>/(a+)+b/</code>
* Input string: <code>aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaacb</code>

It is recommended:
* to fix the hard-coded regex patterns that use CPU intensive features (avoid if possible captures, possessive quantifiers and back-references, for instance replace the above pattern with (/a+b/)).
* when the regex pattern is defined with an user-controlled input, this last should be sanitized in order to escape characters which are part of the https://en.wikipedia.org/wiki/Regular_expression#Syntax[regular expression syntax].

PHP prevents Denial of Service attacks with https://www.php.net/manual/en/pcre.configuration.php[configuration settings] set by default to safe values. lf the _pcre.backtrack_limit_ or _pcre.recursion_limit_ settings are set to higher values than the default values, make sure that it is not too large numbers that will expose the application to Denial of Service in the event of incorrect or malicious regex evaluation. However, despite this mitigation it is recommended to validate/escape user-controlled inputs.

== Noncompliant Code Example

----
$regex = $_GET["regex"];
$input = $_GET["input"];

preg_grep($regex, $input); // Noncompliant
----

== Compliant Solution

----
$regex = $_GET["regex"];
$input = $_GET["input"];

preg_grep(preg_quote($regex), $input); // Compliant
----

include::../see.adoc[]
