Sharing some naming conventions enables teams to collaborate more efficiently. This rule checks that all <code>struct</code> names match a provided regular expression.

Using the default regular expression:  "^[A-Z][a-zA-Z0-9]*$"

== Noncompliant Code Example

----
struct my_struct {
    var one : Int
    var two : Int
}
----

== Compliant Solution

----
struct MyStruct {
    var one : Int
    var two : Int
}
----
