Initializing an alphanumeric field with the literal character <code>' '</code>, <code>'1'</code>, or <code>'0'</code> has the same effect as using the equivalent figurative constant <code>*BLANK</code>, <code>*ON</code>, or <code>*OFF</code>, but the constant version removes ambiguity and makes the code more readable.


== Noncompliant Code Example

----
C                   MOVE      ' '           FLD1
----

----
  /free
    fld1 = ' ';
  /end-free
----


== Compliant Solution

----
C                   MOVE      *BLANK        FLD1
----

----
  /free
    fld1 = *BLANK;
  /end-free
----


