Assignment of <code>false</code> to a pointer type is implicitly converted to a <code>NULL</code> assignment.


== Noncompliant Code Example

----
void foo(bool *p) {
  p = false;  // Noncompliant
}
----

