Use of the bang (<code>!</code>) operator leads to late binding and results in inefficient code. Use the slightly more verbose dot (<code>.</code>) notation instead.


== Noncompliant Code Example

----
MsgBox ObjectWithDefaultProperty!Param
----


== Compliant Solution

----
MsgBox ObjectWithDefaultProperty.DefaultProperty("Param")
----

