Declaring multiple variable on one line is difficult to read and potentially misleading since the <code>As</code> typing only applies to the last variable on the line. 

== Noncompliant Code Example

----
Public AAA, BBB, CCC As String ' AAA and BBB are variants 
----

== Compliant Solution

----
Public AAA as String 
Public BBB as String 
Public CCC as String 
----

include::../see.adoc[]
