include::../description.adoc[]

== Noncompliant Code Example

----
if(condition) 
{
  doSomething();} 
----

== Compliant Solution

----
if(condition) 
{
  doSomething();
} 
----

include::../exceptions.adoc[]
