There are two possible syntaxes for a lambda having only one input parameter with an inferred type: with and without parentheses around that single parameter. The simpler syntax, without parentheses, is more compact and readable than the one with parentheses, and is therefore preferred.

*Note* that this rule is automatically disabled when the project's <code>sonar.java.source</code> is lower than <code>8</code>.


== Noncompliant Code Example

----
(x) -> x * 2
----


== Compliant Solution

----
x -> x * 2
----


