Rather than creating a boxed primitive from a <code>String</code> to extract the primitive value, use the relevant <code>parse</code> method instead. It will be clearer and more efficient.


== Noncompliant Code Example

----
String myNum = "12.2";

float f = (new Float(myNum)).floatValue();  // Noncompliant; creates & discards a Float
----


== Compliant Solution

----
String myNum = "12.2";

float f = Float.parseFloat(myNum);
----


