<code>yield</code> and <code>return</code> only make sense in the context of functions. Using them outside a function raises a <code>SyntaxError</code>. To break out of a loop, use <code>break</code> instead.


== Noncompliant Code Example

----
class MyClass: 
    while True:
        return False #Noncompliant
----


