Sharing some naming conventions is a key point to make it possible for a team to efficiently collaborate.
This rule allows to check that all class names (along with Objective-C interface, protocol and implementation names) match a provided regular expression.

== Noncompliant Code Example

With default provided regular expression <code>^[A-Z][a-zA-Z0-9]*$</code>:

----
// C++
class foo // Noncompliant
{
};

// Objective-C
@interface nonCompliant : NSObject
@end
----

== Compliant Solution

----
// C++
class Foo // Compliant
{
};

// Objective-C
@interface Compliant : NSObject
@end
----
