An Open SQL <code>SELECT</code> statement without an explicit <code>ORDER BY</code> clause will retrieve rows in an unpredictable order. On pool/cluster tables, the current implementation of Open SQL <code>SELECT</code>  returns the result set in the primary key order, but that's not the case for transparent tables. That's why it's safer to always use an <code>ORDER BY</code> clause.


== Noncompliant Code Example

----
OPEN CURSOR C FOR SELECT * FROM SBOOK WHERE CARRID = 'LH '. "NonCompliant
SELECT * FROM FLIGHTS WHERE FLIGHT_NUMBER = 'LH '."NonCompliant
----


== Compliant Solution

----
OPEN CURSOR C FOR SELECT * FROM SBOOK WHERE CARRID = 'LH '
  ORDER BY PRIMARY KEY.
SELECT * FROM FLIGHTS WHERE FLIGHT_NUMBER = 'LH ' ORDER BY PRIMARY KEY.
----

