include::../description.adoc[]

== Noncompliant Code Example

----
def set_name(name)
  name = name
end
----

== Compliant Solution

----
def set_name(name)
  @name = name
end
----

include::../see.adoc[]
