<p>Where the developer uses new versions of standard library functions (e.g. to enhance functionality or add checks of input values), the modified function shall have a new name. However, it is permissible to overload the name to add new parameter types if the functionality is consistent with those of the original. This ensures that the behaviour associated with the name remains consistent. So, for example, if a new version of the &lt;code&gt;sqrt&lt;/code&gt; function is written to check that the input is not negative, the new function shall not be named &lt;code&gt;sqrt&lt;/code&gt;, but shall be given a new name. It is permissible to add a new &lt;code&gt;sqrt&lt;/code&gt; function for a type not present in the library.</p>
<h2>Noncompliant Code Example</h2>
<pre>
int32_t printf ( int32_t a, int32_t b ) // Noncompliant, printf declared in stdio.h
{
  return ( ( a &gt; b ) ? a : b );
}
</pre>
<h2>See</h2>
<ul>
<li>MISRA C++:2008, 17-0-3</li>
</ul>
