<p>&lt;code&gt;@@IDENTITY&lt;/code&gt; returns the last identity column value created on a connection, regardless of the scope. That means it could return the last identity value you produced, or it could return a value generated by a user defined function or trigger, possibly one fired because of your insert. In order to access the last identity value created in your scope, use &lt;code&gt;SCOPE_IDENTITY()&lt;/code&gt; instead.</p>
<h2>Noncompliant Code Example</h2>
<pre>
INSERT ...

SET @id = @@IDENTITY  -- Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
INSERT ...

SET @id = SCOPE_IDENTITY()
</pre>
