include::../description.adoc[]

== Noncompliant Code Example

----
private String code = 'bounteous';

public String calculateCode() {
  doTheThing();
  return code;
}

public String getName() {  // Noncompliant
  doTheThing();
  return code;
}
----

== Compliant Solution

----
private String code = 'bounteous';

public String getCode() {
  doTheThing();
  return code;
}

public String getName() {
  return getCode();
}
----

include::../exceptions.adoc[]
