include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression <code>^[a-z][a-zA-Z0-9]*$</code>:
----
class MyClass {
  var MyField = 1
}
----

== Compliant Solution

----
class MyClass {
  var myField = 1
}
----
