include::../description.adoc[]

== Noncompliant Code Example

----
if ( a == a ) { // always true
  do_z();
}
if ( a != a ) { // always false
  do_y();
}
if ( a == b && a == b ) { // if the first one is true, the second one is too
  do_x();
}
if (a == b || a == b ) { // if the first one is true, the second one is too
  do_w();
}

if (5 / 5) { // always 1
  do_v();
}
if (5 - 5) { // always 0
  do_u();
}
----

== Exceptions

The following are ignored:

* The expression <code>1 << 1</code>
* When an increment or decrement operator is used, ex: <code>*p++ == *p++</code>
* Bitwise operators <code>|, &, ^</code>

include::../see.adoc[]
