Sharing some naming conventions is a key point to make it possible for a team to efficiently collaborate. This rule checks that all variable, property and parameter names match a provided regular expression.

== Noncompliant Code Example

With the default regular expression <code>\^[_$A-Za-z][$A-Za-z0-9]*$|^[_$A-Z][_$A-Z0-9]+$</code>:
----
const foo_bar = 1;
const baz_ = 2;
----

== Compliant Solution

----
const fooBar = 1;
const _baz = 2;
----
