Passing a std::moved variable as a function call argument makes sense only if the variable can actually be moved in that context.
 This is the case if:

* the corresponding argument of the function is an r-value reference.
* the corresponding argument of the function is passed by copy and the argument type has a move constructor.

For other cases, it cannot be achieved or it should be avoided:

* if the corresponding argument is a non-const l-value reference, this is illegal and the code will not compile.
* if the corresponding argument is a const l-value reference, the variable will never be actually moved. So, the std::move is misleading and useless. It should not be used. This rule rule will raise an issue in this case.


== Noncompliant Code Example

----
#include <utility>

class S{};

void f(S s);
void g(const S &s);

void test() {
    S s;
    f(std::move(s));  // compliant: argument copy is constructed by move constructor. s is moved.
    g(std::move(s)); //  noncompliant: s is cast back to const l-value reference. s can never be moved.
}
----


== Compliant Solution

----
#include <utility>

class S{};

void f(S s);
void g(const S &s);

void test() {
    S s;
    f(std::move(s));  // compliant: argument copy is constructed by move constructor. s is moved.
    g(s); //  compliant: no useless and misleading std::move is used.
}
----


== See

* https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es56-write-stdmove-only-when-you-need-to-explicitly-move-an-object-to-another-scope[C++ Core Guidelines - ES.56] - Write "std::move()" only when you need to explicitly move an object to another scope

