include::../description.adoc[]

== Noncompliant Code Example

----
DECLARE
  -- The following line has many trailing whitespaces
  foo VARCHAR2(42) := 'a     
b';
BEGIN
  -- Will misleadingly show 3, counting only the characters 'a', 'b', and the line terminator, but none of the trailing whitespaces
  DBMS_OUTPUT.PUT_LINE(LENGTH(foo));
END;
/
----
