<p>The order in which you &lt;code&gt;close&lt;/code&gt; database-releated resources is crucial. Close a &lt;code&gt;Connection&lt;/code&gt; first, and depending on the database pooling in use, you may no longer be able to truly reach its &lt;code&gt;Statement&lt;/code&gt;s and &lt;code&gt;ResultSet&lt;/code&gt;s to close them, even though the calls are made and execute without error.</p>
<h2>Noncompliant Code Example</h2>
<pre>
Connection conn = null;
PreparedStatement ps = null;
ResultSet rs = null;
try {
  conn = DriverManager.getConnection(connectionString);
  ps = conn.prepareStatement(query);
  rs = ps.executeQuery();
  // ...
} finally {
  try {
    if (conn != null) {
      conn.close();   // Noncompliant; close this last
    }
  } catch (Exception e) {};
  try {
    if (ps != null) {
      ps.close();
    }
  } catch (Exception e) {};
  try {
    if (rs != null) {
      rs.close();
    }
  } catch (Exception e) {};
}
</pre>
<h2>Compliant Solution</h2>
<pre>
Connection conn = null;
PreparedStatement ps = null;
ResultSet rs = null;
try {
  conn = DriverManager.getConnection(connectionString);
  ps = conn.prepareStatement(query);
  rs = ps.executeQuery();
  // ...
} finally {
  try {
    if (rs != null) {
      rs.close();
    }
  } catch (Exception e) {};
  try {
    if (ps != null) {
      ps.close();
    }
  } catch (Exception e) {};
  try {
    if (conn != null) {
      conn.close();
    }
  } catch (Exception e) {};
}
</pre>
