include::../description.adoc[]

== Noncompliant Code Example

----
for i in 1...10 { // noncompliant, loop only executes once
  print("i is \(i)") 
  break
}
----

== Compliant Solution

----
for i in 1...10 {
  print("i is \(i)")
}
----
