include::../description.adoc[]

== Noncompliant Code Example

With a maximum number of 4 parameters:
----
def do_something(param1, param2, param3, param4, param5):
	...
----

== Compliant Solution

----
def do_something(param1, param2, param3, param4):
	...
----
