include::../description.adoc[]

== Noncompliant Code Example

----
MOVE '1' TO OCTET-1 MOVE '2' TO OCTET-2
----

== Compliant Solution

----
MOVE '1' TO OCTET-1 
MOVE '2' TO OCTET-2 
----

== Exceptions

Control flow statements with a single nested statement are ignored.

----
IF x > 0 THEN DISPLAY "positive". *> Compliant
----
