Internal tables can quickly become a source of performance problems if not accessed correctly, <code>SORTED</code> and <code>HASHED</code> tables should always be accessed with the appropriate key or partial key.


== Noncompliant Code Example

----
READ TABLE it INTO work_area INDEX 1.
----


== Compliant Solution

----
READ TABLE it INTO work_area WITH KEY color = 'RED'.
----

