<p>Using the same value on either side of a binary operator is almost always a mistake. In the case of logical operators, it is either a copy/paste error and therefore a bug, or it is simply wasted code, and should be simplified. In the case of bitwise operators and most binary mathematical operators, having the same value on both sides of an operator yields predictable results, and should be simplified.</p>
<h2>Noncompliant Code Example</h2>
<pre>
if ( a == a ) { // always true
  do_z();
}
if ( a != a ) { // always false
  do_y();
}
if ( a == b &amp;&amp; a == b ) { // if the first one is true, the second one is too
  do_x();
}
if (a == b || a == b ) { // if the first one is true, the second one is too
  do_w();
}

if (5 / 5) { // always 1
  do_v();
}
if (5 - 5) { // always 0
  do_u();
}
</pre>
<h2>Exceptions</h2>
<p>The following are ignored:</p>
<ul>
<li>The expression &lt;code&gt;1 &lt;&lt; 1&lt;/code&gt;</li>
<li>When an increment or decrement operator is used, ex: &lt;code&gt;*p == *p&lt;/code&gt;</li>
<li>Bitwise operators &lt;code&gt;|, &amp;, ^&lt;/code&gt;</li>
</ul>
<h2>See</h2>
<ul>
<li><a href="https://www.securecoding.cert.org/confluence/x/NYA5">CERT, MSC12-C.</a> - Detect and remove code that has no effect or is never executed</li>
<li>S1656 - Implements a check on &lt;code&gt;=&lt;/code&gt;.</li>
</ul>
