<p>Using the same value on either side of a binary operator is almost always a mistake. In the case of logical operators, it is either a copy/paste error and therefore a bug, or it is simply wasted code, and should be simplified.</p>
<p>This rule ignores operators &lt;code&gt;+&lt;/code&gt;, &lt;code&gt;*&lt;/code&gt; and &lt;code&gt;||&lt;/code&gt;, and expressions: &lt;code&gt;1=1&lt;/code&gt;, &lt;code&gt;1&lt;&gt;1&lt;/code&gt;, &lt;code&gt;1!=1&lt;/code&gt;, &lt;code&gt;1~=1&lt;/code&gt; and &lt;code&gt;1^=1&lt;/code&gt;.</p>
<h2>Noncompliant Code Example</h2>
<pre>
SELECT code
  FROM Person
  WHERE first_name IS NULL OR first_name IS NULL; -- Noncompliant

SELECT * FROM Users
  INNER JOIN Clients ON Clients.id = Clients.id; -- Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
SELECT code
  FROM Person
  WHERE first_name IS NULL OR last_name IS NULL;

SELECT * FROM Users
  INNER JOIN Clients ON Clients.id = Users.id;
</pre>
<h2>Exceptions</h2>
<p>This rule ignores &lt;code&gt;*&lt;/code&gt;, &lt;code&gt;+&lt;/code&gt;, and &lt;code&gt;=&lt;/code&gt;.</p>
<h2>See</h2>
<ul>
<li><a href="https://www.securecoding.cert.org/confluence/x/NYA5">CERT, MSC12-C.</a> - Detect and remove code that has no effect or is never executed</li>
<li>S1656 - Implements a check on &lt;code&gt;=&lt;/code&gt;.</li>
</ul>
