<p>Using the same value on either side of a binary operator is almost always a mistake. In the case of logical operators, it is either a copy/paste error and therefore a bug, or it is simply wasted code, and should be simplified. In the case of bitwise operators and most binary mathematical operators, having the same value on both sides of an operator yields predictable results, and should be simplified.</p>
<h2>Noncompliant Code Example</h2>
<pre>
     C                   IF        X = X
</pre>
<pre>
      /free
        if a = a; // always true
          doZ();
        endif;
        if a &lt;&gt; a; // always false
          doY();
        endif;
        if a = b and a = b; // if the first one is true, the second one is too
          doX();
        endif;
        if a = b or a = b; // if the first one is true, the second one is too
          doV();
        endif;

        j = 5 / 5; //always 1
        k = 5 - 5; //always 0
      /end-free
</pre>
<h2>Exceptions</h2>
<p>This rule ignores &lt;code&gt;*&lt;/code&gt; and &lt;code&gt;+&lt;/code&gt;.</p>
<h2>See</h2>
<ul>
<li><a href="https://www.securecoding.cert.org/confluence/x/NYA5">CERT, MSC12-C.</a> - Detect and remove code that has no effect or is never executed</li>
<li>S1656 - Implements a check on &lt;code&gt;=&lt;/code&gt;.</li>
</ul>
