<p>Using the same value on either side of a binary operator is almost always a mistake. In the case of logical operators, it is either a copy/paste error and therefore a bug, or it is simply wasted code, and should be simplified. In the case of bitwise operators and most binary mathematical operators, having the same value on both sides of an operator yields predictable results, and should be simplified.</p>
<p>This rule ignores &lt;code&gt;*&lt;/code&gt;, &lt;code&gt;+&lt;/code&gt;.</p>
<h2>Noncompliant Code Example</h2>
<pre>
if a == a { // always true
  doZ()
}
if  a != a  { // always false
  doY()
}
if a == b &amp;&amp; a == b { // if the first one is true, the second one is too
  doX()
}
if a == b || a == b { // if the first one is true, the second one is too
  doW()
}

var j = 5 / 5 //always 1
var k = 5 - 5 //always 0
</pre>
<h2>Exceptions</h2>
<p>Left-shifting 1 onto 1 is common in the construction of bit masks, and is ignored.</p>
<pre>
var i = 1 &lt;&lt; 1; // Compliant
var j = a &lt;&lt; a; // Noncompliant
</pre>
<h2>See</h2>
<ul>
<li><a href="https://www.securecoding.cert.org/confluence/x/NYA5">CERT, MSC12-C.</a> - Detect and remove code that has no effect or is never executed</li>
<li>S1656 - Implements a check on &lt;code&gt;=&lt;/code&gt;.</li>
</ul>
