<p>Multiplying &lt;code&gt;sizeof()&lt;/code&gt; with &lt;code&gt;sizeof()&lt;/code&gt; indicates a logic error.</p>
<h2>Noncompliant Code Example</h2>
<pre>
sizeof(a) * sizeof(b); // Noncompliant
</pre>
