<p>JUnit assertions should not be made from the &lt;code&gt;run&lt;/code&gt; method of a &lt;code&gt;Runnable&lt;/code&gt;, because failed assertions result in &lt;code&gt;AssertionError&lt;/code&gt;s being thrown. If the error is thrown from a thread other than the one that ran the test, the thread will exit but the test wonâ€™t fail.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public void run() {
  // ...
  Assert.assertEquals(expected, actual);  // Noncompliant
}
</pre>
