<p>Sharing some naming conventions is a key point to make it possible for a team to efficiently collaborate. This rule allows to check that all macro names match a provided regular expression.</p>
<h2>Noncompliant Code Example</h2>
<p>With default provided regular expression: ^(<strong>|[a-z0-9_]</strong>)$</p>
<pre>
DEFINE MyMacro.
  ...
END-OF-DEFINITION.
</pre>
<h2>Compliant Solution</h2>
<pre>
DEFINE my_macro.
  ...
END-OF-DEFINITION.
</pre>
