<p>A &lt;code&gt;CROSS JOIN&lt;/code&gt; query will return all records where each row from the first table is combined with each row from the second table. This means that such a query returns the Cartesian product of the sets of rows from the joined tables, which is why it is also know as "Cartesian product query".</p>
<p>Such a query can return a huge amount of data, and therefore should be used only with great caution and only when really needed.</p>
<h2>Noncompliant Code Example</h2>
<pre>
BEGIN
  -- Standard ANSI syntax
  SELECT *
    INTO employeeArray
    FROM employee CROSS JOIN department; -- Noncompliant; explicit cross join
END;
/

BEGIN
  -- Old syntax
  SELECT *
    INTO employeeArray
    FROM employee, department; -- Noncompliant; also a cross join
END;
/
</pre>
