include::../description.adoc[]

== Noncompliant Code Example

----
if equal(myPoint.x, myPoint.x) {  // Noncompliant
  //...
}
----

== Compliant Solution

----
if equal(myPoint.x, myPoint.y) {
  //...
}
----
