<p>Since &lt;code&gt;abstract&lt;/code&gt; classes can’t be instantiated, there’s no point in their having &lt;code&gt;public&lt;/code&gt; or &lt;code&gt;internal&lt;/code&gt; constructors. If there is basic initialization logic that should run when an extending class instance is created, you can by all means put it in a constructor, but make that constructor &lt;code&gt;private&lt;/code&gt; or &lt;code&gt;protected&lt;/code&gt;.</p>
<h2>Noncompliant Code Example</h2>
<pre>
abstract class Base
{
    public Base() // Noncompliant, should be private or protected
    {
      //...
    }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
abstract class Base
{
    protected Base()
    {
      //...
    }
}
</pre>
