The access level defaults to <code>internal</code> if left unspecified. Since that doesn't make sense for most top-level declarations, access levels should always be specified explicitly, even when <code>internal</code> is what's intended.

This rule raises an issue when the access level is not specified on any top-level declaration.


== Noncompliant Code Example

----
class Foo {  // Noncompliant
  // ...
}
----


== Compliant Solution

----
public class Foo {
  // ...
}
----

