<p>The access level defaults to &lt;code&gt;internal&lt;/code&gt; if left unspecified. Since that doesn’t make sense for most top-level declarations, access levels should always be specified explicitly, even when &lt;code&gt;internal&lt;/code&gt; is what’s intended.</p>
<p>This rule raises an issue when the access level is not specified on any top-level declaration.</p>
<h2>Noncompliant Code Example</h2>
<pre>
class Foo {  // Noncompliant
  // ...
}
</pre>
<h2>Compliant Solution</h2>
<pre>
public class Foo {
  // ...
}
</pre>
