<p>A bare &lt;code&gt;raise&lt;/code&gt; statement, i.e. a &lt;code&gt;raise&lt;/code&gt; with no exception provided, will re-raise the last active exception in the current scope. If the "raise" statement is not in an &lt;code&gt;except&lt;/code&gt; or &lt;code&gt;finally&lt;/code&gt; block, no exception is active and a &lt;code&gt;RuntimeError&lt;/code&gt; is raised instead.</p>
<p>If the bare &lt;code&gt;raise&lt;/code&gt; statement is in a function called in an &lt;code&gt;except&lt;/code&gt; statement, the exception caught by the &lt;code&gt;except&lt;/code&gt; will be raised. This works but is hard to understand and maintain. Nothing indicates in the parent &lt;code&gt;except&lt;/code&gt; that the exception will be reraised, and nothing prevents a developer from calling the function in another context.</p>
<p>Note also that using a bare &lt;code&gt;raise&lt;/code&gt; in a &lt;code&gt;finally&lt;/code&gt; block only works when an exception is active, i.e. when an exception from the &lt;code&gt;try&lt;/code&gt; block is not caught or when an exception is raised by an &lt;code&gt;except&lt;/code&gt; block. It will fail in any other case and should not be relied upon. This code smell is handled by rule S5704.</p>
<p>This rule raises an exception when a bare &lt;code&gt;raise&lt;/code&gt; statement is not in an &lt;code&gt;except&lt;/code&gt; or &lt;code&gt;finally&lt;/code&gt; block.</p>
<h2>Noncompliant Code Example</h2>
<pre>
raise  # Noncompliant

def foo():
    raise  # Noncompliant
    try:
        raise  # Noncompliant
    except ValueError as e:
        handle_error()
    except:
        raise
    else:
        raise  # Noncompliant
    finally:
        raise

def handle_error():
    raise  # Noncompliant. This works but is hard to understand.
</pre>
<h2>Compliant Solution</h2>
<pre>
raise ValueError()

def foo():
    raise ValueError()
    try:
        raise ValueError()
    except:
        raise
    else:
        raise ValueError()
    finally:
        raise
</pre>
<h2>See</h2>
<ul>
<li>Python Documentation - <a href="https://docs.python.org/3/reference/simple_stmts.html#raise">The &lt;code&gt;raise&lt;/code&gt; statement</a></li>
</ul>
