A bare <code>raise</code> statement, i.e. a <code>raise</code> with no exception provided, will re-raise the last active exception in the current scope. If the "raise" statement is not in an <code>except</code> or <code>finally</code> block, no exception is active and a <code>RuntimeError</code> is raised instead.

If the bare <code>raise</code> statement is in a function called in an <code>except</code> statement, the exception caught by the <code>except</code> will be raised. This works but is hard to understand and maintain. Nothing indicates in the parent <code>except</code> that the exception will be reraised, and nothing prevents a developer from calling the function in another context.

Note also that using a bare <code>raise</code> in a <code>finally</code> block only works when an exception is active, i.e. when an exception from the <code>try</code> block is not caught or when an exception is raised by an <code>except</code> block. It will fail in any other case and should not be relied upon. This code smell is handled by rule S5704.

This rule raises an exception when a bare <code>raise</code> statement is not in an <code>except</code> or <code>finally</code> block.


== Noncompliant Code Example

----
raise  # Noncompliant

def foo():
    raise  # Noncompliant
    try:
        raise  # Noncompliant
    except ValueError as e:
        handle_error()
    except:
        raise
    else:
        raise  # Noncompliant
    finally:
        raise

def handle_error():
    raise  # Noncompliant. This works but is hard to understand.
----


== Compliant Solution

----
raise ValueError()

def foo():
    raise ValueError()
    try:
        raise ValueError()
    except:
        raise
    else:
        raise ValueError()
    finally:
        raise
----


== See

* Python Documentation - https://docs.python.org/3/reference/simple_stmts.html#raise[The <code>raise</code> statement]

