<p>The execution order of &lt;code&gt;static&lt;/code&gt; initializers is unspecified when they are in different compilation units (files). Relying on a particular initialization order can have nasty repercussions and should therefore be avoided. Even if the code works now, it could change in the future without notice.</p>
<p>If you need to use &lt;code&gt;static&lt;/code&gt; globals, you should put them inside the function that uses them, or create a getter and declare them inside that getter.</p>
<h2>Noncompliant Code Example</h2>
<pre>
static const std::string airports[] = {"GVA", "SFO", "CDG"}; // Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
std::string GetAirportCode(int i) {
  static const std::string airports[] = {"GVA", "SFO", "CDG"};
  return airports[i];
}
</pre>
