The execution order of <code>static</code> initializers is unspecified when they are in different compilation units (files). Relying on a particular initialization order can have nasty repercussions and should therefore be avoided. Even if the code works now, it could change in the future without notice.

If you need to use <code>static</code> globals, you should put them inside the function that uses them, or create a getter and declare them inside that getter.


== Noncompliant Code Example

----
static const std::string airports[] = {"GVA", "SFO", "CDG"}; // Noncompliant
----


== Compliant Solution

----
std::string GetAirportCode(int i) {
  static const std::string airports[] = {"GVA", "SFO", "CDG"};
  return airports[i];
}
----

