There's no point in redundantly defining an <code>abstract</code> method with the same signature as a method in an <code>interface</code> that the class <code>implements</code>. Any concrete child classes will have to implement the method either way.


== Noncompliant Code Example

----
public interface Reportable {
  String getReport();
}

public abstract class AbstractRuleReport implements Reportable{
  public abstract String getReport();  // Noncompliant

  // ...
}
----

