<p>It is a best practice to have variable declarations outside of the loop. Additionally, declaring variables inside a loop is slightly less efficient because memory management is then performed with each iteration of the loop.</p>
<h2>Noncompliant Code Example</h2>
<pre>
BEGIN
  FOR i IN 1..10
  LOOP
    DECLARE
      variableX NUMBER:= 10;
    BEGIN
        variableX:= variableX+i;
        dbms_output.put_line(variableX);
    END;
 END LOOP;
END;
</pre>
<h2>Compliant Solution</h2>
<pre>
DECLARE
  variableX NUMBER:= 0;
BEGIN
  FOR i IN 1..10
  LOOP
    variableX:= variableX+1;
    dbms_output.put_line(variableX);
  END LOOP;
END;
</pre>
