include::../description.adoc[]

== Noncompliant Code Example

With the default threshold of 3:
----
WRITE: / 'Firstname'.
*...
WRITE: / 'Firstname'.
*...
WRITE: / 'Firstname'.
----

== Compliant Solution

----
CONSTANTS: fname    TYPE c LENGTH 9 VALUE 'Firstname',

WRITE: / fname.
*...
WRITE: / fname.
*...
WRITE: / fname.
----

include::../exceptions.adoc[]
