<p>Java’s garbage collection cannot be relied on to clean up everything. Specifically, subclasses of &lt;code&gt;org.eclipse.swt.graphics.Resource&lt;/code&gt; must be manually &lt;code&gt;dispose()&lt;/code&gt;-ed when you’re done with them.
Particularly for the &lt;code&gt;Image&lt;/code&gt; subclass, which retains an open &lt;code&gt;FileHandle&lt;/code&gt; for the life of the instance, failure to properly &lt;code&gt;dispose&lt;/code&gt; of &lt;code&gt;Resource&lt;/code&gt;s can result in a resource leak which could bring first the application and then perhaps the box it’s on to their knees.</p>
<h2>Noncompliant Code Example</h2>
<pre>
import org.eclipse.swt.graphics.Image;

public class MyLeakyView {
  Image myImage = new Image("image/path");  // Noncompliant; not disposed
</pre>
<h2>Compliant Solution</h2>
<pre>
import org.eclipse.swt.graphics.Image;

public class MyView {
  Image myImage = new Image("image/path");

  public void callMeWhenItsDone() {
    myImage.dispose();
  }
</pre>
