Java's garbage collection cannot be relied on to clean up everything. Specifically, subclasses of <code>org.eclipse.swt.graphics.Resource</code> must be manually <code>dispose()</code>-ed when you're done with them. 
Particularly for the <code>Image</code> subclass, which retains an open <code>FileHandle</code> for the life of the instance, failure to properly <code>dispose</code> of <code>Resource</code>s can result in a resource leak which could bring first the application and then perhaps the box it's on to their knees. 


== Noncompliant Code Example

----
import org.eclipse.swt.graphics.Image;

public class MyLeakyView {
  Image myImage = new Image("image/path");  // Noncompliant; not disposed
----


== Compliant Solution

----
import org.eclipse.swt.graphics.Image;

public class MyView {
  Image myImage = new Image("image/path");

  public void callMeWhenItsDone() {
    myImage.dispose();
  }
----

