include::../description.adoc[]

== Noncompliant Code Example

----
if x == 0 {                                        
	doSomething()
} else if x == 1 {
	doSomethingElse()
} 
----

== Compliant Solution

----
if x == 0 {
	doSomething()
} else if x == 1 {
	doSomethingElse()
} else {
	return errors.New("unsupported int")
}
----

include::../see.adoc[]
