<p>While the &lt;code&gt;OCCURS&lt;/code&gt; keyword still works, it is slower than its more modern replacement: &lt;code&gt;DIM&lt;/code&gt;, which should be used instead.</p>
<h2>Noncompliant Code Example</h2>
<pre>
D cussls          ds                  Occurs(200)
D CustNo                         7p 0
D MonthlySls                    11P 2
</pre>
<h2>Compliant Solution</h2>
<pre>
D cussls          ds                  dim(200)
D CustNo                         7p 0
D MonthlySls                    11P 2
</pre>
