While the <code>OCCURS</code> keyword still works, it is slower than its more modern replacement: <code>DIM</code>, which should be used instead.


== Noncompliant Code Example

----
D cussls          ds                  Occurs(200)  
D CustNo                         7p 0                     
D MonthlySls                    11P 2
----


== Compliant Solution

----
D cussls          ds                  dim(200)   
D CustNo                         7p 0                      
D MonthlySls                    11P 2 
----

