include::../description.adoc[]

== Noncompliant Code Example

----
if param = 1.
  Statement.
elseif param = 2.
  Statement.
elseif param = 1.  // Noncompliant
 Statement.
endif.
----

== Compliant Solution

----
if param = 1.
  Statement.
elseif param = 2.
  Statement.
elseif param = 3.
 Statement.
endif.
----

include::../see.adoc[]
