include::../description.adoc[]

== Noncompliant Code Example

With the default threshold of 3:
----
def foo()
  prepare('action random1')    #Noncompliant - "action random1" is duplicated 3 times
  execute('action random1')
  release('action random1')
end
----

== Compliant Solution

----
def foo()
  action1 = 'action random1'
  prepare(action1)
  execute(action1)
  release(action1)
end
----

== Exceptions

To prevent generating some false-positives, literals having 5 or less characters are excluded as well as literals containing only letters, digits and '_'.
