The use of <code>self</code> is optional except when in closure expressions, and when it's needed to distinguish between property names and arguments. For the sake of brevity, <code>self</code> should be omitted when it's not strictly required.

== Noncompliant Code Example

----
class Car {
  var color: Int

  init(color: Int) {
    self.color = color
  }

  func fade() {
    self.color--  // Noncompliant
  }
}
----

== Compliant Solution

----
class Car {
  var color: Int

  init(color: Int) {
    self.color = color
  }

  func fade() {
    color--
  }
}
----
