<p>A regular expression contains a subpattern that will never be matched or. This might happen because a previous alternative always succeeds or already matches everything that this subpattern matches or because a previous repetition consumed anything that this subpattern might match.</p>
<p>In the best case this means that the offending subpattern is merely redundant and should be removed. In the worst case it’s a sign that this regex does not match what it’s intended to and should be reworked.</p>
<h2>Noncompliant Code Example</h2>
<pre>
".*(a?)"   // "a?" will always match the empty string, so the group will always be empty
".*(a*)"   // Same as above
".*(a+)"   // same as ".*(a)", so the "+" is redundant

"a*|b*"    // The "b*" is redundant because "a*" always matches. This will never match a "b".
`a?|b?`    // Same

"[ab]|a"   // The "|a" is redundant because "[ab]" already matches "a"
</pre>
<h2>Compliant Solution</h2>
<pre>
"[^a]*(a?)"
"[^a]*(a*)"
"[^a](a+)"

"a+|b*"    // If there is no a, b* will be tried
`a|b?`     // Same

"[ab]"
</pre>
