<p>This rule allows banning some modules.</p>
<h2>Noncompliant Code Example</h2>
<p>With &lt;code&gt;moduleName&lt;/code&gt; configured with <strong>UT123[a-zA-z]</strong>:</p>
<pre>
CALL UT123.
CALL UT123L.
CALL UT123-L.
CALL WS-UT123.
</pre>
