<p>This rule allows banning certain classes.</p>
<h2>Noncompliant Code Example</h2>
<p>Given parameters:</p>
<ul>
<li>className:System.Console</li>
</ul>
<pre>
System.Console.WriteLine("foo");  // Noncompliant
</pre>
