<p>&lt;code&gt;SQL COUNT(..), MIN(..), MAX(..), SUM(..), AVG(..)&lt;/code&gt; aggregate functions cause the SAP table buffer to be bypassed, so the use of these functions can lead to performance issues.</p>
<h2>Noncompliant Code Example</h2>
<pre>
SELECT COUNT(*)
       FROM persons
       INTO count
       WHERE city = 'NEW YORK'.
</pre>
