<code>SQL COUNT(..), MIN(..), MAX(..), SUM(..), AVG(..)</code> aggregate functions cause the SAP table buffer to be bypassed, so the use of these functions can lead to performance issues.


== Noncompliant Code Example

----
SELECT COUNT(*) 
       FROM persons 
       INTO count 
       WHERE city = 'NEW YORK'. 
----

