According to SQL-92: 
{quote} 
"X BETWEEN Y AND Z" is equivalent to "X>=Y AND X<=Z" 
{quote} 

Even if the <code>BETWEEN</code> predicate is simply syntactic sugar, using it can improve the readability of a SQL WHERE clause, and is therefore preferred. 


== Noncompliant Code Example

----
SELECT * FROM PERSONS 
WHERE AGE >=18 and AGE <=60 
----


== Compliant Solution

----
SELECT * FROM PERSONS 
WHERE AGE BETWEEN 18 and 60 
----

