Empty statements, i.e. <code>:</code>, are usually introduced by mistake, for example because:

* It was meant to be replaced by an actual statement, but this was forgotten.
* There was a typo which resulted in the colon being doubled, i.e. <code>::</code>.

== Noncompliant Code Example

----
Dim i, j As Integer
i = 1 :: j=2   ' Noncompliant - double :
----

== Compliant Solution

----
Dim i, j As Integer
i = 1 : j=2
----

include::../see.adoc[]
