include::../description.adoc[]

== Noncompliant Code Example

With the default threshold of 3:
----
function run() {
  prepare('action1');                              // Non-Compliant - 'action1' is duplicated 3 times
  execute('action1');
  release('action1');
}
----

== Compliant Solution

----
ACTION_1 = 'action1';

function run() {
  prepare(ACTION_1);                             
  execute(ACTION_1);
  release(ACTION_1);
}
----

include::../exceptions.adoc[]
