include::../description.adoc[]

== Noncompliant Code Example

----
public String toString () {
  if (this.collection.isEmpty()) {
    return null; // Noncompliant
  } else {
    // ...
----

== Compliant Solution

----
public String toString () {
  if (this.collection.isEmpty()) {
    return "";
  } else {
    // ...
----

include::../see.adoc[]
