Having two <code>when</code> clauses in a <code>case</code> statement or two branches in an <code>if</code> chain with the same implementation is at best duplicate code, and at worst a coding error. If the same logic is truly needed for both instances, then in an <code>if</code> chain they should be combined, or for a <code>case</code>, duplicates should be refactored. 

== Noncompliant Code Example

----
case i
  when 1
    doFirstThing()
    doSomething()
  when 2
    doSomethingDifferent()
  when 3 # Noncompliant; duplicates case 1's implementation
    doFirstThing()
    doSomething()
  else
    doTheRest()
end

if a >= 0 && a < 10
  doFirstThing()
  doTheThing()
elsif a >= 10 && a < 20
  doTheOtherThing()
elsif a >= 20 && a < 50
  doFirstThing()
  doTheThing()   # Noncompliant; duplicates first condition
else
  doTheRest()
end
----

== Exceptions

Blocks in an <code>if</code> chain that contain a single line of code are ignored, as are blocks in a <code>case</code> statement that contain a single line of code.

----
if a == 
  doSomething()  # no issue, usually this is done on purpose to increase the readability
elsif a == 2
  doSomethingElse()
else
  doSomething()
end
----

But this exception does not apply to <code>if</code> chains without <code>else</code>-s, or to <code>case</code>-es without <code>else</code> clauses when all branches have the same single line of code. In case of <code>if</code> chains with <code>else</code>-s, or of <code>case</code>-es with <code>else</code> clauses, rule S3923 raises a bug. 

----
if a == 1
  doSomething()  # Noncompliant, this might have been done on purpose but probably not
elsif a == 2
  doSomething()
end
----
