<p>This rule allows you to track the usage of the &lt;code&gt;SuppressMessage&lt;/code&gt; attributes and &lt;code&gt;#pragma warning disable&lt;/code&gt; mechanism.</p>
<h2>Noncompliant Code Example</h2>
<pre>
[SuppressMessage("", "S100")]
...

#pragma warning disable S100
...
#pragma warning restore S100
</pre>
