<p>This rule allows you to track the usage of the &lt;code&gt;@SuppressWarnings&lt;/code&gt; mechanism.</p>
<h2>Noncompliant Code Example</h2>
<p>With a parameter value of "unused" :</p>
<pre>
@SuppressWarnings("unused")
@SuppressWarnings("unchecked")  // Noncompliant
</pre>
