<p>There is no reason to concatenate literal strings. Doing so is an exercise is reducing code readability. Instead, the strings should be combined. Similarly, literal strings should not be &lt;code&gt;append&lt;/code&gt;ed to a &lt;code&gt;StringBuffer&lt;/code&gt; or &lt;code&gt;StringBuilder&lt;/code&gt; sequentially, but combined into one call.</p>
<h2>Noncompliant Code Example</h2>
<pre>
String message = "Hello " + "world" + "!";  // Noncompliant
StringBuilder sb = new StringBuilder();
sb.append("I'm pleased").append(" to meet you.");  //Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
String message = "Hello world!";
StringBuilder sb = new StringBuilder();
sb.append("I'm pleased to meet you.");
</pre>
