<p>It is expected that some methods should be called with caution, but others, such as &lt;code&gt;ToString&lt;/code&gt;, are expected to "just work". Throwing an exception from such a method is likely to break callers' code unexpectedly.</p>
<p>An issue is raised when an exception is thrown from any of the following:</p>
<ul>
<li>Event accessors</li>
<li>&lt;code&gt;Object.Equals&lt;/code&gt;</li>
<li>&lt;code&gt;IEquatable.Equals&lt;/code&gt;</li>
<li>&lt;code&gt;GetHashCode&lt;/code&gt;</li>
<li>&lt;code&gt;ToString&lt;/code&gt;</li>
<li>&lt;code&gt;static&lt;/code&gt; constructors</li>
<li>&lt;code&gt;IDisposable.Dispose&lt;/code&gt;</li>
<li>&lt;code&gt;operator ==, !=, &lt;, &gt;, ‚áê, &gt;=&lt;/code&gt;</li>
<li>&lt;code&gt;implicit&lt;/code&gt; cast operators</li>
</ul>
<h2>Noncompliant Code Example</h2>
<pre>
public override string ToString()
{
  if (string.IsNullOrEmpty(Name))
  {
    throw new ArgumentException("...");  // Noncompliant
  }
  //...
</pre>
<h2>Exceptions</h2>
<p>&lt;code&gt;System.NotImplementedException&lt;/code&gt; and its derivatives are ignored.
&lt;code&gt;System.InvalidOperationException&lt;/code&gt;, &lt;code&gt;System.NotSupportedException&lt;/code&gt;, and &lt;code&gt;System.ArgumentException&lt;/code&gt; and their derivatives are ignored in event accessors.</p>
