It is expected that some methods should be called with caution, but others, such as <code>ToString</code>, are expected to "just work". Throwing an exception from such a method is likely to break callers' code unexpectedly.

An issue is raised when an exception is thrown from any of the following: 

* Event accessors
* <code>Object.Equals</code>
* <code>IEquatable.Equals</code>
* <code>GetHashCode</code>
* <code>ToString</code>
* <code>static</code> constructors
* <code>IDisposable.Dispose</code>
* <code>operator ==, !=, <, >, <=, >=</code>
* <code>implicit</code> cast operators


== Noncompliant Code Example

----
public override string ToString()
{
  if (string.IsNullOrEmpty(Name)) 
  {
    throw new ArgumentException("...");  // Noncompliant
  }
  //...
----


== Exceptions

<code>System.NotImplementedException</code> and its derivatives are ignored.
<code>System.InvalidOperationException</code>, <code>System.NotSupportedException</code>, and <code>System.ArgumentException</code> and their derivatives are ignored in event accessors.

