While not technically incorrect, the omission of <code>BEGIN...END</code> can be misleading, and may lead to the introduction of errors during maintenance.

== Noncompliant Code Example

----
IF @flag = 1 -- Noncompliant
  EXEC something;
----

== Compliant Solution

----
IF @flag = 1
    BEGIN
        EXEC something;
    END;
----

include::../see.adoc[]
