<p>Since Java 7 it has been possible to catch multiple exceptions at once. Therefore, when multiple &lt;code&gt;catch&lt;/code&gt; blocks have the same code, they should be combined for better readability.</p>
<p><strong>Note</strong> that this rule is automatically disabled when the projectâ€™s &lt;code&gt;sonar.java.source&lt;/code&gt; is lower than &lt;code&gt;7&lt;/code&gt;.</p>
<h2>Noncompliant Code Example</h2>
<pre>
catch (IOException e) {
  doCleanup();
  logger.log(e);
}
catch (SQLException e) {  // Noncompliant
  doCleanup();
  logger.log(e);
}
catch (TimeoutException e) {  // Compliant; block contents are different
  doCleanup();
  throw e;
}
</pre>
<h2>Compliant Solution</h2>
<pre>
catch (IOException|SQLException e) {
  doCleanup();
  logger.log(e);
}
catch (TimeoutException e) {
  doCleanup();
  throw e;
}
</pre>
