include::../description.adoc[]

== Noncompliant Code Example

----
/* RSA keys set the key length during key generation rather than parameter generation! */
if (type == EVP_PKEY_RSA) {
  if (!EVP_PKEY_CTX_set_rsa_keygen_bits(kctx, 1024)) { // Noncompliant
    goto err;
  }
}
----

== Compliant Solution

----
/* RSA keys set the key length during key generation rather than parameter generation! */
if (type == EVP_PKEY_RSA) {
  if (!EVP_PKEY_CTX_set_rsa_keygen_bits(kctx, 2048)) { // Compliant
    goto err;
  }
}
----

include::../see.adoc[]
