With Java 8, there's no need to write <code>Comparator</code>s that compare primitive values or other <code>Comparable</code>s; they can be generated for you using the <code>Comparator.comparing*</code> functions: <code>comparing</code>, <code>comparingDouble</code>, <code>comparingInt</code>, <code>comparingLong</code>.

*Note* that this rule is automatically disabled when the project's <code>sonar.java.source</code> is lower than <code>8</code>.


== Noncompliant Code Example

----
unparsedFiles.stream()
   .sorted((f1, f2) -> f1.lines - f2.lines)  // Noncompliant
   .limit(30);
----


== Compliant Solution

----
unparsedFiles.stream()
   .sorted(Comparator.comparingInt(UnparsedFile::getLines()))
   .limit(30);
----

