The <code>java.util.function</code> package provides a large array of functional interface definitions for use in lambda expressions and method references. In general it is recommended to use the more specialised form to avoid auto-boxing. For instance <code>IntFunction<Foo></code> should be preferred over <code>Function<Integer, Foo></code>.

This rule raises an issue when any of the following substitution is possible:
||Current Interface||Preferred Interface||
|<code>Function<Integer, R></code>|<code>IntFunction<R></code>|
|<code>Function<Long, R></code>|<code>LongFunction<R></code>|
|<code>Function<Double, R></code>|<code>DoubleFunction<R></code>|
|<code>Function<Double,Integer></code> | <code>DoubleToIntFunction</code> |
|<code>Function<Double,Long></code> | <code>DoubleToLongFunction</code> |
|<code>Function<Long,Double></code> | <code>LongToDoubleFunction</code> |
|<code>Function<Long,Integer></code> | <code>LongToIntFunction</code>|
|<code>Function<R,Integer></code> | <code>ToIntFunction<R></code>|
|<code>Function<R,Long></code> | <code>ToLongFunction<R></code>|
|<code>Function<R,Double></code> | <code>ToDoubleFunction<R></code>|
|<code>Function<T,T></code> | <code>UnaryOperator<T></code> |
|<code>BiFunction<T,T,T></code> | <code>BinaryOperator<T></code> |
|<code>Consumer<Integer></code>|<code>IntConsumer</code>|
|<code>Consumer<Double></code>|<code>DoubleConsumer</code>|
|<code>Consumer<Long></code>|<code>LongConsumer</code>|
|<code>BiConsumer<T,Integer></code> | <code>ObjIntConsumer<T></code>|
|<code>BiConsumer<T,Long></code> | <code>ObjLongConsumer<T></code>|
|<code>BiConsumer<T,Double></code> | <code>ObjDoubleConsumer<T></code>|
|<code>Predicate<Integer></code>|<code>IntPredicate</code>|
|<code>Predicate<Double></code>|<code>DoublePredicate</code>|
|<code>Predicate<Long></code>|<code>LongPredicate</code>|
|<code>Supplier<Integer></code>|<code>IntSupplier</code>|
|<code>Supplier<Double></code>|<code>DoubleSupplier</code>|
|<code>Supplier<Long></code>|<code>LongSupplier</code>|
|<code>Supplier<Boolean></code> | <code>BooleanSupplier</code>|
|<code>UnaryOperator<Integer></code>|<code>IntUnaryOperator</code>|
|<code>UnaryOperator<Double></code>|<code>DoubleUnaryOperator</code>|
|<code>UnaryOperator<Long></code>|<code>LongUnaryOperator</code>|
|<code>BinaryOperator<Integer></code> | <code>IntBinaryOperator</code>|
|<code>BinaryOperator<Long></code> | <code>LongBinaryOperator</code>|
|<code>BinaryOperator<Double></code> | <code>DoubleBinaryOperator</code>|
|<code>Function<T, Boolean></code> | <code>Predicate<T></code>|
|<code>BiFunction<T,U,Boolean></code> | <code>BiPredicate<T,U></code>|


== Noncompliant Code Example

----
public class Foo implements Supplier<Integer> {  // Noncompliant
    @Override
    public Integer get() {
      // ...
    }
}
----


== Compliant Solution

----
public class Foo implements IntSupplier {

  @Override
  public int getAsInt() {
    // ...
  }
}
----


