include::../description.adoc[]

== Noncompliant Code Example

----
IF sy-subrc EQ 42.
  screen-request = 45.
ENDIF.
----

== Compliant Solution

----
answer = 42.
IF sy-subrc EQ answer.
  screen-request = 45.
ENDIF.
----
