include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression: <code>^(_|[a-zA-Z0-9]+)$</code>:

----
func execute_all() {
...
}
----

== Compliant Solution

----
func executeAll() {
...
}
----
