The proper use of white space makes a major contribution to code readability. 

This rule raises an issue when there is not a space character after the beginning and before the end of each comment (<code><!-- ... --></code>), directive (<code><%@ ... %></code>), and expression (<code><% ... %></code>).


== Noncompliant Code Example

----
<!--Do the thing-->  <!-- Noncompliant; missing space at beginning and end of text-->
<%@page import="java.io.*,java.util.*" %> <!-- Noncompliant; missing space at beginning -->
<% String title = "My Page";%> <!-- Noncompliant; missing space at end -->
----


== Compliant Solution

----
<!-- Do the thing -->
<%@ page import="java.io.*,java.util.*" %>
<% String title = "My Page"; %>
----

