<p>The purpose of an abstract class is to provide some heritable behaviors while also defining methods which must be implemented by sub-classes.</p>
<p>A &lt;code&gt;class&lt;/code&gt; with no abstract methods that was made &lt;code&gt;abstract&lt;/code&gt; purely to prevent instantiation should be converted to a concrete &lt;code&gt;class&lt;/code&gt; (i.e. remove the &lt;code&gt;abstract&lt;/code&gt; keyword) with a &lt;code&gt;protected&lt;/code&gt; constructor.</p>
<p>A &lt;code&gt;class&lt;/code&gt; with only &lt;code&gt;abstract&lt;/code&gt; methods and no inheritable behavior should be converted to an &lt;code&gt;interface&lt;/code&gt;.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public abstract class Animal //Noncompliant; should be an interface
{
  abstract void Move();
  abstract void Feed();
}

public abstract class Color //Noncompliant; should be concrete with a private constructor
{
  private int red = 0;
  private int green = 0;
  private int blue = 0;

  public int GetRed()
  {
    return red;
  }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
public interface Animal
{
  void Move();
  void Feed();
}

public class Color
{
  private int red = 0;
  private int green = 0;
  private int blue = 0;

  protected Color()
  {}

  public int GetRed()
  {
    return red;
  }
}

public abstract class Lamp
{
  private bool switchLamp = false;

  public abstract void Glow();

  public void FlipSwitch()
  {
    switchLamp = !switchLamp;
    if (switchLamp)
    {
      Glow();
    }
  }
}
</pre>
