<p>The use of &lt;code&gt;%TYPE&lt;/code&gt; and &lt;code&gt;%ROWTYPE&lt;/code&gt; is an easy way to make sure that a variableâ€™s type always matches the type of the relevant column in an existing table. If the column type changes, then the variable changes with it.</p>
<h2>Noncompliant Code Example</h2>
<pre>
CREATE FUNCTION func1(acc_no IN NUMBER)
   RETURN NUMBER
   IS total NUMBER(11,2); -- Noncompliant
   BEGIN
      SELECT order_total
      INTO total -- total is assigned here: it could be defined as orders.order_total%TYPE
      FROM orders
      WHERE customer_id = acc_no;
      RETURN(total);
    END;
</pre>
<h2>Compliant Solution</h2>
<pre>
CREATE FUNCTION func1(acc_no IN NUMBER)
   RETURN NUMBER
   IS total orders.order_total%TYPE;
   BEGIN
      SELECT order_total
      INTO total
      FROM orders
      WHERE customer_id = acc_no;
      RETURN(total);
   END;
</pre>
