include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression: <code>^[A-Z][a-zA-Z0-9]*</code>
----
enum someEnumeration { // Non-Compliant
    case Bar
}
----

== Compliant Solution

----
enum SomeEnumeration {
    case Bar
}
----
