<p>Because &lt;code&gt;reinterpret_cast&lt;/code&gt; does not perform any type safety validations, it is capable of performing dangerous conversions between unrelated types.</p>
<p>This rule raises an issue when &lt;code&gt;reinterpret_cast&lt;/code&gt; is used.</p>
<h2>Noncompliant Code Example</h2>
<pre>
  class A { public: virtual ~A(){} };
  class B : public A { public: void doSomething(){} };

  void func(A *a) {
    if (B* b = reinterpret_cast&lt;B*&gt;(a)) { // Noncompliant
      b-&gt;doSomething();
    }
  }
</pre>
<h2>Compliant Solution</h2>
<pre>
  class A { public: virtual ~A(){} };
  class B : public A { public: void doSomething(){} };

  void func(A *a) {
    if (B* b = dynamic_cast&lt;B*&gt;(a)) {
      b-&gt;doSomething();
    }
  }
</pre>
<h2>See</h2>
<ul>
<li>CppCoreGuidelines, Type safety profile - Type.1: Donâ€™t use reinterpret_cast.</li>
</ul>
