== Noncompliant Code Example

----
  class A { public: virtual ~A(){} };
  class B : public A { public: void doSomething(){} };

  void func(A *a) {
    if (B* b = reinterpret_cast<B*>(a)) { // Noncompliant
      b->doSomething();
    }
  }
----
