== Noncompliant Code Example

Using the default importing  and returning regular expressions <code>I_\[a-zA-Z_&#92;-0-9\]+</code>, and <code>R_\[a-zA-Z_&#92;-0-9\]+</code>:

----
    METHODS m_fact IMPORTING i1  TYPE i
                       value(i2) TYPE i
                   RETURNING value(factorial) TYPE i.
----
