include::../description.adoc[]

== Noncompliant Code Example

With a maximum number of 4 parameters:
----
public void doSomething(int param1, int param2, int param3, string param4, long param5) 
{
...
}
----

== Compliant Solution

----
public void doSomething(int param1, int param2, int param3, string param4) 
{
...
}
----
