include::../description.adoc[]

== Noncompliant Code Example

----
if(collection.Count >= 0){...}

if(enumerable.Count() < 0){...}

if(array.Length >= 0){...}

bool result = array.Length >=0;
----

== Compliant Solution

----
if (list.Any()) { ... }

if (list.Count > 0) { ... }

if (array.Length >= 42) { ... }
----
