include::../description.adoc[]

== Noncompliant Code Example

----
System.out.println(i>10?"yes":"no");
----

== Compliant Solution

----
if (i > 10) {
  System.out.println(("yes");
} else {
  System.out.println("no");
}
----
