When multiple, adjacent <code>MONITOR</code> statements have duplicate <code>ON-ERROR</code> blocks, they should be merged to consolidate the <code>ON-ERROR</code> logic for cleaner, more readable code. Note that this applies even when there is intervening code outside any <code>MONITOR</code> block.

== Noncompliant Code Example

----
 /free
   monitor; 
      // do something...
   on-error;
      CALLP HandleError(*param);
   endmon;

   // do un-monitored thing

   monitor;
      // do something else...
   on-error;  // Noncompliant
      CALLP HandleError(*param);
   endmon;
 /end-free
----

== Compliant Solution

----
 /free
   monitor; 
      // do something...
      // do un-monitored thing
      // do something else...
   on-error;
      CALLP HandleError(*param);
   endmon;
----
