include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression <code>^([A-Z0-9_]*|[a-z0-9_]*)$</code>:
----
METHOD MyMethod
...
ENDMETHOD.
----

== Compliant Solution

----
METHOD MY_METHOD
...
ENDMETHOD.
----
