include::../description.adoc[]

== Noncompliant Code Example

----
if (x == 0) {
  doSomething();
} else if (x == 1) {
  doSomethingElse();
}
----

== Compliant Solution

----
if (x == 0) {
  doSomething();
} else if (x == 1) {
  doSomethingElse();
} else {
  throw "Unexpected value for x";
}
----

include::../see.adoc[]
