<p>Coding conventions allow teams to collaborate effectively. For maximum standardization and readability, PHP code should use the long &lt;code&gt;&lt;?php ?&gt;&lt;/code&gt; tags or the short-echo &lt;code&gt;&lt;?= ?&gt;&lt;/code&gt; tags; it should not use the other tag variations.</p>
<h2>Noncompliant Code Example</h2>
<pre>
&lt;?
$foo = 1;
?&gt;
</pre>
<h2>Compliant Solution</h2>
<pre>
&lt;?php
$foo = 1;
?&gt;
</pre>
