Coding conventions allow teams to collaborate effectively. For maximum standardization and readability, PHP code should use the long <code><?php ?></code> tags or the short-echo <code><?= ?></code> tags; it should not use the other tag variations.


== Noncompliant Code Example

----
<?
$foo = 1;
?>
----


== Compliant Solution

----
<?php
$foo = 1;
?>
----

