<p>When a test fails due, for example, to infrastructure issues, you might want to ignore it temporarily. But without some kind of notation about why the test is being ignored, it may never be reactivated. Such tests are difficult to address without comprehensive knowledge of the project, and end up polluting their projects.</p>
<p>This rule raises an issue for each ignored test that does not have a &lt;code&gt;WorkItem&lt;/code&gt; attribute nor a comment about why it is being skipped on the right side of the &lt;code&gt;Ignore&lt;/code&gt; attribute.</p>
<h2>Noncompliant Code Example</h2>
<pre>
[TestMethod]
[Ignore]  // Noncompliant
public void Test_DoTheThing()
{
  // ...
}
</pre>
<h2>Compliant Solution</h2>
<pre>
[TestMethod]
[Ignore]  // renable when TCKT-1234 is fixed
public void Test_DoTheThing()
{
  // ...
}
</pre>
<p>or</p>
<pre>
[TestMethod]
[Ignore]
[WorkItem(1234)]
public void Test_DoTheThing()
{
  // ...
}
</pre>
<h2>Exceptions</h2>
<p>The rule doesnâ€™t raise an issue if:
- the test method is also marked with &lt;code&gt;WorkItem&lt;/code&gt; attribute
- there is a comment on the right side of the &lt;code&gt;Ignore&lt;/code&gt; attribute</p>
