When an SWT <code>Image</code> accesses a file directly, it holds the file handle for the life of the image. Do this many times, and the OS may run out of available file handles. At minimum, SWT <code>Image</code>s which directly access files should not be <code>static</code>. At best, they should access their files through <code>ImageDescriptors</code>, which do not hold open file handles.

This rule looks for <code>org.eclipse.swt.graphics.Image</code>s which both directly access a file on the file path and are <code>static</code>.


== Noncompliant Code Example

----
import org.eclipse.swt.graphics.Image;

public class MyView {
  static Image myImage = new Image("path/to/file.png"); // Noncompliant
----


== Compliant Solution

----
import org.eclipse.swt.graphics.Image;
import org.eclipse.jface.resource.ImageDescriptor;

public class MyView {
  static ImageDescription myDescriptor = ImageDescriptor.createFromFile("path/to/file.png");  // Doesn't hold file handle open
  Image myImage = myDescriptor.getImage();
----

