Using the <code>LIKE</code> operator in SQL <code>WHERE</code> conditions can highly impact the performance of the request. The use of this operator should be strongly indicated.


== Noncompliant Code Example

----
SELECT * 
       FROM doktl 
       INTO TABLE text_tab 
       WHERE doktext LIKE srch_str.
----

