<code>Collection.removeIf</code> is more readable and less verbose than using the <code>Iterator.remove</code> idiom. It might also be more performant in some cases, particularly for <code>ArrayList</code> instances.


== Noncompliant Code Example

----
for (Iterator it = items.iterator(); it.hasNext();) {
  if (this.predicate(it.next())) {
    it.remove(); 
  } 
}
----


