<p>When a closable statement contains nested statements, it can quickly become difficult to see which statements are nested and which are not. Thatâ€™s why ending a list of nested statements with &lt;code&gt;END-${STATEMENT-NAME}&lt;/code&gt; is advised.</p>
<h2>Noncompliant Code Example</h2>
<pre>
READ DF-PARAM-SPILOTE AT END
  GO TO F-LECT-SPILOTE.
</pre>
<h2>Compliant Solution</h2>
<pre>
READ DF-PARAM-SPILOTE AT END
  GO TO F-LECT-SPILOTE
END-READ.
</pre>
