When a closable statement contains nested statements, it can quickly become difficult to see which statements are nested and which are not. That's why ending a list of nested statements with <code>END-${STATEMENT-NAME}</code> is advised.


== Noncompliant Code Example

----
READ DF-PARAM-SPILOTE AT END
  GO TO F-LECT-SPILOTE.
----


== Compliant Solution

----
READ DF-PARAM-SPILOTE AT END
  GO TO F-LECT-SPILOTE
END-READ.
----

