According to the specification:
bq. Nonfinal <code>static</code> class fields are disallowed in EJBs because such fields make an enterprise bean difficult or impossible to distribute.

Therefore, all <code>static</code> fields in an EJB should also be <code>final</code>.


== Noncompliant Code Example

----
@Stateless
public class MyEjb
{
  private static String message;  // Noncompliant
}
----

