Functions that use <code>yield</code> are known as "generators". Before Python 3.3, generators cannot <code>return</code> values. Similarly, functions that use <code>return</code> cannot use <code>yield</code>. Doing so will cause a <code>SyntaxError</code>.

Either upgrade your version of Python to a version >= 3.3, or don't use both return and yield in a function.


== Noncompliant Code Example

----
def adder(n):
     num = 0
     while num < n:
         yield num
         num += 1
     return num  #Noncompliant
----

