include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression: ^([A-Z0-9_]*|[a-z0-9_]*)$
----
FUNCTION MyFunction. 
...
ENDFUNCTION.
----

== Compliant Solution

----
FUNCTION MY_FUNCTION. 
...
ENDFUNCTION.
----
