<p>Using a data value clause in the &lt;code&gt;LINKAGE SECTION&lt;/code&gt; can lead to unexpected behavior at runtime.</p>
<h2>Noncompliant Code Example</h2>
<pre>
LINKAGE SECTION.

01 CAR-ID PIC X(20) VALUE IS "VOLVO".   *&gt; Noncompliant

01  EMP-TYPE     PIC X.
   88   FULL-TIME-EMPLOYEE VALUE "F".   *&gt; Compliant; this is a condition name
   88   PART-TIME-EMPLOYEE VALUE "P".

01 TRAIN-ID PIC X(20)
</pre>
