<p>Functions that invoke other &lt;code&gt;inline&lt;/code&gt; functions often become too complex to actually be &lt;code&gt;inline&lt;/code&gt;d, despite the fact that they appear to be small. This problem is especially common with constructors and destructors.</p>
<p>A constructor always invokes the constructors of its base classes and member data before executing its own code.</p>
<p>Code that does not try to &lt;code&gt;inline&lt;/code&gt; constructors and destructors, is more portable because it can be handled by compilers that get confused when generating complex nested &lt;code&gt;inline&lt;/code&gt; functions.</p>
<h2>Noncompliant Code Example</h2>
<pre>
class String
{
  private:
    // ...
  public:
    String();
    ~String();
  };

  inline String::String()      //Noncompliant
  {
    // ...
  }
  inline String::~String()     //Noncompliant
  {
    // ...
  }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
class String
{
  private:
    // ...
  public:
    String();
    ~String();
  };

  String::String()
  {
    // ...
  }
  String::~String()
  {
    // ...
  }
}
</pre>
