Functions that invoke other <code>inline</code> functions often become too complex to actually be <code>inline</code>d, despite the fact that they appear to be small. This problem is especially common with constructors and destructors. 

A constructor always invokes the constructors of its base classes and member data before executing its own code. 

Code that does not try to <code>inline</code> constructors and destructors, is more portable because it can be handled by compilers that get confused when generating complex nested <code>inline</code> functions.


== Noncompliant Code Example

----
class String
{
  private:
    // ...
  public:
    String();
    ~String();
  };

  inline String::String()      //Noncompliant
  {
    // ...
  }
  inline String::~String()     //Noncompliant
  {
    // ...
  }
}
----


== Compliant Solution

----
class String
{
  private:
    // ...
  public:
    String();
    ~String();
  };

  String::String()
  {
    // ...
  }
  String::~String()
  {
    // ...
  }
}
----

