<p>Changing an inherited member to &lt;code&gt;private&lt;/code&gt; will not prevent access to the base class implementation.</p>
<p>This rule raises an issue when a &lt;code&gt;private&lt;/code&gt; method in an unsealed type has a signature that is identical to a &lt;code&gt;public&lt;/code&gt; method declared in a base type.</p>
<h2>Noncompliant Code Example</h2>
<pre>
using System;

namespace MyLibrary
{
  public class Foo
  {
    public void SomeMethod(int count) { }
  }
  public class Bar:Foo
  {
    private void SomeMethod(int count) { } // Noncompliant
  }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
using System;

namespace MyLibrary
{
  public class Foo
  {
    public void SomeMethod(int count) { }
  }
  public sealed class Bar : Foo
  {
    private void SomeMethod(int count) { }
  }
}
</pre>
