Changing an inherited member to <code>private</code> will not prevent access to the base class implementation.

This rule raises an issue when a <code>private</code> method in an unsealed type has a signature that is identical to a <code>public</code> method declared in a base type.


== Noncompliant Code Example

----
using System;

namespace MyLibrary
{
  public class Foo
  {
    public void SomeMethod(int count) { }
  }
  public class Bar:Foo
  {
    private void SomeMethod(int count) { } // Noncompliant
  }
}
----


== Compliant Solution

----
using System;

namespace MyLibrary
{
  public class Foo
  {
    public void SomeMethod(int count) { }
  }
  public sealed class Bar : Foo
  {
    private void SomeMethod(int count) { }
  }
}
----

