<p>The use of the &lt;code&gt;enum.ordinal()&lt;/code&gt; method is regarded as suspicious since conceptually, the significance of an &lt;code&gt;enum&lt;/code&gt; member is in what it represents, not its position within the &lt;code&gt;enum&lt;/code&gt;.</p>
<p>According to the method’s Javadoc:
bq. Most programmers will have no use for this method. It is designed for use by sophisticated enum-based data structures, such as &lt;code&gt;EnumSet&lt;/code&gt; and &lt;code&gt;EnumMap&lt;/code&gt;.</p>
<p>Further, reliance on a particular member’s position within an &lt;code&gt;enum&lt;/code&gt; is a bug waiting to happen since the position of members within an &lt;code&gt;enum&lt;/code&gt; is not generally regarded as significant, and maintainers are just as likely to add new members at the beginning as at the end.</p>
<p>Therefore this rule raises an issue for each use of the &lt;code&gt;ordinal&lt;/code&gt; method.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public enum Fruit {
  APPLE, ORANGE, PLUM, GRAPE;
}

public void doTheThing(Fruit fruit) {

  if (fruit.ordinal() == 2) {  // Noncompliant
    // ...
  }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
public enum Fruit {
  APPLE, ORANGE, PLUM, GRAPE;
}

public void doTheThing(Fruit fruit) {

  if (Fruit.PLUM.equals(fruit)) {
    // ...
  }
}
</pre>
