<p>&lt;code&gt;goto&lt;/code&gt; is an unstructured control flow statement. It makes code less readable and maintainable. Structured control flow statements such as &lt;code&gt;if&lt;/code&gt;, &lt;code&gt;for&lt;/code&gt;, &lt;code&gt;while&lt;/code&gt;, &lt;code&gt;continue&lt;/code&gt; or &lt;code&gt;break&lt;/code&gt; should be used instead.</p>
<h2>Noncompliant Code Example</h2>
<pre>
$i = 0;
loop:
  echo("i = $i");
  $i++;
  if ($i &lt; 10){
    goto loop;
  }
</pre>
<h2>Compliant Solution</h2>
<pre>
for ($i = 0; $i &lt; 10; $i++){
  echo("i = $i");
}
</pre>
