<p>&lt;code&gt;GOTO&lt;/code&gt; is an unstructured control flow statement. It makes code less readable and maintainable, and should only be used to branch to defined &lt;code&gt;ENDSR&lt;/code&gt; points within the source. For normal loop functions use &lt;code&gt;ITER&lt;/code&gt; or &lt;code&gt;LEAVE&lt;/code&gt; instead.</p>
<h2>Noncompliant Code Example</h2>
<pre>
C     LOOP1         TAG
...
C                   GOTO      LOOP1
</pre>
