<p>&lt;code&gt;GoTo&lt;/code&gt; is an unstructured control flow statement. It makes code less readable and maintainable. Structured control flow statements such as &lt;code&gt;If&lt;/code&gt;, &lt;code&gt;For&lt;/code&gt;, &lt;code&gt;While&lt;/code&gt;,  or &lt;code&gt;Exit&lt;/code&gt; should be used instead.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    Sub GoToStatementDemo()
        Dim number As Integer = 1
        Dim sampleString As String
        ' Evaluate number and branch to appropriate label.
        If number = 1 Then GoTo Line1 Else GoTo Line2
Line1:
        sampleString = "Number equals 1"
        GoTo LastLine
Line2:
        ' The following statement never gets executed because number = 1.
        sampleString = "Number equals 2"
LastLine:
        ' Write "Number equals 1" in the Debug window.
        Debug.WriteLine(sampleString)
    End Sub
</pre>
<h2>Compliant Solution</h2>
<pre>
    Sub GoToStatementDemo()
        Dim number As Integer = 1
        Dim sampleString As String
        ' Evaluate number and branch to appropriate label.
        If number = 1 Then
            sampleString = "Number equals 1"
        Else
            sampleString = "Number equals 2"
        End If
        Debug.WriteLine(sampleString)
    End Sub
</pre>
