include::../description.adoc[]

== Noncompliant Code Example

----
public Integer numberOfMinutes(Integer hours) {
  Integer seconds = 0;   // seconds is never used 
  return hours * 60;
}
----

== Compliant Solution

----
public Integer numberOfMinutes(Integer hours) {
  return hours * 60;
}
----
