<p>This rule applies whenever an &lt;code&gt;if&lt;/code&gt; statement is followed by one or more &lt;code&gt;else if&lt;/code&gt; statements; the final &lt;code&gt;else if&lt;/code&gt; should be followed by an &lt;code&gt;else&lt;/code&gt; statement.</p>
<p>The requirement for a final &lt;code&gt;else&lt;/code&gt; statement is defensive programming.
The &lt;code&gt;else&lt;/code&gt; statement should either take appropriate action or contain a suitable comment as to why no action is taken. This is consistent with the requirement to have a final &lt;code&gt;default&lt;/code&gt; clause in a &lt;code&gt;switch&lt;/code&gt; statement.</p>
<h2>Noncompliant Code Example</h2>
<pre>
if (x == 0) {
  doSomething();
} else if (x == 1) {
  doSomethingElse();
}
</pre>
<h2>Compliant Solution</h2>
<pre>
if (x == 0) {
  doSomething();
} else if (x == 1) {
  doSomethingElse();
} else {
  error();
}
</pre>
<h2>Exceptions</h2>
<p>When all branches of an &lt;code&gt;if&lt;/code&gt;-&lt;code&gt;else if&lt;/code&gt; end with &lt;code&gt;return&lt;/code&gt;, &lt;code&gt;break&lt;/code&gt; or &lt;code&gt;throw&lt;/code&gt;, the code that comes after the &lt;code&gt;if&lt;/code&gt; implicitly behaves as if it was in an &lt;code&gt;else&lt;/code&gt; clause. This rule will therefore ignore that case.</p>
<h2>See</h2>
<ul>
<li>MISRA C:2004, 14.10 - All if…​else if constructs shall be terminated with an else clause.</li>
<li>MISRA C++:2008, 6-4-2 - All if…​else if constructs shall be terminated with an else clause.</li>
<li>MISRA C:2012, 15.7 - All if…​else if constructs shall be terminated with an else statement</li>
<li><a href="https://www.securecoding.cert.org/confluence/x/YgE">CERT, MSC01-C.</a> - Strive for logical completeness</li>
<li><a href="https://www.securecoding.cert.org/confluence/x/PQHRAw">CERT, MSC57-J.</a> - Strive for logical completeness</li>
</ul>
