Using a <code><li></code> or <code><dt></code> item tag outside of a <code><ul></code>, <code><ol></code> or <code><dl></code> one does not make sense and indicates a bug.


== Noncompliant Code Example

----
<li>Apple</li>          <!-- Noncompliant -->
<li>Strawberry</li>     <!-- Noncompliant -->

<li>Apple</li>          <!-- Noncompliant -->
<li>Strawberry</li>     <!-- Noncompliant -->

<dt>Apple</dt>          <!-- Noncompliant -->
<dt>Strawberry</dt>     <!-- Noncompliant -->
----


== Compliant Solution

----
<ul>
  <li>Apple</li>
  <li>Strawberry</li>
</ul>

<ol>
  <li>Apple</li>
  <li>Strawberry</li>
</ol>

<dl>
  <dt>Apple</dt>
  <dt>Strawberry</dt>
</dl>
----


