When using some transaction managers like IBM IMS, each COBOL program is in fact considered a sub-program by the transaction manager. The <code>GOBACK</code> statement returns control to the transaction manager, but using <code>STOP RUN</code> might cause unpredictable results or abnormal termination.


== Noncompliant Code Example

----
STOP RUN
----


== Compliant Solution

----
GOBACK
----


