include::../description.adoc[]

== Noncompliant Code Example

----
if (a) {
  if (a) { // Noncompliant
    doSomething();
  }
}
----

== Compliant Solution

----
if (a) {
  if (b) {
    doSomething();
  }
}

// or 
if (a) {
  doSomething();
}
----

include::../see.adoc[]
