<p>Modifying SAP standard tables via an Open SQL statement can result in data corruption and unexpected behavior. This is because a direct modification bypasses all validation mechanisms.
Instead, the use of standard functions or SAP Business Add-Ins (BAdIs) is recommended.</p>
<p>This rule raises an issue on Open SQL statements &lt;code&gt;INSERT&lt;/code&gt;, &lt;code&gt;UPDATE&lt;/code&gt;,&nbsp;&lt;code&gt;MODIFY&lt;/code&gt;, or &lt;code&gt;SELECT …​ FOR UPDATE&lt;/code&gt; which target an SAP table, i.e. a table with a name starting with a character between "A" and "X", case insensitive.</p>
<h2>Noncompliant Code Example</h2>
<pre>
SELECT my_col FROM my_system_table WHERE my_id = 123 FOR UPDATE. " Noncompliant

SELECT my_col FROM z_custom_table WHERE my_id = 123 FOR UPDATE. " Ignored; modifies a custom table
</pre>
