include::../description.adoc[]

== Noncompliant Code Example

----
Class Person
    Private age As Integer

    <Pure> ' Noncompliant. In this case the method makes a possibly visible state change
    Private Sub ConfigureAge(ByVal age As Integer)
        ...
        Me.age = age
    End Sub
End Class
----

== Compliant Solution

----
Class Person
    Private age As Integer

    Private Sub ConfigureAge(ByVal age As Integer)
        Me.age = age
    End Sub
End Class
----
