Always having a <code>RETURN</code> as the last statement in a function is a good practice for two reasons:
* It prevents the <code>ORA-06503 PL/SQL: Function returned without value</code> error.
* It prevents unreachable code from being added after the <code>RETURN</code>.


== Noncompliant Code Example

----
CREATE FUNCTION incorrectFunction1 RETURN PLS_INTEGER IS -- Non-Compliant
BEGIN
  RETURN 42;

   -- This is unreachable code
   NULL;
END;
/

DROP FUNCTION incorrectFunction1;

CREATE FUNCTION incorrectFunction2 RETURN PLS_INTEGER IS -- Non-Compliant
BEGIN
   NULL; -- This function was expected to return a PLS_INTEGER, but did not. Will lead to ORA-06503
END;
/

BEGIN
  DBMS_OUTPUT.PUT_LINE('Ret = ' || incorrectFunction2); -- ORA-06503 PL/SQL: Function returned without value
END;
/

DROP FUNCTION incorrectFunction2;

CREATE FUNCTION correctFunction RETURN PLS_INTEGER IS -- Compliant
BEGIN
  RETURN 42;
END;
/

DROP FUNCTION correctFunction;
----

