This rule checks that the specified attributes are present in HTML tags.


== Noncompliant Code Example

Given a required attribute list of: <code>img.alt,img.height,img.width</code>:

----
<img src="/images/queen.png"> <!-- Noncompliant; missing all required attributes -->
----


== Compliant Solution

----
<img src="/images/queen.png" width="60" height="85" alt="Elizabeth II">
----

