A writable collection property can be replaced by a completely different collection. Making it <code>readonly</code> prevents that while still allowing individual members to be set. If you want to allow the replacement of the whole collection the recommended pattern is to implement a method to remove all the elements (e.g. <code>System.Collections.List<T>.Clear</code>) and a method to populate the collection (e.g. <code>System.Collections.List<T>.AddRange</code>).

This rule raises an issue when an externally visible writable property is of a type that implements <code>System.Collections.ICollection</code> or <code>System.Collections.Generic.ICollection<T></code>.


== Noncompliant Code Example

----
using System;
using System.Collections;

namespace MyLibrary
{
  public class Foo
  {
    List<string> strings;

    public List<string> SomeStrings
    {
      get { return strings; }
      set { strings = value; } // Noncompliant
    }
  }
}
----


== Compliant Solution

----
using System;
using System.Collections;

namespace MyLibrary
{
  public class Foo
  {
    List<string> strings;

    public readonly List<string> SomeStrings
    {
      get { return strings; }
    }
  }
}
----


== Exceptions

This rule does not raise issues for

* <code>string</code>, <code>Array</code> and <code>PermissionSet,</code>
* properties marked as <code>DataMemberAttribute</code>
* classes marked as <code>Serializable</code>
* properties overriding a base class member
* properties implementing interface

Â 

