include::../description.adoc[]

== Noncompliant Code Example

----
class MyClass {
  private $foo = 4;                       //foo is unused

  public function compute($a) {
    return $a * 4;
  }
}
----

== Compliant Solution

----
class MyClass {

  public function compute($a) {
    return $a * 4;
  }
}
----
