The complexity of an expression is defined by the number of <code>AND</code>, <code>OR</code>, <code>XOR</code> and <code>EQUIV</code> operators it contains.
A single expression's complexity should not become too high to keep the code readable.

== Noncompliant Code Example

With the default threshold value of 3

----
IF ((condition1 AND condition2) OR (condition3 AND condition4)) AND condition5.
  ...
ENDIF.
----
