The implementation of certain <code>ResultSet</code> methods is optional for result sets of type <code>TYPE_FORWARD_ONLY</code>. Even if your current JDBC driver does implement those methods, there's no guarantee you won't change drivers in the future. 

This rule looks for invocations of the following methods on <code>TYPE_FORWARD_ONLY</code> <code>ResultSet</code>s:

* <code>isBeforeFirst</code>
* <code>isAfterLast</code>
* <code>isFirst</code>
* <code>getRow</code>


== Noncompliant Code Example

----
Statement stmt = con.createStatement(ResultSet.TYPE_FORWARD_ONLY);
stmt.executeQuery("SELECT name, address FROM PERSON");
ResultSet rs = stmt.getResultSet();

if (rs.isBeforeFirst()) { // Noncompliant

}
----


== See Also

* S2232 - "ResultSet.isLast()" should not be used

