<p>Variables declared without the explicit specification of a data type are &lt;code&gt;Variants&lt;/code&gt;. Variants can be inefficient to use because at each interaction they are converted to the appropriate type for that interaction. Variants may be required for COM interactions, but even then their type should be specified explicitly.</p>
<h2>Noncompliant Code Example</h2>
<pre>
Dim Count
Dim Bool
</pre>
<h2>Compliant Solution</h2>
<pre>
Dim Count As Integer
Dim Bool As Boolean
</pre>
