<p>Returning &lt;code&gt;null&lt;/code&gt; from a non-&lt;code&gt;async&lt;/code&gt; &lt;code&gt;Task&lt;/code&gt;/&lt;code&gt;Task&lt;T&gt;&lt;/code&gt; method will cause a &lt;code&gt;NullReferenceException&lt;/code&gt; at runtime. This problem can be avoided by returning &lt;code&gt;Task.FromResult&lt;T&gt;(null)&lt;/code&gt; instead.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public Task&lt;object&gt; GetFooAsync()
{
    return null; // Noncompliant
}
</pre>
<h2>Compliant Solution</h2>
<pre>
public Task&lt;object&gt; GetFooAsync()
{
    return Task.FromResult&lt;object&gt;(null);
}
</pre>
