include::../description.adoc[]

== Noncompliant Code Example

----
using (var tripleDES = new TripleDESCryptoServiceProvider()) //Noncompliant
{
  //...
}
----

== Compliant Solution

----
using (var aes = new AesCryptoServiceProvider())
{
  //...
}
----

include::../see.adoc[]
