<p>A common anti-pattern is to check that a key does not exist in a dictionary before adding it with a corresponding value. This pattern works but is less readable than the equivalent call to the built-in dictionary method "setdefault()".</p>
<p>Note that if a default value is set for every key of the dictionary it is possible to use pythonâ€™s &lt;code&gt;defaultdict&lt;/code&gt; instead.</p>
<p>This rule raises an issue when a key presence is checked before being set. It only raises an issue when the value is a hard-coded string, number, list, dictionary or tuple. Computed values will not raise an issue as they can have side-effects.</p>
<h2>Noncompliant Code Example</h2>
<pre>
if "key" not in my_dictionary:
    my_dictionary["key"] = ["a", "b", "c"]  # Noncompliant

if "key" not in my_dictionary:
    my_dictionary["key"] = generate_value()  # Compliant. No issue is raised as generate_value() might have some side-effect.
</pre>
<h2>Compliant Solution</h2>
<pre>
my_dictionary.setdefault("key", ["a", "b", "c"])

# OR, if ["a", "b", "c"] is the default value for every key

from collections import defaultdict
my_dictionary = defaultdict(lambda: ["a", "b", "c"])
</pre>
