include::../description.adoc[]

== Noncompliant Code Example

----
public int NumberOfMinutes(int hours)
{
  int seconds = 0;   // seconds is never used 
  return hours * 60;
}
----

== Compliant Solution

----
public int NumberOfMinutes(int hours)
{
  return hours * 60;
}
----

== Exceptions

Unused locally created resources in a <code>using</code> statement are not reported.

----
using(var t = new TestTimer()) // t never used, but compliant.
{
  //...
}
----
