<p>Properties are accessed like fields which makes them easier to use.</p>
<p>This rule raises an issue when the name of a &lt;code&gt;public&lt;/code&gt; or &lt;code&gt;protected&lt;/code&gt; method starts with &lt;code&gt;Get&lt;/code&gt;, takes no parameter, and returns a value that is not an array.</p>
<h2>Noncompliant Code Example</h2>
<pre>
using System;

namespace MyLibrary
{
    public class Foo
    {
        private string name;

        public string GetName()  // Noncompliant
        {
            return name;
        }
    }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
using System;

namespace MyLibrary
{
    public class Foo
    {
        private string name;

        public string Name
        {
            get
            {
                return name;
            }
        }
    }
}
</pre>
<h2>Exceptions</h2>
<p>The rule doesnâ€™t raise an issue when the method:</p>
<ul>
<li>Is a constructor</li>
<li>Is an &lt;code&gt;override&lt;/code&gt;</li>
<li>Is an interface implementation</li>
<li>Is &lt;code&gt;async&lt;/code&gt;</li>
<li>Returns &lt;code&gt;Task&lt;/code&gt;, &lt;code&gt;Task&lt;T&gt;&lt;/code&gt;</li>
<li>Is named &lt;code&gt;GetEnumerator&lt;/code&gt;, &lt;code&gt;GetAwaiter&lt;/code&gt;</li>
</ul>
