Properties are accessed like fields which makes them easier to use.

This rule raises an issue when the name of a <code>public</code> or <code>protected</code> method starts with <code>Get</code>, takes no parameter, and returns a value that is not an array.


== Noncompliant Code Example

----
using System;

namespace MyLibrary
{
    public class Foo
    {
        private string name;

        public string GetName()  // Noncompliant
        {
            return name;
        }
    }
}
----


== Compliant Solution

----
using System;

namespace MyLibrary
{
    public class Foo
    {
        private string name;

        public string Name
        {
            get
            {
                return name;
            }
        }
    }
}
----


== Exceptions

The rule doesn't raise an issue when the method:

* Is a constructor
* Is an <code>override</code>
* Is an interface implementation
* Is <code>async</code>
* Returns <code>Task</code>, <code>Task<T></code>
* Is named <code>GetEnumerator</code>, <code>GetAwaiter</code>

