include::../description.adoc[]

== Noncompliant Code Example

----
for ($i = 0; $i < $length; $i--) { // Noncompliant
  //...
}
----

== Compliant Solution

----
for ($i = 0; $i < $length; $i++) {
  //...
}
----

include::../see.adoc[]
