include::../description.adoc[]

== Noncompliant Code Example

----
// the two statements seems to be attached to the if statement, but that is only true for the first one:
if (condition)
  ExecuteSomething();
  CheckSomething();
----

== Compliant Solution

----
if (condition)
{
  ExecuteSomething();
  CheckSomething();
}
----

include::../see.adoc[]
