Copybooks should be used only to share data definitions or logic. The following keywords relate to the nature or structure of a COBOL program, and should be defined directly in the source code of the COBOL program:

* <code>IDENTIFICATION DIVISION.</code>
* <code>PROGRAM-ID xxxxxxxx.</code>
* <code>AUTHOR. yyyyyyyyyyy.</code>
* <code>INSTALLATION.  zzzzzz.</code>
* <code>DATE-WRITTEN. zzzzzz.</code>
* <code>DATE-COMPILED. zzzzzz.</code>
* <code>ENVIRONNEMENT DIVISION.</code>
* <code>CONFIGURATION SECTION.</code>
* <code>SOURCE-COMPUTER. xxxxxx.</code>
* <code>OBJECT-COMPUTER. xxxxxx.</code>
* <code>SPECIAL-NAMES. DECIMAL-POINT IS COMMA.</code>
* <code>I-O CONTROL.</code>
* <code>FILE-CONTROL.</code>
* <code>SELECT â€¦</code>
* <code>DATA DIVISION.</code>
* <code>FILE SECTION.</code>
* <code>WORKING-STORAGE SECTION.</code>
* <code>SCREEN.</code>
* <code>REPORT.</code>
* <code>INPUT-OUTPUT SECTION.</code>
* <code>LINKAGE SECTION.</code>
* <code>PROCEDURE DIVISION.</code>


== Noncompliant Code Example

----
LINKAGE SECTION.
COPY CSCEMOD1.
MOVE A TO B
----


== Compliant Solution

----
LINKAGE SECTION.
COPY CSCEMOD1.
PROCEDURE DIVISION
COPY CSCEMOD2.
MOVE A TO B
----


