<p>Calling &lt;code&gt;Environment.Exit(exitCode)&lt;/code&gt; or &lt;code&gt;Application.Exit()&lt;/code&gt; terminates the process and returns an exit code to the operating system..
Each of these methods should be used with extreme care, and only when the intent is to stop the whole application.</p>
<h2>Noncompliant Code Example</h2>
<pre>
Environment.Exit(0);
Application.Exit();
</pre>
<h2>Exceptions</h2>
<p>These methods are ignored inside &lt;code&gt;Main&lt;/code&gt;.</p>
