The use of comparison operators (<code><</code>, <code><=</code>, <code>>=</code>, <code>></code>) with strings is not likely to yield the expected results. Make sure the intention was to compare strings and not numbers.


== Noncompliant Code Example

----
var appleNumber = "123";
var orangeNumber = "45";
if (appleNumber < orangeNumber) {  // Noncompliant, this condition is true
  alert("There are more oranges");
}
----


== Compliant Solution

----
var appleNumber = "123";
var orangeNumber = "45";
if (Number(appleNumber) < Number(orangeNumber)) {
  alert("There are more oranges");
}
----

