<p>Declaring that a method &lt;code&gt;throws&lt;/code&gt; an exception type that it doesn’t is like a wart on the code: seemingly ugly but harmless. Unfortunately, like warts, those un-thrown exceptions will spread as each calling method is forced to deal with the &lt;code&gt;throws&lt;/code&gt; declaration. Eventually, your code is so covered in warts that you can’t see the form beneath.</p>
<p>Which is why this rule raises an issue when a method declares an exception type it doesn’t actually throw.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public void output(String text) throws NumberFormatException {  // Noncompliant
  System.out.println(text);
}
</pre>
<h2>Compliant Solution</h2>
<pre>
public void output(String text) {
  System.out.println(text);
}
</pre>
