<p>&lt;code&gt;$this&lt;/code&gt; refers to the current class instance. But static methods can be accessed without instantiating the class, and &lt;code&gt;$this&lt;/code&gt; is not available to them. Using &lt;code&gt;$this&lt;/code&gt; in a static context will result in a fatal error at runtime.</p>
<h2>Noncompliant Code Example</h2>
<pre>
class Clazz {
  $name=NULL;  // instance variable

  public static function foo(){
    if ($this-&gt;name != NULL) {
      // ...
    }
  }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
class Clazz {
  $name=NULL;  // instance variable

  public static function foo($nameParam){
    if ($nameParam != NULL) {
      // ...
    }
  }
}
</pre>
