<p>Marking a parameter for output means that callers will expect its value to be updated with a result from the execution of the procedure. Failing to update the parameter before the procedure returns is surely an error.</p>
<h2>Noncompliant Code Example</h2>
<pre>
CREATE PROCEDURE greet
  @Name varchar(20),
  @Greeting varchar(25) OUTPUT  -- Noncompliant
AS
  DECLARE @Message VARCHAR(45)
  SET @Message = N'Hello ' + RTRIM(@Name);
  PRINT @Message
GO
</pre>
<h2>Compliant Solution</h2>
<pre>
CREATE PROCEDURE greet
  @Name varchar(20),
  @Greeting varchar(25) OUTPUT
AS
  SELECT @Greeting = N'Hello ' + RTRIM(@Name);
GO
</pre>
