<p>Given a consistent set of data, the results from any given database query will be consistent, but any random subset of that data may not be. Any time you perform a query that retrieves only the first n rows, without having ordered the data in the query, you’re grabbing that random subset and putting the reliability of your program at risk. Thus queries that use &lt;code&gt;FETCH FIRST …​ ROWS&lt;/code&gt; must always include an &lt;code&gt;ORDER BY&lt;/code&gt;.</p>
<h2>Noncompliant Code Example</h2>
<pre>
SELECT name   -- Noncompliant
FROM employee
WHERE sick_days &gt; 5
FETCH FIRST 10 ROWS ONLY
</pre>
<h2>Compliant Solution</h2>
<pre>
SELECT name
FROM employee
WHERE sick_days &gt; 5
ORDER BY sick_days desc
FETCH FIRST 10 ROWS ONLY
</pre>
