<p>The complexity of an expression is defined by the number of &lt;code&gt;&amp;&amp;&lt;/code&gt;, &lt;code&gt;||&lt;/code&gt; and &lt;code&gt;condition ? ifTrue : ifFalse&lt;/code&gt; operators it contains.
A single expressionâ€™s complexity should not become too high to keep the code readable.</p>
<h2>Noncompliant Code Example</h2>
<pre>
       IDENTIFICATION DIVISION.
       PROGRAM-ID. foo.

       PROCEDURE DIVISION.

      * Noncompliant, 4 operands are found, higher than the maximum allowed 3
           IF WS-FOO(1) = 1 OR
              WS-FOO(2) = 2 OR
              WS-FOO(3) = 3 OR
              WS-BAR = 4 OR
              WS-BAZ = 5 OR
              WS-QUX = 42
           END-IF.
       END PROGRAM foo.
</pre>
