<p>The complexity of an expression is defined by the number of &lt;code&gt;&amp;&amp;&lt;/code&gt;, &lt;code&gt;||&lt;/code&gt; and &lt;code&gt;condition ? ifTrue : ifFalse&lt;/code&gt; operators it contains.
A single expressionâ€™s complexity should not become too high to keep the code readable.</p>
<h2>Noncompliant Code Example</h2>
<p>With the default threshold value of 3:</p>
<pre>
if (((condition1 &amp;&amp; condition2) || (condition3 &amp;&amp; condition4)) &amp;&amp; condition5) { ... }
</pre>
<h2>Compliant Solution</h2>
<pre>
if ( (myFirstCondition() || mySecondCondition()) &amp;&amp; myLastCondition()) { ... }
</pre>
<h2>Exceptions</h2>
<p>No issue is reported inside &lt;code&gt;equals&lt;/code&gt; methods, because it is common to compare all the fields of a class for equality inside this kind of method.</p>
