Currently, <code>VARCHAR</code> and <code>VARCHAR2</code> are identical data types. But to prevent future changes in behavior, Oracle recommends the use of <code>VARCHAR2</code>.


== Noncompliant Code Example

----
DECLARE
  var VARCHAR(42);  -- Noncompliant
BEGIN
  NULL;
END;
/
----


== Compliant Solution

----
DECLARE
  var VARCHAR2(42);  -- Noncompliant
BEGIN
  NULL;
END;
/
----

