According to the IBM documentation,

bq. The Enterprise COBOL compiler now uses the Language Environment STACK storage work area for several functions, including <code>UPPER-CASE</code>, <code>LOWER-CASE</code>, and <code>NATIONAL-OF</code>. If the STACK is allocated below the 16-MB line and a large DSA (Dynamic Save Area) is needed, an insufficient storage error might occur.

To aid in migrations to IBM Enterprise COBOL 5.x, this rule raises an issue when <code>UPPER-CASE</code>, <code>LOWER-CASE</code>, or <code>NATIONAL-OF</code> is used.


== Noncompliant Code Example

----
        MOVE FUNCTION UPPER-CASE(FIRST-NAME) to FIRST-NAME2.  *> Noncompliant    
        MOVE FUNCTION LOWER-CASE(FIRST-NAME) to FIRST-NAME2.  *> Noncompliant
----


