<p>Setting the creation of an anonymous class inside a loop means a new instance will be created each time through the loop. But since the anonymous class is likely independent of any values that change within the loop, itâ€™s more efficient to create it once, outside the loop and use the same instance for each iteration.</p>
<h2>Noncompliant Code Example</h2>
<pre>
private void setUpListeners(List&lt;GuiObject&gt; list) {
  for (GuiObject gObj : list) {
    gObj.addListener(new Listener() {   // Noncompliant
      public void onChange(Event e) {
        // ...
      }
    });
  }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
private void setUpListeners(List&lt;GuiObject&gt; list) {

  Listener listener = new Listener() {
    public void onChange(Event e) {
      // ...
    }
  }

  for (GuiObject gObj : list) {
    gObj.addListener(listener);
  }
}
</pre>
