The contract and expectation of a <code>get</code> function is that it will return something. Failing to return a value on all paths is surely an error.


== Noncompliant Code Example

----
var x = {
  _prop : "foo",
  get prop(){  // Noncompliant
    console.log("prop getter");
  }
}

x.prop; // returns undefined
----


== Compliant Solution

----
var x = {
  _prop : "foo",
  get prop(){
    console.log("prop getter");
    return _prop;
  }
}

x.prop; // returns "foo"
----

