<p>&lt;code&gt;dynamic_cast&lt;/code&gt; allow to convert pointers from one class to another, following the class inheritance hierarchy (note that in many cases, the need to perform a &lt;code&gt;dynamic_cast&lt;/code&gt; is an indicator of poor class design, for instance not following the Liskov substitution principle).</p>
<p>If the requested conversion is not possible, the result of the conversion is a null pointer to the requested class.
An implication of this is that result of &lt;code&gt;dynamic_cast&lt;/code&gt; should be tested before using it to avoid dereferencing a null pointer.</p>
<h2>Noncompliant Code Example</h2>
<pre>
struct Base {
    virtual ~Base() = default;
};

struct Derived: Base {
    virtual void method() {}
};

void f(Base *b) {
    auto d = dynamic_cast&lt;Derived*&gt;(b);
    d-&gt;method(); // Noncompliant, d could be null
}
</pre>
<h2>Compliant Solution</h2>
<pre>
struct Base {
    virtual ~Base() = default;
};

struct Derived: Base {
    virtual void method() {}
};

void f(Base *b) {
    if (auto d = dynamic_cast&lt;Derived*&gt;(b)) {
        d-&gt;method();
    }
}
</pre>
<h2>See</h2>
<ul>
<li><a href="https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#c148-use-dynamic_cast-to-a-pointer-type-when-failure-to-find-the-required-class-is-considered-a-valid-alternative">C++ Core Guidelines - C.148</a> - Use dynamic_cast to a pointer type when failure to find the required class is considered a valid alternative</li>
</ul>
