<code>dynamic_cast</code> allow to convert pointers from one class to another, following the class inheritance hierarchy (note that in many cases, the need to perform a <code>dynamic_cast</code> is an indicator of poor class design, for instance not following the Liskov substitution principle).

If the requested conversion is not possible, the result of the conversion is a null pointer to the requested class.
An implication of this is that result of <code>dynamic_cast</code> should be tested before using it to avoid dereferencing a null pointer.


== Noncompliant Code Example

----
struct Base {
    virtual ~Base() = default;
};

struct Derived: Base {
    virtual void method() {}
};

void f(Base *b) {
    auto d = dynamic_cast<Derived*>(b);
    d->method(); // Noncompliant, d could be null
}
----


== Compliant Solution

----
struct Base {
    virtual ~Base() = default;
};

struct Derived: Base {
    virtual void method() {}
};

void f(Base *b) {
    if (auto d = dynamic_cast<Derived*>(b)) {
        d->method();
    }
}
----


== See

* https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#c148-use-dynamic_cast-to-a-pointer-type-when-failure-to-find-the-required-class-is-considered-a-valid-alternative[C++ Core Guidelines - C.148] - Use dynamic_cast to a pointer type when failure to find the required class is considered a valid alternative

