include::../description.adoc[]

== Noncompliant Code Example

----
public void SetName(string name)
{
  name = name;
}
----

== Compliant Solution

----
public void SetName(string name)
{
  this.name = name;
}
----

include::../see.adoc[]
