include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression <code>YIF_[A-Za-z]+</code>:
----
INTERFACE MyInterface. "Noncompliant
...
ENDINTERFACE.
----

== Compliant Solution

----
INTERFACE YIF_myinterface. 
...
ENDINTERFACE.
----
