<p>The <a href="https://docs.python.org/3/reference/compound_stmts.html#the-with-statement">&lt;code&gt;with&lt;/code&gt; statement</a> should only be used with <a href="https://docs.python.org/3/reference/datamodel.html#context-managers">context managers</a>. A context manager should have an &lt;code&gt;<em>enter</em>&lt;/code&gt; and an &lt;code&gt;<em>exit</em>&lt;/code&gt; method.</p>
<p>Note that only &lt;code&gt;async with&lt;/code&gt; statements should be used on <a href="https://docs.python.org/3/reference/datamodel.html#async-context-managers">asynchronous context managers</a>.</p>
<p>This rule raises an issue when a &lt;code&gt;with&lt;/code&gt; statement is used on something else than a context manager.</p>
<h2>Noncompliant Code Example</h2>
<pre>
from contextlib import contextmanager, asynccontextmanager

def generator():
    yield 42

with generator() as e:  # Noncompliant
    print(e)

class Empty:
    pass

with Empty():  # Noncompliant
    pass

@asynccontextmanager
async def async_context_manager():
    yield 42

with async_context_manager() as context:  # Noncompliant Missing "async"
    print(context)
</pre>
<h2>Compliant Solution</h2>
<pre>
from contextlib import contextmanager, asynccontextmanager

@contextmanager
def simple_contextmanager():
    print("enter")
    yield
    print("exit")

with simple_contextmanager() as e:
    print(e)

class SimpleContextManager:
    def __enter__(self):
        print("enter")
    def __exit__(self, type_, value, traceback):
        print("exit")

with SimpleContextManager():
    pass

@asynccontextmanager
async def async_context_manager():
    yield 42

async def call_async_context_manager():
    async with async_context_manager() as context:
        print(context)

asyncio.run(call_async_context_manager())
</pre>
