Using element type in class selectors is slower than using only the class selector.


== Noncompliant Code Example

----
var $products = $("div.products");    // Noncompliant - slow
----


== Compliant Solution

----
var $products = $(".products");    // Compliant - fast
----

