<code>linear-gradient</code> was standardized with CSS3. Before that, it was possible to use different non-standard values to define the gradient's direction. Because these values are not standard, they are not supported in all browsers and therefore they should no longer be used in order to get the expected gradient in the latest browser versions that support CSS3.

This rule raises an issue when the first parameter of a <code>linear-gradient</code> is not a valid <code><side-or-corner></code> or <code>angle</code>.


== Noncompliant Code Example

----
.foo { 
  background: -webkit-linear-gradient(to top, #fff, #000);
  background: linear-gradient(top, #fff, #000);
}

.bar {
  background: linear-gradient(45, #fff, #000);
}
----


== Compliant Solution

----
.foo { 
  background: -webkit-linear-gradient(top, #fff, #000);
  background: linear-gradient(to top, #fff, #000);
}

.bar {
  background: linear-gradient(45deg, #fff, #000);
}
----


== See

* https://developer.mozilla.org/en-US/docs/Web/CSS/linear-gradient[Mozilla Web Technology for Developers] - <code>linear-gradient</code>

