<p>The comparison of numeric values of different formats is inefficient. For instance, comparing a &lt;code&gt;COMP-3&lt;/code&gt; with a &lt;code&gt;COMP-4&lt;/code&gt; causes a performance drag because conversions are required under the covers before the comparison.</p>
<p>This rule raises an issue when variables with different &lt;code&gt;USAGE&lt;/code&gt; clauses, or different numbers of decimal places are compared.</p>
<h2>Noncompliant Code Example</h2>
<pre>
01 SUB1 PIC 9999 BINARY
01 WS-DISPLAY-1	PIC 9(12)
01 WS-PACKED-DEC PIC 9(12)V9(2) COMP-3
01 WS-BIN PIC S9999 COMP-4
01 WS-DISPLAY-2	PIC 9(4)

PERFORM VARYING SUB1 FROM WS-DISPLAY-1
BY WS-PACKED-DEC
UNTIL WS-BIN &gt; WS-DISPLAY-2  *&gt; Noncompliant
* ...
END-PERFORM
</pre>
<h2>Compliant Solution</h2>
<pre>
01 SUB1 PIC 9999 BINARY
01 WS-DISPLAY-1	PIC 9(12)
01 WS-PACKED-DEC PIC 9(12)V9(2) COMP-4
01 WS-BIN PIC S9999 COMP-4
01 WS-DISPLAY-2	PIC 9(4)

PERFORM VARYING SUB1 FROM WS-DISPLAY-1
BY WS-PACKED-DEC
UNTIL WS-BIN &gt; WS-DISPLAY-2
* ...
END-PERFORM
</pre>
