<p>When there is only one statement in a chain, the chain syntax can be omitted, which simplifies the code.</p>
<h2>Noncompliant Code Example</h2>
<pre>
CLEAR: w_alvvr.
</pre>
<h2>Compliant Solution</h2>
<pre>
CLEAR w_alvvr.
</pre>
