When there is only one statement in a chain, the chain syntax can be omitted, which simplifies the code.


== Noncompliant Code Example

----
CLEAR: w_alvvr.
----


== Compliant Solution

----
CLEAR w_alvvr.
----

