<p>Shared naming conventions allow teams to collaborate efficiently. This rule checks that all function names match a provided regular expression.</p>
<h2>Noncompliant Code Example</h2>
<p>With default provided regular expression &lt;code&gt;^(<strong>|[a-z0-9_]</strong>)$&lt;/code&gt;:</p>
<pre>
METHOD MyMethod
...
ENDMETHOD.
</pre>
<h2>Compliant Solution</h2>
<pre>
METHOD MY_METHOD
...
ENDMETHOD.
</pre>
