<p>Shared naming conventions allow teams to collaborate efficiently. This rule checks that all function names match a provided regular expression.</p>
<h2>Noncompliant Code Example</h2>
<p>With default provided regular expression: &lt;code&gt;^(_|[a-zA-Z0-9]+)$&lt;/code&gt;:</p>
<pre>
func execute_all() {
...
}
</pre>
<h2>Compliant Solution</h2>
<pre>
func executeAll() {
...
}
</pre>
