<p>Shared naming conventions allow teams to collaborate efficiently. This rule checks that all function names match a provided regular expression.</p>
<h2>Noncompliant Code Example</h2>
<p>With default provided regular expression: &lt;code&gt;^[a-z][a-zA-Z0-9]*$&lt;/code&gt;:</p>
<pre>
function DoSomething(){...}
</pre>
<h2>Compliant Solution</h2>
<pre>
function doSomething(){...}
</pre>
<h2>Exceptions</h2>
<p>Methods with an &lt;code&gt;@inheritdoc&lt;/code&gt; annotation, as well as magic methods (&lt;code&gt;<em>construct()&lt;/code&gt;, &lt;code&gt;</em>destruct()&lt;/code&gt;, &lt;code&gt;<em>call()&lt;/code&gt;, &lt;code&gt;</em>callStatic()&lt;/code&gt;, &lt;code&gt;<em>get()&lt;/code&gt;, &lt;code&gt;</em>set()&lt;/code&gt;, &lt;code&gt;<em>isset()&lt;/code&gt;, &lt;code&gt;</em>unset()&lt;/code&gt;, &lt;code&gt;<em>sleep()&lt;/code&gt;, &lt;code&gt;</em>wakeup()&lt;/code&gt;, &lt;code&gt;<em>toString()&lt;/code&gt;, &lt;code&gt;</em>invoke()&lt;/code&gt;, &lt;code&gt;<em>set\_state()&lt;/code&gt;, &lt;code&gt;</em>clone()&lt;/code&gt;, &lt;code&gt;__debugInfo()&lt;/code&gt;) are ignored.</p>
<pre>
function __construct(){...}
function __destruct(){...}

/**
 * {@inheritdoc}
 */
function myFunc(){...}
</pre>
