Coding conventions allow teams to work efficiently together. This rule checks that the <code>public</code> section of a class is declared first, followed by the <code>protected</code> section, and ending with the <code>private</code> section.


== Noncompliant Code Example

----
class Point
{
  private: 
    String _color; 
    ...

  protected:
    ...

  public: 
    String getColor(); 
    String setColor(); 
}; 
----


== Compliant Solution

----
class Point
{
  public: 
    String getColor(); 
    String setColor(); 

  protected:
    ...

  private: 
    String _color; 
    ...
}; 
----

