<p>If a compiler implementation detects that a call to a copy constructor is redundant, then it is permitted to omit that call, even if the copy constructor has a side effect other than the construction of a copy of the object. This is called "copy elision".
It is therefore important to ensure that a copy constructor does not modify the program state, since the number of such modifications may be indeterminate.</p>
<h2>Noncompliant Code Example</h2>
<pre>
class A
{
public:
  A ( A const &amp; rhs )
  : m_i ( rhs.m_i )
  {
    ++m_static; // Non-compliant, might not be executed
  }
private:
  int32_t m_i;
  static int32_t m_static;
};
</pre>
<h2>See</h2>
<ul>
<li>MISRA C++:2008, 12-8-1</li>
<li><a href="https://www.securecoding.cert.org/confluence/x/EwCICg">CERT, OOP58-CPP.</a> - Copy operations must not mutate the source object</li>
</ul>
