<p>Empty statements, i.e. &lt;code&gt;;&lt;/code&gt;, are usually introduced by mistake, for example because:</p>
<ul>
<li>It was meant to be replaced by an actual statement, but this was forgotten.</li>
<li>There was a typo which lead the semicolon to be doubled, i.e. &lt;code&gt;;;&lt;/code&gt;.</li>
</ul>
<h2>Noncompliant Code Example</h2>
<pre>
void doSomething() {
  ;                                                       // Noncompliant - was used as a kind of TODO marker
}
</pre>
<h2>Compliant Solution</h2>
<pre>
void doSomething() {
}
</pre>
<h2>Exceptions</h2>
<p>In the case of empty expanded macro and in the case of 2 consecutive semi-colons when one of the two is part of a macro-definition then the issue is not raised.
Example:</p>
<pre>
#define A(x) x;
#define LOG(x)

void fun() {
  A(5);
  LOG(X);
}
</pre>
<h2>See</h2>
<ul>
<li>MISRA C:2004, 14.3 - Before preprocessing, a null statement shall only occur on a line by itself; it may be followed by a comment provided that the first character following the null statement is a white-space character.</li>
<li>MISRA C++:2008, 6-2-3 - Before preprocessing, a null statement shall only occur on a line by itself; it may be followed by a comment, provided that the first character following the null statement is a white-space character.</li>
<li><a href="https://www.securecoding.cert.org/confluence/x/NYA5">CERT, MSC12-C.</a> - Detect and remove code that has no effect or is never executed</li>
<li><a href="https://www.securecoding.cert.org/confluence/x/7gCTAw">CERT, MSC51-J.</a> - Do not place a semicolon immediately following an if, for, or while condition</li>
<li><a href="https://www.securecoding.cert.org/confluence/x/i4FtAg">CERT, EXP15-C.</a> - Do not place a semicolon on the same line as an if, for, or while statement</li>
</ul>
