<p>The &lt;code&gt;&lt;!DOCTYPE&gt;&lt;/code&gt; declaration tells the web browser which (X)HTML version is being used on the page, and therefore how to interpret the various elements.
Validators also rely on it to know which rules to enforce.
It should always preceed the &lt;code&gt;&lt;html&gt;&lt;/code&gt; tag.</p>
<h2>Noncompliant Code Example</h2>
<pre>
&lt;html&gt;  &lt;!-- Noncompliant --&gt;
...
&lt;/html&gt;
</pre>
<h2>Compliant Solution</h2>
<pre>
&lt;!DOCTYPE html&gt;
&lt;html&gt;  &lt;!-- Compliant --&gt;
...
&lt;/html&gt;
</pre>
