include::../description.adoc[]

== Noncompliant Code Example

With provided regular expression <code>^[a-z][a-zA-Z0-9_]*$</code>:
----
public void DoSomething(){...}
----

== Compliant Solution

----
public void doSomething(){...}
----
