To prevent portability issues <code>!=</code>, <code>!></code> and <code>!<</code> operators should be replaced by the ANSI standard operators: <code><></code>, <code><=</code> or <code>>=</code>.


== Noncompliant Code Example

----
SELECT C1 FROM S1TESTMD WHERE BIRTHDATE != 2000
----


== Compliant Solution

----
SELECT C1 FROM S1TESTMD WHERE BIRTHDATE <> 2000
----


