PHP lets you make static calls to non-static methods, ex: <code>A::f();</code>. But the fact that you can doesn't mean you _should_. While such calls will work when there's no <code>$this</code> reference in the called method, a fatal error will occur when there is. Furthermore, such calls have been deprecated in PHP 7, and their support may be removed from future releases.


== Noncompliant Code Example

----
class A {
  function f() {
    // ...
  }
}

A::f();      // Noncompliant, "f" is non-static
----


== Compliant Solution

----
class A {
  function f() {
    // ...
  }
}

$a = new A();
$a->f();
----


