There are two ways to write lambdas that contain single statement, but one is definitely more compact and readable than the other.

*Note* that this rule is automatically disabled when the project's <code>sonar.java.source</code> is lower than <code>8</code>.


== Noncompliant Code Example

----
x -> {System.out.println(x+1);}
(a, b) -> { return a+b; }
----


== Compliant Solution

----
x -> System.out.println(x+1)
(a, b) -> a+b    //For return statement, the return keyword should also be dropped
----


