Overriding <code>operator new</code> typically indicates that custom memory allocation is required for the class. When that's the case, it must be balanced with a custom memory deallocation in a matching <code>operator delete</code> method. Otherwise memory leaks or memory corruption will result.


== Noncompliant Code Example

----
class AirPlane
{
public:
  void* operator new(size_t size);
  void fly();
};
----


== Compliant Solution

----
class AirPlane
{
public:
  void* operator new(size_t size);
  void operator delete(void* deadObject, size_t size);
  void fly();
};
----


== See

* https://www.securecoding.cert.org/confluence/x/FYCpAQ[CERT, DCL54-CPP.] - Overload allocation and deallocation functions as a pair in the same scope
* https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#r15-always-overload-matched-allocationdeallocation-pairs[C++ Core Guidelines - R.15] - Always overload matched allocation/deallocation pairs

