<p>Unused parameters are misleading. Whatever the values passed to such parameters, the behavior will be the same.</p>
<p>This rule raises an issue when a &lt;code&gt;private&lt;/code&gt; method of a class/struct takes a parameter without using it.</p>
<h2>Noncompliant Code Example</h2>
<pre>
private void DoSomething(int a, int b) // "b" is unused
{
    Compute(a);
}

private void DoSomething2(int a) // value of "a" is unused
{
    a = 10;
    Compute(a);
}
</pre>
<h2>Compliant Solution</h2>
<pre>
private void DoSomething(int a)
{
    Compute(a);
}

private void DoSomething2()
{
    var a = 10;
    Compute(a);
}
</pre>
<h2>Exceptions</h2>
<p>This rule doesnâ€™t raise any issue in the following contexts:</p>
<ul>
<li>The &lt;code&gt;this&lt;/code&gt; parameter of extension methods.</li>
<li>Methods decorated with attributes.</li>
<li>Empty methods.</li>
<li>Methods which only throw &lt;code&gt;NotImplementedException&lt;/code&gt;.</li>
<li>Main methods.</li>
<li>&lt;code&gt;virtual&lt;/code&gt;, &lt;code&gt;override&lt;/code&gt; methods.</li>
<li>interface implementations.</li>
</ul>
<h2>See</h2>
<ul>
<li><a href="https://www.securecoding.cert.org/confluence/x/NYA5">CERT, MSC12-C.</a> - Detect and remove code that has no effect or is never executed</li>
</ul>
