include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression: ^[A-Z][a-zA-Z0-9_]*$

----
Private Sub doSomething()
...
End Sub
----

== Compliant Solution

----
Private Sub DoSomething()
...
End Sub
----
