<p>When storing numeric fields, using an odd number of digits allows the sign to be included in the storage area without wasting any space.
For example the number -1234567, when packed as 7 digits is stored as follows.</p>
<pre>
1357
246-
</pre>
<p>thus taking up only 4 data slots. but when packed as 8 characters it uses 5 data slots</p>
<pre>
02467
0135-
</pre>
<p>or when 7 digits, non-packed, it will be stored as 8 data slots, including the sign.</p>
<pre>
FFFFFFF6
12345670
</pre>
<h2>Noncompliant Code Example</h2>
<pre>
     D NUMFLD          S              8P 0
     D NUMFLD          S              7S 0
</pre>
<h2>Compliant Solution</h2>
<pre>
     D NUMFLD          S              7P 0
</pre>
