include::../description.adoc[]

== Noncompliant Code Example

----
Public Function NumberOfMinutes(ByVal hours As Integer) As Integer
    Dim seconds As Integer = 0 ' Seconds never used
    Return hours * 60
End Function
----

== Compliant Solution

----
Public Function NumberOfMinutes(ByVal hours As Integer) As Integer
    Return hours * 60
End Function
----

== Exceptions

Unused locally created resources in a <code>Using</code> statement are not reported.

----
Using t = New TestTimer()
End Using
----
