include::../description.adoc[]

== Noncompliant Code Example

----
for (i = 0; i < 10; j++) // Noncompliant
{  
  // ...
}
----

== Compliant Solution

----
for (i = 0; i < 10; i++)
{
  // ...
}
----
