When implementing the <code>Comparable<T>.compareTo</code> method, the parameter's type has to match the type used in the <code>Comparable</code> declaration. When a different type is used this creates an overload instead of an override, which is unlikely to be the intent.

This rule raises an issue when the parameter of the <code>compareTo</code> method of a class implementing <code>Comparable<T></code> is not same as the one used in the <code>Comparable</code> declaration.


== Noncompliant Code Example

----
public class Foo {
  static class Bar implements Comparable<Bar> {
    public int compareTo(Bar rhs) {
      return -1;
    }
  }

  static class FooBar extends Bar {
    public int compareTo(FooBar rhs) {  // Noncompliant: Parameter should be of type Bar
      return 0;
    }
  }
}
----


== Compliant Solution

----
public class Foo {
  static class Bar implements Comparable<Bar> {
    public int compareTo(Bar rhs) {
      return -1;
    }
  }

  static class FooBar extends Bar {
    public int compareTo(Bar rhs) {
      return 0;
    }
  }
}
----

