Having all branches in an <code>EVALUATE</code> or <code>IF</code> chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be an <code>EVALUATE</code>/<code>IF</code> chain at all.

== Noncompliant Code Example

----
EVALUATE X *> Noncompliant
  WHEN 1
    PERFORM SECTION1
  WHEN OTHER
    PERFORM SECTION1
END-EVALUATE.

IF X = 1 THEN *> Noncompliant
  PERFORM SECTION1
ELSE
  PERFORM SECTION1
END-IF.
----

== Exceptions

Note that this rule does not apply to <code>IF</code> chains without <code>ELSE</code>-s, or to <code>EVALUATE</code>-s without <code>WHEN OTHER</code> clauses.

----
IF X = 1 THEN *> //no issue, this could have been done on purpose to make the code more readable
  PERFORM SECTION1
ELSE-IF X = 2 THEN
  PERFORM SECTION1
END-IF.
----
