<p>Code is clearest when each statement has its own line. Nonetheless, it is a common pattern to combine on the same line an &lt;code&gt;if&lt;/code&gt; and its resulting <em>then</em> statement. However, when an &lt;code&gt;if&lt;/code&gt; is placed on the same line as the closing &lt;code&gt;}&lt;/code&gt; from a preceding  <em>then</em>, <em>else</em> or <em>else if</em> part, it is either an error - &lt;code&gt;else&lt;/code&gt; is missing - or the invitation to a future error as maintainers fail to understand that the two statements are unconnected.</p>
<h2>Noncompliant Code Example</h2>
<pre>
if (condition1) {
  // ...
} if (condition2) {  // Noncompliant
  //...
}
</pre>
<h2>Compliant Solution</h2>
<pre>
if (condition1) {
  // ...
} else if (condition2) {
  //...
}
</pre>
<p>Or</p>
<pre>
if (condition1) {
  // ...
}

if (condition2) {
  //...
}
</pre>
