A <code>org.assertj.core.configuration.Configuration</code> will be effective only once you call <code>Configuration.apply()</code> or <code>Configuration.applyAndDisplay()</code>.

This rule raises an issue when configurations are set without the appropriate call to apply them.


== Noncompliant Code Example

----
Configuration configuration = new Configuration(); // Noncompliant, this configuration will not be applied.
configuration.setComparingPrivateFields(true);{code}
----

== Compliant Solution

----
Configuration configuration = new Configuration();
configuration.setComparingPrivateFields(true);
configuration.applyAndDisplay();
// Alternatively: configuration.apply();
----


== See

* https://assertj.github.io/doc/#assertj-configuration[AssertJ configuration documentation]

