include::../description.adoc[]

== Noncompliant Code Example

----
$data = $_GET["data"];
error_log($data); // Noncompliant
----

== Compliant Solution

----
$data = $_GET["data"];
$badchars = array("\n", "\r", "\t");
$safedata = str_replace($badchars, "", $data);
error_log($safedata);
----

include::../see.adoc[]
