The point of calling <code>notify</code>, or more prefereably <code>notifyAll</code>, is to alert other threads that the state of <code>this</code> object has changed. Doing so without a state change seems pointless, and may indicate a logic error.


== Noncompliant Code Example

----
public void run(){
  synchronized(this){
    int total = 0;
    for(int i=0; i<100 ; i++){
      total += i;
    }
    notify();  // Noncompliant; only local variables have been modified. What's the point?
  }
}
----

