include::../description.adoc[]

== Noncompliant Code Example

With the default comment pattern <code>^'\s*\S+\s*$</code>, which ignores single word comments:

----
Module Module1
  Sub Main()
    Console.WriteLine("Hello, world!") ' Noncompliant - My first program!
    Console.WriteLine("Hello, world!") ' CompliantOneWord
  End Sub
End Module
----

== Compliant Solution

----
Module Module1
  Sub Main()
    ' Compliant - My first program!
    Console.WriteLine("Hello, world!")
    Console.WriteLine("Hello, world!") ' CompliantOneWord
  End Sub
End Module
----
