include::../description.adoc[]

== Noncompliant Code Example

----
void process(int a) {
   if(a = BAD_OPS) {...}
}
----

== Exceptions

In C++, it is common to null-check a pointer by using it as a condition. In such cases, if the pointer is <code>NULL</code>, the condition evaluates to <code>false</code>, and otherwise it is <code>true</code>. Such pointer assignments are ignored: 

----
int *ip;
if (ip = get_status() ) {...}
----

CLang is able to detect the same kind of quality issues and allows to switch-off those issues by surrounding the non-boolean assignment with some parentheses. To prevent raising some issues which were intentionally switched off, this rule supports the same coding convention: 

----
if((my_status = get_status())) {...}
----

include::../see.adoc[]
