<p>Using more than one &lt;code&gt;OF&lt;/code&gt; clause to access a data item can quickly decrease the readability of the source code. Either some &lt;code&gt;OF&lt;/code&gt; clauses are optional and should be removed, or there are too many intersections between several data structures and those intersections should be removed.</p>
<h2>Noncompliant Code Example</h2>
<pre>
01 EMPLOYEE.
  05 MOTHER-IN-LAW.
    10 NAME PIC X(20).
  05 FATHER-IN-LAW.
    10 NAME PIC X(20).
...
01 CUSTOMER.
  05 MOTHER-IN-LAW.
    10 NAME PIC X(20).
  05 FATHER-IN-LAW.
    10 NAME PIC X(20).
...

MOVE MY_VALUE TO NAME OF MOTHER-IN-LAW OF CUSTOMER
</pre>
