include::../description.adoc[]

== Noncompliant Code Example

With the default provided regular expression <code>^([A-Z0-9_]*|[a-z0-9_]*)$</code>:

----
CLASS MyClass DEFINITION. 
...
ENDCLASS.
----

== Compliant Solution

----
CLASS MY_CLASS DEFINITION. 
...
ENDCLASS.
----
