include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression <code>^[_a-z][_a-z0-9]*$</code>:
----
class MyClass:
  myField = 1
----

== Compliant Solution

----
class MyClass:
  my_field = 1
----
