include::../description.adoc[]

== Noncompliant Code Example

----
if (x < 0)
  new ArgumentException("x must be nonnegative");
----

== Compliant Solution

----
if (x < 0)
  throw new ArgumentException("x must be nonnegative");
----
