<code>Symbol</code> is a primitive type introduced in ECMAScript2015. Its instances are mainly used as unique property keys.
An instance can only be created by using <code>Symbol</code> as a function. Using <code>Symbol</code> with the <code>new</code> operator will raise a <code>TypeError</code>.


== Noncompliant Code Example

----
const sym = new Symbol("foo");   // Noncompliant
----


== Compliant Solution

----
const sym = Symbol("foo");
----


