<p>Having a single declaration of a type, object or function allows the compiler to detect incompatible types for the same entity.</p>
<p>Normally, this will mean declaring an external identifier in a header file that will be included in any file where the identifier is defined or used.</p>
<h2>Noncompliant Code Example</h2>
<pre>
// header.hpp
extern int16_t a;

// file1.cpp
#include "header.hpp"

extern int16_t b;

// file2.cpp
#include "header.hpp"
extern int32_t b; // Noncompliant, compiler may not detect the error
int32_t a; // Compliant, compiler will detect the error
</pre>
<h2>Compliant Solution</h2>
<pre>
// header.hpp
extern int16_t a; // Compliant, declared once in a header
extern int32_t b; // Compliant, declared once in a header

// file1.cpp
#include "header.hpp"

// file2.cpp
#include "header.hpp"
</pre>
<h2>See</h2>
<ul>
<li>MISRA C++:2008, 3-2-3</li>
</ul>
