<p>When rethrowing an exception, you should do it by simply calling &lt;code&gt;throw;&lt;/code&gt; and not &lt;code&gt;throw exc;&lt;/code&gt;, because the stack trace is reset with the second syntax, making debugging a lot harder.</p>
<h2>Noncompliant Code Example</h2>
<pre>
try
{}
catch(ExceptionType1 exc)
{
  Console.WriteLine(exc);
  throw exc; // Noncompliant; stacktrace is reset
}
catch (ExceptionType2 exc)
{
  throw new Exception("My custom message", exc);  // Compliant; stack trace preserved
}
</pre>
<h2>Compliant Solution</h2>
<pre>
try
{}
catch(ExceptionType1 exc)
{
  Console.WriteLine(exc);
  throw;
}
catch (ExceptionType2 exc)
{
  throw new Exception("My custom message", exc);
}
</pre>
