include::../description.adoc[]

== Noncompliant Code Example

----
$i = 0;
loop:
  echo("i = $i");
  $i++;
  if ($i < 10){
    goto loop;     
  }
----

== Compliant Solution

----
for ($i = 0; $i < 10; $i++){
  echo("i = $i");
}
----
