include::../description.adoc[]

== Noncompliant Code Example

----
if (true) {  
  doSomething(); 
}
...
if (false) {  
  doSomethingElse(); 
}
----

include::../compliant.adoc[]

include::../see.adoc[]
