A serialVersionUID field is required in a <code>Serializable</code> class. In a non-<code>Serializable</code>, it's just confusing.


== Noncompliant Code Example

----
public class MyClass {
  private static final long serialVersionUID = -1L;  // Noncompliant
}
----

