include::../description.adoc[]

== Noncompliant Code Example

----
try {
  doSomething();
} catch (ex) {  // Noncompliant
  throw ex;
}
----

== Compliant Solution

----
try {
  doSomething();
} catch (ex) {
  console.err(ex);
  throw ex;
}
----
or
----
doSomething();
----
