A cleanly coded web application will have a  clear separation of concerns, with business logic in the <code>@Service</code> layer, and communication with other systems in the data access layer.

To help enforce such a separation of concerns, this rule raises an issue when a <code>@Service</code> class has <code>RestTemplate</code>, <code>JmsTemplate</code>, <code>WebServiceTemplate</code>, <code>JdbcTemplate</code>, or <code>DataSource</code> member.


== Noncompliant Code Example

----
@Service ("greetingmanager")
public class GreetingManagerImpl implements GreetingManager
{
    @Autowired
    DataSource ds;  // Noncompliant
----


== Compliant Solution

----
@Service ("greetingmanager")
public class GreetingManagerImpl implements GreetingManager
{
    @Autowired
    GreetingDao gdao;  // DataSource and its use have been moved here
----


