Heading tags are used by search engines and screen reader softwares to construct an outline of the page.
Starting at <code><h1></code> and not skipping any level eases this automatic construction.


== Noncompliant Code Example

----
<h2>My Title</h2>     <!-- Non-Compliant - h1 is not being used -->

<h3>My Sub Title</h3> <!-- Compliant -->
----


== Compliant Solution

----
<h1>My Title</h1>     <!-- Compliant -->

<h2>My Sub Title</h2> <!-- Compliant -->
----

