If you've gone to the trouble of writing an <code>iterator</code> method in a class that doesn't implement <code>Iterable</code>, that trivial omission is costing you half the benefit of the method because you can't use the class in enhanced <code>for</code> loops. 


== Noncompliant Code Example

----
public class MyList {  // Noncompliant
  public Iterator iterator() {
    //...
  }
}

public class MyClass {

  public void doSomething(MyList myList) {
    Iterator itr = myList.iterator();
    while (itr.hasNext() {
      Object obj = itr.next(); 
      processObj(obj);
    }
  }
}
----


== Compliant Solution

----
public class MyList implements Iterable{
  public Iterator iterator() {
    //...
  }
}

public class MyClass {

  public void doSomething(MyList myList) {
    for(Object obj : myList) {
      processObj(obj);
    }
  }
}
----


