<p>By contract, chaining the 'Address of' operator &lt;code&gt;&amp;&lt;/code&gt; with the 'Indirection' operator &lt;code&gt;*&lt;/code&gt; results in a return to the initial value. Thus, such combinations are confusing at best, and bugs at worst.</p>
<h2>Noncompliant Code Example</h2>
<pre>
int *ptr = ...;
int *result1 = &amp;(*ptr); //Noncompliant
int *result2 = &amp;*ptr; //Noncompliant

int value = 4;
int result3 = *(&amp;value); //Noncompliant
int result4 = *&amp;value; //Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
int *ptr = ...;
int *result1 = ptr;
int *result2 = ptr;

int value = 4;
int result3 = value;
int result4 = value;
</pre>
<h2>Exceptions</h2>
<p>No issue is raised when the &lt;code&gt;*&lt;/code&gt; or &lt;code&gt;&amp;&lt;/code&gt; operators are overloaded or when both operators are not located in the same piece of code (one being generated by a macro expansion and the other one located in the main source code for instance).</p>
