<p>The use of trailing closure syntax can make code clearer, but it should only be used when the call only requires a single function type parameter.</p>
<p>This rule raises an issue when trailing closure syntax is not used for the last argument in a call that requires only a single function type parameter.</p>
<h2>Noncompliant Code Example</h2>
<pre>
UIView.animateWithDuration(1.0, animations: {  // Noncompliant
  self.myView.alpha = 0
})
</pre>
<h2>Compliant Solution</h2>
<pre>
UIView.animateWithDuration(1.0) {
  self.myView.alpha = 0
}
</pre>
