<p>OS/VS COBOL accepted the &lt;code&gt;EXHIBIT&lt;/code&gt; statement, but IBM Enterprise COBOL does not. With IBM Enterprise COBOL, the &lt;code&gt;DISPLAY&lt;/code&gt; statement must be used instead.</p>
<h2>Noncompliant Code Example</h2>
<pre>
       IDENTIFICATION DIVISION.
       PROGRAM-ID. foo.

       DATA DIVISION.

       WORKING-STORAGE SECTION.
         01 WS-FOO PIC X(42).
         01 WS-BAR PIC X(42).

       PROCEDURE DIVISION.
      * Non-Compliant
         EXHIBIT NAMED WS-FOO WS-BAR.
       END PROGRAM foo.
</pre>
<h2>Compliant Solution</h2>
<pre>
       IDENTIFICATION DIVISION.
       PROGRAM-ID. foo.

       DATA DIVISION.

       WORKING-STORAGE SECTION.
         01 WS-FOO PIC X(42).
         01 WS-BAR PIC X(42).

       PROCEDURE DIVISION.
      * Compliant
         DISPLAY "WS-FOO = " WS-FOO
             "WS-BAR = " WS-BAR.
       END PROGRAM foo.
</pre>
