OS/VS COBOL accepted the <code>EXHIBIT</code> statement, but IBM Enterprise COBOL does not. With IBM Enterprise COBOL, the <code>DISPLAY</code> statement must be used instead.


== Noncompliant Code Example

----
       IDENTIFICATION DIVISION.
       PROGRAM-ID. foo.

       DATA DIVISION.

       WORKING-STORAGE SECTION.
         01 WS-FOO PIC X(42).
         01 WS-BAR PIC X(42).

       PROCEDURE DIVISION.
      * Non-Compliant
         EXHIBIT NAMED WS-FOO WS-BAR.
       END PROGRAM foo.
----


== Compliant Solution

----
       IDENTIFICATION DIVISION.
       PROGRAM-ID. foo.

       DATA DIVISION.

       WORKING-STORAGE SECTION.
         01 WS-FOO PIC X(42).
         01 WS-BAR PIC X(42).

       PROCEDURE DIVISION.
      * Compliant
         DISPLAY "WS-FOO = " WS-FOO
             "WS-BAR = " WS-BAR.
       END PROGRAM foo.
----

