<p>The &lt;code&gt;Files.exists&lt;/code&gt; method has noticeably poor performance in JDK 8, and can slow an application significantly when used to check files that don’t actually exist.
The same goes for &lt;code&gt;Files.notExists&lt;/code&gt;, &lt;code&gt;Files.isDirectory&lt;/code&gt; and &lt;code&gt;Files.isRegularFile&lt;/code&gt;.</p>
<p><strong>Note</strong> that this rule is automatically disabled when the project’s &lt;code&gt;sonar.java.source&lt;/code&gt; is not 8.</p>
<h2>Noncompliant Code Example</h2>
<pre>
Path myPath;
if(java.nio.Files.exists(myPath)) {  // Noncompliant
 // do something
}
</pre>
<h2>Compliant Solution</h2>
<pre>
Path myPath;
if(myPath.toFile().exists())) {
 // do something
}
</pre>
<h2>See</h2>
<ul>
<li><a href="https://bugs.openjdk.java.net/browse/JDK-8153414">https://bugs.openjdk.java.net/browse/JDK-8153414</a></li>
<li><a href="https://bugs.openjdk.java.net/browse/JDK-8154077">https://bugs.openjdk.java.net/browse/JDK-8154077</a></li>
</ul>
