The <code>Files.exists</code> method has noticeably poor performance in JDK 8, and can slow an application significantly when used to check files that don't actually exist. 
The same goes for <code>Files.notExists</code>, <code>Files.isDirectory</code> and <code>Files.isRegularFile</code>.

*Note* that this rule is automatically disabled when the project's <code>sonar.java.source</code> is not 8.


== Noncompliant Code Example

----
Path myPath;
if(java.nio.Files.exists(myPath)) {  // Noncompliant
 // do something
}
----


== Compliant Solution

----
Path myPath;
if(myPath.toFile().exists())) { 
 // do something
}
----


== See

* https://bugs.openjdk.java.net/browse/JDK-8153414[https://bugs.openjdk.java.net/browse/JDK-8153414]
* https://bugs.openjdk.java.net/browse/JDK-8154077[https://bugs.openjdk.java.net/browse/JDK-8154077]

