Having all branches in a <code>case</code> or <code>if</code> chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldn't be a <code>case</code>/<code>if</code> chain at all.

== Noncompliant Code Example

----
if b == 0  # Noncompliant
  doOneMoreThing()
else
  doOneMoreThing()
end

b = a > 12 ? 4 : 4;  # Noncompliant

case i  # Noncompliant
  when 1
    doSomething()
  when 2
    doSomething()
  when 3
    doSomething()
  else 
    doSomething()
end
----

== Exceptions

This rule does not apply to <code>if</code> chains or  <code>case</code>-es without <code>else</code>-s.

----
if b == 0 # no issue, this could have been done on purpose to make the code more readable
  doSomething()
elsif b == 1
  doSomething()
end
----
