Taking the address of an object of incomplete type, where the complete type contains a user declared <code>operator &</code> leads to undefined behavior.


== Noncompliant Code Example

----
// A.h
class A
{
public:
  A * operator & ( ); // Noncompliant
};

// f1.cc
class A;
void f ( A & a )
{
  &a; // uses built-in operator &
}

// f2.cc
#include "A.h"
void f2 ( A & a )
{
  &a; // use user-defined operator &
}
----


== See

* MISRA C++ 2008, 5-3-3 - The unary & operator shall not be overloaded. 

