<p>The &lt;code&gt;TO_DATE&lt;/code&gt; and &lt;code&gt;TO_TIMESTAMP&lt;/code&gt; functions are converting char of &lt;code&gt;CHAR&lt;/code&gt;, &lt;code&gt;VARCHAR2&lt;/code&gt;, &lt;code&gt;NCHAR&lt;/code&gt;, or &lt;code&gt;NVARCHAR2&lt;/code&gt; datatype to respectively a value of &lt;code&gt;DATE&lt;/code&gt; or &lt;code&gt;TIMESTAMP&lt;/code&gt; datatype.
Without providing the format of the input char, &lt;code&gt;TO_DATE&lt;/code&gt; will consider the provided char is compliant with the default date format. Relying on a default configuration is a source of error because it creates a dependency between the code and the configuration of the ORACLE server where this code is deployed.
According to the Oracleâ€™s documentation "the default date format is determined implicitly by the NLS_TERRITORY initialization parameter or can be set explicitly by the NLS_DATE_FORMAT parameter.". It means the behaviour of the &lt;code&gt;TO_DATE&lt;/code&gt; function will certainly not be the expected one if the configuration of the ORACLE server is changing.</p>
<h2>Noncompliant Code Example</h2>
<pre>
SELECT TO_DATE( 'January 15, 2018, 11:00 A.M.')
FROM DUAL;
</pre>
<h2>Compliant Solution</h2>
<pre>
SELECT TO_DATE( 'January 15, 2018, 11:00 A.M.', 'Month dd, YYYY, HH:MI A.M.')
FROM DUAL;
</pre>
