include::../description.adoc[]

== Noncompliant Code Example

----
if condition1 {                                   
	if condition2 {
		fmt.Println("Hello World")
	}
}
----

== Compliant Solution

----
if condition1 && condition2 {
	fmt.Println("Hello World")
}
----
