include::../description.adoc[]

== Noncompliant Code Example

With a maximum number of 4 parameters:

----
void doSomething(int param1, int param2, int param3, int param4, int param5) { 
  ...
}
----

== Compliant Solution

----
void doSomething(int param1, int param2, int param3, int param4) { 
  ...
}
----

== See

* https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#i23-keep-the-number-of-function-arguments-low[C++ Core Guidelines I.23]: Keep the number of function arguments low
