include::../description.adoc[]

== Noncompliant Code Example

----
foo: proc options(main);
  declare i fixed decimal init(42); put list (i); /* Noncompliant - there are two statements */
end;
----

== Compliant Solution

----
foo: proc options(main);
  /* Compliant */
  declare i fixed decimal init(42);
  put list (i);
end;
----
