<p>In order to protect shared resources, Salesforce enforces a maximum number of SOQL queries which can be executed inside a single transaction. This is part of Governor limits.</p>
<p>Limiting the number of queries executed is necessary, and SOQL queries which have no chance of returning a result should be avoided. This is why SOQL Query of the form &lt;code&gt;SELECT …​ FROM …​ WHERE …​ in :filterValues&lt;/code&gt; should first check that the Set/Map (in this example named "filterValues") is not empty.</p>
<p>This rule raises an issue when such an SOQL query is not enclosed in one of</p>
<ul>
<li>&lt;code&gt;if (!filtervalues.isEmpty())&lt;/code&gt;</li>
<li>&lt;code&gt;if (!filtervalues.size() != 0)&lt;/code&gt;</li>
<li>&lt;code&gt;if (!filtervalues.size() &gt; 0)&lt;/code&gt;</li>
</ul>
<h2>Noncompliant Code Example</h2>
<pre>
public class fooClass{
        Set&lt;Id&gt; setCasesId = new Set();

        //Noncompliant - SOQL Query executed without checking whether the Set is Empty
        for(Case c :[SELECT Id FROM Case WHERE Id IN :setCasesId]){
              //something
        }
}


public class fooClass{
        Map&lt;Id, String&gt; mapCasesId = new Map();

        //Noncompliant - SOQL Query executed without checking whether the Map is Empty
        for(Case c :[SELECT Id FROM Case WHERE Id IN :mapCasesId.keyset()]){
              // ...
        }
}

public class fooClass{
        Map&lt;Id, String&gt; mapCasesId = new Map();

        //Noncompliant - SOQL Query executed without checking whether the Map is Empty or Set is Empty
        for(Case c :[SELECT Id, Subject FROM Case WHERE Id IN :mapCasesId.keyset() OR  Id IN :setCasesId]){
              // ...
        }
}


public class fooClass{
        Map&lt;Id, String&gt; mapCasesId = new Map();

        //Noncompliant - SOQL Query executed without checking whether the Map is Empty and Set is Empty
        for(Case c :[SELECT Id, Subject FROM Case WHERE Id IN :mapCasesId.keyset() AND  Id IN :setCasesId]){
              // ...
        }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
public class fooClass{
        Set&lt;Id&gt; setCasesId = new Set();
        if(!setCasesId.isEmpty()){
          for(Case c :[SELECT Id FROM Case WHERE Id IN :setCasesId]){
                //something
          }
       }
       // OR
       if(!setCasesId.size() &gt; 0){
          for(Case c :[SELECT Id FROM Case WHERE Id IN :setCasesId]){
                //something
          }
       }
       // OR
       if(!setCasesId.size() != 0){
          for(Case c :[SELECT Id FROM Case WHERE Id IN :setCasesId]){
                //something
          }
       }
}


public class fooClass{
        Set&lt;Id&gt; setCasesId = new Set();
        if(!mapCasesId.isEmpty()){
          for(Case c :[SELECT Id FROM Case WHERE Id IN :mapCasesId.keyset()]){
                //something
          }
       }
}


public class fooClass{
        Map&lt;Id, String&gt; mapCasesId = new Map();
        if((!setCasesId.isEmpty()) || (!mapCasesId.isEmpty())){
        for(Case c :[SELECT Id, Subject FROM Case WHERE Id IN :mapCasesId.keyset() OR  Id IN :setCasesId]){
              //something
        }
    }
}


public class fooClass{
        Map&lt;Id, String&gt; mapCasesId = new Map();
        //Recommended - SOQL Query executed after checking that the Map is Not Empty or Set is Not Empty
        if((!setCasesId.isEmpty()) &amp;&amp; (!mapCasesId.isEmpty())){
        for(Case c :[SELECT Id, Subject FROM Case WHERE Id IN :mapCasesId.keyset()  AND  Id IN :setCasesId]){
              //something
        }
    }
}
</pre>
