Using <code>count()</code> to test for emptiness works, but using <code>empty()</code> makes the code more readable and can be more performant

== Noncompliant Code Example

----
if (count($a) > 0) { // Noncompliant
 echo $a[0];
}
----

== Compliant Solution

----
if (!empty($a)) {
  echo $a[0];
}
----
