<p>If a local variable is declared but not used, it is dead code and should be removed. Doing so will improve maintainability because developers will not wonder what the variable is used for.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public func numberOfMinutes(hours:Int) -&gt; Int {
  var seconds = 0   // seconds is never used
  return hours * 60;
}
</pre>
<h2>Compliant Solution</h2>
<pre>
public func numberOfMinutes(hours:Int) -&gt; Int{
  return hours * 60
}
</pre>
<h2>Exceptions</h2>
<p>Simple &lt;code&gt;for-in&lt;/code&gt; loop counters are ignored by this rule because while they are often legitimately unused, their declaration is required by the syntax.</p>
<pre>
for i in 1...10 {  // Ignored
  print("Hello! ");
}

for (a, b) in someElements {  // Noncompliant; b unused
  print(a)
}
</pre>
