include::../description.adoc[]

== Noncompliant Code Example

----
$data = $_GET["data"];
eval("echo \$data;");
----

== Compliant Solution

----
$data = $_GET["data"];
if (in_array($data, $whitelist)) {
  eval("echo \$data;");
}
----

include::../see.adoc[]
