<p>Use of the LIKE statement for fields with the same specification that are being used together in the same statements makes the relevance of individual fields more understandable, and gives easy reference information, and makes relationships between those fields more obvious.</p>
<h2>Noncompliant Code Example</h2>
<pre>
D NAME            S             20A
D PNAM            S             20A
</pre>
<h2>Compliant Solutions</h2>
<pre>
D NAME            S             20A
D PNAM            S             LIKE(NAME)
</pre>
