It may seem tidy to add your new declarations to the <code>std</code> or <code>posix</code> namespaces, but doing so results in undefined behavior. The C++14 Standard, \[namespace.std\] (ISO/IEC 14882-2014 ยง17.6.4.2.1), paragraphs 1 and 2 states:

{quote}
# The behavior of a C++ program is undefined if it adds declarations or definitions to namespace std or to a namespace within namespace std unless otherwise specified. A program may add a template specialization for any standard library template to namespace std only if the declaration depends on a user-defined type and the specialization meets the standard library requirements for the original template and is not explicitly prohibited.
# The behavior of a C++ program is undefined if it declares:
#* an explicit specialization of any member function of a standard library class template, or
#* an explicit specialization of any member function template of a standard library class or class template, or
#* an explicit or partial specialization of any member class template of a standard library class or class template.
{quote}

In addition to restricting extensions to the <code>std</code> namespace, the C++14 Standard goes on in ยง17.6.4.2.2 to say:
{quote}
# The behavior of a C++ program is undefined if it adds declarations or definitions to namespace posix or to a namespace within namespace posix unless otherwise specified. The namespace posix is reserved for use by ISO/IEC 9945 and other POSIX standards.
{quote}

This rule raises an issue for any modification of the standard <code>std</code> and <code>posix</code> namespaces.


== Noncompliant Code Example

----
namespace std { // Noncompliant
  int x;
  void swap(MyNamespace::MyType &m1, MyNamespace::MyType &m2);
}
----


== Compliant Solution

----
namespace expanded_std {
  int x;
}
namespace MyNamespace {
  void swap(MyType &m1, MyType &m2);
}
----

To override <code>swap</code> for your own types, the override should not be placed into the namespace <code>std</code>, but in the namespace of your type (and preferably in the same header file). Then, when you make an unqualified call to <code>swap</code>, argument dependant lookup will ensure that your overload is found by overload resolution (as a consequence, you should never make qualified calls to <code>std::swap</code>, because it will miss some overloads).


== Exceptions

A namespace fragment that only contains template specializations or explicit instantiations is ignored by this rule.


== See

* https://www.securecoding.cert.org/confluence/x/LwDTAQ[CERT, DCL58-CPP.] - Do not modify the standard namespaces

