The <code>default</code> clause should take appropriate action. Having an empty <code>default</code> is a waste of keystrokes.


== Noncompliant Code Example

----
enum Fruit
{
  Apple,
  Orange,
  Banana
}

void PrintName(Fruit fruit)
{
  switch(fruit)
  {
    case Fruit.Apple:
      Console.WriteLine("apple");
      break;
    default:  //Noncompliant
      break;
  }
}
----


== Compliant Solution

----
enum Fruit
{
  Apple,
  Orange,
  Banana
}

void PrintName(Fruit fruit)
{
  switch(fruit)
  {
    case Fruit.Apple:
      Console.WriteLine("apple");
      break;
    default:
      throw new NotSupportedException();
  }
}
----
or
----
void PrintName(Fruit fruit)
{
  switch(fruit)
  {
    case Fruit.Apple:
      Console.WriteLine("apple");
      break;
  }
}
----


== Exceptions

<code>default</code> clauses containing only a comment are ignored with the assumption that they are empty on purpose and the comment documents why.

