== Noncompliant Code Example

With the table <code>MY_TABLE</code> having a <code>NOT NULL</code> column <code>N2</code> without default value and a <code>NOT NULL</code> column <code>N3</code> with default value:
----
EXEC SQL
  INSERT INTO MY_TABLE  *> Noncompliant; N2 value omitted
  (
    N1
  )
  VALUES 
  (
    :ITQ1-NUMBER,
  )
END-EXEC.
----
