<p>The semicolon (&lt;code&gt;;&lt;/code&gt;) is optional as a statement separator except in traditional &lt;code&gt;for&lt;/code&gt; loops and when several statements are combined on one line (which is a bad practice). For cleaner code, semicolons should be omitted.</p>
<h2>Noncompliant Code Example</h2>
<pre>
int a;  // Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
int a
</pre>
