== Noncompliant Code Example

----
class Account /* Non-Compliant - should have a constructor to initialize "balance" */
{
public:
  void credit(int amount)
  {
    balance = balance + amount;
  }

private:
  int balance; /* This variable is left uninitialized, and could have any value */
};

int main(int argc, char* argv[])
{
  Account* acc = new Account;
  acc->credit(100); /* The balance is not guaranteed to be 100, it could be really anything */
  return 0;
}
----
