A Spring <code>singleton</code> bean may be used by many threads at once, and the use of instance (non-<code>static</code>) variables could cause concurrency issues. 

This rule applies to classes with the following annotations: <code>@Service</code>, <code>@Component</code>, <code>@Repository</code>, <code>@Scope("singleton")</code>


== Noncompliant Code Example

----
@Service("animalService")
public class AnimalService {
private int age = 1;              // Noncompliant
private static int count = 0; // Compliant; static

@Inject
private AnimalDAO animalDAO;  // Compliant; managed by Spring
...
}
----

