<code>UPDATE dbtab SET ...</code> without a <code>WHERE</code> condition changes all the entries of the table. Check whether dataset to be changed can be limited by a suitable <code>WHERE</code> condition.


== Noncompliant Code Example

----
UPDATE COUNTRIES SET NAME=country_name.
----


== Compliant Solution

----
UPDATE COUNTRIES SET NAME=country_name WHERE CODE=country_code.
----

