While it's possible for inheritance to be non-<code>public</code>, it is rarely justified, and complicates the use of the derived class. For instance, inherited member visibility is diminished, and implicit and <code>static_cast</code> casts from the base class to the derived class won't work. 

It is sometimes used to limit the base class functionality available in the derived class. When that's the desire, composition should be used instead.


== Noncompliant Code Example

----
class B : private A {  // Noncompliant
  // ...
}
----


== Compliant Solution

----
class B : public A {
  // ...
}
----
or

----
class B {
  private: 
  A a;
  // ...
}
----

