This <code>BYPASSING BUFFER</code> clause explicitly switches off SAP table buffering, so the <code>SELECT</code> reads data directly from the database.
By definition, using this clause can lead to performance issues, which is why its use must be strongly indicated.


== Noncompliant Code Example

----
SELECT * 
INTO US_PERSONS
FROM PERSONS
BYPASSING BUFFER
WHERE CITY EQ 'US'
----

