The <code><strong></code>/<code><b></code> and <code><em></code>/<code><i></code> tags have exactly the same effect in most web browsers, but there is a fundamental difference between them: <code><strong></code> and <code><em></code> have a semantic meaning whereas <code><b></code> and <code><i></code> only convey styling information like CSS.  

While <code><b></code> can have simply no effect on a some devices with limited display or when a screen reader software is used by a blind person, <code><strong></code> will:

* Display the text bold in normal browsers
* Speak with lower tone when using a screen reader such as Jaws

Consequently:

* in order to convey semantics, the <code><b></code> and <code><i></code> tags shall never be used,
* in order to convey styling information, the <code><b></code> and <code><i></code> should be avoided and CSS should be used instead.


== Noncompliant Code Example

----
<i>car</i>             <!-- Noncompliant -->
<b>train</b>         <!-- Noncompliant -->         
----


== Compliant Solution

----
<em>car</em>
<strong>train</strong>
----


== Exceptions

This rule is relaxed in case of https://www.w3.org/WAI/GL/wiki/Using_aria-hidden%3Dtrue_on_an_icon_font_that_AT_should_ignore[icon fonts] usage.

----
<i class="..." aria-hidden="true" />    <!-- Compliant icon fonts usage -->
----


