<p>The &lt;code&gt;&lt;strong&gt;&lt;/code&gt;/&lt;code&gt;&lt;b&gt;&lt;/code&gt; and &lt;code&gt;&lt;em&gt;&lt;/code&gt;/&lt;code&gt;&lt;i&gt;&lt;/code&gt; tags have exactly the same effect in most web browsers, but there is a fundamental difference between them: &lt;code&gt;&lt;strong&gt;&lt;/code&gt; and &lt;code&gt;&lt;em&gt;&lt;/code&gt; have a semantic meaning whereas &lt;code&gt;&lt;b&gt;&lt;/code&gt; and &lt;code&gt;&lt;i&gt;&lt;/code&gt; only convey styling information like CSS.</p>
<p>While &lt;code&gt;&lt;b&gt;&lt;/code&gt; can have simply no effect on a some devices with limited display or when a screen reader software is used by a blind person, &lt;code&gt;&lt;strong&gt;&lt;/code&gt; will:</p>
<ul>
<li>Display the text bold in normal browsers</li>
<li>Speak with lower tone when using a screen reader such as Jaws</li>
</ul>
<p>Consequently:</p>
<ul>
<li>in order to convey semantics, the &lt;code&gt;&lt;b&gt;&lt;/code&gt; and &lt;code&gt;&lt;i&gt;&lt;/code&gt; tags shall never be used,</li>
<li>in order to convey styling information, the &lt;code&gt;&lt;b&gt;&lt;/code&gt; and &lt;code&gt;&lt;i&gt;&lt;/code&gt; should be avoided and CSS should be used instead.</li>
</ul>
<h2>Noncompliant Code Example</h2>
<pre>
&lt;i&gt;car&lt;/i&gt;             &lt;!-- Noncompliant --&gt;
&lt;b&gt;train&lt;/b&gt;         &lt;!-- Noncompliant --&gt;
</pre>
<h2>Compliant Solution</h2>
<pre>
&lt;em&gt;car&lt;/em&gt;
&lt;strong&gt;train&lt;/strong&gt;
</pre>
<h2>Exceptions</h2>
<p>This rule is relaxed in case of <a href="https://www.w3.org/WAI/GL/wiki/Using_aria-hidden%3Dtrue_on_an_icon_font_that_AT_should_ignore">icon fonts</a> usage.</p>
<pre>
&lt;i class="..." aria-hidden="true" /&gt;    &lt;!-- Compliant icon fonts usage --&gt;
</pre>
