include::../description.adoc[]

== Noncompliant Code Example

----
SELECT * FROM db_persons INTO us_persons
----

== Compliant Solution

----
SELECT * FROM db_persons INTO us_persons WHERE country IS 'US'
----

== Exceptions

Not having a WHERE clause is acceptable in read-only cursors as results are generally sorted and it is possible to stop processing in the middle.
