When a closure contains only a <code>return</code> statement, the <code>return</code> itself can be omitted.


== Noncompliant Code Example

----
someList.sort { a, b in
  return a > b
}
----


== Compliant Solution

----
someList.sort { a, b in a > b }
----

