include::../description.adoc[]

== Noncompliant Code Example

----
ServicePointManager.ServerCertificateValidationCallback +=
    (sender, certificate, chain, errors) => {
        return true; // Noncompliant: trust all certificates
    };
----

== Compliant Solution

----
ServicePointManager.ServerCertificateValidationCallback +=
    (sender, certificate, chain, errors) =>
    {
        if (development) return true; // for development, trust all certificates
        return errors == SslPolicyErrors.None
            && validCerts.Contains(certificate.GetCertHashString()); // Compliant: trust only some certificates
    };
----

include::../see.adoc[]
