<p>The <a href="https://docs.python.org/3/reference/datamodel.html#object">https://docs.python.org/3/reference/datamodel.html#object</a>.<em>len</em>[&lt;code&gt;_\_len\_\_&lt;/code&gt;] special method enables to define the length of an object. As such it should always return an integer greater than or equal to 0. When the &lt;code&gt;len(…​)&lt;/code&gt; builtin is called on an object, any other return type will raise a &lt;code&gt;TypeError&lt;/code&gt; and a negative integer will raise a &lt;code&gt;ValueError&lt;/code&gt;.</p>
<p>This rule raises an issue when a &lt;code&gt;_\_len\_\_&lt;/code&gt; method returns an expression which is not an integer greater than or equal to 0.</p>
<h2>Noncompliant Code Example</h2>
<pre>
class A:
    def __len__(self):
        return 2.0  # Noncompliant

len(A())  # This will raise "TypeError: 'float' object cannot be interpreted as an integer"

class B:
    def __len__(self):
        return -2  # Noncompliant

len(B())  # This will raise "ValueError: __len__() should return &gt;= 0"
</pre>
<h2>Compliant Solution</h2>
<pre>
class A:
    def __len__(self):
        return 2

len(A())  # This will return 2
</pre>
<h2>Exceptions</h2>
<p>This rule raises no issue when the type returned by &lt;code&gt;_\_len\__&lt;/code&gt; is a class implementing the <a href="https://docs.python.org/3/reference/datamodel.html#object">https://docs.python.org/3/reference/datamodel.html#object</a>.<em>index</em>[&lt;code&gt;\__index\_\_&lt;/code&gt;] special method.</p>
<pre>
class Ind:
    def __index__(self):
        return 3

class A:
    def __len__(self):
        return Ind()

len(A())  # This will return 3
</pre>
