The https://docs.python.org/3/reference/datamodel.html#object.__len__[<code>\_\_len\_\_</code>] special method enables to define the length of an object. As such it should always return an integer greater than or equal to 0. When the <code>len(...)</code> builtin is called on an object, any other return type will raise a <code>TypeError</code> and a negative integer will raise a <code>ValueError</code>.

This rule raises an issue when a <code>\_\_len\_\_</code> method returns an expression which is not an integer greater than or equal to 0.


== Noncompliant Code Example

----
class A:
    def __len__(self):
        return 2.0  # Noncompliant

len(A())  # This will raise "TypeError: 'float' object cannot be interpreted as an integer"

class B:
    def __len__(self):
        return -2  # Noncompliant

len(B())  # This will raise "ValueError: __len__() should return >= 0"
----


== Compliant Solution

----
class A:
    def __len__(self):
        return 2

len(A())  # This will return 2
----


== Exceptions

This rule raises no issue when the type returned by <code>\_\_len\_\_</code> is a class implementing the https://docs.python.org/3/reference/datamodel.html#object.__index__[<code>\_\_index\_\_</code>] special method.

----
class Ind:
    def __index__(self):
        return 3

class A:
    def __len__(self):
        return Ind()

len(A())  # This will return 3
----

