<p>&lt;code&gt;include_once&lt;/code&gt; and &lt;code&gt;include&lt;/code&gt; only generate a warning if an error occurs during the operation. Because of this they should only be used after possible error conditions have been checked.</p>
<h2>Noncompliant Code Example</h2>
<pre>
include_once $user.'_history.php'; // Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
if (is_member($user)) {
  include_once $user.'_history.php';
}
</pre>
