<code>include_once</code> and <code>include</code> only generate a warning if an error occurs during the operation. Because of this they should only be used after possible error conditions have been checked.


== Noncompliant Code Example

----
include_once $user.'_history.php'; // Noncompliant
----


== Compliant Solution

----
if (is_member($user)) {
  include_once $user.'_history.php';
}
----


