<p>Backticks are a deprecated alias for &lt;code&gt;repr()&lt;/code&gt;. Donâ€™t use them any more, the syntax was removed in Python 3.0.</p>
<h2>Noncompliant Code Example</h2>
<pre>
return `num`  # Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
return repr(num)
</pre>
