<p>Shared coding conventions allow teams to collaborate effectively. This rule allows to check that all class names match a provided regular expression.</p>
<h2>Noncompliant Code Example</h2>
<p>With the default provided regular expression &lt;code&gt;^(<strong>|[a-z0-9_]</strong>)$&lt;/code&gt;:</p>
<pre>
CLASS MyClass DEFINITION.
...
ENDCLASS.
</pre>
<h2>Compliant Solution</h2>
<pre>
CLASS MY_CLASS DEFINITION.
...
ENDCLASS.
</pre>
