<p>Shared coding conventions allow teams to collaborate effectively. This rule allows to check that all class names match a provided regular expression.</p>
<h2>Noncompliant Code Example</h2>
<p>With default provided regular expression &lt;code&gt;^[A-Z][a-zA-Z0-9_]*$&lt;/code&gt;:</p>
<pre>
class myClass {}
</pre>
<h2>Compliant Solution</h2>
<pre>
class MyClass {}
</pre>
