<p>Shared coding conventions allow teams to collaborate effectively. This rule allows to check that all class names  (and interfaces for TypeScript) match a provided regular expression.</p>
<h2>Noncompliant Code Example</h2>
<p>With default provided regular expression &lt;code&gt;^[A-Z][a-zA-Z0-9]*$&lt;/code&gt;:</p>
<pre>
class my_class {...}
</pre>
<h2>Compliant Solution</h2>
<pre>
class MyClass {...}
</pre>
