The use of a <code>StringBuilder</code> or <code>StringBuffer</code> is supposed to make <code>String</code> assembly more efficient than plain concatenation. So don't ruin the effect by concatenating the arguments to <code>append</code>.


== Noncompliant Code Example

----
StringBuilder sb = new StringBuilder();
sb.append("foo is: " + getFoo()); // Noncompliant
----


== Compliant Solution

----
StringBuilder sb = new StringBuilder();
sb.append("foo is: ").append(getFoo());
----

