<p>Shared coding conventions allow teams to collaborate efficiently. This rule checks that reserved words are written in upper case.</p>
<h2>Noncompliant Code Example</h2>
<pre>
BEGIN
  null; -- Noncompliant
END;
/
</pre>
<h2>Compliant Solution</h2>
<pre>
BEGIN
  NULL;
END;
/
</pre>
