Shared coding conventions allow teams to collaborate efficiently. This rule checks that reserved words are written in upper case.


== Noncompliant Code Example

----
BEGIN
  null; -- Noncompliant
END;
/
----


== Compliant Solution

----
BEGIN
  NULL;
END;
/
----

