Instantiating a <code>StringBuilder</code> or a <code>StringBuffer</code> with a character is misleading because most Java developers would expect the character to be the initial value of the <code>StringBuffer</code>. 
What actually happens is that the int representation of the character is used to determine the initial size of the <code>StringBuffer</code>.


== Noncompliant Code Example

----
StringBuffer foo = new StringBuffer('x');   //equivalent to StringBuffer foo = new StringBuffer(120);
----


== Compliant Solution

----
StringBuffer foo = new StringBuffer("x");
----

