include::../description.adoc[]

== Noncompliant Code Example

----
class MyClass {
public :
  static void Mymethod() {
    // ...
  }
};

MyClass* pmyclass = new MyClass();
pmyclass->Mymethod(); // Noncompliant
----

== Compliant Solution

----
class MyClass {
public :
  static Mymethod() {
    // ...
  }
};

Myclass::Mymethod();
----
