While you can perform arithmetic on a <code>time_t</code> type, you may not get the results you expect from the operation because the way that a time is encoded within the type is unspecified. Therefore there is no safe way to manually perform arithmetic on <code>time_t</code> variables despite the fact that the ISO C standard defines <code>time_t</code> as an "arithmetic type". The relevant function calls should be used instead.


== Noncompliant Code Example

----
int max_seconds = 30;
time_t start = time(NULL);

while (time(NULL) < start + max_seconds) {  // Noncompliant
  //...
}
----


== Compliant Solution

----
int max_seconds = 30;
time_t start = time(NULL);
time_t now = start;

while (difftime(now, start) < max_seconds) {
  now = time(NULL);
  //...
}
----


== See

* https://www.securecoding.cert.org/confluence/x/VgAV[CERT, MSC05-C.] - Do not manipulate <code>time_t</code> typed values directly


