include::../description.adoc[]

== Noncompliant Code Example

----
fun doSomething() {
    var name = ""
    // ...
    name = name
}
----

== Compliant Solution

----
fun doSomething() {
    var name = ""
    // ...
    this.name = name
}
----

include::../see.adoc[]
