<p>Access modifiers define which classes can access properties, variables, methods, and other classes. If an access modifier is not specified, the access level defaults to &lt;code&gt;internal&lt;/code&gt;, which grants access to all classes in the same package. This may be what is intended, but it should be specified explicitly to avoid confusion.</p>
<p>Available access modifiers are:</p>
<ul>
<li>&lt;code&gt;internal&lt;/code&gt; - access allowed within the same package</li>
<li>&lt;code&gt;private&lt;/code&gt; - access allowed only within the same class</li>
<li>&lt;code&gt;protected&lt;/code&gt; - access allowed to the class and its child classes</li>
<li>&lt;code&gt;public&lt;/code&gt; - unfettered access by all</li>
</ul>
<h2>Noncompliant Code Example</h2>
<pre>
function checkResources():Boolean {
  ...
  return true;
}
</pre>
<h2>Compliant Solution</h2>
<pre>
public function checkResources():Boolean {
  ...
  return true;
}
</pre>
