include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression <code>^[a-z][a-zA-Z0-9]*$</code>:
----
void doSomething(int my_param) {
  int LOCAL;
  ...
}
----

== Compliant Solution

----
void doSomething(int myParam) {
  int local;
  ...
}
----

== Exceptions

Loop counters and <code>const</code> variables are ignored by this rule.
