<code>SELECT INTO TABLE</code> is much more efficient than <code>SELECT ... ENDSELECT</code>. <code>SELECT INTO TABLE</code> needs more memory to hold the result set, but in normal situations, this is not a concern. When memory is a concern, the result set can be divided into smaller sets.


== Noncompliant Code Example

----
SELECT * FROM T006 INTO X006_WA.
  ...
ENDSELECT.
----


== Compliant Solution

----
SELECT * FROM T006 INTO TABLE X006.
LOOP AT X006 INTO X006_WA.
  ...
ENDLOOP.
----

