<p>The use of the &lt;code&gt;ZonedDateTime&lt;/code&gt; class introduced in Java 8 to truncate a date can be significantly faster than the &lt;code&gt;DateUtils&lt;/code&gt; class from Commons Lang.</p>
<p><strong>Note</strong> that this rule is automatically disabled when the projectâ€™s &lt;code&gt;sonar.java.source&lt;/code&gt; is lower than &lt;code&gt;8&lt;/code&gt;.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public Date trunc(Date date) {
  return DateUtils.truncate(date, Calendar.SECOND);  // Noncompliant
}
</pre>
<h2>Compliant Solution</h2>
<pre>
public Date trunc(Date date) {
  Instant instant = date.toInstant();
  ZonedDateTime zonedDateTime = instant.atZone(ZoneId.systemDefault());
  ZonedDateTime truncatedZonedDateTime = zonedDateTime.truncatedTo(ChronoUnit.SECONDS);
  Instant truncatedInstant = truncatedZonedDateTime.toInstant();
  return Date.from(truncatedInstant);
}
</pre>
