<p>The &lt;code&gt;/EJECT&lt;/code&gt; compiler directive adds a page break when the code is printed, and should be used at the end of each &lt;code&gt;F&lt;/code&gt;, &lt;code&gt;D&lt;/code&gt;, and &lt;code&gt;C&lt;/code&gt; specification section to make the structure of the code clearer and to enhance overall readability.</p>
<h2>Noncompliant Code Example</h2>
<pre>
F* Conversion Job Activity File (CV)
F*
FCVJBAVP   IF   E           K DISK
F* Conversion Job Dependency File (CZ)
F*
FCVJBDPP   IF   E           K DISK
F*
DP0RTCD           S              1
DP0JBNM           S             10A
DP0ERMG           S            100A
DK0JBNM           S                   LIKE(CVJBNM)
DK1JBNM           S                   LIKE(CVJBNM)
DJST              S             10A   DIM(10)
DI                S              3  0 INZ
C
C                   EXSR      SR999
C
C                   EXSR      SR100
</pre>
<h2>Compliant Solution</h2>
<pre>
F* Conversion Job Activity File (CV)
F*
FCVJBAVP   IF   E           K DISK
F* Conversion Job Dependency File (CZ)
F*
FCVJBDPP   IF   E           K DISK
F*
F/EJECT
DP0RTCD           S              1
DP0JBNM           S             10A
DP0ERMG           S            100A
DK0JBNM           S                   LIKE(CVJBNM)
DK1JBNM           S                   LIKE(CVJBNM)
DJST              S             10A   DIM(10)
DI                S              3  0 INZ
D/EJECT
C
C                   EXSR      SR999
C
C                   EXSR      SR100
C/EJECT
</pre>
