<code>readonly</code> properties can only be assigned in a class constructor or at the point of declaration. If a class has a property that's not marked <code>readonly</code> but is only set in the constructor, it could cause confusion about the property's intended use. To avoid confusion, such properties should be marked <code>readonly</code> to make their intended use explicit, and to prevent future maintainers from inadvertently changing their use.

== Noncompliant Code Example

----
class Person {
  private _birthYear: number;  // Noncompliant
  constructor(birthYear: number) {
    this._birthYear = birthYear;
  }
}
----

== Compliant Solution

----
class Person {
  private readonly _birthYear: number;
  constructor(birthYear: number) {
    this._birthYear = birthYear;
  }
}
----
