include::../description.adoc[]

== Noncompliant Code Example

----
public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {

  if (ServletFileUpload.isMultipartContent(request)) {

    FileItemFactory factory = new DiskFileItemFactory();
    ServletFileUpload upload = new ServletFileUpload(factory);  // Noncompliant
    // ...
----

include::../see.adoc[]
