include::../description.adoc[]

== Noncompliant Code Example

----
if (x == 0) {
  doSomething();
} else if (x == 1) {
  doSomethingElse();
} 
----

== Compliant Solution

----
if (x == 0) {
  doSomething();
} else if (x == 1) {
  doSomethingElse();
} else {
  throw new MyException('Illegal state');
}
----

include::../see.adoc[]
