In order to be accessible to visually impaired users, it is important that tables provides a description of its content before the data is accessed.

The simplest way to do it, and also the one https://www.w3.org/TR/WCAG20-TECHS/H39[recommended by WCAG2] is to add a <code><caption></code> element inside the <code><table></code>.

Other technics this rule accepts are:

* adding a concise description via https://www.w3.org/TR/wai-aria/#aria-label[<code>aria-label</code>] or https://www.w3.org/TR/wai-aria/#aria-labelledby[<code>aria-labelledby</code>] attributes in the <code><table></code>.
* referencing a description element with an https://www.w3.org/TR/wai-aria/#aria-describedby[<code>aria-describedby</code>] attribute in the <code><table></code>.
* embedding the <code><table></code> inside a <code><figure></code> which also contains a <code><figcaption></code>.
* adding a <code>summary</code> attribute to the <code><table></code> tag. However note that this attribute has been deprecated in HTML5.

See https://www.w3.org/WAI/tutorials/tables/tips/[W3C WAI Web Accessibility Tutorials] for more information.

This rule raises an issue when a <code><table></code> has neither of the previously mentioned description mechanisms.


== Noncompliant Code Example

----
<table> <!-- Noncompliant -->
  ...
<table>
----


== Compliant Solution

Adding a <code><caption></code> element.

----
<table>
  <caption>New York City Marathon Results 2013</caption>
  ...
</table>
----
Adding an <code>aria-describedby</code> attribute.

----
<p id="mydesc">New York City Marathon Results 2013</p>
<table aria-describedby="mydesc">
  ...
</table>
----
Embedding the table in a <code><figure></code> which also contains a <code><figcaption></code>.

----
<figure>
  <figcaption>New York City Marathon Results 2013</figcaption>
  <table>
    ...
  </table>
</figure>
----
Adding a <code>summary</code> attribute.

----
<table summary="New York City Marathon Results 2013">
  ...
</table>
----


== Exceptions

No issue will be raised on <code><table></code> used for layout purpose, i.e. when it contains a <code>role</code> attribute set to <code>"presentation"</code> or <code>"none"</code>. Note that using <code><table></code> for layout purpose is a bad practice.

No issue will be raised either on <code><table></code> containing an <code>aria-hidden</code> attribute set to <code>"true"</code>.


== See

* https://www.w3.org/WAI/WCAG21/quickref/?versions=2.0#qr-content-structure-separation-programmatic[WCAG2, 1.3.1] - Info and Relationships
* https://www.w3.org/TR/WCAG20-TECHS/H39[WCAG2, H39] - Using caption elements to associate data table captions with data tables

