<p>&lt;code&gt;@VisibleForTesting&lt;/code&gt;&nbsp;can be used to mark methods and fields whose visibility restrictions have been relaxed more than necessary for the API to allow for easier unit testing.</p>
<p>Access to such methods and fields only possible thanks to this relaxed visibility is fine for test code, but it should be avoided in production code.</p>
<p>Supported framework (TBD):</p>
<ul>
<li><a href="https://guava.dev/releases/19.0/api/docs/com/google/common/annotations/VisibleForTesting.html">Guava</a></li>
<li><a href="https://joel-costigliola.github.io/assertj/core/api/org/assertj/core/util/VisibleForTesting.html">AssertJ</a></li>
</ul>
<h2>Noncompliant Code Example</h2>
<pre>
/** src/main/java/MyObject.java */

@VisibleForTesting String foo;

/** src/main/java/Service.java */

new MyObject().foo; // Noncompliant, foo is accessed from production code
</pre>
<h2>Compliant Solution</h2>
<pre>
/** src/main/java/MyObject.java */

@VisibleForTesting String foo;

/** src/test/java/MyObjectTest.java */

new MyObject().foo; // Compliant, foo is accessed from test code
</pre>
