<code>@VisibleForTesting</code>Â can be used to mark methods and fields whose visibility restrictions have been relaxed more than necessary for the API to allow for easier unit testing.

Access to such methods and fields only possible thanks to this relaxed visibility is fine for test code, but it should be avoided in production code.

Supported framework (TBD):

* https://guava.dev/releases/19.0/api/docs/com/google/common/annotations/VisibleForTesting.html[Guava]
* https://joel-costigliola.github.io/assertj/core/api/org/assertj/core/util/VisibleForTesting.html[AssertJ]


== Noncompliant Code Example

----
/** src/main/java/MyObject.java */

@VisibleForTesting String foo;

/** src/main/java/Service.java */

new MyObject().foo; // Noncompliant, foo is accessed from production code
----


== Compliant Solution

----
/** src/main/java/MyObject.java */

@VisibleForTesting String foo;

/** src/test/java/MyObjectTest.java */

new MyObject().foo; // Compliant, foo is accessed from test code
----

