Throwing such general exceptions as <code>Exception</code>, <code>SystemException</code>, <code>ApplicationException</code>, <code>IndexOutOfRangeException</code>, <code>NullReferenceException</code>, <code>OutOfMemoryException</code> and <code>ExecutionEngineException</code> prevents calling methods from handling true, system-generated exceptions differently than application-generated errors. 

== Noncompliant Code Example

----
public void DoSomething(object obj)
{
  if (obj == null)
  {
    throw new NullReferenceException("obj");  // Noncompliant
  }
  // ...
}
----

== Compliant Solution

----
public void DoSomething(object obj)
{
  if (obj == null)
  {
    throw new ArgumentNullException("obj");
  }
  // ...
}
----

include::../see.adoc[]
