Use of the <code>exec</code> statement could be dangerous, and should be avoided. Moreover, the <code>exec</code> statement was removed in Python 3.0. Instead, the built-in <code>exec()</code> function can be used.


== Noncompliant Code Example

----
exec 'print 1' # Noncompliant
----


== Compliant Solution

----
exec('print 1')
----

