<p>An &lt;code&gt;Externalizable&lt;/code&gt; class is one which handles its own &lt;code&gt;Serialization&lt;/code&gt; and deserialization. During deserialization, the first step in the process is a default instantiation using the class' no-argument constructor. Therefore an &lt;code&gt;Externalizable&lt;/code&gt; class without a no-arg constructor cannot be deserialized.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public class Tomato implements Externalizable {  // Noncompliant; no no-arg constructor

  public Tomato (String color, int weight) { ... }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
public class Tomato implements Externalizable {

  public Tomato() { ... }
  public Tomato (String color, int weight) { ... }
}
</pre>
