include::../description.adoc[]

== Noncompliant Code Example

----
CREATE TABLE mytable(col1 INTEGER NOT NULL);
INSERT INTO mytable VALUES (value) -- FIXME ensure value is not null
----

include::../see.adoc[]
