<p>The W3C specifications define the valid &lt;code&gt;at-rules&lt;/code&gt;. Only the official and browser-specific &lt;code&gt;at-rules&lt;/code&gt; should be used to get the expected impact in the final rendering.</p>
<h2>Noncompliant Code Example</h2>
<pre>
@encoding "utf-8";
</pre>
<h2>Compliant Solution</h2>
<pre>
@charset "utf-8";
</pre>
