The W3C specifications define the valid <code>at-rules</code>. Only the official and browser-specific <code>at-rules</code> should be used to get the expected impact in the final rendering.


== Noncompliant Code Example

----
@encoding "utf-8";
----


== Compliant Solution

----
@charset "utf-8";
----

