include::../description.adoc[]

== Noncompliant Code Example

----
CREATE OR REPLACE PROCEDURE greet(
  name     IN  VARCHAR2,
  greeting OUT VARCHAR2) -- Noncompliant
AS
  message VARCHAR2(45);
BEGIN
  SELECT 'Hello ' || RTRIM(name) INTO message FROM DUAL;
END;
----

== Compliant Solution

----
CREATE OR REPLACE PROCEDURE greet(
  name     IN  VARCHAR2,
  greeting OUT VARCHAR2)
AS
BEGIN
  SELECT 'Hello ' || RTRIM(name) INTO greeting FROM DUAL;
END;
----
