While <code>:<element_type></code> and <code>[type="<element_type>"]</code> can both be used in jQuery to select elements by their type, <code>[type="<element_type>"]</code> is far faster because it can take advantage of the native DOM <code>querySelectorAll()</code> method in modern browsers. 

This rule raises an issue when following selectors are used:
* <code>:checkbox</code>
* <code>:file</code>
* <code>:image</code>
* <code>:password</code>
* <code>:radio</code>
* <code>:reset</code>
* <code>:text</code>


== Noncompliant Code Example

----
var input = $( "form input:radio" ); // Noncompliant
----


== Compliant Solution

----
var input = $( "form input[type=radio]" ); // Compliant
----

