<p>A primary key uniquely identifies a row in a database table, and should be considered immutable. Primary key values may be used in foreign keys in other tables, as well as in external systems. Changing such a value, even with the best of motivations, is likely to wreak havoc on the systemâ€™s data integrity and potentially across other systems as well.</p>
<p><strong>Note</strong> That this rule raises issues only when a database catalog is provided during the SonarQube analysis.</p>
<h2>Noncompliant Code Example</h2>
<pre>
UPDATE USERS
  SET USER_ID = :new-id, USER_NAME = :new-name  *&gt; Noncompliant
  WHERE USER_ID = :input
</pre>
