When a non-existent variable is referenced a <code>ReferenceError</code> is raised.

Due to the dynamic nature of JavaScript this can happen in a number of scenarios:

* When typo was made in a symbol's name.
* When using variable declared with <code>let</code> or <code>const</code> before declaration (unlike <code>var</code>-declarations, they are not hoisted to the top of the scope).
* Due to confusion with scopes of <code>let</code>- and <code>const</code>-declarations (they have block scope, unlike <code>var</code>-declarations, having function scope).
* When accessing a property in the wrong scope (e.g. forgetting to specify <code>this.</code>).

This rule does not raise issues on global variables which are defined with <code>sonar.javascript.globals</code> and <code>sonar.javascript.environments</code> properties.

== Noncompliant Code Example

----
var john = { 
  firstName: "john", 
  show: function() { console.log(firstName); } // Noncompliant: firstName is not defined
}
john.show(); 
----

== Compliant Solution

----
var john = { 
  firstName: "john", 
  show: function() { console.log(this.firstName); }
}
john.show(); 
----
