include::../description.adoc[]

== Noncompliant Code Example

With a maximum number of 4 parameters:

----
public function addData(p1 : int, p2 : int, p3 : int, p4 : int, p5 : int): void  { 
...
}
----

== Compliant Solution

----
public function addData(p1 : int, p2 : int, p3 : int, p4 : int): void  { 
...
}
----
