<p>The &lt;code&gt;PartCreationPolicyAttribute&lt;/code&gt; attribute, which is part of the Managed Extensibility Framework (MEF), is used to specify how the exported object will be created. Therefore it doesnâ€™t make sense not to export this a class with this attribute using the &lt;code&gt;ExportAttribute&lt;/code&gt; attribute.</p>
<p>This rule raises an issue when a class is marked as shared with a &lt;code&gt;PartCreationPolicyAttribute&lt;/code&gt; but lacks a &lt;code&gt;ExportAttribute&lt;/code&gt;.</p>
<h2>Noncompliant Code Example</h2>
<pre>
[PartCreationPolicy(CreationPolicy.Any)] // Noncompliant
public class FooBar : IFooBar
{
}
</pre>
<h2>Compliant Solution</h2>
<pre>
[Export(typeof(IFooBar))]
[PartCreationPolicy(CreationPolicy.Any)]
public class FooBar : IFooBar
{
}
</pre>
