<p>The &lt;code&gt;PartCreationPolicyAttribute&lt;/code&gt; attribute, which is part of the Managed Extensibility Framework (MEF), is used to specify how the exported object will be created. Therefore it doesnâ€™t make sense not to export this a class with this attribute using the &lt;code&gt;ExportAttribute&lt;/code&gt; attribute.</p>
<p>This rule raises an issue when a class is marked as shared with a &lt;code&gt;PartCreationPolicyAttribute&lt;/code&gt; but lacks a &lt;code&gt;ExportAttribute&lt;/code&gt;.</p>
<h2>Noncompliant Code Example</h2>
<pre>
&lt;PartCreationPolicy(CreationPolicy.Any)&gt; ' Noncompliant
Public Class FooBar
    Inherits IFooBar
End Class
</pre>
<h2>Compliant Solution</h2>
<pre>
&lt;Export(GetType(IFooBar))&gt;
&lt;PartCreationPolicy(CreationPolicy.Any)&gt;
Public Class FooBar
    Inherits IFooBar
End Class
</pre>
