include::../description.adoc[]

== Noncompliant Code Example

----
var a1 = b + c // This is a trailing comment that can be very very long
----

== Compliant Solution

----
// This very long comment is better placed before the line of code
var a2 = b + c
----
