<p>Indicators should not be used on "IF" statements because they foster poor logic flow and result in code that is difficult to read and maintain.</p>
<h2>Noncompliant Code Example</h2>
<pre>
C   99FLD1          IFEQ      FLD2
</pre>
<h2>Compliant Solution</h2>
<pre>
C     FLD1          IFEQ      FLD2
C     *IN99         ANDEQ     *ON
</pre>
