Indicators should not be used on "IF" statements because they foster poor logic flow and result in code that is difficult to read and maintain.


== Noncompliant Code Example

----
C   99FLD1          IFEQ      FLD2
----


== Compliant Solution

----
C     FLD1          IFEQ      FLD2
C     *IN99         ANDEQ     *ON
----

