This rule allows banning some modules.

== Noncompliant Code Example

With <code>moduleName</code> configured with [a-zA-z\-]*UT123[a-zA-z]*:

----
CALL UT123.
CALL UT123L.
CALL UT123-L.
CALL WS-UT123.
----
