Code annotated as deprecated should not be used since it will be removed sooner or later.

== Noncompliant Code Example

----
// C++14 attribute
[[deprecated]]
void fun();

// GNU attribute
__attribute__((deprecated))
void fun();

// Microsoft attribute
__declspec(deprecated)
void fun();

void example() {
  fun(); // Noncompliant
}
----

include::../see.adoc[]
