include::../description.adoc[]

== Noncompliant Code Example

----
def divide(numerator, denominator)
  return numerator / denominator             # FIXME denominator value might be 0
end
----

include::../see.adoc[]
