Running finalizers on JVM exit is disabled by default. It can be enabled with <code>System.runFinalizersOnExit</code> and <code>Runtime.runFinalizersOnExit</code>, but both methods are deprecated because they are are inherently unsafe. 

According to the Oracle Javadoc:
{quote}
It may result in finalizers being called on live objects while other threads are concurrently manipulating those objects, resulting in erratic behavior or deadlock.
{quote}

If you really want to be execute something when the virtual machine begins its shutdown sequence, you should attach a shutdown hook. 


== Noncompliant Code Example

----
public static void main(String [] args) {
  ...
  System.runFinalizersOnExit(true);  // Noncompliant
  ...
}

protected void finalize(){
  doSomething();
}
----


== Compliant Solution

----
public static void main(String [] args) {
  Runtime.addShutdownHook(new Runnable() {
    public void run(){
      doSomething();
    }
  });
  //...
----


== See

* https://www.securecoding.cert.org/confluence/x/H4cbAQ[CERT, MET12-J.] - Do not use finalizers

