While keywords introduced in later standards can legally be used as identifiers in code compiled to earlier standards, doing so will eventually cause problems. Such code will cause compile errors if (when) the compiler is upgraded, and fixing those errors could be difficult and painful.  

Additionally, such misuse of keywords has the potential to thoroughly confuse people who are unfamiliar with the code base, possibly leading them to introduce additional errors. 

For these reasons, the earlier this practice is stopped, the better.

This rule flags instances of the following keywords used as identifiers:
*C99*
<code>inline</code>, <code>restrict</code>, <code>_Bool</code>, <code>_Complex</code>, <code>_Noreturn</code>, <code>_Static_assert</code>, <code>_Thread_local</code>
*C11*
<code>_Alignas</code>, <code>_Alignof</code>, <code>_Atomic</code>, <code>_Generic</code>, <code>_Imaginary</code>
*C++11*
<code>alignas</code>, <code>alignof</code>, <code>char16_t</code>, <code>char32_t</code>, <code>constexpr</code>, <code>decltype</code>, <code>noexcept</code>, <code>nullptr</code>, <code>static_assert</code>, <code>thread_local</code>
*C++20*
<code>concept</code>, <code>requires</code>


== Noncompliant Code Example

----
int inline = 0;
----


== Compliant Solution

----
int inline_count = 0;
----

