Use either the <code>// ...</code> or <code>/* ... */</code> comment syntax, but be consistent and do not mix them within the same file.


== Noncompliant Code Example

----
/* Noncompliant; both comment syntaxes are used in the same file */
// Foo
/* Bar */
----


== Compliant Solution

----
// Compliant; uniform comment syntax
// Foo
// Bar
----


== See

* https://www.securecoding.cert.org/confluence/x/DQBWB[CERT, MSC55-J.] - Use comments consistently and in a readable fashion

