This rule applies whenever an <code>IF</code> statement is followed by one or 
more <code>ELSEIF</code> statements; the final <code>ELSEIF</code> should be followed by an <code>ELSE</code> statement.

The requirement for a final <code>ELSE</code> statement is defensive programming. 
The <code>ELSE</code> statement should either take appropriate action or contain 
a suitable comment as to why no action is taken. This is consistent with the 
requirement to have a final <code>OTHERS</code> clause in a <code>CASE</code>
statement. 

== Noncompliant Code Example

----
IF RESULT > 0.
  PERFORM do_something.
ELSEIF RESULT = 0.
  PERFORM do_something_else.
ENDIF.
----

== Compliant Solution

----
IF RESULT > 0.
  PERFORM do_something.
ELSEIF RESULT = 0.
  PERFORM do_something_else.
ELSE.
  PERFORM error.
ENDIF.
----

include::../see.adoc[]
