There is no point in creating a <code>const</code> reference to a literal numeric value. Most likely the intent was not to create a reference, but a constant value.


== Noncompliant Code Example

----
const int & weekdayCount = 7;  // Noncompliant 
----


== Compliant Solution

----
const int weekdayCount = 7;
----

