<p>Namespace prefixes that are declared but not used simply clutter the document and confuse maintainers. Such prefixes should be removed.</p>
<h2>Noncompliant Code Example</h2>
<pre>
&lt;xs:schema targetNamespace="http://www.codeSamples.com/fruit"
           xmlns:xs="http://www.w3.org/2001/XMLSchema"
           xmlns:animal="http://www.codeSamples.com/animal"
           elementFormDefault="qualified" version="1"&gt;  &lt;!-- Noncompliant --&gt;
  &lt;xs:element name="fruit"&gt;
    &lt;xs:complexType&gt;
      &lt;xs:sequence&gt;
        ...                        &lt;!-- assume that no use of prefix "animal" is made here --&gt;
      &lt;/xs:sequence&gt;
    &lt;/xs:complexType&gt;
  &lt;/xs:element&gt;
&lt;/xs:schema&gt;
</pre>
<h2>Compliant Solution</h2>
<pre>
&lt;xs:schema targetNamespace="http://www.codeSamples.com/fruit"
           xmlns:xs="http://www.w3.org/2001/XMLSchema"
           elementFormDefault="qualified" version="1"&gt;
  &lt;xs:element name="fruit"&gt;
    &lt;xs:complexType&gt;
      &lt;xs:sequence&gt;
        ...                        &lt;!-- assume that no use of prefix "animal" is made here --&gt;
      &lt;/xs:sequence&gt;
    &lt;/xs:complexType&gt;
  &lt;/xs:element&gt;
&lt;/xs:schema&gt;
</pre>
