PL/SQL does not support nested C-style (<code>/* ... */</code>) comments.

== Noncompliant Code Example

----
/*
  This is a comment block, for which the ending tag was omitted
  It may be difficult to figure out that the following line of code is actually commented


variable = function_call();
/* variable contains the result, this is not allowed, as it is an attempt to create an inner comment */
----

include::../see.adoc[]
