Jump statements (<code>return</code>, <code>break</code>, <code>continue</code>, and <code>fallthrough</code>) move control flow out of the current code block. So any statements that come after a jump are dead code.

== Noncompliant Code Example

----
func fun(a:Int)->Int{
  var i = 10;
  return i + a;
  i++;             // this is never executed
}
----

== Compliant Solution

----
func fun(a:Int)->Int{
  var i = 10;
  return i + a;
}
----

include::../see.adoc[]
