Shared naming conventions allow teams to collaborate efficiently. This rule checks that all [report/]program names match a provided regular expression.


== Noncompliant Code Example

With the regular expression <code>^([A-Z0-9_]*|[a-z0-9_]*)$</code>:

----
    IDENTIFICATION DIVISION.
       PROGRAM-ID. MY-PROGRAM.      *> Noncompliant
----


== Compliant Solution

----
    IDENTIFICATION DIVISION.
       PROGRAM-ID. MY_PROGRAM.
----

