<p>Cumulatively, saved keystrokes can add up to a lot of saved time. Or at least it can feel that way. So the difference between using &lt;code&gt;L#macro_arg&lt;/code&gt; and &lt;code&gt;L###macro_arg&lt;/code&gt; to create a wide string literal may seem perfectly justified because MSVC yields the same result for each. But if you ever need to switch to another compiler, that saved time - and then some - will be lost. So itâ€™s best to use the cross-compiler standard from the start.</p>
<h2>Noncompliant Code Example</h2>
<pre>
#define MY_WIDE_CONSTANT_MAP(x) my_wide_constant_string_map[x] = L#x;
</pre>
<h2>Compliant Solution</h2>
<pre>
#define MY_WIDE_CONSTANT_MAP(x) my_wide_constant_string_map[x] = L###x;
</pre>
