<p>Using &lt;code&gt;auto&lt;/code&gt; when the type that would be deduced is a pointer type can cause confusion. It is much better to specify the pointer part outside of &lt;code&gt;auto&lt;/code&gt;.</p>
<h2>Noncompliant Code Example</h2>
<pre>
auto item = new Item(); // Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
auto* item = new Item();
</pre>
