<p>You could you use preprocessor directives to do some pretty wild things, up to and including redefining the syntax of the language. But doing so <em>could</em> result in unexpected behavior, and definitely <em>would</em> result in impossible-to-read code. Therefore the acceptable uses of macros are strictly limited.</p>
<p>Specifically, macros may only be used do define a:</p>
<ul>
<li>braced initialzer</li>
<li>constant</li>
<li>parenthesized expression</li>
<li>type qualifier</li>
<li>storage class specifier</li>
<li>&lt;code&gt;do&lt;/code&gt;-&lt;code&gt;while (0)&lt;/code&gt; construct</li>
</ul>
<p>Macros should not be used to define statements or parts of statements except for the use of the &lt;code&gt;do&lt;/code&gt;-&lt;code&gt;while (0)&lt;/code&gt; construct, or to redefine the syntax of the language. All brackets ( (), {}, and [] ) in a macro must be balanced</p>
<h2>Noncompliant Code Example</h2>
<pre>
#define int32_t long /* Noncompliant; use typedef instead */
#define STRTIF if( /* Noncompliant; unbalanced () and language redefinition */
#define XSTAL 10000000 /* Compliant; constant */
#define CLOCK (XSTAL/16) /* Compliant; constant expression */
#define PLUS2(X) ((X) + 2) /* Compliant; macro expanding to expression */
#define STOR extern /* Compliant; storage class specifier */
#define INIT(value) { (value), 0, 0} /* Compliant; braced initialiser */
#define READ_TIME_32() \  /* Compliant; example of do-while-zero */
  do { \
    DISABLE_INTERRUPTS (); \
    time_now = (uint32_t)TIMER_HI &lt;&lt; 16; \
    time_now = time_now | (uint32_t)TIMER_LO; \
    ENABLE_INTERRUPTS (); \
  } while (0)  /* note the semicolon must be omitted here at the end */
...

STRTIF my_bool )  /* Huh? This is unreadable */
{
  ...
}
</pre>
<h2>Compliant Solution</h2>
<pre>
#define XSTAL 10000000
#define CLOCK (XSTAL/16)
#define PLUS2(X) ((X) + 2)
#define STOR extern
#define INIT(value) { (value), 0, 0}
#define READ_TIME_32() \
  do { \
    DISABLE_INTERRUPTS (); \
    time_now = (uint32_t)TIMER_HI &lt;&lt; 16; \
    time_now = time_now | (uint32_t)TIMER_LO; \
    ENABLE_INTERRUPTS (); \
  } while (0)

typedef int32_t long;

...

if (my_bool)
{
  ...
}
</pre>
<h2>See</h2>
<ul>
<li>MISRA C:2004, 19.4</li>
</ul>
