include::../description.adoc[]

== Noncompliant Code Example

----
void called(int foo) {
  foo==1; // Noncompliant
  if (foo==1) {
    printf("foo\n");
  }
}

int main() {
  called(2);
  return 0;
}
----

== Compliant Solution

----
void called(int foo) {
  foo=1;
  if (foo==1) {
    printf("foo\n");
  }
}

int main() {
  called(2);
  return 0;
}
----

include::../see.adoc[]
