include::../description.adoc[]

== Noncompliant Code Example

----
if true 
  doSomething()
end
...
if false 
  doSomethingElse()
end
----

== Compliant Solution

----
doSomething()
...
----

include::../see.adoc[]
