<p>Sharing some naming conventions is a key point to make it possible for a team to efficiently collaborate. This rule allows to check that field names match a provided regular expression.</p>
<h2>Noncompliant Code Example</h2>
<p>With the default regular expression &lt;code&gt;^[a-z][a-zA-Z0-9]*$&lt;/code&gt;:</p>
<pre>
class MyClass {
  var MyField = 1
}
</pre>
<h2>Compliant Solution</h2>
<pre>
class MyClass {
  var myField = 1
}
</pre>
