<p>The &lt;code&gt;typeof&lt;/code&gt; operator returns a string indicating the type of its argument, and the set of returned values is limited:</p>
<ul>
<li>"undefined"</li>
<li>"boolean"</li>
<li>"number"</li>
<li>"string"</li>
<li>"symbol" (since ECMAScript 2015)</li>
<li>"function"</li>
<li>"object" (for &lt;code&gt;null&lt;/code&gt; and any other object)</li>
</ul>
<p>Compare a &lt;code&gt;typeof&lt;/code&gt; expression to anything else, and the result is predefined: &lt;code&gt;false&lt;/code&gt;.</p>
<h2>Noncompliant Code Example</h2>
<pre>
function someFunc(x: any): boolean {
  return typeof x === "Number"; // Noncompliant, function will always return 'false'
}
</pre>
<h2>Compliant Solution</h2>
<pre>
function someFunc(x: any): boolean {
  return typeof x === "number";
}
</pre>
