The <code>typeof</code> operator returns a string indicating the type of its argument, and the set of returned values is limited:

* "undefined"
* "boolean"
* "number"
* "string"
* "symbol" (since ECMAScript 2015)
* "function"
* "object" (for <code>null</code> and any other object)

Compare a <code>typeof</code> expression to anything else, and the result is predefined: <code>false</code>.


== Noncompliant Code Example

----
function someFunc(x: any): boolean {
  return typeof x === "Number"; // Noncompliant, function will always return 'false'
}
----


== Compliant Solution

----
function someFunc(x: any): boolean {
  return typeof x === "number";
}
----

