<p>The &lt;code&gt;ServiceContract&lt;/code&gt; attribute specifies that a class or interface defines the communication contract of a Windows Communication Foundation (WCF) service. The service operations of this class or interface are defined by &lt;code&gt;OperationContract&lt;/code&gt; attributes added to methods. It doesnâ€™t make sense to define a contract without any service operations; thus, in a &lt;code&gt;ServiceContract&lt;/code&gt; class or interface at least one method should be annotated with &lt;code&gt;OperationContract&lt;/code&gt;. Similarly, WCF only serves &lt;code&gt;OperationContract&lt;/code&gt; methods that are defined inside &lt;code&gt;ServiceContract&lt;/code&gt; classes or interfaces; thus, this rule also checks that &lt;code&gt;ServiceContract&lt;/code&gt; is added to the containing type of &lt;code&gt;OperationContract&lt;/code&gt; methods.</p>
<h2>Noncompliant Code Example</h2>
<pre>
[ServiceContract]
interface IMyService // Noncompliant
{
  int MyServiceMethod();
}
</pre>
<h2>Compliant Solution</h2>
<pre>
[ServiceContract]
interface IMyService
{
  [OperationContract]
  int MyServiceMethod();
}
</pre>
