This rule checks that the specified child tag does not appear as a direct child of the specified parent.


== Noncompliant Code Example

Assuming a parent/child combination of <code>head</code>/<code>body</code>:

----
<head>
  ...
  <body>  <!-- Noncompliant -->
    ...
  </body>
</head>
----


== Compliant Solution

----
<head>
  ...
</head>
----

