Because the value returned never changes, it is inefficient to call <code>Asc</code>/<code>AscW</code> on a String constant. Use the numeric value instead.


== Noncompliant Code Example

----
intCode = Asc("*")
----


== Compliant Solution

----
intCode = 42
----

