include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression <code>^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$</code>:

----
public static const first:String = "first";
----

== Compliant Solution

----
public static const FIRST:String = "first";
----
