<p>To aid code readability, all the &lt;code&gt;#include&lt;/code&gt; directives in a particular code file should be grouped together near the top of the file. The only items which may precede an &lt;code&gt;#include&lt;/code&gt; in a file are other preprocessor directives or comments.</p>
<h2>Noncompliant Code Example</h2>
<pre>
#include &lt;h1.h&gt; /* Compliant */
int32_t i;
#include &lt;f2.h&gt; /* Noncompliant */
</pre>
<h2>Compliant Solution</h2>
<pre>
#include &lt;h1.h&gt;
#include &lt;f2.h&gt;

int32_t i;
</pre>
<h2>See</h2>
<ul>
<li>MISRA C:2004, 19.1 - #include statements in a file should only be preceded by other preprocessor directives or comments.</li>
<li>MISRA C++:2008, 16-0-1 - #include directives in a file shall only be preceded by other preprocessor directives or comments.</li>
<li>MISRA C:2012, 20.1 - #include directives should only be preceded by preprocessor directives or comments</li>
</ul>
