Assembling a <code>StringBuilder</code> or <code>StringBuffer</code> into a <code>String</code> merely to see if it's empty is a waste of cycles. Instead, jump right to the heart of the matter and get its <code>.length()</code> instead.


== Noncompliant Code Example

----
StringBuilder sb = new StringBuilder();
// ...
if ("".equals(sb.toString()) { // Noncompliant
  // ...
}
----


== Compliant Solution

----
StringBuilder sb = new StringBuilder();
// ...
if (sb.length() == 0) {
  // ...
}
----


