include::../description.adoc[]

== Noncompliant Code Example

With the default provided regular expression: ^[a-z_][a-z0-9_]{2,30}$
----
def MyFunction(a,b):
    ...
----

== Compliant Solution

----
def my_function(a,b):
    ...
----
