<p>Level 77 identifies data items that are not subdivisions of other items, and that have no subdivisions. They are atomic by declaration. To make future subdivision possible, level 01 should be used instead of level 77.</p>
<h2>Noncompliant Code Example</h2>
<pre>
77 CAR            PIC 999.
</pre>
<h2>Compliant Solution</h2>
<pre>
01 CAR            PIC 999.
</pre>
