include::../description.adoc[]

== Noncompliant Code Example

----
func (user *User) rename(name string) {
  name = name  // Noncompliant
}
----

== Compliant Solution

----
func (user *User) rename(name string) {
  user.name = name
}
----

include::../see.adoc[]
