There is no reason to concatenate literal strings. Doing so is an exercise is reducing code readability. Instead, the strings should be combined. Similarly, literal strings should not be <code>append</code>ed to a <code>StringBuffer</code> or <code>StringBuilder</code> sequentially, but combined into one call.

== Noncompliant Code Example

----
String message = "Hello " + "world" + "!";  // Noncompliant
StringBuilder sb = new StringBuilder();
sb.append("I'm pleased").append(" to meet you.");  //Noncompliant
----

== Compliant Solution

----
String message = "Hello world!";
StringBuilder sb = new StringBuilder();
sb.append("I'm pleased to meet you.");  
----
