<code>GOTO</code> is an unstructured control flow statement. It makes code less readable and maintainable, and should only be used to branch to defined <code>ENDSR</code> points within the source. For normal loop functions use <code>ITER</code> or <code>LEAVE</code> instead.

== Noncompliant Code Example

----
C     LOOP1         TAG 
... 
C                   GOTO      LOOP1 
----
