The <code>delete</code> operator expects a pointer argument. Passing it an object may compile and seem to run (with an implicit cast to pointer type) but it can result in unexpected behavior at runtime.


== Noncompliant Code Example

----
class CString {
public:
  operator const char*();
  // ...
};

void fun() {
  CString str;
  // ...
  delete str;  // Noncompliant
}
----


== Compliant Solution

----
void fun() {
  CString *pstr = new CString;
  // ...
  delete pstr;
}
----

