<code>xs:any</code> and <code>xs:anyAttribute</code> are wildcards; they allow the XML document author to use any element or attribute. By specifying <code>nameSpace</code>s for them, you can restrict what's used to elements in the specified name spaces. By default, what's used for <code>xs:any</code> and <code>xs:anyAttribute</code> will be validated in a strict manner (<code>processContents="strict"</code>). That is, validation will fail if the element used isn't recognized. That makes sense if you've provided a namespace, but not if you haven't. Conversely, <code>processContents="lax"</code> allows unrecognized elements to pass validation even if they're not recognized. That makes sense if no namespace has been specified, but not if one has been specified.

This rule raises an issue when <code>strict</code> processing is used (either explicitly or implicitly) when no namespace is provided, and conversely when <code>lax</code> processing is used with a namespace.


== Noncompliant Code Example

----
ToDo
----


== Compliant Solution

----
ToDo
----

