<p>Naming conventions are an important tool in efficient team collaboration. This rule checks that all form names match a regular expression naming convention.</p>
<h2>Noncompliant Code Example</h2>
<p>With the default regular expression:</p>
<pre>
FORM MyForm.
...
ENDFORM.
</pre>
<h2>Compliant Solution</h2>
<pre>
FORM MY_FORM.
...
ENDFORM.
</pre>
