Naming conventions are an important tool in efficient team collaboration. This rule checks that all form names match a regular expression naming convention.


== Noncompliant Code Example

With the default regular expression:

----
FORM MyForm. 
...
ENDFORM.
----


== Compliant Solution

----
FORM MY_FORM. 
...
ENDFORM.
----


