include::../description.adoc[]

== Noncompliant Code Example

----
<Serializable>
Public Class Foo ' Noncompliant
    <OptionalField(VersionAdded:=2)>
    Private optionalField As Integer = 5
End Class
----

== Compliant Solution

----
<Serializable>
Public Class Foo
    <OptionalField(VersionAdded:=2)>
    Private optionalField As Integer = 5

    <OnDeserializing>
    Private Sub OnDeserializing(ByVal context As StreamingContext)
        optionalField = 5
    End Sub

    <OnDeserialized>
    Private Sub OnDeserialized(ByVal context As StreamingContext)
    End Sub
End Class
----
