<p>Weak &lt;code&gt;REF CURSOR&lt;/code&gt; types are harder to work with than ones with a return type. Indeed, the compiler’s type-checker is unable to make some verifications, which are then delayed till runtime.</p>
<p>When the use of weak &lt;code&gt;REF CURSOR&lt;/code&gt; is required, it is best to use the &lt;code&gt;SYS_REFCURSOR&lt;/code&gt; built-in type instead of defining a new one.
This rule’s sysRefCursorAllowed parameter can be used to control whether or not the usage of &lt;code&gt;SYS_REFCURSOR&lt;/code&gt; is allowed.</p>
<h2>Noncompliant Code Example</h2>
<pre>
DECLARE
  TYPE dualCursorType IS REF CURSOR;                      -- Noncompliant
  dualCursor dualCursorType;

  otherCursor SYS_REFCURSOR;                              -- Compliant or non-compliant, depending on the "sysRefCursorAllowed" parameter
BEGIN
  otherCursor := dualCursor;                              -- Works
END;
/
</pre>
<h2>Compliant Solution</h2>
<pre>
DECLARE
  TYPE dualCursorType IS REF CURSOR RETURN DUAL%ROWTYPE;
  dualCursor dualCursorType;

  TYPE otherCursorType IS REF CURSOR RETURN a%ROWTYPE;
  otherCursor otherCursorType;
BEGIN
  otherCursor := dualCursor;                              -- raises PLS-00382: expression is of wrong type, which makes debugging easier
END;
/
</pre>
