Weak <code>REF CURSOR</code> types are harder to work with than ones with a return type. Indeed, the compiler's type-checker is unable to make some verifications, which are then delayed till runtime.

When the use of weak <code>REF CURSOR</code> is required, it is best to use the <code>SYS_REFCURSOR</code> built-in type instead of defining a new one.
This rule's sysRefCursorAllowed parameter can be used to control whether or not the usage of <code>SYS_REFCURSOR</code> is allowed.


== Noncompliant Code Example

----
DECLARE
  TYPE dualCursorType IS REF CURSOR;                      -- Noncompliant
  dualCursor dualCursorType;

  otherCursor SYS_REFCURSOR;                              -- Compliant or non-compliant, depending on the "sysRefCursorAllowed" parameter
BEGIN
  otherCursor := dualCursor;                              -- Works
END;
/
----


== Compliant Solution

----
DECLARE
  TYPE dualCursorType IS REF CURSOR RETURN DUAL%ROWTYPE;
  dualCursor dualCursorType;

  TYPE otherCursorType IS REF CURSOR RETURN a%ROWTYPE;
  otherCursor otherCursorType;
BEGIN
  otherCursor := dualCursor;                              -- raises PLS-00382: expression is of wrong type, which makes debugging easier
END;
/
----

