include::../description.adoc[]

== Noncompliant Code Example

----
void setValue(int value) {
  value = value;
}
----

== Compliant Solution

----
void setValue(int value) {
  this->value = value;
}
----

include::../see.adoc[]
