<p>The &lt;code&gt;==&lt;/code&gt; and &lt;code&gt;!=&lt;/code&gt; operators do type coercion before comparing values. This is bad because it can mask type errors. For example, it evaluates &lt;code&gt;' \t\r\n' == 0&lt;/code&gt; as &lt;code&gt;true&lt;/code&gt;.
It is best to always use the side-effect-less &lt;code&gt;===&lt;/code&gt; and &lt;code&gt;!==&lt;/code&gt; operators instead.</p>
<h2>Noncompliant Code Example</h2>
<pre>
if (var == 'howdy') {...} // Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
if (var === 'howdy') {...}
</pre>
<h2>Exceptions</h2>
<p>Even if testing the equality of a variable against null doesn’t do exactly what most JavaScript developers believe, usage of &lt;code&gt;==&lt;/code&gt; or &lt;code&gt;!=&lt;/code&gt; is tolerated in such context. In the following case, if &lt;code&gt;foo&lt;/code&gt; hasn’t been initialized, its default value is not &lt;code&gt;null&lt;/code&gt; but &lt;code&gt;undefined&lt;/code&gt;. Nevertheless &lt;code&gt;undefined == null&lt;/code&gt;, so JavaScript developers get the expected behavior.</p>
<pre>
if(foo == null) {...}
</pre>
