Implementing the special method <code>\_\_ne\_\_</code> is not equivalent to implementing the special method <code>\_\_eq\_\_</code>. By default <code>\_\_ne\_\_</code> will call <code>\_\_eq\_\_</code>, but the default implementation of <code>\_\_eq\_\_</code> does not call <code>\_\_ne\_\_</code>.

This rule raises an issue when the special method <code>\_\_ne\_\_</code> is implemented but not the <code>\_\_eq\_\_</code> method.


== Noncompliant Code Example

----
class Ne:
    def __ne__(self, other):   # Noncompliant.
        return False

myvar = Ne() == 1  # False. __ne__ is not called
myvar = 1 == Ne()  # False. __ne__ is not called
myvar = Ne() != 1  # False
myvar = 1 != Ne()  # False
----


== Compliant Solution

----
# Recommended, as __ne__ delegates to __eq__ by default
class Eq:
    def __eq__(self, other):
        return True

myvar = Eq() == 1  # True
myvar = 1 == Eq()  # True
myvar = Eq() != 1  # False. __eq__ is called
myvar = 1 != Eq()  # False. __eq__ is called

# OR if __ne__ needs a special implementation

class Eq:
    def __eq__(self, other):
        return True

    def __ne__(self, other):
        return False
----


== See

* Python documentation - https://docs.python.org/3/reference/datamodel.html#object.__ne__[the \_\_ne\_\_ special method]

