<p>In Spring applications, application components that expose interfaces should be package protected at most, not &lt;code&gt;public&lt;/code&gt;. Such reduced visibility helps ensure that the interface is only accessed through the container and not directly.</p>
<h2>Noncompliant Code Example</h2>
<pre>
// TODO
</pre>
<h2>Compliant Solution</h2>
<pre>
// TODO
</pre>
