<p>A loop with at most one iteration is equivalent to the use of an &lt;code&gt;if&lt;/code&gt; statement to conditionally execute one piece of code. No developer expects to find such a use of a loop statement. If the initial intention of the author was really to conditionally execute one piece of code, an &lt;code&gt;if&lt;/code&gt; statement should be used instead.</p>
<p>At worst that was not the initial intention of the author and so the body of the loop should be fixed to use the nested &lt;code&gt;return&lt;/code&gt;, &lt;code&gt;break&lt;/code&gt; or &lt;code&gt;throw&lt;/code&gt; statements in a more appropriate way.</p>
<h2>Noncompliant Code Example</h2>
<pre>
for i in 1...10 { // noncompliant, loop only executes once
  print("i is \(i)")
  break
}
</pre>
<h2>Compliant Solution</h2>
<pre>
for i in 1...10 {
  print("i is \(i)")
}
</pre>
