include::../description.adoc[]

== Noncompliant Code Example

----
If Collection.Count >= 0 Then ...

If Enumerable.Count < 0 Then ...

If array.Length >= 0 Then ...

Dim result As Boolean = Array.Length >= 0
----

== Compliant Solution

----
If list.Count = 0 Then ...
If array.Length >= 42 Then ...
----
