<p>Using the standard &lt;code&gt;getClassLoader()&lt;/code&gt; may not return the <em>right</em> class loader in a JEE context. Instead, go through the &lt;code&gt;currentThread&lt;/code&gt;.</p>
<h2>Noncompliant Code Example</h2>
<pre>
ClassLoader cl = this.getClass().getClassLoader();  // Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
ClassLoader cl = Thread.currentThread().getContextClassLoader();
</pre>
