Using the standard <code>getClassLoader()</code> may not return the _right_ class loader in a JEE context. Instead, go through the <code>currentThread</code>.


== Noncompliant Code Example

----
ClassLoader cl = this.getClass().getClassLoader();  // Noncompliant
----


== Compliant Solution

----
ClassLoader cl = Thread.currentThread().getContextClassLoader();
----

