<p>Sharing some naming conventions enables teams to collaborate more efficiently. This rule checks that all &lt;code&gt;union&lt;/code&gt; names match a provided regular expression.</p>
<h2>Noncompliant Code Example</h2>
<p>Using the default regular expression &lt;code&gt;^[A-Z][a-zA-Z0-9]*$&lt;/code&gt;:</p>
<pre>
union my_union {
    int one;
    int two;
};
</pre>
<h2>Compliant Solution</h2>
<pre>
union MyUnion {
    int one;
    int two;
};
</pre>
