Sharing some naming conventions enables teams to collaborate more efficiently. This rule checks that all <code>union</code> names match a provided regular expression.


== Noncompliant Code Example

Using the default regular expression <code>^[A-Z][a-zA-Z0-9]*$</code>:
----
union my_union {
    int one;
    int two;
};
----


== Compliant Solution

----
union MyUnion {
    int one;
    int two;
};
----


