include::../description.adoc[]

== Noncompliant Code Example

----
if (file != null) {
  if (file.isFile || file.isDirectory) {
    /* ... */
  }
}
----

== Compliant Solution

----
if (file != null && isFileOrDirectory(file)) {
  /* ... */
}

def isFileOrDirectory(file: File): Boolean = file.isFile || file.isDirectory
----
