include::../description.adoc[]

== Noncompliant Code Example

----
Function Divide(ByVal numerator As Integer, ByVal denominator As Integer) As Integer
    Return numerator / denominator  ' FIXME denominator value might be  0
End Function
----

include::../see.adoc[]
