include::../description.adoc[]

== Noncompliant Code Example

----
var myNumber = 010;   // Noncompliant. myNumber will hold 8, not 10 - was this really expected?
----

== Compliant Solution

----
var myNumber = 8;
----

include::../see.adoc[]
