When a method doesn't match it's <code>super</code> method in visibility (<code>public</code>, <code>protected</code>, ...), malicious callers could take advantage of the over-broad access offered by the child class to undermine the application.


== Noncompliant Code Example

----
public class Parent {

  protected void foo() {
    //...
  }
}

public class Child extends Parent {

  public void foo() {  // Noncompliant
    // ...
    super.foo();
  }
}
----


== Compliant Solution

----
public class Parent {

  protected void foo() {
    //...
  }
}

public class Child extends Parent {

  protected void foo() {
    // ...
    super.foo();
  }
}
----


== See

* https://www.securecoding.cert.org/confluence/x/LIYbAQ[CERT, MET04-J.] - Do not increase the accessibility of overridden or hidden methods

