include::../description.adoc[]

== Noncompliant Code Example

----
int <T> Add(int a, int b) // Noncompliant; <T> is ignored
{
  return a + b; 
}
----

== Compliant Solution

----
int Add(int a, int b)
{
  return a + b; 
}
----
