<p>The DML events clause of a trigger is not meant to be used with multiple &lt;code&gt;OF&lt;/code&gt; conditions. When it is, only the last one will actually be taken into account, without any error message being produced. This can lead to counter-intuitive code.</p>
<p>Only the &lt;code&gt;UPDATE&lt;/code&gt; event should have an &lt;code&gt;OF&lt;/code&gt; condition, and there should be at most one occurence of it.</p>
<h2>Noncompliant Code Example</h2>
<pre>
CREATE OR REPLACE TRIGGER myTrigger
  BEFORE UPDATE OF firstName OR UPDATE OF lastName -- Noncompliant - will *only* be triggered on updates of lastName!
  ON myTable
  FOR EACH ROW
BEGIN
  NULL;
END;
/
</pre>
<h2>Compliant Solution</h2>
<pre>
CREATE OR REPLACE TRIGGER myTrigger
  BEFORE UPDATE OF firstName, lastName             -- Compliant - triggered on updates of firstName or/and lastName
  ON myTable
  FOR EACH ROW
BEGIN
  NULL;
END;
/
</pre>
