Classes should only hold <code>weak</code> references to delegate fields with <code>class</code> type. Otherwise, the owning class will have a <code>strong</code> reference to its delegate, and vice versa, and the OS won't be able to deallocate either of them..

Note that this only applies to non-computed delegate fields in classes, and not to fields in <code>struct</code>s and <code>enum</code>s.


== Noncompliant Code Example

----
class MyClass {
  var delegate: ConventionDelegate?  // Noncompliant
}
----


== Compliant Solution

----
class MyClass {
  weak var delegate: ConventionDelegate?
}
----

