Since databases don't offer "Are you sure?" dialogs, it's best to be very certain of what you're changing before you do it. <code>UPDATE</code> and <code>DELETE</code> statements that don't precisely limit their effects to single rows risk changing more than was intended. That's why they should be reviewed carefully.

This rule raises an issue when an <code>UPDATE</code> or <code>DELETE</code> statement's <code>WHERE</code> clause does not use precisely either a unique index or all parts of the table's primary key. That includes both cases where they are omitted in whole or in part, and when they are used but could still describe multiple rows. E.G. <code>WHERE AGE = 34</code>, and <code>WHERE TABLE_ID > 0 AND TABLE_ID < 40</code>.

*Note* That this rule raises issues only when a database catalog is provided during the SonarQube analysis.


== Noncompliant Code Example

----
CREATE table my_table (
     compound_a integer not null,
     compound_b integer not null,
     column_c varchar(50),
     primary key (compound_a, compound_b)
);

DELETE FROM my_table
WHERE compound_b=4;  -- Noncompliant
----


== Exceptions

Statements using a cursor and <code>WHERE CURRENT OF</code> are ignored.

