Merging collapsible <code>IF</code> statements increases the code's readability.

== Noncompliant Code Example

----
       IF CONDITION1 THEN
         IF CONDITION2 THEN
           ...
         END-IF
       END-IF.
----

== Compliant Solution

----
       IF CONDITION1 AND CONDITION2 THEN
         ...
       END-IF.
----
