The <code>IEquatable<T></code> interface has only one method in it: <code>Equals(<T>)</code>. If you've already written <code>Equals(T)</code>, there's no reason not to explicitly implement <code>IEquatable<T></code>. Doing so expands the utility of your class by allowing it to be used where an <code>IEquatable</code> is called for.

**Note**: Classes that implement <code>IEquatable<T></code> should also be <code>sealed</code>.


== Noncompliant Code Example

----
class MyClass  // Noncompliant
{
  public bool Equals(MyClass other)
  {
    //...
  }
}
----


== Compliant Solution

----
sealed class MyClass : IEquatable<MyClass>
{
  public override bool Equals(object other)
  {
    return Equals(other as MyClass);
  }

  public bool Equals(MyClass other) 
  {
    //...
  }
}
----

