include::../description.adoc[]

== Noncompliant Code Example

With the default threshold of 3:

----
def fun():          # Noncompliant as there are 4 return statements
  if condition1:
    return True
  elif condition2:
    return False
  else:
    return True
  return False
----
