It is more efficient to skip the function invocation and use the predefined string constants rather than calling <code>Chr$()</code> or <code>ChrW$()</code> for the following numbers:

* 0 - <code>vbNullChar</code>
* 8 - <code>vbBack</code>
* 9 - <code>vbTab</code>
* 10 - <code>vbLf</code>
* 11 - <code>vbVerticalTab</code>
* 12 - <code>vbFormFeed</code>
* 13 - <code>vbCr</code>
* 13+10 - <code>vbCrLf</code> | <code>vbNewLine</code>
* 34 - <code>""</code>


== Noncompliant Code Example

----
Dim myChar As Char 
myChar = Chr$(0)
----


== Compliant Solution

----
Dim myChar As Char 
myChar = vbNullChar
----


