<p>Every function should be commented to explain its goal and how it works. This non-empty comment must be located before the function definition.</p>
<h2>Noncompliant Code Example</h2>
<pre>
FUNCTION my_function.
  ...
ENDFUNCTION.
</pre>
<h2>Compliant Solution</h2>
<pre>
* here is
* my comment
FUNCTION my_function.
  ...
ENDFUNCTION.
</pre>
