Shared coding conventions allow teams to collaborate efficiently. This rule checks that paragraph names match a provided regular expression.


== Noncompliant Code Example

With a regular expression of <code>[A-Z-]+</code>:

----
 PROCEDURE DIVISION.

    Do_The_Thing.           *> Noncompliant
----


== Compliant Solution

----
 PROCEDURE DIVISION.

    DO-THE-THING           *> Noncompliant
----

