Offering the same experience with the mouse and the keyboard allow users to pick their preferred devices.
 Additionally, users of assistive technology will also be able to browse the site even if they cannot use the mouse.

This rule raises an issue when:
* an HTML element with an <code>onMouseover</code> attribute doesn't also have an <code>onFocus attribute.</code>
* an HTML element with an <code>onMouseout</code> attribute doesn't also have an <code>onBlur attribute.</code>
* an HTML element with an <code>onClick</code> attribute doesn't also have one of the following attributes: <code>onKeyDown</code>, <code>onKeyUp</code>, <code>onKeyPress</code>.

Note that in the case of <code>onClick</code>, the equivalent keyboard handler should support both the "Enter" and "Space" keys as these are usually used by screen-readers.


== Noncompliant Code Example

----
<div onClick="doSomething();" ...>                                <!-- Noncompliant - 'onKeyDown/onKeyUp/onKeyPress' missing -->
<a onMouseover="doSomething();" ...>                            <!-- Noncompliant - 'onFocus' missing -->
<a onMouseout="doSomething();" ...>                             <!-- Noncompliant - 'onBlur' missing -->
----


== Compliant Solution

Note that setting the <code>tabindex</code> attribute is necessary to make the <code><div></code> element focusable.
----
<div onClick="doSomething();" onKeyDown="doSomething();" tabindex="0" ...>    <!-- Compliant -->
<a onMouseover="doSomething();" onFocus="doSomething();" ...>   <!-- Compliant -->
<a onMouseout="doSomething();" onBlur="doSomething();" ...>     <!-- Compliant -->
----


== Exceptions

For the following elements, https://www.w3.org/TR/WCAG20-TECHS/SCR35.html[pressing a key will trigger the <code>onClick</code> attribute]: <code><input type="button"></code>, <code><input type="submit"></code>, <code><button></code>, <code><a></code>. Thus no issue will be raised when an <code>onClick</code> attribute is found in these elements without a <code>onKeyDown/onKeyUp/onKeyPress</code>.

An issue will still be raised for https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role[elements with the <code>role="button"</code> attribute] as they don't behave the same way.


== See

* https://www.w3.org/TR/WCAG20-TECHS/SCR2.html[SCR2: Using redundant keyboard and mouse event handlers]

