Character classes in regular expressions are a convenient way to match one of several possible characters by listing the allowed characters or ranges of characters. If the same character is listed twice in the same character class or if the character class contains overlapping ranges, this has no effect.

Thus duplicate characters in a character class are either a simple oversight or a sign that the author misunderstood how character classes work and wanted to match more than one character. A common example of the latter mistake is trying to use a range like <code>[0-99]</code> to match numbers of up to two digits, when in fact it is equivalent to <code>[0-9]</code>.


== Noncompliant Code Example

----
str.matches("[0-99]") // Noncompliant, this won't actually match strings with two digits
----


== Compliant Solution

----
str.matches("[0-9]{1,2}")
----


