<p>There are valid cases for passing a variable multiple times into the same function or procedure call, but usually doing so is a mistake, and something else was intended for one of the arguments.</p>
<h2>Noncompliant Code Example</h2>
<pre>
SET @result = dbo.MyAdd(@val1, @val1) -- Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
SET @result = dbo.MyAdd(@val1, @val2)
</pre>
