Member functions can only be used with an instance of a class. But <code>friend</code> functions can be used with an implicitly converted type. So loosening access privileges to <code>friend</code> on overloaded binary operators makes them more flexible. Specifically, with a <code>friend</code> function, the class instance can be on either the right or the left of the operator, but with a member function, it can only be on the left.

This rule raises an issue for all non-<code>friend</code> overloaded binary operators except:

* "<code>=</code>", "<code>[ ]</code>", and "<code>-></code>", which cannot be overloaded as <code>friend</code> functions.
* "<code>+=</code>", "<code>-=</code>", "<code>*=</code>", "<code>/=</code>", "<code>%=</code>", "<code>^=</code>", "<code>&=</code>", "<code>|=</code>", "<code><<=</code>", and "<code>>>=</code>", which are not symmetric operators.


== Noncompliant Code Example

----
bool operator==(const MyClass &RHS);  // Noncompliant
----


== Compliant Solution

----
friend bool operator==(const MyClass &LHS, const MyClass &RHS);
----


== See

* https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#c161-use-nonmember-functions-for-symmetric-operators[C++ Core Guidelines C.161] Use nonmember functions for symmetric operators

