According to http://docs.oracle.com/javase/specs/jls/se7/html/jls-8.html#jls-8.9[the docs]:
bq. Nested enum types are implicitly <code>static</code>.

So there's no need to declare them <code>static</code> explicitly.


== Noncompliant Code Example

----
public class Flower {
  static enum Color { // Noncompliant; static is redundant here
    RED, YELLOW, BLUE, ORANGE
  }

  // ...
}
----


== Compliant Solution

----
public class Flower {
  enum Color { // Compliant
    RED, YELLOW, BLUE, ORANGE
  }

  // ...
}
----


