include::../description.adoc[]

== Noncompliant Code Example

----
IF something THEN
  IF something_else THEN
    -- ...
  END IF;
END IF;
----

== Compliant Solution

----
IF something AND something_else THEN
  -- ...
END IF;
----
