Using an <code>ORDER BY</code> clause in a <code>SELECT</code> helps ensure runtime reproducibility. Without one, _some_ databases _may_ return results in a consistent order, but it is likely to be one that appears random to users and is not optimal for processing the data. 


== Noncompliant Code Example

----
SELECT model   -- Noncompliant
FROM car
WHERE wheel_count = 4
----


== Compliant Solution

----
SELECT model
FROM car
WHERE wheel_count = 4
ORDER BY year, make
----

