When the size of a variable-length table is <code>DEPENDING ON</code> a non-<code>BINARY</code>/<code>COMP</code> variable, use of that table is inefficient because a conversion must be done every time the table is used.


== Noncompliant Code Example

----
01 VARS
    05 TABLE_SIZE   PIC 9(4).
    05 MY_TABLE OCCURS 1 TO 10
                        DEPENDING ON TABLE_SIZE  *> Noncompliant; TABLE-SIZE isn't BINARY or COMP
                        PIC X(10).
----


== Compliant Solution

----
01 VARS
    05 TABLE_SIZE   PIC 9(4) BINARY.
    05 MY_TABLE OCCURS 1 TO 10
                        DEPENDING ON TABLE_SIZE 
                        PIC X(10).
----


