include::../description.adoc[]

== Noncompliant Code Example

----
if param == 1:
  openWindow()
elif param == 2:
  closeWindow()
elif param == 1:            # Noncompliant
  moveWindowToTheBackground()
----

== Compliant Solution

----
if param == 1:
  openWindow()
elif param == 2:
  closeWindow()
elif param == 3:        
  moveWindowToTheBackground()
----

include::../see.adoc[]
