When error conditions occur, it is usually a bad idea to simply ignore them. Instead, it is better to handle them properly, or at least to log them.

== Noncompliant Code Example

----
EXEC CICS
  ...
  INGNORE CONDITION ERROR *> Noncompliant
END-EXEC.
----

include::../compliant.adoc[]

include::../see.adoc[]
