Adding <code>IBInspectable</code> to a properly-typed variable makes it available in Xcode's Interface Builder. But that only works if variable type is declared explicitly as one of the following: 

* <code>Int</code> types, <code>Double</code>, <code>Float</code>, <code>Bool</code>
* <code>String</code> (or its optional)
* <code>CGFloat</code>, <code>CGPoint</code>, <code>CGSize</code>, <code>CGRect</code>
* <code>UIColor</code>, <code>UIImage</code> (and their optionals)
* <code>NSString</code>,  <code>NSColor</code>, <code>NSImage</code> (and their optionals)
* <code>NSRect</code>, <code>NSPoint</code>, <code>NSSize</code>, 


== Noncompliant Code Example

----
@IBInspectable  // Noncompliant; type is implicit
public var cornerRadius = 2.0 {
  didSet {
     //...
  }
}
----


== Compliant Solution

----
@IBInspectable
public var cornerRadius: CGFloat  = 2.0 {
  didSet {
     //...
  }
}
----


