A global variable can be modified from anywhere in the program. A first, this may look convenient, but in fact it makes programs very hard to understand: When you see a function call, you cannot know if the function will affect the value of the variable or not. You have lost the ability to reason locally on you code, and must always have the whole program in mind.

Additionally, in multi-threaded environments, global variables are often the subject of race conditions.

Some global variables defined in external libraries (such as <code>std::cout</code>, <code>std::cin</code>, <code>std::cerr</code>) are fine to use, but you should not have a good reason to create your own very often. If you do, make sure that they can be safely accessed concurrently.

This rule detects all declarations of global variables (in the global namespace or in any namespace) that are not constant.


== Noncompliant Code Example

----
double oneFoot = 0.3048;
double userValue;
void readValue();
void writeResult();

int main() {
  readValue();
  writeResult();
}
----


== Compliant Solution

----
constexpr double footToMeter = 0.3048;

double readValue();
void writeResult(double);

int main() {
  auto userValue = readValue();
  writeResult(userValue * footToMeter);
}
----


== See

* https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#i2-avoid-non-const-global-variables[C++ Core Guidelines - I.2] - Avoid non-const global variables

