Checker to find use of single quote where double quote is preferred.


== Noncompliant Code Example

----
&lt;div id='header'&gt;&lt;/div&gt;
----


== Compliant Solution

----
&lt;div id="header"&gt;&lt;/div&gt;
----

