<p>Using &lt;code&gt;LOOP…​INTO&lt;/code&gt;  with &lt;code&gt;MODIFY&lt;/code&gt; statements will put the required record in a work area, process it and put it back in the internal table. It is more efficient to modify the internal table directly by using &lt;code&gt;LOOP…​ASSIGNING&lt;/code&gt; and field-symbols.</p>
<p>This rule raises an issue when a &lt;code&gt;LOOP…​INTO&lt;/code&gt; contains one or more &lt;code&gt;MODIFY&lt;/code&gt; statements.</p>
<h2>Noncompliant Code Example</h2>
<pre>
LOOP AT i_bseg INTO wa_bseg.
    ...
    wa_bseg-sgtxt = 'VALUE'.
    MODIFY i_bseg FROM wa_bseg.
ENDLOOP.
</pre>
<h2>Compliant Solution</h2>
<pre>
LOOP AT i_bseg ASSIGNING &lt;fs_bseg&gt;.
  ...
  &lt;fs_bseg&gt;-sgtxt = 'VALUE'.
ENDLOOP.
</pre>
<h2>See</h2>
<ul>
<li><a href="http://zevolving.com/2009/12/use-of-field-symbols-vs-work-area/">Use of Field-symbols vs Work area</a></li>
</ul>
