Using <code>LOOP...INTO</code>  with <code>MODIFY</code> statements will put the required record in a work area, process it and put it back in the internal table. It is more efficient to modify the internal table directly by using <code>LOOP...ASSIGNING</code> and field-symbols.

This rule raises an issue when a <code>LOOP...INTO</code> contains one or more <code>MODIFY</code> statements.


== Noncompliant Code Example

----
LOOP AT i_bseg INTO wa_bseg.
    ...
    wa_bseg-sgtxt = 'VALUE'.
    MODIFY i_bseg FROM wa_bseg.
ENDLOOP.
----


== Compliant Solution

----
LOOP AT i_bseg ASSIGNING <fs_bseg>.
  ...
  <fs_bseg>-sgtxt = 'VALUE'.
ENDLOOP.
----


== See

* http://zevolving.com/2009/12/use-of-field-symbols-vs-work-area/[Use of Field-symbols vs Work area]

