<code>java.util.concurrent.locks.Lock</code> offers far more powerful and flexible locking operations than are available with <code>synchronized</code> blocks. So synchronizing on a <code>Lock</code> throws away the power of the object, and is just silly. Instead, such objects should be locked and unlocked using <code>tryLock()</code> and <code>unlock()</code>.


== Noncompliant Code Example

----
Lock lock = new MyLockImpl();
synchronized(lock) {  // Noncompliant
  //...
}
----


== Compliant Solution

----
Lock lock = new MyLockImpl();
lock.tryLock();
//...
----


== See

* https://www.securecoding.cert.org/confluence/x/cQCaAg[CERT, LCK03-J.] - Do not synchronize on the intrinsic locks of high-level concurrency objects

