<p>Regardless of the logging framework in use (logback, log4j, commons-logging, java.util.logging, …​), loggers should be:</p>
<ul>
<li>&lt;code&gt;private&lt;/code&gt;: never be accessible outside of its parent class. If another class needs to log something, it should instantiate its own logger.</li>
<li>&lt;code&gt;static&lt;/code&gt;: not be dependent on an instance of a class (an object). When logging something, contextual information can of course be provided in the messages but the logger should be created at class level to prevent creating a logger along with each object.</li>
<li>&lt;code&gt;final&lt;/code&gt;: be created once and only once per class.</li>
</ul>
<h2>Noncompliant Code Example</h2>
<p>With a default regular expression of &lt;code&gt;LOG(?:GER)?&lt;/code&gt;:</p>
<pre>
public Logger logger = LoggerFactory.getLogger(Foo.class);  // Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
private static final Logger LOGGER = LoggerFactory.getLogger(Foo.class);
</pre>
<h2>Exceptions</h2>
<p>Variables of type &lt;code&gt;org.apache.maven.plugin.logging.Log&lt;/code&gt; are ignored.</p>
