Directives in the php.ini files can be of type <code>boolean</code>, <code>integer</code> or <code>string</code>.
For <code>boolean</code> acceptable values are <code>0</code>, <code>1</code>, <code>true</code>, <code>false</code>, <code>yes</code>, <code>no</code>, <code>on</code> and <code>off</code>.
For <code>integers</code> they can be qualified with <code>k</code>, <code>m</code> or <code>g</code>. E.g. 16k means 16000.
The complete list of directive is at http://php.net/manual/en/ini.list.php


== Noncompliant Code Example

----
allow_url_include = 42; Noncompliant, should be a boolean
max_execution_time = "1024"; Noncompliant, should be an integer
----


== Compliant Solution

----
allow_url_include = false
max_execution_time = 1024
----

