When using the Backbone.js framework, the internal hash containing the model's state, <code>model.attributes</code>, should not be accessed directly. Doing so bypasses the listeners set on the model and could put your application in a bad state.

Instead, you should use <code>set</code> to update, <code>get</code> to read, and <code>_.clone(model.attributes)</code> to obtain a copy.


== Noncompliant Code Example

----
myModel.attributes.someAttribute = 1;
----


== Compliant Solution

----
myModel.set({ someAttribute: 1 });
----

