<code>sizeof()</code> is an alias of <code>count()</code> so the two offer the same results, but <code>sizeof()</code> has a different meaning in other languages and may confuse developers who are new to PHP.


== Noncompliant Code Example

----
$fruit=("apple","pear","raspberry","gooseberry","red currant");
echo sizeof($fruit);  // Noncompliant
----


== Compliant Solution

----
$fruit=("apple","pear","raspberry","gooseberry","red currant");
echo count($fruit);
----


