include::../description.adoc[]

== Noncompliant Code Example

----
if ($x > 0)  // Noncompliant
doTheThing();
doTheOtherThing();

foo();
----

== Compliant Solution

----
if ($x > 0) {
  doTheThing();
  doTheOtherThing();
}

foo();
----
or
----
if ($x > 0)
    doTheThing();
doTheOtherThing();

foo();
----
