<p>Proper indentation is a simple and effective way to improve the codeâ€™s readability. Consistent indentation among the developers within a team also reduces the differences that are committed to source control systems, making code reviews easier.</p>
<p>By default this rule checks that each block of code is indented, although it does not check the size of the indent. Parameter "indentSize" allows the expected indent size to be defined. Only the first line of a badly indented section is reported.</p>
<h2>Noncompliant Code Example</h2>
<p>With an indent size of 2:</p>
<pre>
  Public Sub OutputObject(oOutput As Object)
    Set oObject = oOutput

  If TypeOf oObject Is Form Then
  oObject.Cls
  ElseIf TypeOf oObject Is PictureBox Then
        oObject.Cls
    ElseIf TypeOf oObject Is ListBox Then
        oObject.Clear
    End If
End Sub
</pre>
<h2>Compliant Solution</h2>
<pre>
Public Sub OutputObject(oOutput As Object)
    Set oObject = oOutput

    If TypeOf oObject Is Form Then
        oObject.Cls
    ElseIf TypeOf oObject Is PictureBox Then
        oObject.Cls
    ElseIf TypeOf oObject Is ListBox Then
        oObject.Clear
    End If
End Sub
</pre>
