include::../description.adoc[]

== Noncompliant Code Example

----
Public Class Foo
End Class

Public Structure Bar
End Structure
----

== Compliant Solution

----
Namespace SomeSpace
    Public Class Foo
    End Class

    Public Structure Bar
    End Structure
End Namespace
----
