include::../description.adoc[]

== Noncompliant Code Example

----
x := (y / 2 + 1); -- Compliant even if the parentheses are ignored
IF (x > 0) AND ((x+y > 0)) THEN -- Noncompliant
  -- ...
END IF;
----

== Compliant Solution

----
x := (y / 2 + 1);
IF (x > 0) AND (x+y > 0) THEN
  -- ...
END IF;
----
