include::../description.adoc[]

== Noncompliant Code Example

----
try 
{
  /* some work which end up throwing an exception */
  throw new ArgumentException();
} 
finally 
{
  /* clean up */
  throw new InvalidOperationException();       // Noncompliant; will mask the ArgumentException
}
----

== Compliant Solution

----
try
{
  /* some work which end up throwing an exception */
  throw new ArgumentException();
} 
finally 
{
  /* clean up */                       // Compliant
}
----

include::../see.adoc[]
