<p>Jump statements, such as &lt;code&gt;return&lt;/code&gt;, &lt;code&gt;break&lt;/code&gt; and &lt;code&gt;continue&lt;/code&gt; let you change the default flow of program execution, but jump statements that direct the control flow to the original direction are just a waste of keystrokes.</p>
<h2>Noncompliant Code Example</h2>
<pre>
function redundantJump(x) {
  if (x == 1) {
    console.log("x == 1");
    return; // Noncompliant
  }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
function redundantJump(x) {
  if (x == 1) {
    console.log("x == 1");
  }
}
</pre>
<h2>Exceptions</h2>
<p>&lt;code&gt;break&lt;/code&gt; and &lt;code&gt;return&lt;/code&gt; inside &lt;code&gt;switch&lt;/code&gt; statement are ignored, because they are often used for consistency. &lt;code&gt;continue&lt;/code&gt; with label is also ignored, because label is usually used for clarity. Also a jump statement being a single statement in a block is ignored.</p>
