<p>Jump statements, such as &lt;code&gt;RETURN&lt;/code&gt; and &lt;code&gt;CONTINUE&lt;/code&gt; let you change the default flow of program execution, but jump statements that direct the control flow to the original direction are just a waste of keystrokes.</p>
<h2>Noncompliant Code Example</h2>
<pre>
CREATE PROCEDURE print_numbers AS
BEGIN
  FOR i in 1..4 LOOP
    DBMS_OUTPUT.PUT_LINE(i);
    CONTINUE; -- Noncompliant
  END LOOP;
  RETURN;     -- Noncompliant
END;
</pre>
<h2>Compliant Solution</h2>
<pre>
CREATE PROCEDURE print_numbers AS
BEGIN
  FOR i in 1..4 LOOP
    DBMS_OUTPUT.PUT_LINE(i);
  END LOOP;
END;
</pre>
