<p>Jump statements such as &lt;code&gt;return&lt;/code&gt; and &lt;code&gt;continue&lt;/code&gt; let you change the default flow of program execution, but jump statements that direct the control flow to the original direction are just a waste of keystrokes.</p>
<h2>Noncompliant Code Example</h2>
<pre>
var i:Int
for i in 1...10 {
    print("i is \(i)")
    continue  // this is meaningless; the loop would continue anyway
}
</pre>
<h2>Compliant Solution</h2>
<pre>
var i:Int
for i in 1...10 {
    print("i is \(i)")
}
</pre>
