include::../description.adoc[]

== Noncompliant Code Example

----
if (condition)  // Noncompliant
DoTheThing();

DoTheOtherThing();
SomethingElseEntirely();

Foo();
----

== Compliant Solution

----
if (condition)
  DoTheThing();

DoTheOtherThing();
SomethingElseEntirely();

Foo();
----
