<p>Both &lt;code&gt;php_sapi_name()&lt;/code&gt; and the &lt;code&gt;PHP_SAPI&lt;/code&gt; constant give the same value. But calling the method is less efficient that simply referencing the constant.</p>
<h2>Noncompliant Code Example</h2>
<pre>
if (php_sapi_name() == 'test') { ... }
</pre>
<h2>Compliant Solution</h2>
<pre>
if (PHP_SAPI == 'test') { ... }
</pre>
