Both <code>php_sapi_name()</code> and the <code>PHP_SAPI</code> constant give the same value. But calling the method is less efficient that simply referencing the constant. 


== Noncompliant Code Example

----
if (php_sapi_name() == 'test') { ... } 
----


== Compliant Solution

----
if (PHP_SAPI == 'test') { ... } 
----

