<p>When declaring a Windows Communication Foundation (WCF) &lt;code&gt;OperationContract&lt;/code&gt; method one-way, that service method won’t return any result, not even an underlying empty confirmation message. These are fire-and-forget methods that are useful in event-like communication. Specifying a return type therefore does not make sense.</p>
<h2>Noncompliant Code Example</h2>
<pre>
[ServiceContract]
interface IMyService
{
  [OperationContract(IsOneWay = true)]
  int SomethingHappened(int parameter); // Noncompliant
}
</pre>
<h2>Compliant Solution</h2>
<pre>
[ServiceContract]
interface IMyService
{
  [OperationContract(IsOneWay = true)]
  void SomethingHappened(int parameter);
}
</pre>
<h2>Exceptions</h2>
<p>The rule doesn’t report if &lt;code&gt;OperationContractAttribute.AsyncPattern&lt;/code&gt; is set to &lt;code&gt;true&lt;/code&gt;.</p>
