Proper use of the <code>angular</code> object is crucial when using the AngularJS framework. Misspell an <code>angular</code> function, and depending on the browser you may not get any errors at all, or the error you do get will be distinctly unhelpful.

This rule is intended to catch misspellings of the recognized <code>angular</code> functions. It checks that only the following functions are used:

* <code>directive</code>
* <code>module</code>
* <code>controller</code>
* <code>provider</code>
* <code>filter</code>


== Noncompliant Code Example

----
angular.moudle('docsSimpleDirectiveLD', []);  // Noncompliant
----


== Compliant Solution

----
angular.module('docsSimpleDirectiveLD', []);  // Compliant
----

