<p>There is such a thing as software that’s too helpful, and that’s the case with DB2’s implicit casting of host variables used in &lt;code&gt;WHERE&lt;/code&gt; clauses. If for instance you compare a varchar column with the value of a numeric host variable, starting with version 10, DB2 will silently convert the numeric value to a string - at a potentially huge performance cost.</p>
<p>This rule raises an issue when the type of a variable used with a SQL &lt;code&gt;WHERE&lt;/code&gt; clause does not match the underlying type of the column to which it is compared.</p>
<h2>Noncompliant Code Example</h2>
<pre>
TODO
</pre>
