There is such a thing as software that's too helpful, and that's the case with DB2's implicit casting of host variables used in <code>WHERE</code> clauses. If for instance you compare a varchar column with the value of a numeric host variable, starting with version 10, DB2 will silently convert the numeric value to a string - at a potentially huge performance cost. 

This rule raises an issue when the type of a variable used with a SQL <code>WHERE</code> clause does not match the underlying type of the column to which it is compared.


== Noncompliant Code Example

----
TODO
----

