<p>On the principle that clearer code is better code, you should explicitly &lt;code&gt;import&lt;/code&gt; the things you want to use in a module. Using &lt;code&gt;import *&lt;/code&gt; imports everything in the module, and runs the risk of confusing maintainers. Similarly, &lt;code&gt;export * from "module";&lt;/code&gt; imports and then re-exports everything in the module, and runs the risk of confusing not just maintainers but also users of the module.</p>
<h2>Noncompliant Code Example</h2>
<pre>
import * as Imported from "aModule";  // Noncompliant
</pre>
