Changing the configuration of database options <code>ANSI_NULLS</code>, <code>ANSI_PADDING</code> and <code>CONCAT_NULL_YIELDS_NULL</code> is deprecated. The future versions of SQL Server will only support the <code>ON</code> value, and the <code>SET</code> statement for those options to <code>OFF</code> will eventually generate an error.


== Noncompliant Code Example

----
SET ANSI_NULLS OFF -- Noncompliant
SELECT column1 FROM table1 WHERE id = NULL

SET ANSI_PADDING OFF -- Noncompliant
SET CONCAT_NULL_YIELDS_NULL ON -- Noncompliant

SET ANSI_PADDING ON -- "ON" is ignored
----


== Compliant Solution

----
SELECT column1 FROM table1 WHERE id IS NULL
----

