include::../description.adoc[]

== Noncompliant Code Example

----
if ((str = cont.substring(pos1, pos2)) != '') { // Noncompliant
  //...
}
----

== Compliant Solution

----
str = cont.substring(pos1, pos2);
if (str != '') {
  //...
}
----

include::../see.adoc[]
