include::../description.adoc[]

== Noncompliant Code Example

----
if (x < 0)
  new IllegalArgumentException("x must be nonnegative");
----

== Compliant Solution

----
if (x < 0)
  throw new IllegalArgumentException("x must be nonnegative");
----
