include::../description.adoc[]

== Noncompliant Code Example

----
for (int i = 0; i < 10; i++) {
  ...
  i = i - 1; // Noncompliant; counter updated in the body of the loop
  ...
} 
----

== Compliant Solution

----
for (int i = 0; i < 10; i++) {...} 
----
