If a class method doesn't access properties of that class (i.e. it doesn't use <code>this</code>), it should be made <code>static</code>.

== Noncompliant Code Example

----
class Foo {
  constructor() {
    this.i = 1;
  }

  bar() {
    window.focus();
  }
}
----

== Compliant Solution

----
class Foo {
  constructor() {
    this.i = 1;
  }

  static bar() {
    window.focus();
  }
}
----

== Exceptions

The constructor is exempt of this rule.
