include::../description.adoc[]

== Noncompliant Code Example

----
def doSomething() = {
    var name = ""
    // ...
    name = name
}
----

== Compliant Solution

----
def doSomething() = {
    var name = ""
    // ...
    this.name = name
}
----

include::../see.adoc[]
