Variant functions are inefficient by nature, and should not be used with strings when string-specific versions, denoted by a <code>$</code> at the end of the name, are available.

This rule flags instances of these functions:
<code>Left()</code>, <code>Mid()</code>, <code>Right()</code>, <code>Chr()</code>, <code>ChrW()</code>, <code>UCase()</code>, <code>LCase()</code>, <code>LTrim()</code>, <code>RTrim()</code>, <code>Trim()</code>, <code>Space()</code>, <code>String()</code>, <code>Format()</code>, <code>Hex()</code>, <code>Oct(),</code> <code>Str()</code>, <code>Error</code>


== Noncompliant Code Example

----
TrimString = Trim(TestString)
----


== Compliant Solution

----
TrimString = Trim$(TestString)
----

