<p>OS/VS COBOL accepted the &lt;code&gt;NOTE&lt;/code&gt; statement, but IBM Enterprise COBOL does not. Therefore all &lt;code&gt;NOTE&lt;/code&gt; statements should be replaced by standard comment lines.</p>
<h2>Noncompliant Code Example</h2>
<pre>
       IDENTIFICATION DIVISION.
       PROGRAM-ID. foo.

       PROCEDURE DIVISION.
      * Non-Compliant
         NOTE This is a comment.

      * This is a compliant comment.
       END PROGRAM foo.
</pre>
<h2>Compliant Solution</h2>
<pre>
       IDENTIFICATION DIVISION.
       PROGRAM-ID. foo.

       PROCEDURE DIVISION.
      * Compliant
      * This is a comment.
       END PROGRAM foo.
</pre>
