OS/VS COBOL accepted the <code>NOTE</code> statement, but IBM Enterprise COBOL does not. Therefore all <code>NOTE</code> statements should be replaced by standard comment lines.


== Noncompliant Code Example

----
       IDENTIFICATION DIVISION.
       PROGRAM-ID. foo.

       PROCEDURE DIVISION.
      * Non-Compliant
         NOTE This is a comment.

      * This is a compliant comment.
       END PROGRAM foo.
----


== Compliant Solution

----
       IDENTIFICATION DIVISION.
       PROGRAM-ID. foo.

       PROCEDURE DIVISION.
      * Compliant
      * This is a comment.
       END PROGRAM foo.
----

