While <code>END</code> will adequately close a statement, it is less clear than the use of the relevant, statement-specific <code>ENDxx</code>. Thus, the statement-specific version is preferred to facilitate code maintenance and enhance clarity.

This rule is applied to the following operations: <code>CASxx</code>, <code>DO</code>, <code>DOU</code>, <code>DOUxx</code>, <code>DOW</code>, <code>DOWxx</code>, <code>FOR</code>, <code>IF</code>, <code>IFxx</code> and <code>SELECT</code> groups.


== Noncompliant Code Example

----
C     W0PKEX        DOUEQ     W0ON   
...
C                   END     

C     W0PKEX        IFEQ      W0ON   
...
C                   END  
----

----
 /free
  if x = 1;
    ...
  end;
 /end-free
----


== Compliant Solution

----
C     W0PKEX        DOUEQ     W0ON   
...
C                   ENDDO     

C     W0PKEX        IFEQ      W0ON   
...
C                   ENDIF     
----

----
 /free
  if x = 1;
    ...
  endif;
 /end-free
----


