After calling CICS commands with the <code>RESP</code> or <code>NOHANDLE</code> options, the return code should be tested.


== Noncompliant Code Example

----
  EXEC CICS DELETEQ TS        *> Noncompliant; WS-STATUS should have been tested before the MOVE
    QNAME(WS-TS5FTARF-NAME)
    RESP(WS-STATUS)        
  END-EXEC.                   
  MOVE WS-EIBTASKN (4:4) TO WS-TS5FTAR1-NAME-TSKID.
----


== Compliant Solution

----
  EXEC CICS DELETEQ TS
    QNAME(WS-TS5FTARF-NAME)
    RESP(WS-STATUS)        
  END-EXEC.                   
  IF WS-STATUS ...

  MOVE WS-EIBTASKN (4:4) TO WS-TS5FTAR1-NAME-TSKID.
----

