By default, the Maven Surefire plugin only executes test classes with names that end in "Test" or "TestCase". Name your class "TestClassX.java", for instance, and it will be skipped.

This rule raises an issue for each test class with a name not ending in "Test" or "TestCase".


== Noncompliant Code Example

----
public class TestClassX {  // Noncompliant
  @Test
  public void testDoTheThing() {
    //...
----


== Compliant Solution

----
public class ClassXTest {
  @Test
  public void testDoTheThing() {
    //...
----


