PHP has two sets of logical operators: <code>&&</code> / <code>||</code>, and <code>and</code> / <code>or</code>. The difference between the sets is precedence. Because <code>and</code> / <code>or</code> have a lower precedence than almost any other operator, using them instead of <code>&&</code> / <code>||</code> may not have the result you expect.


== Noncompliant Code Example

----
$have_time = true;
$have_money = false;
$take_vacation = $have_time and $have_money;  // Noncompliant. $take_vacation == true.
----


== Compliant Solution

----
$have_time = true;
$have_money = false;
$take_vacation = $have_time && $have_money;  // $take_vacation == false.
----


