include::../description.adoc[]

== Noncompliant Code Example

----
private void DoSomething()
{
    // ...
    Console.WriteLine("so far, so good..."); // Noncompliant
    // ...
}
----

== Exceptions

The following are ignored by this rule:
* Console Applications
* Calls in methods decorated with <code>[Conditional ("DEBUG")]</code>
* Calls included in DEBUG preprocessor branches (<code>#if DEBUG</code>)

include::../see.adoc[]
