<p>Merging collapsible &lt;code&gt;IF&lt;/code&gt; statements increases the codeâ€™s readability.</p>
<h2>Noncompliant Code Example</h2>
<pre>
       IF CONDITION1 THEN
         IF CONDITION2 THEN
           ...
         END-IF
       END-IF.
</pre>
<h2>Compliant Solution</h2>
<pre>
       IF CONDITION1 AND CONDITION2 THEN
         ...
       END-IF.
</pre>
