<p>Merging collapsible &lt;code&gt;if&lt;/code&gt; statements increases the codeâ€™s readability.</p>
<h2>Noncompliant Code Example</h2>
<pre>
IF something THEN
  IF something_else THEN
    -- ...
  END IF;
END IF;
</pre>
<h2>Compliant Solution</h2>
<pre>
IF something AND something_else THEN
  -- ...
END IF;
</pre>
