<p>Merging collapsible &lt;code&gt;if&lt;/code&gt; statements increases the codeâ€™s readability.</p>
<h2>Noncompliant Code Example</h2>
<pre>
if condition1 {
  if condition2 {
    doSomething()
  }
}

if let y = someOptional {
  if x &gt; 0 {
    doSomething()
  }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
if condition1 &amp;&amp; condition2 {
  doSomething()
}

if let y = someOptional where x &gt; 0 {
  doSomething()
}

if x &gt; 0, let y = someOptional {
  doSomething()
}
</pre>
