<p>Merging collapsible &lt;code&gt;if&lt;/code&gt; statements increases the codeâ€™s readability.</p>
<h2>Noncompliant Code Example</h2>
<pre>
IF something
  IF something_else
    -- ...
</pre>
<h2>Compliant Solution</h2>
<pre>
IF something AND something_else
  -- ...
</pre>
