<p>Merging collapsible &lt;code&gt;if&lt;/code&gt; statements increases the codeâ€™s readability.</p>
<h2>Noncompliant Code Example</h2>
<pre>
If a And b Then
  If c And d Then
    doSomething()
  End If
End If
</pre>
<h2>Compliant Solution</h2>
<pre>
If a And b And c And d Then
  doSomething()
End If
</pre>
