Because <code>RESULT_CACHE</code>-enabled functions increase memory consumption, one should double-check that the gain in performances is significant, and avoid over-using this feature in general.


== Noncompliant Code Example

----
CREATE FUNCTION myFastFunction RETURN PLS_INTEGER RESULT_CACHE AS -- Noncompliant
BEGIN
  RETURN 42;
END;
/

DROP FUNCTION myFastFunction;
----

