C++ does not support polymorphic copy or move assignment operators. For example, the signature of a copy assignment operator on a "Base" class would be <code>Base& operator=(const Base& other)</code>. 

And on a "Derived" class that extends "Base", it would be <code>Derived& operator=(const Derived& other)</code>.

Because these are two entirely different method signatures, the second method does not override the first, and adding <code>virtual</code> to the "Base" signature does not change which method is called. 

It is possible to add an <code>operator=</code> override in a derived class, but doing so is an indication that you may need to reexamine your application architecture.


== Noncompliant Code Example

----
class Base {
public:
  virtual Base& operator=(const Base& other); // Noncompliant
};

class Derived : public Base {
public:
  Derived& operator=(const Derived& other);
};
----


== Compliant Solution

----
class Base {
protected:
  Base& operator=(const Base& other); // not virtual
};

class Derived : public Base {
public:
  Derived& operator=(const Derived& other);
};
----


== See

* https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#c60-make-copy-assignment-non-virtual-take-the-parameter-by-const-and-return-by-non-const[C++ Core Guidelines - C.60] - Make copy assignment non-virtual, take the parameter by const&, and return by non-const&
* https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#c63-make-move-assignment-non-virtual-take-the-parameter-by--and-return-by-non-const-[C++ Core Guidelines - C.63] - Make move assignment non-virtual, take the parameter by &&, and return by non-const &

