include::../description.adoc[]

== Noncompliant Code Example

----
void doSomething(Integer a, Integer b) {     // "b" is unused
  compute(a);
}
----

== Compliant Solution

----
void doSomething(Integer a) {
  compute(a);
}
----

include::../see.adoc[]
