<p>String literals written on multiple lines should not be used, especially since the new line characters are <strong>not</strong> part of the string anyway. The concatenation operator &lt;code&gt;||&lt;/code&gt; should be used instead. Moreover, if trailing comments are present, they can be automatically removed by text editors and lead to unexpected results.</p>
<h2>Noncompliant Code Example</h2>
<pre>
foo: proc options(main);
  put list ('Hello,      /* Noncompliant; trailing space is not readable and may be stripped */
world');
end;
</pre>
<h2>Compliant Solution</h2>
<pre>
foo: proc options(main);
  put list ('Hello, ' ||
'world');
end;
</pre>
