include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression <code>^[A-Z][a-zA-Z0-9]*$</code>:

----
public class myClass {...}
----

== Compliant Solution

----
public class MyClass {...}
----
