<p>Unlike class and instance methods, static methods don’t receive an implicit first argument. Nonetheless naming the first argument &lt;code&gt;self&lt;/code&gt; or &lt;code&gt;cls&lt;/code&gt; guarantees confusion - either on the part of the original author, who may never understand why the arguments don’t hold the values he expected, or on that of future maintainers.</p>
<h2>Noncompliant Code Example</h2>
<pre>
class MyClass:
  @staticmethod
  def s_meth(self, arg1, arg2):  #Noncompliant
    # ...
</pre>
<h2>Compliant Solution</h2>
<pre>
class MyClass:
  @staticmethod
  def s_meth(arg1, arg2):  #Noncompliant
    # ...
</pre>
