<p>If a &lt;code&gt;RuntimeException&lt;/code&gt; is thrown while code is unit tested, it’s likely to be thrown again in production. Therefore, test methods should be allowed to &lt;code&gt;throw RuntimeException&lt;/code&gt;s so you’re alerted early to problems in the code.</p>
<h2>Noncompliant Code Example</h2>
<pre>
@Test
public void testTheThing() {
  try {
    // ...
  catch (RuntimeExeption e) {  // Noncompliant
    // ...
  }
}
</pre>
