Some exception classes are designed to be used only as base classes to more specific exceptions, for instance <code>std::exception</code> (the base class of all standard C++ exceptions), <code>std::logic_error</code> or <code>std::runtime_error</code>.

Catching such a generic exception types is a usually bad idea, because it implies that the "catch" block is clever enough to handle any type of exception.

== Noncompliant Code Example

----
try {
  /* code that may throw std::system_error */
} catch (const std::exception &ex) { // Noncompliant
  /*...*/
}
----

== Compliant Solution

----
try {
  /* code that may throw std::system_error */
} catch (const std::system_error &ex) {
  /*...*/
}
----

== Exceptions

There are cases though where you want to catch all exceptions, because no exceptions should be allowed to escape the function, and generic <code>catch</code> handlers are excluded from the rule:
* In the main function
* In a <code>noexcept</code> function
* In an <code>extern "C"</code> function

Additionally, if the <code>catch</code> handler is throwing an exception (either the same as before, with <code>throw;</code> or a new one that may make more sense to the callers of the function), or is never exiting (because it calls a <code>noreturn</code> function, for instance <code>exit</code>), then the accurate type of the exception usually does not matter any longer: this case is excluded too.

include::../see.adoc[]
