<code>ID</code> and <code>IDREF</code> are constructs inherited from DTDs, and they suffer from painful drawbacks, mainly:

* <code>ID</code>s are too widely scoped: an <code>ID</code> must be unique in the XML document. This means that if an XML document contains <code><Person></code> elements and <code><Dog></code> elements, both with a "name" attribute of <code>type="xs:ID"</code>, then two people can't have the same name. Neither can two dogs, or a person and a dog.
* IDs must comply with tight naming scheme. For example, "123" is not a valid value for an ID.

Instead, <code>KEY</code> and <code>KEYREF</code> should be used. And if you really need to insure uniqueness, use <code>xs:unique</code> to add a uniqueness constraint.


== Noncompliant Code Example

----
ToDo
----


== Compliant Solution

----
ToDo
----

