<p>Property getters should be simple operations that are always safe to call. If exceptions need to be thrown, it is best to convert the property to a method.</p>
<p>It is valid to throw exceptions from indexed property getters and from property setters, which are not detected by this rule.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public int Foo
{
    get
    {
        throw new Exception(); // Noncompliant
    }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
public int Foo
{
    get
    {
        return 42;
    }
}
</pre>
<h2>Exceptions</h2>
<p>No issue is raised when the thrown exception derives from or is of type &lt;code&gt;NotImplementedException&lt;/code&gt;, &lt;code&gt;NotSupportedException&lt;/code&gt; or &lt;code&gt;InvalidOperationException&lt;/code&gt;.</p>
