This rule applies whenever an <code>IF</code> statement is followed by one or more <code>ELSE IF</code> statements; the final <code>ELSE IF</code> should be followed by an <code>ELSE</code> statement.

The requirement for a final <code>ELSE</code> statement is defensive programming.
The <code>ELSE</code> statement should either take appropriate action or contain a suitable comment as to why no action is taken. This is consistent with the requirement to have a final <code>ELSE</code> clause in a <code>CASE</code> statement.

== Noncompliant Code Example

----
IF @x = 1
  PRINT 'A'
ELSE IF @x = 2
  PRINT 'B'
ELSE IF @x = 3
  PRINT 'C'
-- Noncompliant; final ELSE is missing
----

== Compliant Solution

----
IF @x = 1
  PRINT 'A'
ELSE IF @x = 2
  PRINT 'B'
ELSE IF @x = 3
  PRINT 'C'
ELSE
  PRINT 'Z'
----

include::../see.adoc[]
