include::../description.adoc[]

== Noncompliant Code Example

----
if (condition1) {
  if (condition2) {             // NonCompliant
    /* ... */
  }
}
----

== Compliant Solution

----
if (condition1 && condition2) { // Compliant
  /* ... */
}
----
