Each element should be on a line to itself.


== Noncompliant Code Example

----
<parent><child /></parent>  <!-- Noncompliant -->
----


== Compliant Solution

----
<parent>
  <child />
</parent>
----

