include::../description.adoc[]

== Noncompliant Code Example

With a threshold of 5:
----
switch ($var) {
  case 0:  // 6 lines till next case
    methodCall1();
    methodCall2();
    methodCall3();
    methodCall4();
    break;
  default:
    break;
}
----

== Compliant Solution

----
switch ($var) {
  case 0:
    doSomething();
    break;
  default:
    break;  
}

function doSomething(){
  methodCall1("");
  methodCall2("");
  methodCall3("");
  methodCall4("");
}
----
