<p>Declaring a member function static or const limits its access to the non-static data members.</p>
<p>This helps to prevent unintentional modification of the data, and facilitates compliance with MISRA C++ 2008 Rule 7–1–1 (A variable which is not modified shall be const qualified).</p>
<h2>Noncompliant Code Example</h2>
<pre>
class A
{
public:
  int16_t f1 ( ) // Non-compliant – can be const
  {
    return m_i;
  }
  int16_t f2 ( ) // Non-compliant – can be static
  {
    return m_s;
  }
private:
  int16_t m_i;
  static int16_t m_s;
};
</pre>
<h2>Compliant Solution</h2>
<pre>
class A
{
public:
  int16_t const f1 ( ) // Compliant
  {
    return m_i;
  }
  static int16_t f2 ( ) // Compliant
  {
    return m_s;
  }
  int16_t f3 ( ) // Compliant – cannot be const or static
  {
    return ++m_i;
  }
private:
  int16_t m_i;
  static int16_t m_s;
};
</pre>
<h2>See</h2>
<ul>
<li>MISRA C++:2008, 9-3-3</li>
</ul>
