<p>Calling &lt;code&gt;Iterator.hasNext()&lt;/code&gt; is not supposed to have any side effects, and therefore should not change the state of the iterator. &lt;code&gt;Iterator.next()&lt;/code&gt; advances the iterator by one item. So calling it inside &lt;code&gt;Iterator.hasNext()&lt;/code&gt;, breaks the &lt;code&gt;hasNext()&lt;/code&gt; contract, and will lead to unexpected behavior in production.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public class FibonacciIterator implements Iterator&lt;Integer&gt;{
...
@Override
public boolean hasNext() {
  if(next() != null) {
    return true;
  }
  return false;
}
...
}
</pre>
