include::../description.adoc[]

== Noncompliant Code Example

----
func doNothing() { // Noncompliant
}
----

== Compliant Solution

----
func doNothing() {
  // Do nothing because of X and Y.
}
----
