Leftover empty blocks are usually introduced by mistake. They are useless and prevent readability of the code. They should be removed or completed with real code.


== Noncompliant Code Example

----
a { }
----


== Compliant Solution

----
a { color: pink; }
----

