<p>It is possible in the same statement, to declare a user-defined type (&lt;code&gt;class&lt;/code&gt;, &lt;code&gt;struct&lt;/code&gt;, &lt;code&gt;union&lt;/code&gt; or &lt;code&gt;enum&lt;/code&gt;) followed by variable declarations of this type. But mixing more than one concern in a single statement is confusing for maintainers.</p>
<p>This rule raises an issue when a variable is declared at the end of a user-defined type declaration statement.</p>
<h2>Noncompliant Code Example</h2>
<pre>
struct Container { int size; } container; // Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
struct Container { int size; };
Container container;
</pre>
