It is possible in the same statement, to declare a user-defined type (<code>class</code>, <code>struct</code>, <code>union</code> or <code>enum</code>) followed by variable declarations of this type. But mixing more than one concern in a single statement is confusing for maintainers.

This rule raises an issue when a variable is declared at the end of a user-defined type declaration statement.


== Noncompliant Code Example

----
struct Container { int size; } container; // Noncompliant
----


== Compliant Solution

----
struct Container { int size; };
Container container;
----


