<p>When you create a &lt;code&gt;DataTable&lt;/code&gt; or &lt;code&gt;DataSet&lt;/code&gt;, you should set the locale explicitly. By default, the locale for these types is the current culture. For data that is stored in a database or file and is shared globally, the locale should ordinarily be set to the invariant culture (&lt;code&gt;CultureInfo.InvariantCulture&lt;/code&gt;).</p>
<p>This rule raises an issue when &lt;code&gt;System.Data.DataTable&lt;/code&gt; or &lt;code&gt;System.Data.DataSet&lt;/code&gt; instances are created without explicitly setting the locale property (&lt;code&gt;DataTable.Locale&lt;/code&gt; or &lt;code&gt;DataSet.Locale&lt;/code&gt;).</p>
<h2>Noncompliant Code Example</h2>
<pre>
using System;
using System.Data;

namespace MyLibrary
{
    public class Foo
    {
        public DataTable CreateTable()
        {
            DataTable table = new DataTable("Customers"); // Noncompliant table.Locale not set
            DataColumn key = table.Columns.Add("ID", typeof(Int32));

            key.AllowDBNull = false;
            key.Unique = true;
            table.Columns.Add("LastName", typeof(String));
            table.Columns.Add("FirstName", typeof(String));
            return table;
        }
    }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
using System;
using System.Data;
using System.Globalization;

namespace MyLibrary
{
    public class Foo
    {
        public DataTable CreateTable()
        {
            DataTable table = new DataTable("Customers");
            table.Locale = CultureInfo.InvariantCulture;
            DataColumn key = table.Columns.Add("ID", typeof(Int32));

            key.AllowDBNull = false;
            key.Unique = true;
            table.Columns.Add("LastName", typeof(String));
            table.Columns.Add("FirstName", typeof(String));
            return table;
        }
    }
}
</pre>
