<p>Sharing some naming conventions is a key point to make it possible for a team to efficiently collaborate. This rule allows to check that all variable names match a provided regular expression.</p>
<h2>Noncompliant Code Example</h2>
<pre>
foo: proc options(main);
  declare bar# fixed decimal; /* Noncompliant */
end;
</pre>
<h2>Compliant Solution</h2>
<pre>
foo: proc options(main);
  declare bar fixed decimal;  /* Compliant */
end;
</pre>
