There's no need to end a <code>void</code> method with a <code>return</code> statement. Since nothing is actually being returned, the <code>return</code> can be omitted; <code>void</code> will be returned implicitly.


== Noncompliant Code Example

----
public void incrementCount() {
  this.count++;
  return;  // Noncompliant
}
----


== Compliant Solution

----
public void incrementCount() {
  this.count++;
}
----


