<p>It is possible to use the increment operator &lt;code&gt;&lt;/code&gt;, to set the value of a &lt;code&gt;bool&lt;/code&gt;(C) or &lt;code&gt;_Bool&lt;/code&gt;Â© variable to &lt;code&gt;true&lt;/code&gt;. But this feature has been deprecated in C++ since the 1998 version of the standard, and even where allowed, is simply confusing.</p>
<h2>Noncompliant Code Example</h2>
<pre>
bool alive;
...
alive++;
</pre>
<h2>Compliant Solution</h2>
<pre>
bool alive;
...
alive = true;
</pre>
<h2>See</h2>
<ul>
<li>ISO/IEC 14882:1998, 5.3.2</li>
</ul>
