It is possible to use the increment operator <code>++</code>, to set the value of a <code>bool</code>(C++) or <code>_Bool</code>(C) variable to <code>true</code>. But this feature has been deprecated in C++ since the 1998 version of the standard, and even where allowed, is simply confusing. 


== Noncompliant Code Example

----
bool alive;
...
alive++;
----


== Compliant Solution

----
bool alive;
...
alive = true;
----


== See

*  ISO/IEC 14882:1998, 5.3.2

