<p>Thereâ€™s no need for multiple, consecutive logging calls. Instead, they should be combined, to save I/O time, especially when the logs are written to file. Further, doing so can improve log output readability in concurrent applications; adjacent logger calls will not necessarily result in adjacent log entries in multithreaded environments.</p>
<h2>Noncompliant Code Example</h2>
<pre>
int result = doSomething();
LOGGER.log(LEVEL.info, "doSomething() called");
LOGGER.log(LEVEL.debug, "doSomething response: %n", result);  // Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
int result = doSomething();
LOGGER.log(LEVEL.info, "doSomething() called. Response: %n", result);
</pre>
