Queries with contradictory <code>WHERE</code> clauses will always return empty result sets. This is clearly a bug.


== Noncompliant Code Example

----
SELECT *
FROM fruit
WHERE type='apple' AND type='orange'  -- Noncompliant
----

