Schema versioning is a big topic, with best practices rather than hard-and-fast rules. One of these best practices is to include the <code>version</code> attribute in the <code>xsd:schema</code> and increment it to indicate minor changes. Nothing breaks if the value changes, but it does give the user a relatively prominent indication that something has changed.


== Noncompliant Code Example

----
<xs:schema targetNamespace="http://www.codeSamples.com/fruit"
           xmlns:xs="http://www.w3.org/2001/XMLSchema"
           elementFormDefault="qualified">  <!-- Noncompliant -->
...
----


== Compliant Solution

----
<xs:schema targetNamespace="http://www.codeSamples.com/fruit"
           xmlns:xs="http://www.w3.org/2001/XMLSchema"
           elementFormDefault="qualified" version="1">
...
----

