Python interprets <code>False</code> as a "false" boolean value, and <code>True</code> as a "true" boolean value. Assigning them other values could lead to unexpected behaviors.

This rule raises an issue when "True" and "False", with any case, are used as identifiers.


== Noncompliant Code Example

----
True = "red"  # Noncompliant
----


== Compliant Solution

----
color = "red"  # Noncompliant
----

