<p>The path provided here doesnâ€™t match the actual path on this file system (e.g. the case is different). While this may work on a particular environment, this is not portable and may fail on a different environment.</p>
<h2>Noncompliant Code Example</h2>
<pre>
#include "Foo.h" // Noncompliant, the file name is "foo.h"
</pre>
