== Noncompliant Code Example

----
class MyException: public std::exception
{
public:
  explicit MyException(const char* message):
    msg_(message)
    {
      //...
      if (bad_thing) {
        throw std::exception("A bad thing happened");  // Noncompliant
      }
    }
----