There are two classes in the Java standard library that deal with iterations: <code>Iterable<T></code> and <code>Iterator<T></code>. An <code>Iterable<T></code> represents a data structure that can be the target of the "for-each loop" statement, and an <code>Iterator<T></code> represents the state of an ongoing traversal. An <code>Iterable<T></code> is generally expected to support multiple traversals.
An <code>Iterator<T></code> that also implements <code>Iterable<t></code> by returning itself as its <code>iterator()</code> will not support multiple traversals since its state will be carried over.

This rule raises an issue when the <code>iterator()</code> method of a class implementing both <code>Iterable<T></code> and <code>Iterator<t></code> returns <code>this</code>.


== Noncompliant Code Example

----
class FooIterator implements Iterator<Foo>, Iterable<Foo> {
  private Foo[] seq;
  private int idx = 0;

  public boolean hasNext() {
    return idx < seq.length;
  }

  public Foo next() {
    return seq[idx++];
  }

  public Iterator<Foo> iterator() {
    return this; // Noncompliant
  }
  // ...
}
----


== Compliant Solution

----
class FooSequence implements Iterable<Foo> {
  private Foo[] seq;

  public Iterator<Foo> iterator() {
    return new Iterator<Foo>() {
      private int idx = 0;

      public boolean hasNext() {
        return idx < seq.length;
      }

      public Foo next() {
        return seq[idx++];
      }
    };
  }
  // ...
}
----

