include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression <code>^[_a-z][a-zA-Z0-9]*$</code>:
----
public function doSomething(my_param:int):void
{  
  var LOCAL:int;
  ...
}
----

== Compliant Solution

----
public function doSomething(myParam):void
{  
  var local;
  ...
}
----
