Control characters aren't visible to maintainers, so they should be escaped.

== Noncompliant Code Example

----
const char* foo = "A	B";  // Noncompliant, contains a tabulation
----

== Compliant Solution

----
const char* foo = "A\tB";  // Compliant, use escaped value
----

== Exceptions

Raw string literals has no escape character mechanism.
