<p>Thereâ€™s no point in forcing the overhead of a function or method call for a function that always returns the same constant value. Even worse, the fact that a function call must be made will likely mislead developers who call the method thinking that something more is done. Declare a constant instead.</p>
<p>This rule raises an issue on functions that contain only one  statement: the &lt;code&gt;return&lt;/code&gt; of a constant value.</p>
<h2>Noncompliant Code Example</h2>
<pre>
func getBestNumber() -&gt; Int {
  return 12  // Noncompliant
}
</pre>
<h2>Compliant Solution</h2>
<pre>
let bestNumber = 12;
</pre>
<h2>Exceptions</h2>
<p>Methods which are members of a class having a type inheritance clause are ignored.</p>
