It is far faster to compare the length of a string to 0 than it is to compare the string itself to empty string. Further, the <code>LenB</code> implementation is faster than the <code>Len</code> implementation, and is therefore preferred.


== Noncompliant Code Example

----
If Text = "" Then
If Text <> "" Then
----


== Compliant Solution

----
If LenB(Text) = 0 Then
If LenB(Text) <> 0 Then
----

