include::../description.adoc[]

== Noncompliant Code Example

----
a = b + c;   // This is a trailing comment that could be very very long
----

== Compliant Solution

----
// This very long comment is better placed before the line of code
a = b + c;
----
