This rule applies whenever an <code>IF</code> statement is followed by one or more <code>ELSEIF</code> statements; the final <code>ELSEIF</code> should be followed by an <code>ELSE</code> statement.
The requirement for a final <code>ELSE</code> statement is defensive programming. 
The <code>ELSE</code> statement should either take appropriate action or contain a suitable comment as to why no action is taken. This is consistent with the requirement to have a final <code>OTHER</code> clause in a <code>SELECT</code> statement. 

== Noncompliant Code Example

----
C                   IF         RESULT = 0
C                   ...
C                   ELSEIF     RESULT > 0
C                   ...
C                   ENDIF
----

----
 /free
   if result = 0;
     ...
   elseif result > 0;
     ...
   endif;
 /end-free
----

== Compliant Solution

----
C                   IF         RESULT = 0
C                   ...
C                   ELSEIF     RESULT > 0
C                   ...
C                   ELSE
C                   EXSR       ERRO
C                   ENDIF
----

----
 /free
   if result = 0;
     ...
   elseif result > 0;
     ...
   else;
     exsr ERR0;
   endif;
 /end-free
----

include::../see.adoc[]
