Delegate event handlers (i.e. delegates used as type of an event) should have a very specific signature:

* Return type <code>void</code>.
* First argument of type <code>System.Object</code> and named 'sender'.
* Second argument of type <code>System.EventArgs</code> (or any derived type) and is named 'e'.

This rule raises an issue whenever a <code>delegate</code> declaration doesn't match that signature.


== Noncompliant Code Example

----
public delegate void AlarmEventHandler(object s);

public class Foo
{
    public event AlarmEventHandler AlarmEvent; // Noncompliant
}
----


== Compliant Solution

----
public delegate void AlarmEventHandler(object sender, AlarmEventArgs e);

public class Foo
{
    public event AlarmEventHandler AlarmEvent; // Compliant
}
----


== See

https://msdn.microsoft.com/en-us/library/edzehd2t.aspx[Handling and Raising Events]

