include::../description.adoc[]

== Noncompliant Code Example

----
IF NOT (@a = 2) -- Noncompliant
BEGIN
  ...
END

IF NOT (@b < 10) -- Noncompliant
BEGIN
  ...
END
----

== Compliant Solution

----
IF @a <> 2
BEGIN
  ...
END

IF @b >= 10
BEGIN
  ...
END
----
