<p>In a Zen-like manner, "NULL" is never equal to anything, even itself. Therefore comparisons using equality operators will always return &lt;code&gt;False&lt;/code&gt;, even when the value actually &lt;code&gt;IS NULL&lt;/code&gt;.</p>
<p>For that reason, comparison operators should never be used to make comparisons with &lt;code&gt;NULL&lt;/code&gt;; &lt;code&gt;IS NULL&lt;/code&gt; and &lt;code&gt;IS NOT NULL&lt;/code&gt; should be used instead. This extends as well to empty string (""), which is equivalent to &lt;code&gt;NULL&lt;/code&gt; for some database engines.</p>
<h2>Noncompliant Code Example</h2>
<pre>
SET SERVEROUTPUT ON

DECLARE
  name VARCHAR2(42) := ''; /* This is equivalent to name := NULL; */
BEGIN
  IF name = '' THEN /* Noncompliant; equivalent to name = NULL &amp; therefore always FALSE*/
    DBMS_OUTPUT.PUT_LINE('True'); /* Unreachable */
  ELSE
    DBMS_OUTPUT.PUT_LINE('False');
  END IF;
END;
/
</pre>
<h2>Compliant Solution</h2>
<pre>
SET SERVEROUTPUT ON

DECLARE
  name VARCHAR2(42) := ''; /* This is equivalent to name := NULL; */
BEGIN
  IF name IS NULL THEN /* Compliant */
    DBMS_OUTPUT.PUT_LINE('True');
  ELSE
    DBMS_OUTPUT.PUT_LINE('False');
  END IF;
END;
/
</pre>
