include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression <code>^[_a-z][a-zA-Z0-9]*$</code>:

----
class MyClass {
  public var my_field:int;
}
----

== Compliant Solution

----
public class MyClass {
   public var myField:int;
}
----
