include::../description.adoc[]

== Noncompliant Code Example

With default provided regular expression <code>^[A-Z][a-zA-Z0-9]*$</code>:

----
enum someEnumeration { // Noncompliant
};
----

== Compliant Solution

----
enum SomeEnumeration {
};
----
