Shared coding conventions allow teams to collaborate efficiently. This rule checks that type specifiers always appear in the following order:
# <code>typedef</code>
# type name, spelling of built-in types with more than one type-specifier:
## signedness - <code>signed</code> or <code>unsigned</code>
## last single type-specifier or
##* <code>short int</code>
##* <code>long int</code>
##* <code>long long int</code>
##* <code>long double</code>

Since the positioning of the <code>const</code> keyword is controversial, this rule does not check it.


== Noncompliant Code Example

----
int typedef T;

double long d;
char unsigned ch;
long signed int i;

----


== Compliant Solution

----
typedef int T;

long double d;
unsigned char ch;
signed long int i;

----


