<p>Omitting the data type of an attribute or element leaves it to default to &lt;code&gt;xs:any&lt;/code&gt; for elements, and &lt;code&gt;xs:anyType&lt;/code&gt; for attributes. However there is no reason for making such a significant design decision implicit..</p>
<h2>Noncompliant Code Example</h2>
<pre>
&lt;xs:element name="fruit"&gt;
  &lt;xs:complexType&gt;
    &lt;xs:sequence&gt;
      &lt;xs:element name="color"/&gt; &lt;!-- Noncompliant --&gt;
    &lt;/xs:sequence&gt;
  &lt;/xs:complexType&gt;
&lt;/xs:element&gt;
</pre>
<h2>Compliant Solution</h2>
<pre>
&lt;xs:element name="fruit"&gt;
  &lt;xs:complexType&gt;
    &lt;xs:sequence&gt;
      &lt;xs:element name="color" type="xs:string"/&gt;
    &lt;/xs:sequence&gt;
  &lt;/xs:complexType&gt;
&lt;/xs:element&gt;
</pre>
