include::../description.adoc[]

== Noncompliant Code Example

----
foo: proc options(main);
  declare bar# fixed decimal; /* Noncompliant */
end;
----

== Compliant Solution

----
foo: proc options(main);
  declare bar fixed decimal;  /* Compliant */
end;
----
