<p>When an assembly uses Windows Forms (classes and interfaces from the &lt;code&gt;System.Windows.Forms&lt;/code&gt; namespace) its entry point should be marked with the &lt;code&gt;STAThreadAttribute&lt;/code&gt; to indicate that the threading model should be "Single-Threaded Apartment" (STA) which is the only one supported by Windows Forms.</p>
<p>This rule raises an issue when the entry point (&lt;code&gt;Shared Sub Main&lt;/code&gt; method) of an assembly using Windows Forms is not marked as STA.</p>
<h2>Noncompliant Code Example</h2>
<pre>
Imports System.Windows.Forms

Public Class Foo
  Shared Sub Main()
    Dim winForm As Form = New Form
    Application.Run(winForm)
  End Sub
End Class
</pre>
<h2>Compliant Solution</h2>
<pre>
Imports System.Windows.Forms

Public Class Foo
  &lt;STAThread()&gt; Shared Sub Main()
    Dim winForm As Form = New Form
    Application.Run(winForm)
  End Sub
End Class
</pre>
