include::../description.adoc[]

== Noncompliant Code Example

----
int i = 0;
while (true)
{
  if (i == 10)
  {
    break;      // Non-Compliant
  }

  Console.WriteLine(i);
  i++;
}
----

== Compliant Solution

----
int i = 0;
while (i != 10) // Compliant
{
  Console.WriteLine(i);
  i++;
}
----
