<p>&lt;code&gt;IMPORT&lt;/code&gt;ed values should be treated as read-only. Doing otherwise could result in unpleasant surprises down the line.</p>
<h2>Noncompliant Code Example</h2>
<pre>
D VAR2             S              5  0 IMPORT
 * Noncompliant
C           MOVE       MYVAR     VAR2

C/FREE
  // Noncompliant
  VAR2=MYVAR;
/END-FREE
</pre>
