While most <code>compareTo</code> methods return -1, 0, or 1, some do not, and testing the result of a <code>compareTo</code> against a specific value other than 0 could result in false negatives.


== Noncompliant Code Example

----
if (myClass.compareTo(arg) == -1) {  // Noncompliant
  // ...
}
----


== Compliant Solution

----
if (myClass.compareTo(arg) < 0) {
  // ...
}
----


