In the past, it was required to load a JDBC driver before creating a <code>java.sql.Connection</code>. Nowadays, when using JDBC 4.0 drivers, this is no longer required and <code>Class.forName()</code> can be safely removed because JDBC 4.0 (JDK 6) drivers available in the classpath are automatically loaded.

This rule raises an issue when <code>Class.forName()</code> is used with one of the following values:

* <code>com.mysql.jdbc.Driver</code>
* <code>oracle.jdbc.driver.OracleDriver</code>
* <code>com.ibm.db2.jdbc.app.DB2Driver</code>
* <code>com.ibm.db2.jdbc.net.DB2Driver</code>
* <code>com.sybase.jdbc.SybDriver</code>
* <code>com.sybase.jdbc2.jdbc.SybDriver</code>
* <code>com.teradata.jdbc.TeraDriver</code>
* <code>com.microsoft.sqlserver.jdbc.SQLServerDriver</code>
* <code>org.postgresql.Driver</code>
* <code>sun.jdbc.odbc.JdbcOdbcDriver</code>
* <code>org.hsqldb.jdbc.JDBCDriver</code>
* <code>org.h2.Driver</code>
* <code>org.firebirdsql.jdbc.FBDriver</code>
* <code>net.sourceforge.jtds.jdbc.Driver</code>


== Noncompliant Code Example

----
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

public class Demo {
  private static final String DRIVER_CLASS_NAME = "org.postgresql.Driver";
  private final Connection connection;

  public Demo(String serverURI) throws SQLException, ClassNotFoundException {
    Class.forName(DRIVER_CLASS_NAME); // Noncompliant; no longer required to load the JDBC Driver using Class.forName()
    connection = DriverManager.getConnection(serverURI);
  }
}
----


== Compliant Solution

----
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

public class Demo {
    private final Connection connection;

    public Demo(String serverURI) throws SQLException {
        connection = DriverManager.getConnection(serverURI);
    }
}
----

