Float data types, such as <code>FLOAT</code>, <code>DOUBLE PRECISION</code>, and <code>REAL</code> cannot have a scale constraint. Trying to specify a scale results in the exception <code>PLS-00510: Float cannot have scale being raised.</code>


== Noncompliant Code Example

----
DECLARE
  foo FLOAT(10, 3); -- Noncompliant - raises PLS-00510
BEGIN
  NULL;
END;
/
----


== Compliant Solution

----
DECLARE
  foo FLOAT(10); -- Compliant
BEGIN
  NULL;
END;
/
----

