include::../description.adoc[]

== Noncompliant Code Example

----
label: -- Noncompliant
  PRINT 'hello world';
GO
----

== Compliant Solution

----
PRINT 'hello world';
GO
----

include::../see.adoc[]
