When calling a CICS command other than <code>RETURN</code>, <code>ADDRESS</code>, or <code>ABEND</code>, either <code>RESP</code> should be used to specify where the response code should be written, or <code>NOHANDLE</code> should be used to specify that abnormal conditions should be ignored.


== Noncompliant Code Example

----
           EXEC CICS DELETEQ TS        *> Noncompliant
             QNAME(WS-TS5FTARF-NAME)
           END-EXEC.
----


== Compliant Solution

----
           EXEC CICS DELETEQ TS        
             QNAME(WS-TS5FTARF-NAME)
             RESP(WS-STATUS)        
           END-EXEC.  
----

