include::../description.adoc[]

== Noncompliant Code Example

----
class Example {
public:
  void foo(int x); // NonCompliant
  void notFoo(int x);
  void foo(double x); // Should be moved next to its overload
private:
  void foo(long x); // Okay since the function has different access specifier
};
----

== Compliant Solution

----
class Example {
public:
  void foo(int x); // Compliant
  void foo(double x);
  void notFoo(int x);
private:
  void foo(long x); // Okay since the function has different access specifier
};
----
