include::../description.adoc[]

== Noncompliant Code Example

----
public function setName($name) {
    $name = $name;
}
----

== Compliant Solution

----
public function setName($name) {
    $this->name = $name;
}
----

include::../see.adoc[]
