include::../description.adoc[]

== Noncompliant Code Example

----
SET SERVEROUTPUT ON

BEGIN
  DBMS_OUTPUT.PUT_LINE('Hello!'); DBMS_OUTPUT.PUT_LINE('This is unreadable!'); -- Noncompliant
END;
/
----

== Compliant Solution

----
SET SERVEROUTPUT ON

BEGIN
  DBMS_OUTPUT.PUT_LINE('Hello!');
  DBMS_OUTPUT.PUT_LINE('This is much better!');
END;
/
----
