<p>Creating a new &lt;code&gt;Exception&lt;/code&gt; without actually throwing it is useless and is probably due to a mistake.</p>
<h2>Noncompliant Code Example</h2>
<pre>
if (x &lt; 0)
{
  new ArgumentException("x must be nonnegative");
}
</pre>
<h2>Compliant Solution</h2>
<pre>
if (x &lt; 0)
{
  throw new ArgumentException("x must be nonnegative");
}
</pre>
