include::../description.adoc[]

== Noncompliant Code Example

----
DECLARE
  maxAge PLS_INTEGER := 60;
BEGIN
  UPDATE employee SET status = 'retired'; -- Noncompliant - the WHERE was forgotten
END;
/
----

== Compliant Solution

----
DECLARE
  maxAge PLS_INTEGER := 60;
BEGIN
  UPDATE employee SET status = 'retired' WHERE age > maxAge;
END;
/
----
