include::../description.adoc[]

== Noncompliant Code Example

----
class Foo {
  public var myField:int;

  public function doSomething():String {
    var myField:int = 0;
    ...
  }
}
----

include::../see.adoc[]
