include::../description.adoc[]

== Noncompliant Code Example

----
string actual = "Hello World!";
actual.Should();  // Noncompliant
----

== Compliant Solution

----
string actual = "Hello World!";
actual.Should().Contain("Hello");
----
