Shared coding conventions allow teams to collaborate effectively. This rule checks that labels match a provided regular expression.


== Noncompliant Code Example

With default provided regular expression <code>^[A-Z][A-Z0-9_]*$</code>:
----
exit:  // Noncompliant
  doCleanup();
----


== Compliant Solution

----
EXIT:  // Compliant
  doCleanup();
----

