include::../description.adoc[]

== Noncompliant Code Example

With the default regular expression <code>^[a-z][a-zA-Z0-9]*$</code>:

----
public function doSomething($my_param){  
  $LOCAL;
  ...
}
----

== Compliant Solution

----
public function doSomething($myParam){  
  $local;
  ...
}
----
