Code is clearest when each statement has its own line. Nonetheless, it is a common pattern to combine on the same line an <code>IF</code> and its resulting _then_ statement. However, when an <code>IF</code> is placed on the same line as the closing <code>END</code> from a preceding <code>ELSE</code> or <code>ELSE IF</code>, it is either an error - <code>ELSE</code> is missing - or the invitation to a future error as maintainers fail to understand that the two statements are unconnected.

== Noncompliant Code Example

----
IF (condition1) BEGIN
  EXEC something
END IF (condition2) BEGIN  -- Noncompliant
  EXEC something
END
----

== Compliant Solution

----
IF (condition1) BEGIN
  EXEC something
END ELSE IF (condition2) BEGIN
  EXEC something
END
----
Or

----
IF (condition1) BEGIN
  EXEC something
END

IF (condition2) BEGIN
  EXEC something
END
----
