Unused parameters are misleading. Whatever the values passed to such parameters, the behavior will be the same.

This rule raises an issue when a <code>private</code> procedure of a <code>Class</code>, <code>Module</code> or <code>Structure</code> takes a parameter without using it.

== Noncompliant Code Example

----
Private Sub DoSomething(ByVal a As Integer, ByVal b as Integer) ' "b" is unused
    Compute(a)
End Sub

Private Function DoSomething2(ByVal a As Integer, ByVal b As Integer) As Integer ' "a" is unused 
    Compute(b)
    Return b
End Function
----

== Compliant Solution

----
Private Sub DoSomething(ByVal a As Integer)
    Compute(a)
End Sub

Private Function DoSomething2(ByVal b As Integer) As Integer
    Compute(b)
    Return b
End Function
----

== Exceptions

This rule doesn't raise any issue in the following contexts:

* Procedures decorated with attributes.
* Empty procedures.
* Procedures which only throw <code>NotImplementedException</code>.
* Main methods.
* <code>virtual</code>, <code>override</code> procedures.
* Interface implementations.
* Event handlers.

include::../see.adoc[]
