<p>Shared coding conventions allow teams to collaborate efficiently. This rule checks that object attribute names match the provided regular expression.</p>
<h2>Noncompliant Code Example</h2>
<p>With the default regular expression &lt;code&gt;[a-zA-Z](_?[a-zA-Z0-9])*&lt;/code&gt; :</p>
<pre>
CREATE TYPE my_type AS OBJECT(
  foo__bar INTEGER             -- Non-Compliant
);
/

DROP TYPE my_type;
</pre>
<h2>Compliant Solution</h2>
<pre>
CREATE TYPE my_type AS OBJECT(
  foo_bar INTEGER              -- Compliant
);
/

DROP TYPE my_type;
</pre>
