Shared coding conventions allow teams to collaborate efficiently. This rule checks that object attribute names match the provided regular expression.


== Noncompliant Code Example

With the default regular expression <code>[a-zA-Z](_?+[a-zA-Z0-9])*+</code> :

----
CREATE TYPE my_type AS OBJECT(
  foo__bar INTEGER             -- Non-Compliant
);
/

DROP TYPE my_type;
----


== Compliant Solution

----
CREATE TYPE my_type AS OBJECT(
  foo_bar INTEGER              -- Compliant
);
/

DROP TYPE my_type;
----

