include::../description.adoc[]

== Noncompliant Code Example

----
var i:Int
for i in 1...10 {
    print("i is \(i)")
    continue  // this is meaningless; the loop would continue anyway
}
----

== Compliant Solution

----
var i:Int
for i in 1...10 {
    print("i is \(i)")
}
----
