include::../description.adoc[]

== Noncompliant Code Example

----
WRITE 'Hello World'. WRITE 'You can use multiple statements on a line'.
----

== Compliant Solution

----
WRITE 'Hello World'. 
WRITE 'You can use multiple statements on a line'.
WRITE 'But you shouldn't.'.
----
