The bitwise operators <code>&</code>, <code>|</code> can be mistaken for the boolean operators <code>&&</code> and <code>||</code>. 

This rule raises an issue when <code>&</code> or <code>|</code> is used in a boolean context.


== Noncompliant Code Example

----
if (a & b) { ... } // Noncompliant; & used in error
----


== Compliant Solution

----
if (a && b) { ... }
----


== Exceptions

When a file contains other bitwise operations, (<code>\^</code>, <code><<</code>, <code>>>></code>, <code>>></code>, <code>~</code>, <code>&=</code>, <code>^=</code>, <code>|=</code>, <code><<=</code>, <code>>>=</code>, <code>>>>=</code> and <code>&</code> or <code>|</code> used with a numeric literal as the right operand) all issues in the file are ignored, because it is evidence that bitwise operations are truly intended in the file.

