<p>The binary algorithm used by &lt;code&gt;SEARCH ALL&lt;/code&gt; is far more efficient for large tables than the one used by &lt;code&gt;SEARCH&lt;/code&gt;. While itâ€™s not always possible to use &lt;code&gt;SEARCH ALL&lt;/code&gt;, it should be the preferred algorithm.</p>
<p>This rule raises an issue when tables with more than the specified number of possible entries are searched using &lt;code&gt;SEARCH&lt;/code&gt;.</p>
<h2>Noncompliant Code Example</h2>
<p>Using the default threshold of 500:</p>
<pre>
01  MY-TABLE.
05 MY-TAB-ELEM OCCURS 300000
     INDEXED BY MY-TAB-IND.
  10 MY-ATTR1                        PIC X(07).
  10 MY-ATTR2                        PIC X(07).
  10 MY-ATTR3                        PIC X(07).

01  MY-TAB2.
05 MY-TAB2-ELEM          OCCURS 300000
     ASCENDING MY-ATTR1  *&gt; Key is defined. Why not use it?
     INDEXED BY MY-TAB-IND.
  10 MY-ATTR1                        PIC X(07).
  10 MY-ATTR2                        PIC X(07).
  10 MY-ATTR3                        PIC X(07).

01  MY-TAB-IND             PIC 9(08).


SEARCH MY-TAB-ELEM.  *&gt; Noncompliant; define a key &amp; use binary search
       AT END...

SEARCH MY-TAB2-ELEM.  *&gt; Noncompliant
       AT END...
</pre>
<h2>Compliant Solution</h2>
<pre>
01  MY-TABLE.
05 MY-TAB-ELEM OCCURS 300000
     ASCENDING MY-ATTR1
     INDEXED BY MY-TAB-IND.
  10 MY-ATTR1                        PIC X(07).
  10 MY-ATTR2                        PIC X(07).
  10 MY-ATTR3                        PIC X(07).

01  MY-TAB2.
05 MY-TAB2-ELEM          OCCURS 300000
     ASCENDING MY-ATTR1
     INDEXED BY MY-TAB-IND.
  10 MY-ATTR1                        PIC X(07).
  10 MY-ATTR2                        PIC X(07).
  10 MY-ATTR3                        PIC X(07).

01  MY-TAB-IND             PIC 9(08).


SEARCH ALL MY-TAB-ELEM.
       AT END...

SEARCH ALL MY-TAB2-ELEM.
       AT END...
</pre>
