<p>&lt;code&gt;java.lang.Error&lt;/code&gt; and its subclasses represent abnormal conditions, such as &lt;code&gt;OutOfMemoryError&lt;/code&gt;, which should only be encountered by the Java Virtual Machine.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public class MyException extends Error { /* ... */ }       // Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
public class MyException extends Exception { /* ... */ }   // Compliant
</pre>
