<code>java.lang.Error</code> and its subclasses represent abnormal conditions, such as <code>OutOfMemoryError</code>, which should only be encountered by the Java Virtual Machine.


== Noncompliant Code Example

----
public class MyException extends Error { /* ... */ }       // Noncompliant
----


== Compliant Solution

----
public class MyException extends Exception { /* ... */ }   // Compliant
----


