<p>The use of &lt;code&gt;Option Base&lt;/code&gt; to change the lower bound of an arrayâ€™s index values results in confusing code.</p>
<h2>Noncompliant Code Example</h2>
<pre>
Option Explicit
Option Base 1

'...
    Dim MyArray(1 To 3) As Integer
    For I = 1 To 3
        MsgBox MyArray(I)
    Next I
</pre>
<h2>Compliant Solution</h2>
<pre>
Option Explicit

'...
    Dim MyArray(0 To 2) As Integer
    For I = 0 To 2
        MsgBox MyArray(I)
    Next I
</pre>
