<code>SELECT</code> can contain an <code>OTHERWISE</code> clause for various reasons: to handle unexpected values, to show that all the cases were properly considered.
For readability purpose, to help a developer to quickly find the default behavior of a <code>SELECT</code> statement, it is recommended to put the <code>OTHERWISE</code> clause at the end of the <code>SELECT</code> statement. This rule raises an issue if the <code>OTHERWISE</code> clause is not the first or the last one of the <code>SELECT</code>'s cases.

== Noncompliant Code Example

----
foo: proc options(main);
  declare i fixed decimal init (42);

  select (i);
    when (0) put list ('i = 0');
    otherwise put list ('i = 2'); /* Non-Compliant */
    when (1) put list ('i = 1');
  end;
end;
----

== Compliant Solution

----
foo: proc options(main);
  declare i fixed decimal init (42);

  select (i); /* Compliant */
    when (0) put list ('i = 0');
    when (1) put list ('i = 1');
    otherwise put list ('i = 2');
  end;
end;
----
