In Python, special methods corresponding to numeric operators, rich comparison operators and the <code>\_\_length_hint\_\_</code> method should return <code>NotImplemented</code> when the operation is not supported. These methods should not raise <code>NotImplementedError</code> as callers don't expect it and won't catch this exception.

For example <code>A + B</code> is equivalent to calling <code>A.\_\_add\_\_(B)</code>. If this binary operation is not supported by class A, <code>A.\_\_add\_\_(B)</code> should return <code>NotImplemented</code>. The interpreter will then try the reverse operation, i.e. <code>B.\_\_radd\_\_(A)</code>. This enables adding new operations by changing only one class instead of two.

This rule raises an issue when one of the following methods raises <code>NotImplementedError</code> instead of returning <code>NotImplemented</code>:

* \_\_lt\_\_(self, other)
* \_\_le\_\_(self, other)
* \_\_eq\_\_(self, other)
* \_\_ne\_\_(self, other)
* \_\_gt\_\_(self, other)
* \_\_ge\_\_(self, other)
* \_\_add\_\_(self, other)
* \_\_sub\_\_(self, other)
* \_\_mul\_\_(self, other)
* \_\_matmul\_\_(self, other)
* \_\_truediv\_\_(self, other)
* \_\_floordiv\_\_(self, other)
* \_\_mod\_\_(self, other)
* \_\_divmod\_\_(self, other)
* \_\_pow\_\_(self, other[, modulo])
* \_\_lshift\_\_(self, other)
* \_\_rshift\_\_(self, other)
* \_\_and\_\_(self, other)
* \_\_xor\_\_(self, other)
* \_\_or\_\_(self, other)
* \_\_radd\_\_(self, other)
* \_\_rsub\_\_(self, other)
* \_\_rmul\_\_(self, other)
* \_\_rmatmul\_\_(self, other)
* \_\_rtruediv\_\_(self, other)
* \_\_rfloordiv\_\_(self, other)
* \_\_rmod\_\_(self, other)
* \_\_rdivmod\_\_(self, other)
* \_\_rpow\_\_(self, other[, modulo])
* \_\_rlshift\_\_(self, other)
* \_\_rrshift\_\_(self, other)
* \_\_rand\_\_(self, other)
* \_\_rxor\_\_(self, other)
* \_\_ror\_\_(self, other)
* \_\_iadd\_\_(self, other)
* \_\_isub\_\_(self, other)
* \_\_imul\_\_(self, other)
* \_\_imatmul\_\_(self, other)
* \_\_itruediv\_\_(self, other)
* \_\_ifloordiv\_\_(self, other)
* \_\_imod\_\_(self, other)
* \_\_ipow\_\_(self, other[, modulo])
* \_\_ilshift\_\_(self, other)
* \_\_irshift\_\_(self, other)
* \_\_iand\_\_(self, other)
* \_\_ixor\_\_(self, other)
* \_\_ior\_\_(self, other)
* \_\_length_hint\_\_(self)


== Noncompliant Code Example

----
class MyClass:
    def __add__(self, other):
        raise NotImplementedError()  # Noncompliant
    def __radd__(self, other):
        raise NotImplementedError()  # Noncompliant

class MyOtherClass:
    def __add__(self, other):
        return 42
    def __radd__(self, other):
        return 42

MyClass() + MyOtherClass()  # This will raise NotImplementedError
----


== Compliant Solution

----
class MyClass:
    def __add__(self, other):
        return NotImplemented
    def __radd__(self, other):
        return NotImplemented

class MyOtherClass:
    def __add__(self, other):
        return 42
    def __radd__(self, other):
        return 42

MyClass() + MyOtherClass()  # This returns 42
----


== See

* Python documentation - https://docs.python.org/3/library/constants.html#NotImplemented[Built-in Constants - NotImplemented]
* Python documentation - https://docs.python.org/3/library/numbers.html#implementing-the-arithmetic-operations[Implementing the arithmetic operations]

