<p>A &lt;code&gt;NullPointerException&lt;/code&gt; should indicate that a &lt;code&gt;null&lt;/code&gt; value was unexpectedly encountered. Good programming practice dictates that code is structured to avoid NPE’s.</p>
<p>Explicitly throwing &lt;code&gt;NullPointerException&lt;/code&gt; forces a method’s callers to explicitly catch it, rather than coding to avoid it. Further, it makes it difficult to distinguish between the unexpectedly-encountered &lt;code&gt;null&lt;/code&gt; value and the condition which causes the method to purposely throw an NPE.</p>
<p>If an NPE is being thrown to indicate that a parameter to the method should not have been null, use the &lt;code&gt;@NotNull&lt;/code&gt; annotation instead.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public void doSomething (String aString) throws NullPointerException {
     throw new NullPointerException();
}
</pre>
<h2>Compliant Solution</h2>
<pre>
public void doSomething (@NotNull String aString) {
}
</pre>
