When a non-<code>static</code> class field is annotated with <code>ThreadStatic</code>, the code seems to show that the field can have different values for different calling threads, but that's not the case, since the <code>ThreadStatic</code> attribute is simply ignored on non-<code>static</code> fields. 

So <code>ThreadStatic</code> should either be removed or replaced with a use of the <code>ThreadLocal<T></code> class, which gives a similar behavior for non-<code>static</code> fields.


== Noncompliant Code Example

----
public class MyClass 
{
  [ThreadStatic]  // Noncompliant
  private int count = 0;

  // ...
}
----


== Compliant Solution

----
public class MyClass 
{
  private int count = 0;

  // ...
}
----
or

----
public class MyClass 
{
  private readonly ThreadLocal<int> count = new ThreadLocal<int>();
  public int Count
  {
    get { return count.Value; }
    set { count.Value = value; }
  }
  // ...
}
----


