Since C++11, declaring a variable, class, or function in an unnamed <code>namespace</code> gives it internal linkage. Similarly, marking a declaration <code>static</code> also gives it internal linkage. Because both mechanisms have the same effect (although <code>static</code> has a narrower application) using them together is clearly redundant.


== Noncompliant Code Example

----
namespace {
  static int i = 3;  // Noncompliant
}
----


== Compliant Solution

----
namespace {
  int i = 3;
}
----


== See

* ISO/IEC 14882:2011 ยง3.5 paragraph 4

