<p>A method using the &lt;code&gt;VarArgs&lt;/code&gt; calling convention is not Common Language Specification (CLS) compliant and might not be accessible across programming languages, while the &lt;code&gt;params&lt;/code&gt; keyword works the same way and is CLS compliant.</p>
<p>This rule raises an issue when a &lt;code&gt;public&lt;/code&gt; or &lt;code&gt;protected&lt;/code&gt; type contains a &lt;code&gt;public&lt;/code&gt; or &lt;code&gt;protected&lt;/code&gt; method that uses the &lt;code&gt;VarArgs&lt;/code&gt; calling convention.</p>
<h2>Noncompliant Code Example</h2>
<pre>
using System;

namespace MyLibrary
{
    public class Foo
    {
        public void Bar(__arglist) // Noncompliant
        {
            ArgIterator argumentIterator = new ArgIterator(__arglist);
            for(int i = 0; i &lt; argumentIterator.GetRemainingCount(); i++)
            {
                Console.WriteLine(
                    __refvalue(argumentIterator.GetNextArg(), string));
            }
        }
    }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
using System;

[assembly: CLSCompliant(true)]
namespace MyLibrary
{
    public class Foo
    {
        public void Bar(params string[] wordList)
        {
            for(int i = 0; i &lt; wordList.Length; i++)
            {
                Console.WriteLine(wordList[i]);
            }
        }
    }
}
</pre>
<h2>Exceptions</h2>
<p>Interop methods using &lt;code&gt;VarArgs&lt;/code&gt; calling convention do not raise an issue.</p>
<pre>
[DllImport("msvcrt40.dll")]
public static extern int printf(string format, __arglist); // Compliant
</pre>
