A method using the <code>VarArgs</code> calling convention is not Common Language Specification (CLS) compliant and might not be accessible across programming languages, while the <code>params</code> keyword works the same way and is CLS compliant.

This rule raises an issue when a <code>public</code> or <code>protected</code> type contains a <code>public</code> or <code>protected</code> method that uses the <code>VarArgs</code> calling convention.


== Noncompliant Code Example

----
using System;

namespace MyLibrary
{
    public class Foo 
    {
        public void Bar(__arglist) // Noncompliant
        { 
            ArgIterator argumentIterator = new ArgIterator(__arglist);
            for(int i = 0; i < argumentIterator.GetRemainingCount(); i++) 
            { 
                Console.WriteLine(
                    __refvalue(argumentIterator.GetNextArg(), string));
            } 
        }
    }
}
----


== Compliant Solution

----
using System;

[assembly: CLSCompliant(true)]
namespace MyLibrary
{
    public class Foo 
    {
        public void Bar(params string[] wordList)
        { 
            for(int i = 0; i < wordList.Length; i++) 
            { 
                Console.WriteLine(wordList[i]);
            } 
        }
    }
}
----


== Exceptions

Interop methods using <code>VarArgs</code> calling convention do not raise an issue.

----
[DllImport("msvcrt40.dll")]
public static extern int printf(string format, __arglist); // Compliant
----

