<p>Characters like &lt;code&gt;'é'&lt;/code&gt; can be expressed either as a single code point or as a cluster of the letter &lt;code&gt;'e'&lt;/code&gt; and a combining accent mark. Without the &lt;code&gt;CANON_EQ&lt;/code&gt; flag, a regex will only match a string in which the characters are expressed in the same way.</p>
<h2>Noncompliant Code Example</h2>
<pre>
String s = "e\u0300";
Pattern p = Pattern.compile("é|ë|è"); // Noncompliant
System.out.println(p.matcher(s).replaceAll("e")); // print 'é'
</pre>
<h2>Compliant Solution</h2>
<pre>
String s = "e\u0300";
Pattern p = Pattern.compile("é|ë|è", Pattern.CANON_EQ);
System.out.println(p.matcher(s).replaceAll("e")); // print 'e'
</pre>
