The <code><xs:all></code> group element isn't supported universally among web service stacks. That means that if you use it, you may end up failing to transmit your data (in the resulting XML files) to systems that can't process it. For that reason, <code><xs:sequence></code> should be used instead. The only constraint imposed by the use of an <code><xs:sequence></code> is the order it imposes, but that is usually a benefit for human readers.


== Noncompliant Code Example

----
<xs:complexType name="fruitType">
  <xs:all>  <!-- Noncompliant -->
    <xs:element name="name" type="xs:string"/>
    <xs:element name="color" type="xs:string"/>
  </xs:all>
</xs:complexType>
----


== Compliant Solution

----
<xs:complexType name="fruitType">
  <xs:sequence>
    <xs:element name="name" type="xs:string"/>
    <xs:element name="color" type="xs:string"/>
  </xs:sequence>
</xs:complexType>
----

