Doing an operation on a string without using the result of the operation is useless and is certainly due to a misunderstanding. 

== Noncompliant Code Example

----
var str = "..."
str.toUpperCase(); // Noncompliant
----

== Compliant Solution

----
var str = "..."
str = str.toUpperCase();
----

include::../see.adoc[]
