include::../description.adoc[]

== Noncompliant Code Example

----
for i := 0; i < 10; i++ { // noncompliant, loop only executes once
	fmt.Println(i)
	break
}
----

== Compliant Solution

----
for i := 0; i < 10; i++ {
	fmt.Println(i)
}
----
