include::../description.adoc[]

== Noncompliant Code Example

With the default threshold of 3:

----
function myFunction():boolean { // Noncompliant as there are 4 return statements
  if (condition1) {
    return true;
  } else {
    if (condition2) {
      return false;
    } else {
      return true;
    }
  }
  return false;
}
----
