If a variable is declared but not used in the program, it can be considered dead code and should therefore be removed. This will improve maintainability because developers will not wonder what the variable is used for.

== Noncompliant Code Example

----
D I               S              5P 0      INZ
D* Noncompliant; X is not used in C-Specs
D X               S              5P 0      INZ

C                   EVAL      I = I + 1
----

== Compliant Solution

----
D I               S              5P 0      INZ

C                   EVAL      I = I + 1
----
