Methods that override parent class implementations should end with calls to the <code>super</code> implementations. Otherwise, you risk leaving the job half done.

This rule raises an issue when the following <code>Activity</code> methods do not end with a call to <code>super</code>:

* <code>onDestroy</code>
* <code>onPause</code>
* <code>onSaveInstanceState</code>
* <code>onStop</code>
* <code>onTerminate</code>


== Noncompliant Code Example

----
//TODO
----


== Compliant Solution

----
//TODO
----

