<p>For better readability, it is preferable to use &lt;code&gt;DECLARE&lt;/code&gt; instead of the unpronounceable &lt;code&gt;DCL&lt;/code&gt; abbreviation.</p>
<h2>Noncompliant Code Example</h2>
<pre>
foo: proc options(main);
  dcl i fixed decimal init (0); /* Noncompliant */
end;
</pre>
<h2>Compliant Solution</h2>
<pre>
foo: proc options(main);
  declare i fixed decimal init (0); /* Compliant */
end;
</pre>
