For better readability, it is preferable to use <code>DECLARE</code> instead of the unpronounceable <code>DCL</code> abbreviation.


== Noncompliant Code Example

----
foo: proc options(main);
  dcl i fixed decimal init (0); /* Noncompliant */
end;
----


== Compliant Solution

----
foo: proc options(main);
  declare i fixed decimal init (0); /* Compliant */
end;
----

