Using several "--" or "++" unary operators in the same arithmetic expression can quickly make the expression unreadable. 


== Noncompliant Code Example

----
var j:int = foo++ - --bar;
----


== Compliant Solution

----
bar--;
var j:int = foo++ - bar;
----

