Postfix increment and decrement typically involves making a copy of the object being incremented or decremented, whereas its prefix form does not. Therefore the prefix form is usually the more efficient form, and should be preferred.


== Noncompliant Code Example

----
void myFunc(int lim)
{
  int i;
  for (i = 0; i < lim; i++)
  {
    // do something
  }
}
----


== Compliant Solution

----
void myFunc(int lim)
{
  int i;
  for (i = 0; i < lim; ++i)
  {
    // do something
  }
}
----


