<p>Having all branches in a &lt;code&gt;CASE&lt;/code&gt; or &lt;code&gt;IF&lt;/code&gt; chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldnâ€™t be a &lt;code&gt;CASE&lt;/code&gt;/&lt;code&gt;IF&lt;/code&gt; chain at all.</p>
<h2>Noncompliant Code Example</h2>
<pre>
IF a &gt;0.
    doSomething.
ELSE IF b&gt; 0.
    doSomething.
ELSE.
    doSomething.
ENDIF.

CASE i.
  WHEN 1 OR 3.
    doSomething.
  WHEN 2.
    doSomething.
  WHEN OTHERS.
        doSomething.
ENDCASE.
</pre>
<h2>Exceptions</h2>
<p>Note that this rule does not apply to &lt;code&gt;IF&lt;/code&gt; chains without final &lt;code&gt;ELSE&lt;/code&gt;-s, or to &lt;code&gt;CASE&lt;/code&gt;-s without &lt;code&gt;WHEN OTHERS&lt;/code&gt; clauses.</p>
<pre>
IF a &gt;0.   "no issue, this could have been done on purpose to make the code more readable
    doSomething.
ELSEIF b&gt; 0.
    doSomething.
ENDIF.
</pre>
