<p>Having all branches in an &lt;code&gt;EVALUATE&lt;/code&gt; or &lt;code&gt;IF&lt;/code&gt; chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldnâ€™t be an &lt;code&gt;EVALUATE&lt;/code&gt;/&lt;code&gt;IF&lt;/code&gt; chain at all.</p>
<h2>Noncompliant Code Example</h2>
<pre>
EVALUATE X *&gt; Noncompliant
  WHEN 1
    PERFORM SECTION1
  WHEN OTHER
    PERFORM SECTION1
END-EVALUATE.

IF X = 1 THEN *&gt; Noncompliant
  PERFORM SECTION1
ELSE
  PERFORM SECTION1
END-IF.
</pre>
<h2>Exceptions</h2>
<p>Note that this rule does not apply to &lt;code&gt;IF&lt;/code&gt; chains without &lt;code&gt;ELSE&lt;/code&gt;-s, or to &lt;code&gt;EVALUATE&lt;/code&gt;-s without &lt;code&gt;WHEN OTHER&lt;/code&gt; clauses.</p>
<pre>
IF X = 1 THEN *&gt; //no issue, this could have been done on purpose to make the code more readable
  PERFORM SECTION1
ELSE-IF X = 2 THEN
  PERFORM SECTION1
END-IF.
</pre>
