<p>Having all branches in a &lt;code&gt;CASE&lt;/code&gt; or &lt;code&gt;IF&lt;/code&gt;/&lt;code&gt;ELSIF&lt;/code&gt; chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldnâ€™t be a &lt;code&gt;CASE&lt;/code&gt;/&lt;code&gt;IF&lt;/code&gt;/&lt;code&gt;ELSIF&lt;/code&gt; chain at all.</p>
<h2>Noncompliant Code Example</h2>
<pre>
IF param = 1 THEN
  result := 'A';
ELSIF param = 2 THEN
  result := 'A';
ELSE
  result := 'A';
END IF;

result := CASE param
   WHEN 1 THEN 'A'
   WHEN 2 THEN 'A'
   ELSE 'A'
END;
</pre>
<h2>Compliant Solution</h2>
<pre>
IF param = 1 THEN
  result := 'A';
ELSIF param = 2 THEN
  result := 'B';
ELSE
  result := 'C';
END IF;

result := CASE param
   WHEN 1 THEN 'A'
   WHEN 2 THEN 'B'
   ELSE 'C'
END;
</pre>
<h2>Exceptions</h2>
<p>This rule does not apply to &lt;code&gt;IF&lt;/code&gt;/&lt;code&gt;CASE&lt;/code&gt; chains without &lt;code&gt;ELSE&lt;/code&gt; clauses.</p>
<pre>
IF param = 1 THEN   -- no issue, this could have been done on purpose to make the code more readable
  result := 'A';
ELSIF param = 2 THEN
  result := 'A';
END IF;
</pre>
