<p>Having the same implementation in every branch of an &lt;code&gt;if&lt;/code&gt; chain is an error. Either a copy-paste error was made and something different should be executed, or there shouldnâ€™t be an &lt;code&gt;if&lt;/code&gt; chain at all.</p>
<h2>Noncompliant Code Example</h2>
<pre>
if b == 0:  # Noncompliant
    doOneMoreThing()
elif b == 1:
    doOneMoreThing()
else:
    doOneMoreThing()

b = 4 if a &gt; 12 else 4  # Noncompliant
</pre>
<h2>Exceptions</h2>
<p>This rule does not apply to &lt;code&gt;if&lt;/code&gt; chains without &lt;code&gt;else&lt;/code&gt;-s.</p>
<pre>
if b == 0:  # no issue, this could have been done on purpose to make the code more readable
    doOneMoreThing()
elif b == 1:
    doOneMoreThing()
</pre>
