<p>Having all branches in a &lt;code&gt;case&lt;/code&gt; or &lt;code&gt;if&lt;/code&gt; chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldnâ€™t be a &lt;code&gt;case&lt;/code&gt;/&lt;code&gt;if&lt;/code&gt; chain at all.</p>
<h2>Noncompliant Code Example</h2>
<pre>
if b == 0  # Noncompliant
  doOneMoreThing()
else
  doOneMoreThing()
end

b = a &gt; 12 ? 4 : 4;  # Noncompliant

case i  # Noncompliant
  when 1
    doSomething()
  when 2
    doSomething()
  when 3
    doSomething()
  else
    doSomething()
end
</pre>
<h2>Exceptions</h2>
<p>This rule does not apply to &lt;code&gt;if&lt;/code&gt; chains or  &lt;code&gt;case&lt;/code&gt;-es without &lt;code&gt;else&lt;/code&gt;-s.</p>
<pre>
if b == 0 # no issue, this could have been done on purpose to make the code more readable
  doSomething()
elsif b == 1
  doSomething()
end
</pre>
