<p>Having all branches in a &lt;code&gt;CASE&lt;/code&gt;, &lt;code&gt;IF&lt;/code&gt; or &lt;code&gt;IIF&lt;/code&gt; chain with the same implementation is an error. Either a copy-paste error was made and something different should be executed, or there shouldnâ€™t be a &lt;code&gt;CASE&lt;/code&gt;/&lt;code&gt;IF&lt;/code&gt;/&lt;code&gt;IIF&lt;/code&gt; chain at all.</p>
<h2>Noncompliant Code Example</h2>
<pre>
IF @x &lt; 25 -- Noncompliant
  PRINT 'A'
ELSE IF @x &lt; 10
  PRINT 'A'
ELSE
  PRINT 'A'

SELECT
  CASE col1 -- Noncompliant
    WHEN 1 THEN 'A'
    WHEN 2 THEN 'A'
    ELSE        'A'
  END,
  IIF(col1 &lt; 25, 'A', 'A') -- Noncompliant
 FROM table1
</pre>
<h2>Exceptions</h2>
<p>This rule does not apply to &lt;code&gt;IF&lt;/code&gt;/&lt;code&gt;CASE&lt;/code&gt; chains without &lt;code&gt;ELSE&lt;/code&gt; clauses.</p>
<pre>
IF @x &lt; 25 -- no issue, this could have been done on purpose to make the code more readable
  PRINT 'A'
ELSE IF @x &gt; 10
  PRINT 'A'
</pre>
