include::../description.adoc[]

== Noncompliant Code Example

----
      /free
        name = name;
      /end-free
----

== Compliant Solution

----
      /free
        customer.name = name;
      /end-free
----

include::../see.adoc[]
