The first argument to <code>super</code> must be the name of the class making the call. If it's not, the result will be a runtime error.


== Noncompliant Code Example

----
class Person(object):
  #...

class PoliceOfficer(Person):
  def __init__(self, name):
    super().__init__(name)  // Noncompliant
----


== Compliant Solution

----
class Person(object):
  #...

class PoliceOfficer(Person):
  def __init__(self, name):
    super(PoliceOfficer).__init__(name)
----


