<p>Having all the pieces in front of you makes it easier to understand and debug or maintain a piece of code. Unfortunately, thatâ€™s often not the case with compile-time arrays, since they are often separated from the code that uses them by many, many lines. Instead, data structures are often a better option.</p>
<h2>Noncompliant Code Example</h2>
<pre>
     D directions      S              5  0 dim(4) ctdata perrcd(1)
</pre>
<h2>Compliant Solution</h2>
<pre>
     D dirData         DS
     D                                5    Inz('North')
     D                                5    Inz('South')
     D                                5    Inz('East')
     D                                5    Inz('West')
     D directions                     5    Dim(4) Overlay(dirData)
</pre>
