<p>Making a public constant just &lt;code&gt;const&lt;/code&gt; as opposed to &lt;code&gt;static const&lt;/code&gt; leads to duplicating its value for every instance of the class, uselessly increasing the amount of memory required to execute the application.</p>
<h2>Noncompliant Code Example</h2>
<pre>
public class Myclass
{
  public const THRESHOLD:int = 3;
}
</pre>
<h2>Compliant Solution</h2>
<pre>
public class Myclass
{
  public static const THRESHOLD:int = 3;
}
</pre>
