Calling <code>Class.newInstance</code> invokes the class' default constructor. Unfortunately, since it's not a direct call to the constructor, compile-time checking will be unable to detect the possibility. This means that your code will compile even if you haven't put the invocation in a <code>try</code> block.

On the other hand, <code>Construtor.newInstance</code> handles exceptions by wrapping them in an <code>InvocationTargetException</code> and explicitly throwing them. 

This rule raises an issue when <code>Class.newInstance</code> is used to invoke a constructor that throws checked exceptions.


== Noncompliant Code Example

----
Foo f = Foo.class.newInstance();  // Noncompliant
----


== Compliant Solution

----
Foo f = Foo.class.getConstructor().newInstance();
----

