An <code>INSERT</code> statement that does not explicitly list the columns being inserted into, as well as the values being inserted, is dependent for correct functioning on the structure of the table not changing. Additionally, not having the explicit column list degrades the readability and understandability of the code. Therefore, <code>INSERT</code> statements should always contain an explicit column list.


== Noncompliant Code Example

----
INSERT INTO PERSONS VALUES (1, 'DUPONT', 'Marcel')
----


== Compliant Solution

----
INSERT INTO PERSONS (ID, LAST_NAME, FIRST_NAME) 
VALUES (1, 'DUPONT', 'Marcel')
----

