include::../description.adoc[]

== Noncompliant Code Example

----
System.out.println("My Message");  // Noncompliant
----

== Compliant Solution

----
logger.log("My Message");
----

include::../see.adoc[]
