C++11 improves the specification of the parser so that at the end of a template, multiple <code>></code>s will be interpreted as closing the template argument list rather than as the right shift operator or stream extraction operator. It is no longer required to place a space between the two <code>></code>s. 

This rule raises an issue when such useless spaces exist.


== Noncompliant Code Example

----
vector<pair<int,int> > s; // Noncompliant
----


== Compliant Solution

----
vector<pair<int,int>> s;
----

