<p>Shared coding conventions allow teams to collaborate efficiently. This rule checks that type names match the provided regular expression.</p>
<h2>Noncompliant Code Example</h2>
<p>With the default regular expression &lt;code&gt;[a-zA-Z]([a-zA-Z0-9_]*[a-zA-Z0-9])?&lt;/code&gt;:</p>
<pre>
DECLARE
  TYPE Collection-type_ IS VARRAY(42) OF PLS_INTEGER; -- Noncompliant
BEGIN
  NULL;
END;
/
</pre>
<h2>Compliant Solution</h2>
<pre>
DECLARE
  TYPE collectionType IS VARRAY(42) OF PLS_INTEGER;
BEGIN
  NULL;
END;
/ {code}
</pre>
