Shared coding conventions allow teams to collaborate efficiently. This rule checks that type names match the provided regular expression.


== Noncompliant Code Example

With the default regular expression <code>[a-zA-Z]([a-zA-Z0-9_]*[a-zA-Z0-9])?</code>:

----
DECLARE
  TYPE Collection-type_ IS VARRAY(42) OF PLS_INTEGER; -- Noncompliant
BEGIN
  NULL;
END;
/
----


== Compliant Solution

----
DECLARE
  TYPE collectionType IS VARRAY(42) OF PLS_INTEGER;
BEGIN
  NULL;
END;
/ {code}
----

