Shared coding conventions allow teams to collaborate effectively. This rule allows to check that all class names match a provided regular expression.

The convention in go is to use mixedCaps rather than underscores. See https://golang.org/doc/effective_go.html#names[Go documentation] for the complete naming conventions.

Note that the casing of the first character determines if the type is exported or not.

== Noncompliant Code Example

With default provided regular expression <code>^[a-zA-Z][a-zA-Z0-9]*$</code>:

----
type my_struct struct {...}
----

== Compliant Solution

----
type myStruct struct {...}
----
