Frequent commits are widely understood to negatively impact performance. Thus, committing inside a loop (even when only executed conditionally once every n iterations) is highly likely to cause unwanted performance impacts. 

Further, in general use <code>COMMIT</code> should only be used at the end of a transaction. Code that is not structured to have one transaction per loop iteration could yield unexpected results if <code>COMMIT</code> is nonetheless used inside the loop. Code that _is_ structured to have one transaction per loop iteration should probably be reconsidered.

Note that when dealing with very large data sets, a <code>COMMIT</code> may be required every n iterations, but the goal should be to avoid <code>COMMIT</code>s inside loops.


== Noncompliant Code Example

----
FOR item IN itemlist
LOOP
  -- ...   
  COMMIT;  -- Noncompliant
END LOOP;
----


== Compliant Solution

----
FOR item IN itemlist
LOOP
  -- ...   
END LOOP;
COMMIT;
----

