include::../description.adoc[]

== Noncompliant Code Example

----
public void doSomething() {
}
----

== Compliant Solution

----
public void doSomething() {
  // Do nothing because of X and Y.
}
----
